The aim of this project is to assess the degree to which metastasis of NSCLC can be accurately predicted using genomic data from TRACERx and elucidate the factors that confer metasatic potential in cancer evolution.

This repository contains the code used to process and analyse data from the TRACERx 421 cohort using machine learning models, as well as the prediction results, performance metrics, and plots.

  - The [Matrices](https://github.com/jasminechau/INIM0038-research-project/tree/main/Matrices) folder contains the data used in the original analyses, as well as those with cumulative features.
  - Jupyter Notebooks contain the code specifically written for this project:
    - [code_template](https://github.com/jasminechau/INIM0038-research-project/blob/main/code_template.ipynb) contains templates of logistic regression, SVM, and MLP models which were can be used to make predictions of recurrence with different matrices as input. 
    - [TRACERx 4 genes](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%204%20genes.ipynb) and [TRACERx 100-10 genes](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20100-10%20genes.ipynb) focus on the models using only the presence or absence of frequent mutations to make predictions of metastasis.
    - [TRACERx LUAD_LUSC](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20LUAD_LUSC.ipynb), [TRACERx Ploidy](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20Ploidy.ipynb), [TRACERx Timing](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20Timing.ipynb), [TRACERx copy number](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20copy%20number.ipynb), and [TRACERx cumulative](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20cumulative.ipynb) contain models that build on those from [TRACERx 100-10 genes](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20100-10%20genes.ipynb), incorporating various combinations of genetic and evolutionary features to determine their predictiveness for metastasis.
    - [TRACERx metastatic patterns](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20metastatic%20patterns.ipynb) and [TRACERx feature selection](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20feature%20selection.ipynb) contain models trained on the presence/absence of genes identified as metastatic patterns in the [MSK-MET cohort](https://doi.org/10.1016/j.cell.2022.01.003), and driver mutations with the greatest increase in frequency between recurrence and no-recurrence TRACERx patient cohorts.
  - Performance metrics for all models can be found in [Prediction results](https://github.com/jasminechau/INIM0038-research-project/blob/main/Prediction%20results.csv) - which was used to generate plots in [TRACERx plots](https://github.com/jasminechau/INIM0038-research-project/blob/main/TRACERx%20plots.ipynb).

The [figurecode](https://github.com/jasminechau/INIM0038-research-project/tree/main/figurecode) and [transcriptomics_scripts_data_updated](https://github.com/jasminechau/INIM0038-research-project/tree/main/transcriptomics_scripts_data_updated/20221014_transcriptomic_DATA) folders are from open source TRACERx data which can be found here: https://github.com/McGranahanLab/LCCE_workshop_2023/wiki.
