{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/y8/66w8_8kj0dx2znq80cdbl_rw0000gn/T/ipykernel_39093/4191223103.py:2: DeprecationWarning: \n",
      "Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),\n",
      "(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)\n",
      "but was not found to be installed on your system.\n",
      "If this would cause problems for you,\n",
      "please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466\n",
      "        \n",
      "  import pandas as pd\n"
     ]
    }
   ],
   "source": [
    "import collections\n",
    "import pandas as pd\n",
    "import pyreadr\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import seaborn as sns\n",
    "from venn import venn\n",
    "import numpy as np\n",
    "import math\n",
    "import scipy\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cruk_id</th>\n",
       "      <th>tumour_id_muttable_cruk</th>\n",
       "      <th>tumour_id_per_patient</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>packyears</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>ECOG_PS</th>\n",
       "      <th>pathologyTNM</th>\n",
       "      <th>pT_stage_per_patient</th>\n",
       "      <th>pN_stage_per_patient</th>\n",
       "      <th>LVI_per_patient</th>\n",
       "      <th>PL_per_patient</th>\n",
       "      <th>margin_status_per_patient</th>\n",
       "      <th>size_pathology_per_patient</th>\n",
       "      <th>Surgery_type</th>\n",
       "      <th>histology_lesion1</th>\n",
       "      <th>histology_lesion1_merged</th>\n",
       "      <th>lesion1_sampled</th>\n",
       "      <th>histology_lesion2</th>\n",
       "      <th>lesion2_sampled</th>\n",
       "      <th>histology_multi_full</th>\n",
       "      <th>histology_multi_full_genomically.confirmed</th>\n",
       "      <th>LUAD_pred_subtype</th>\n",
       "      <th>adjuvant_treatment_YN</th>\n",
       "      <th>adjuvant_treatment_given</th>\n",
       "      <th>num_cycle_na.added</th>\n",
       "      <th>CHMPlatDgName_cleaned</th>\n",
       "      <th>CHMOthDgName_cleaned</th>\n",
       "      <th>AdjRadStartTime_manual</th>\n",
       "      <th>AdjRadEndTime_manual</th>\n",
       "      <th>Recurrence_time_use</th>\n",
       "      <th>newPrim_time_use</th>\n",
       "      <th>first_dfs_any_event_rec.or.new.primary</th>\n",
       "      <th>first_event_during_followup</th>\n",
       "      <th>cens_os</th>\n",
       "      <th>os_time</th>\n",
       "      <th>cens_dfs</th>\n",
       "      <th>dfs_time</th>\n",
       "      <th>cens_dfs_any_event</th>\n",
       "      <th>dfs_time_any_event</th>\n",
       "      <th>cens_lung_event</th>\n",
       "      <th>lung_event_time</th>\n",
       "      <th>Relapse_cat</th>\n",
       "      <th>Relapse_cat_new</th>\n",
       "      <th>tx100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>30</td>\n",
       "      <td>30.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>0</td>\n",
       "      <td>1120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>81</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>5.0</td>\n",
       "      <td>35</td>\n",
       "      <td>8.75</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>217.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>2304</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>69</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>325.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>1323</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>74</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>56</td>\n",
       "      <td>56.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>0</td>\n",
       "      <td>649</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>65</td>\n",
       "      <td>Female</td>\n",
       "      <td>Middle eastern</td>\n",
       "      <td>10.0</td>\n",
       "      <td>45</td>\n",
       "      <td>22.50</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1b</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>30</td>\n",
       "      <td>15.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>66</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>54</td>\n",
       "      <td>54.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>883.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>893</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881_Tumour1;CRUK0881_Tumour2</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>Lesion1: collision tumour</td>\n",
       "      <td>False</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>758</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>15.0</td>\n",
       "      <td>43</td>\n",
       "      <td>32.25</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>421 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      cruk_id tumour_id_muttable_cruk              tumour_id_per_patient  age  \\\n",
       "88   CRUK0001                CRUK0001                           CRUK0001   84   \n",
       "79   CRUK0002                CRUK0002                           CRUK0002   81   \n",
       "26   CRUK0003                CRUK0003                           CRUK0003   78   \n",
       "23   CRUK0004                CRUK0004                           CRUK0004   69   \n",
       "7    CRUK0005                CRUK0005                           CRUK0005   74   \n",
       "..        ...                     ...                                ...  ...   \n",
       "414  CRUK0861                CRUK0861                           CRUK0861   65   \n",
       "418  CRUK0864                CRUK0864                           CRUK0864   78   \n",
       "416  CRUK0872                CRUK0872                           CRUK0872   66   \n",
       "420  CRUK0881        CRUK0881_Tumour1  CRUK0881_Tumour1;CRUK0881_Tumour2   84   \n",
       "415  CRUK0893                CRUK0893                           CRUK0893   59   \n",
       "\n",
       "        sex       ethnicity  cigs_perday  years_smoking  packyears  \\\n",
       "88     Male  White- British         20.0             30      30.00   \n",
       "79     Male  White- British          5.0             35       8.75   \n",
       "26   Female  White- British          0.0              0       0.00   \n",
       "23   Female  White- British         10.0             27      13.50   \n",
       "7    Female  White- British         20.0             56      56.00   \n",
       "..      ...             ...          ...            ...        ...   \n",
       "414  Female  Middle eastern         10.0             45      22.50   \n",
       "418  Female  White- British         10.0             30      15.00   \n",
       "416    Male  White- British         20.0             54      54.00   \n",
       "420    Male  White- British         20.0             70      70.00   \n",
       "415  Female  White- British         15.0             43      32.25   \n",
       "\n",
       "    smoking_status_merged  is.family.lung  ECOG_PS pathologyTNM  \\\n",
       "88              Ex-Smoker           False      1.0          IIA   \n",
       "79              Ex-Smoker           False      1.0           IB   \n",
       "26           Never Smoked           False      0.0         IIIA   \n",
       "23              Ex-Smoker            True      0.0          IIB   \n",
       "7               Ex-Smoker           False      1.0         IIIA   \n",
       "..                    ...             ...      ...          ...   \n",
       "414                Smoker           False      1.0           IA   \n",
       "418             Ex-Smoker            True      0.0          IIA   \n",
       "416             Ex-Smoker           False      1.0         IIIA   \n",
       "420                Smoker           False      1.0          IIA   \n",
       "415                Smoker           False      1.0           IA   \n",
       "\n",
       "    pT_stage_per_patient  pN_stage_per_patient LVI_per_patient PL_per_patient  \\\n",
       "88                    2a                     1              No            Yes   \n",
       "79                    2a                     0             Yes             No   \n",
       "26                    2b                     2             Yes             No   \n",
       "23                     3                     0              No            Yes   \n",
       "7                     2a                     2             Yes            Yes   \n",
       "..                   ...                   ...             ...            ...   \n",
       "414                   1b                     0              No             No   \n",
       "418                   1b                     1              No             No   \n",
       "416                   1b                     2              No             No   \n",
       "420                   2b                     0             Yes            Yes   \n",
       "415                   1a                     0             Yes             No   \n",
       "\n",
       "    margin_status_per_patient  size_pathology_per_patient Surgery_type  \\\n",
       "88                         R0                        26.0    Lobectomy   \n",
       "79                         R0                        35.0    Lobectomy   \n",
       "26                         R0                        52.0    Lobectomy   \n",
       "23                         R0                        47.0    Lobectomy   \n",
       "7                          R0                        43.0    Lobectomy   \n",
       "..                        ...                         ...          ...   \n",
       "414                        R0                        25.0    Lobectomy   \n",
       "418                        R0                        27.0    Lobectomy   \n",
       "416                        R0                        25.0    Lobectomy   \n",
       "420                        R0                        62.0    Lobectomy   \n",
       "415                        R0                        17.0    Lobectomy   \n",
       "\n",
       "           histology_lesion1 histology_lesion1_merged  lesion1_sampled  \\\n",
       "88   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "79   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "26   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "23   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "7    Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "..                       ...                      ...              ...   \n",
       "414  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "418  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "416  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "420  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "415  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "\n",
       "             histology_lesion2 lesion2_sampled histology_multi_full  \\\n",
       "88                         NaN             NaN                 LUAD   \n",
       "79                         NaN             NaN                 LUAD   \n",
       "26                         NaN             NaN                 LUAD   \n",
       "23                         NaN             NaN                 LUAD   \n",
       "7                          NaN             NaN                 LUAD   \n",
       "..                         ...             ...                  ...   \n",
       "414                        NaN             NaN                 LUSC   \n",
       "418                        NaN             NaN                 LUAD   \n",
       "416                        NaN             NaN                 LUSC   \n",
       "420  Lesion1: collision tumour           False               LUADx2   \n",
       "415                        NaN             NaN                 LUAD   \n",
       "\n",
       "    histology_multi_full_genomically.confirmed LUAD_pred_subtype  \\\n",
       "88                                        LUAD            acinar   \n",
       "79                                        LUAD            acinar   \n",
       "26                                        LUAD            acinar   \n",
       "23                                        LUAD            acinar   \n",
       "7                                         LUAD            acinar   \n",
       "..                                         ...               ...   \n",
       "414                                       LUSC               NaN   \n",
       "418                                       LUAD             solid   \n",
       "416                                       LUSC               NaN   \n",
       "420                                     LUADx2            acinar   \n",
       "415                                       LUAD            acinar   \n",
       "\n",
       "    adjuvant_treatment_YN adjuvant_treatment_given num_cycle_na.added  \\\n",
       "88            No adjuvant                      NaN                  0   \n",
       "79            No adjuvant                      NaN                  0   \n",
       "26               Adjuvant           Platinum chemo                  4   \n",
       "23               Adjuvant           Platinum chemo                  4   \n",
       "7             No adjuvant                      NaN                  0   \n",
       "..                    ...                      ...                ...   \n",
       "414           No adjuvant                      NaN                  0   \n",
       "418              Adjuvant           Platinum chemo                  4   \n",
       "416           No adjuvant                      NaN                  0   \n",
       "420           No adjuvant                      NaN                  0   \n",
       "415           No adjuvant                      NaN                  0   \n",
       "\n",
       "                               CHMPlatDgName_cleaned  \\\n",
       "88                                               NaN   \n",
       "79                                               NaN   \n",
       "26   Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "23           Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "7                                                NaN   \n",
       "..                                               ...   \n",
       "414                                              NaN   \n",
       "418  Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "416                                              NaN   \n",
       "420                                              NaN   \n",
       "415                                              NaN   \n",
       "\n",
       "                                CHMOthDgName_cleaned  AdjRadStartTime_manual  \\\n",
       "88                                               NaN                     NaN   \n",
       "79                                               NaN                     NaN   \n",
       "26   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "23   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "7                                                NaN                     NaN   \n",
       "..                                               ...                     ...   \n",
       "414                                              NaN                     NaN   \n",
       "418  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "416                                              NaN                     NaN   \n",
       "420                                              NaN                     NaN   \n",
       "415                                              NaN                     NaN   \n",
       "\n",
       "     AdjRadEndTime_manual  Recurrence_time_use newPrim_time_use  \\\n",
       "88                    NaN                  NaN              NaN   \n",
       "79                    NaN                  NaN              NaN   \n",
       "26                    NaN                217.0              NaN   \n",
       "23                    NaN                325.0              NaN   \n",
       "7                     NaN                  NaN              NaN   \n",
       "..                    ...                  ...              ...   \n",
       "414                   NaN                  NaN              NaN   \n",
       "418                   NaN                  NaN              NaN   \n",
       "416                   NaN                883.0              NaN   \n",
       "420                   NaN                748.0              NaN   \n",
       "415                   NaN                  NaN              NaN   \n",
       "\n",
       "    first_dfs_any_event_rec.or.new.primary first_event_during_followup  \\\n",
       "88                                     NaN                       Death   \n",
       "79                                     NaN                         NaN   \n",
       "26                              recurrence                  recurrence   \n",
       "23                              recurrence                  recurrence   \n",
       "7                                      NaN                       Death   \n",
       "..                                     ...                         ...   \n",
       "414                                    NaN                         NaN   \n",
       "418                                    NaN                         NaN   \n",
       "416                             recurrence                  recurrence   \n",
       "420                             recurrence                  recurrence   \n",
       "415                                    NaN                         NaN   \n",
       "\n",
       "     cens_os  os_time  cens_dfs  dfs_time  cens_dfs_any_event  \\\n",
       "88         1     1120         1      1120                   1   \n",
       "79         0     1901         0      1901                   0   \n",
       "26         0     2304         1       217                   1   \n",
       "23         1     1323         1       325                   1   \n",
       "7          1      649         1       649                   1   \n",
       "..       ...      ...       ...       ...                 ...   \n",
       "414        0     1084         0      1084                   0   \n",
       "418        0     1434         0      1434                   0   \n",
       "416        1      893         1       883                   1   \n",
       "420        1      758         1       748                   1   \n",
       "415        0     1001         0      1001                   0   \n",
       "\n",
       "     dfs_time_any_event  cens_lung_event  lung_event_time    Relapse_cat  \\\n",
       "88                 1120                0             1120            NaN   \n",
       "79                 1901                0             1901            NaN   \n",
       "26                  217                1              217  Intra & Extra   \n",
       "23                  325                1              325  Intra & Extra   \n",
       "7                   649                0              649            NaN   \n",
       "..                  ...              ...              ...            ...   \n",
       "414                1084                0             1084            NaN   \n",
       "418                1434                0             1434            NaN   \n",
       "416                 883                1              883  Intra & Extra   \n",
       "420                 748                1              748  Intrathoracic   \n",
       "415                1001                0             1001            NaN   \n",
       "\n",
       "    Relapse_cat_new  tx100  \n",
       "88           No rec   True  \n",
       "79           No rec   True  \n",
       "26    Intra & Extra   True  \n",
       "23    Intra & Extra   True  \n",
       "7            No rec   True  \n",
       "..              ...    ...  \n",
       "414          No rec  False  \n",
       "418          No rec  False  \n",
       "416   Intra & Extra  False  \n",
       "420   Intrathoracic  False  \n",
       "415          No rec  False  \n",
       "\n",
       "[421 rows x 50 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/figurecode/data/20221109_TRACERx421_all_patient_df.rds')\n",
    "clinical_df = pd.DataFrame(clinical_df[None])\n",
    "clinical_df = clinical_df.sort_values(by=['cruk_id'])\n",
    "clinical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>clinical_sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Histology_per_tumour_id_muttable</th>\n",
       "      <th>histology_3</th>\n",
       "      <th>LUAD_pred_subtype_with.IMA_per_tumour</th>\n",
       "      <th>site_per_lesion</th>\n",
       "      <th>pT_stage_per_lesion</th>\n",
       "      <th>pN_stage_per_lesion</th>\n",
       "      <th>size_pathology_per_lesion</th>\n",
       "      <th>vascular_invasion_per_lesion</th>\n",
       "      <th>pleural_invasion_per_lesion</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>pack_years</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>pathologyTNM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>Male</td>\n",
       "      <td>84</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>30</td>\n",
       "      <td>30.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>Male</td>\n",
       "      <td>81</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Left Lower Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>5.0</td>\n",
       "      <td>35</td>\n",
       "      <td>8.75</td>\n",
       "      <td>False</td>\n",
       "      <td>IB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>Female</td>\n",
       "      <td>78</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>52</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>Female</td>\n",
       "      <td>69</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Left Upper Lobe</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>10.0</td>\n",
       "      <td>27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>True</td>\n",
       "      <td>IIB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>Female</td>\n",
       "      <td>74</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Lower Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>43</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>56</td>\n",
       "      <td>56.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>Female</td>\n",
       "      <td>78</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Right Lower Lobe</td>\n",
       "      <td>1b</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>10.0</td>\n",
       "      <td>30</td>\n",
       "      <td>15.00</td>\n",
       "      <td>True</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>Male</td>\n",
       "      <td>66</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>1b</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>54</td>\n",
       "      <td>54.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>Male</td>\n",
       "      <td>84</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>CRUK0881_Tumour2</td>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>Male</td>\n",
       "      <td>84</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>Female</td>\n",
       "      <td>59</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Left Upper Lobe</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>15.0</td>\n",
       "      <td>43</td>\n",
       "      <td>32.25</td>\n",
       "      <td>False</td>\n",
       "      <td>IA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>432 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            tumour_id patient_id clinical_sex  age       ethnicity  \\\n",
       "89           CRUK0001   CRUK0001         Male   84  White- British   \n",
       "80           CRUK0002   CRUK0002         Male   81  White- British   \n",
       "26           CRUK0003   CRUK0003       Female   78  White- British   \n",
       "23           CRUK0004   CRUK0004       Female   69  White- British   \n",
       "7            CRUK0005   CRUK0005       Female   74  White- British   \n",
       "..                ...        ...          ...  ...             ...   \n",
       "428          CRUK0864   CRUK0864       Female   78  White- British   \n",
       "426          CRUK0872   CRUK0872         Male   66  White- British   \n",
       "430  CRUK0881_Tumour1   CRUK0881         Male   84  White- British   \n",
       "431  CRUK0881_Tumour2   CRUK0881         Male   84  White- British   \n",
       "425          CRUK0893   CRUK0893       Female   59  White- British   \n",
       "\n",
       "    Histology_per_tumour_id_muttable histology_3  \\\n",
       "89           Invasive adenocarcinoma        LUAD   \n",
       "80           Invasive adenocarcinoma        LUAD   \n",
       "26           Invasive adenocarcinoma        LUAD   \n",
       "23           Invasive adenocarcinoma        LUAD   \n",
       "7            Invasive adenocarcinoma        LUAD   \n",
       "..                               ...         ...   \n",
       "428          Invasive adenocarcinoma        LUAD   \n",
       "426          Squamous cell carcinoma        LUSC   \n",
       "430          Invasive adenocarcinoma        LUAD   \n",
       "431          Invasive adenocarcinoma        LUAD   \n",
       "425          Invasive adenocarcinoma        LUAD   \n",
       "\n",
       "    LUAD_pred_subtype_with.IMA_per_tumour   site_per_lesion  \\\n",
       "89                                 acinar  Right Upper Lobe   \n",
       "80                                 acinar   Left Lower Lobe   \n",
       "26                                 acinar  Right Upper Lobe   \n",
       "23                                 acinar   Left Upper Lobe   \n",
       "7                                  acinar  Right Lower Lobe   \n",
       "..                                    ...               ...   \n",
       "428                                 solid  Right Lower Lobe   \n",
       "426                                   NaN  Right Upper Lobe   \n",
       "430                                acinar  Right Upper Lobe   \n",
       "431                                acinar  Right Upper Lobe   \n",
       "425                                acinar   Left Upper Lobe   \n",
       "\n",
       "    pT_stage_per_lesion pN_stage_per_lesion size_pathology_per_lesion  \\\n",
       "89                   2a                   1                        26   \n",
       "80                   2a                   0                        35   \n",
       "26                   2b                   2                        52   \n",
       "23                    3                   0                        47   \n",
       "7                    2a                   2                        43   \n",
       "..                  ...                 ...                       ...   \n",
       "428                  1b                   1                        27   \n",
       "426                  1b                   2                        25   \n",
       "430                 NaN                   0                       NaN   \n",
       "431                 NaN                   0                       NaN   \n",
       "425                  1a                   0                        17   \n",
       "\n",
       "    vascular_invasion_per_lesion pleural_invasion_per_lesion  \\\n",
       "89                            No                         Yes   \n",
       "80                           Yes                          No   \n",
       "26                           Yes                          No   \n",
       "23                            No                         Yes   \n",
       "7                            Yes                         Yes   \n",
       "..                           ...                         ...   \n",
       "428                           No                          No   \n",
       "426                           No                          No   \n",
       "430                          Yes                         NaN   \n",
       "431                           No                         NaN   \n",
       "425                          Yes                          No   \n",
       "\n",
       "    smoking_status_merged  cigs_perday  years_smoking  pack_years  \\\n",
       "89              Ex-Smoker         20.0             30       30.00   \n",
       "80              Ex-Smoker          5.0             35        8.75   \n",
       "26           Never Smoked          0.0              0        0.00   \n",
       "23              Ex-Smoker         10.0             27       13.50   \n",
       "7               Ex-Smoker         20.0             56       56.00   \n",
       "..                    ...          ...            ...         ...   \n",
       "428             Ex-Smoker         10.0             30       15.00   \n",
       "426             Ex-Smoker         20.0             54       54.00   \n",
       "430                Smoker         20.0             70       70.00   \n",
       "431                Smoker         20.0             70       70.00   \n",
       "425                Smoker         15.0             43       32.25   \n",
       "\n",
       "     is.family.lung pathologyTNM  \n",
       "89            False          IIA  \n",
       "80            False           IB  \n",
       "26            False         IIIA  \n",
       "23             True          IIB  \n",
       "7             False         IIIA  \n",
       "..              ...          ...  \n",
       "428            True          IIA  \n",
       "426           False         IIIA  \n",
       "430           False          IIA  \n",
       "431           False          IIA  \n",
       "425           False           IA  \n",
       "\n",
       "[432 rows x 20 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumor_df = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/LCCE_converted_data/20221109_TRACERx421_all_tumour_df.rds')\n",
    "tumor_df = pd.DataFrame(tumor_df[None])\n",
    "tumor_df = tumor_df.sort_values(by=['patient_id'])\n",
    "tumor_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>mutation_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>ref</th>\n",
       "      <th>var</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>func</th>\n",
       "      <th>exonic.func</th>\n",
       "      <th>NucleotideChange</th>\n",
       "      <th>AAChange</th>\n",
       "      <th>GL_VAF</th>\n",
       "      <th>GL_nAlt</th>\n",
       "      <th>GL_depth</th>\n",
       "      <th>ITHState</th>\n",
       "      <th>PyCloneClonal_SC</th>\n",
       "      <th>PyCloneCluster_SC</th>\n",
       "      <th>cleanCluster_SC</th>\n",
       "      <th>combTiming_SC</th>\n",
       "      <th>DriverMut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:34291428:G</td>\n",
       "      <td>chr19</td>\n",
       "      <td>34291428</td>\n",
       "      <td>34291428</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KCTD15</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>187</td>\n",
       "      <td>1</td>\n",
       "      <td>S</td>\n",
       "      <td>3.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:9084770:C</td>\n",
       "      <td>chr19</td>\n",
       "      <td>9084770</td>\n",
       "      <td>9084770</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>MUC16</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G7045T</td>\n",
       "      <td>p.V2349L</td>\n",
       "      <td>0.39</td>\n",
       "      <td>2</td>\n",
       "      <td>510</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:1:2160418:G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2160418</td>\n",
       "      <td>2160418</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SKI</td>\n",
       "      <td>exonic</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>c.G213T</td>\n",
       "      <td>p.P71P</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:40095981:G</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40095981</td>\n",
       "      <td>40095981</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>LGALS13</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G256T</td>\n",
       "      <td>p.G86C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>659</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:41313703:C</td>\n",
       "      <td>chr19</td>\n",
       "      <td>41313703</td>\n",
       "      <td>41313703</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>RAB4B-EGLN2</td>\n",
       "      <td>ncRNA_intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>894</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>5.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379003</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:10:120513946:C</td>\n",
       "      <td>chr10</td>\n",
       "      <td>120513946</td>\n",
       "      <td>120513946</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CACUL1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G329A</td>\n",
       "      <td>p.S110N</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>659</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>7.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379004</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:14:99872776:T</td>\n",
       "      <td>chr14</td>\n",
       "      <td>99872776</td>\n",
       "      <td>99872776</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SETD3</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>539</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>7.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379005</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:8:110131566:C</td>\n",
       "      <td>chr8</td>\n",
       "      <td>110131566</td>\n",
       "      <td>110131566</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TRHR</td>\n",
       "      <td>exonic</td>\n",
       "      <td>stopgain SNV</td>\n",
       "      <td>c.C1079A</td>\n",
       "      <td>p.S360X</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>712</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>7.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379006</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:1:51032853:G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>51032853</td>\n",
       "      <td>51032853</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>FAF1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.C1164G</td>\n",
       "      <td>p.N388K</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>398</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>9.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379007</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:20:24524120:G</td>\n",
       "      <td>chr20</td>\n",
       "      <td>24524120</td>\n",
       "      <td>24524120</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SYNDIG1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>c.G387A</td>\n",
       "      <td>p.P129P</td>\n",
       "      <td>0.13</td>\n",
       "      <td>1</td>\n",
       "      <td>785</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>379008 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id tumour_id              mutation_id    chr      start  \\\n",
       "0        CRUK0005  CRUK0005   CRUK0005:19:34291428:G  chr19   34291428   \n",
       "1        CRUK0005  CRUK0005    CRUK0005:19:9084770:C  chr19    9084770   \n",
       "2        CRUK0005  CRUK0005     CRUK0005:1:2160418:G   chr1    2160418   \n",
       "3        CRUK0005  CRUK0005   CRUK0005:19:40095981:G  chr19   40095981   \n",
       "4        CRUK0005  CRUK0005   CRUK0005:19:41313703:C  chr19   41313703   \n",
       "...           ...       ...                      ...    ...        ...   \n",
       "379003   CRUK0872  CRUK0872  CRUK0872:10:120513946:C  chr10  120513946   \n",
       "379004   CRUK0872  CRUK0872   CRUK0872:14:99872776:T  chr14   99872776   \n",
       "379005   CRUK0872  CRUK0872   CRUK0872:8:110131566:C   chr8  110131566   \n",
       "379006   CRUK0872  CRUK0872    CRUK0872:1:51032853:G   chr1   51032853   \n",
       "379007   CRUK0872  CRUK0872   CRUK0872:20:24524120:G  chr20   24524120   \n",
       "\n",
       "             stop ref var  Hugo_Symbol            func        exonic.func  \\\n",
       "0        34291428   G   T       KCTD15        intronic                NaN   \n",
       "1         9084770   C   A        MUC16          exonic  nonsynonymous SNV   \n",
       "2         2160418   G   T          SKI          exonic     synonymous SNV   \n",
       "3        40095981   G   T      LGALS13          exonic  nonsynonymous SNV   \n",
       "4        41313703   C   G  RAB4B-EGLN2  ncRNA_intronic                NaN   \n",
       "...           ...  ..  ..          ...             ...                ...   \n",
       "379003  120513946   C   T       CACUL1          exonic  nonsynonymous SNV   \n",
       "379004   99872776   T   C        SETD3        intronic                NaN   \n",
       "379005  110131566   C   A         TRHR          exonic       stopgain SNV   \n",
       "379006   51032853   G   C         FAF1          exonic  nonsynonymous SNV   \n",
       "379007   24524120   G   A      SYNDIG1          exonic     synonymous SNV   \n",
       "\n",
       "       NucleotideChange  AAChange GL_VAF GL_nAlt GL_depth ITHState  \\\n",
       "0                   NaN       NaN      0       0      187        1   \n",
       "1              c.G7045T  p.V2349L   0.39       2      510        1   \n",
       "2               c.G213T    p.P71P      0       0      114        1   \n",
       "3               c.G256T    p.G86C      0       0      659        1   \n",
       "4                   NaN       NaN      0       0      894        3   \n",
       "...                 ...       ...    ...     ...      ...      ...   \n",
       "379003          c.G329A   p.S110N      0       0      659        3   \n",
       "379004              NaN       NaN      0       0      539        3   \n",
       "379005         c.C1079A   p.S360X      0       0      712        3   \n",
       "379006         c.C1164G   p.N388K      0       0      398        3   \n",
       "379007          c.G387A   p.P129P   0.13       1      785        3   \n",
       "\n",
       "       PyCloneClonal_SC  PyCloneCluster_SC cleanCluster_SC combTiming_SC  \\\n",
       "0                     S                3.0            True     subclonal   \n",
       "1                     C                2.0            True         early   \n",
       "2                     C                2.0            True         early   \n",
       "3                     C                2.0            True         early   \n",
       "4                     S                5.0            True     subclonal   \n",
       "...                 ...                ...             ...           ...   \n",
       "379003                S                7.0            True     subclonal   \n",
       "379004                S                7.0            True     subclonal   \n",
       "379005                S                7.0            True     subclonal   \n",
       "379006                S                9.0            True     subclonal   \n",
       "379007              NaN                NaN             NaN            NA   \n",
       "\n",
       "        DriverMut  \n",
       "0           False  \n",
       "1           False  \n",
       "2           False  \n",
       "3           False  \n",
       "4           False  \n",
       "...           ...  \n",
       "379003      False  \n",
       "379004      False  \n",
       "379005      False  \n",
       "379006      False  \n",
       "379007      False  \n",
       "\n",
       "[379008 rows x 22 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_table = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/LCCE_converted_data/20221109_TRACERx421_mutation_table.rds')\n",
    "mut_table = pd.DataFrame(mut_table[None])\n",
    "mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>RegionID</th>\n",
       "      <th>chr</th>\n",
       "      <th>startpos</th>\n",
       "      <th>endpos</th>\n",
       "      <th>n.het</th>\n",
       "      <th>cnTotal</th>\n",
       "      <th>nMajor</th>\n",
       "      <th>nMinor</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>ACF</th>\n",
       "      <th>nAraw</th>\n",
       "      <th>nBraw</th>\n",
       "      <th>cpn_event_vs_ploidy</th>\n",
       "      <th>MSAI</th>\n",
       "      <th>samp_used_to_phase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>2252860</td>\n",
       "      <td>73</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>2.463756</td>\n",
       "      <td>1.184761</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>2252970</td>\n",
       "      <td>2527412</td>\n",
       "      <td>26</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>1.645818</td>\n",
       "      <td>0.284001</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>2527437</td>\n",
       "      <td>2700215</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>2.187291</td>\n",
       "      <td>0.956066</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>2700332</td>\n",
       "      <td>28531815</td>\n",
       "      <td>600</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>1.725844</td>\n",
       "      <td>0.708290</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>28534782</td>\n",
       "      <td>29649999</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>1.711109</td>\n",
       "      <td>0.695610</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419823</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>36962306</td>\n",
       "      <td>39176976</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>4.207645</td>\n",
       "      <td>0.208286</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419824</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>39262577</td>\n",
       "      <td>41604353</td>\n",
       "      <td>41</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>2.385038</td>\n",
       "      <td>0.092194</td>\n",
       "      <td>loss</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419825</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>41604399</td>\n",
       "      <td>41612295</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>2.331273</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>loss</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419826</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>41648860</td>\n",
       "      <td>42899151</td>\n",
       "      <td>37</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>3.403717</td>\n",
       "      <td>0.186030</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419827</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>42912377</td>\n",
       "      <td>51219216</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>1.608646</td>\n",
       "      <td>0.238268</td>\n",
       "      <td>loss</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>419828 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id         tumour_id     RegionID  chr  startpos    endpos  \\\n",
       "0        CRUK0005          CRUK0005  CRUK0005:R1    1    762273   2252860   \n",
       "1        CRUK0005          CRUK0005  CRUK0005:R1    1   2252970   2527412   \n",
       "2        CRUK0005          CRUK0005  CRUK0005:R1    1   2527437   2700215   \n",
       "3        CRUK0005          CRUK0005  CRUK0005:R1    1   2700332  28531815   \n",
       "4        CRUK0005          CRUK0005  CRUK0005:R1    1  28534782  29649999   \n",
       "...           ...               ...          ...  ...       ...       ...   \n",
       "419823   CRUK0555  CRUK0555_Tumour2          :R2   22  36962306  39176976   \n",
       "419824   CRUK0555  CRUK0555_Tumour2          :R2   22  39262577  41604353   \n",
       "419825   CRUK0555  CRUK0555_Tumour2          :R2   22  41604399  41612295   \n",
       "419826   CRUK0555  CRUK0555_Tumour2          :R2   22  41648860  42899151   \n",
       "419827   CRUK0555  CRUK0555_Tumour2          :R2   22  42912377  51219216   \n",
       "\n",
       "        n.het  cnTotal  nMajor  nMinor    Ploidy   ACF     nAraw     nBraw  \\\n",
       "0          73        3       2       1  3.131282  0.15  2.463756  1.184761   \n",
       "1          26        2       2       0  3.131282  0.15  1.645818  0.284001   \n",
       "2          12        3       2       1  3.131282  0.15  2.187291  0.956066   \n",
       "3         600        3       2       1  3.131282  0.15  1.725844  0.708290   \n",
       "4          11        3       2       1  3.131282  0.15  1.711109  0.695610   \n",
       "...       ...      ...     ...     ...       ...   ...       ...       ...   \n",
       "419823     55        4       4       0  3.796640  0.19  4.207645  0.208286   \n",
       "419824     41        2       2       0  3.796640  0.19  2.385038  0.092194   \n",
       "419825      6        2       2       0  3.796640  0.19  2.331273  0.000000   \n",
       "419826     37        3       3       0  3.796640  0.19  3.403717  0.186030   \n",
       "419827    226        2       2       0  3.796640  0.19  1.608646  0.238268   \n",
       "\n",
       "       cpn_event_vs_ploidy   MSAI samp_used_to_phase  \n",
       "0                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "1                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "2                  neutral  False  CRUK0005_SU_T1-R3  \n",
       "3                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "4                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "...                    ...    ...                ...  \n",
       "419823             neutral  False  CRUK0555_SU_T1-R1  \n",
       "419824                loss  False  CRUK0555_SU_T1-R2  \n",
       "419825                loss  False  CRUK0555_SU_T1-R1  \n",
       "419826             neutral  False  CRUK0555_SU_T1-R1  \n",
       "419827                loss  False  CRUK0555_SU_T1-R1  \n",
       "\n",
       "[419828 rows x 17 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read table into memory\n",
    "cna_table = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/LCCE_converted_data/20221109_TRACERx421_scna_table.rds')\n",
    "# Obtan dataframe and rename column sample to RegionID\n",
    "cna_table = pd.DataFrame(cna_table[None])\n",
    "cna_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>RegionID</th>\n",
       "      <th>is_lymphnode</th>\n",
       "      <th>mutation_id</th>\n",
       "      <th>var_count</th>\n",
       "      <th>depth</th>\n",
       "      <th>Is.present.region</th>\n",
       "      <th>CCF_pyclone_SC</th>\n",
       "      <th>CCF_phylo_SC</th>\n",
       "      <th>mut_cpn_SC</th>\n",
       "      <th>MajorCPN_SC</th>\n",
       "      <th>MinorCPN_SC</th>\n",
       "      <th>PyCloneCluster_SC</th>\n",
       "      <th>PyCloneClonal_SC</th>\n",
       "      <th>cleanCluster_SC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005:19:34291428:G</td>\n",
       "      <td>5.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.54</td>\n",
       "      <td>0.54</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>S</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005:19:9084770:C</td>\n",
       "      <td>149.0</td>\n",
       "      <td>498.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.16</td>\n",
       "      <td>1.16</td>\n",
       "      <td>4.57</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005:1:2160418:G</td>\n",
       "      <td>27.0</td>\n",
       "      <td>171.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.96</td>\n",
       "      <td>2.37</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005:19:40095981:G</td>\n",
       "      <td>47.0</td>\n",
       "      <td>507.0</td>\n",
       "      <td>True</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.25</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>C</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005:19:41313703:C</td>\n",
       "      <td>0.0</td>\n",
       "      <td>616.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>S</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1403373</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:R4</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0872:10:120513946:C</td>\n",
       "      <td>13.0</td>\n",
       "      <td>469.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.43</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>S</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1403374</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:R4</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0872:14:99872776:T</td>\n",
       "      <td>13.0</td>\n",
       "      <td>510.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.46</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7.0</td>\n",
       "      <td>S</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1403375</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:R4</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0872:8:110131566:C</td>\n",
       "      <td>16.0</td>\n",
       "      <td>650.0</td>\n",
       "      <td>True</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.43</td>\n",
       "      <td>0.43</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7.0</td>\n",
       "      <td>S</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1403376</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:R4</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0872:1:51032853:G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>428.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>9.0</td>\n",
       "      <td>S</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1403377</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:R4</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0872:20:24524120:G</td>\n",
       "      <td>0.0</td>\n",
       "      <td>621.0</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1403378 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        tumour_id     RegionID  is_lymphnode              mutation_id  \\\n",
       "0        CRUK0005  CRUK0005:R1         False   CRUK0005:19:34291428:G   \n",
       "1        CRUK0005  CRUK0005:R1         False    CRUK0005:19:9084770:C   \n",
       "2        CRUK0005  CRUK0005:R1         False     CRUK0005:1:2160418:G   \n",
       "3        CRUK0005  CRUK0005:R1         False   CRUK0005:19:40095981:G   \n",
       "4        CRUK0005  CRUK0005:R1         False   CRUK0005:19:41313703:C   \n",
       "...           ...          ...           ...                      ...   \n",
       "1403373  CRUK0872  CRUK0872:R4         False  CRUK0872:10:120513946:C   \n",
       "1403374  CRUK0872  CRUK0872:R4         False   CRUK0872:14:99872776:T   \n",
       "1403375  CRUK0872  CRUK0872:R4         False   CRUK0872:8:110131566:C   \n",
       "1403376  CRUK0872  CRUK0872:R4         False    CRUK0872:1:51032853:G   \n",
       "1403377  CRUK0872  CRUK0872:R4         False   CRUK0872:20:24524120:G   \n",
       "\n",
       "         var_count  depth  Is.present.region  CCF_pyclone_SC  CCF_phylo_SC  \\\n",
       "0              5.0  159.0               True            0.54          0.54   \n",
       "1            149.0  498.0               True            1.16          1.16   \n",
       "2             27.0  171.0               True            0.96          0.96   \n",
       "3             47.0  507.0               True            1.21          1.21   \n",
       "4              0.0  616.0              False            0.00          0.00   \n",
       "...            ...    ...                ...             ...           ...   \n",
       "1403373       13.0  469.0               True            0.43          0.43   \n",
       "1403374       13.0  510.0               True            0.46          0.46   \n",
       "1403375       16.0  650.0               True            0.43          0.43   \n",
       "1403376        0.0  428.0              False            0.00          0.00   \n",
       "1403377        0.0  621.0              False             NaN           NaN   \n",
       "\n",
       "         mut_cpn_SC MajorCPN_SC MinorCPN_SC  PyCloneCluster_SC  \\\n",
       "0              0.54           4           2                3.0   \n",
       "1              4.57           4           0                2.0   \n",
       "2              2.37           2           1                2.0   \n",
       "3              1.25           2           0                2.0   \n",
       "4              0.00           2           0                5.0   \n",
       "...             ...         ...         ...                ...   \n",
       "1403373        0.43           2           0                7.0   \n",
       "1403374        0.46           2           2                7.0   \n",
       "1403375        0.43           2           2                7.0   \n",
       "1403376        0.00           2           2                9.0   \n",
       "1403377         NaN         NaN         NaN                NaN   \n",
       "\n",
       "        PyCloneClonal_SC cleanCluster_SC  \n",
       "0                      S            True  \n",
       "1                      C            True  \n",
       "2                      C            True  \n",
       "3                      C            True  \n",
       "4                      S            True  \n",
       "...                  ...             ...  \n",
       "1403373                S            True  \n",
       "1403374                S            True  \n",
       "1403375                S            True  \n",
       "1403376                S            True  \n",
       "1403377              NaN             NaN  \n",
       "\n",
       "[1403378 rows x 15 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read table into memory\n",
    "mut_table_region = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/LCCE_converted_data/20221123_TRACERx421_mutation_table_region.rds')\n",
    "\n",
    "mut_table_region = pd.DataFrame(mut_table_region[None])\n",
    "mut_table_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/100_gene_matrix.csv')\n",
    "driver_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/100_driver_matrix.csv')\n",
    "fifty_gene_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/50_gene_matrix.csv')\n",
    "fifty_driver_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/50_driver_matrix.csv')\n",
    "twenty_gene_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/20_gene_matrix.csv')\n",
    "twenty_driver_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/20_driver_matrix.csv')\n",
    "ten_driver_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/10_driver_matrix.csv')\n",
    "\n",
    "gene_matrix = gene_matrix.set_index(gene_matrix.columns[0])\n",
    "driver_matrix = driver_matrix.set_index(driver_matrix.columns[0])\n",
    "fifty_gene_matrix = fifty_gene_matrix.set_index(fifty_gene_matrix.columns[0])\n",
    "fifty_driver_matrix = fifty_driver_matrix.set_index(fifty_driver_matrix.columns[0])\n",
    "twenty_gene_matrix = twenty_gene_matrix.set_index(twenty_gene_matrix.columns[0])\n",
    "twenty_driver_matrix = twenty_driver_matrix.set_index(twenty_driver_matrix.columns[0])\n",
    "ten_driver_matrix = ten_driver_matrix.set_index(ten_driver_matrix.columns[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# separate clinical_df into two dataframes, one for patients where first_dfs_any_event_rec.or.new.primary is 'recurrence' and one for patients where first_dfs_any_event_rec.or.new.primary is not 'recurrence'\n",
    "recurrence_clinical_df = clinical_df[clinical_df['first_dfs_any_event_rec.or.new.primary'] == 'recurrence']\n",
    "no_recurrence_clinical_df = clinical_df[clinical_df['first_dfs_any_event_rec.or.new.primary'] != 'recurrence']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cruk_id</th>\n",
       "      <th>tumour_id_muttable_cruk</th>\n",
       "      <th>tumour_id_per_patient</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>packyears</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>ECOG_PS</th>\n",
       "      <th>pathologyTNM</th>\n",
       "      <th>pT_stage_per_patient</th>\n",
       "      <th>pN_stage_per_patient</th>\n",
       "      <th>LVI_per_patient</th>\n",
       "      <th>PL_per_patient</th>\n",
       "      <th>margin_status_per_patient</th>\n",
       "      <th>size_pathology_per_patient</th>\n",
       "      <th>Surgery_type</th>\n",
       "      <th>histology_lesion1</th>\n",
       "      <th>histology_lesion1_merged</th>\n",
       "      <th>lesion1_sampled</th>\n",
       "      <th>histology_lesion2</th>\n",
       "      <th>lesion2_sampled</th>\n",
       "      <th>histology_multi_full</th>\n",
       "      <th>histology_multi_full_genomically.confirmed</th>\n",
       "      <th>LUAD_pred_subtype</th>\n",
       "      <th>adjuvant_treatment_YN</th>\n",
       "      <th>adjuvant_treatment_given</th>\n",
       "      <th>num_cycle_na.added</th>\n",
       "      <th>CHMPlatDgName_cleaned</th>\n",
       "      <th>CHMOthDgName_cleaned</th>\n",
       "      <th>AdjRadStartTime_manual</th>\n",
       "      <th>AdjRadEndTime_manual</th>\n",
       "      <th>Recurrence_time_use</th>\n",
       "      <th>newPrim_time_use</th>\n",
       "      <th>first_dfs_any_event_rec.or.new.primary</th>\n",
       "      <th>first_event_during_followup</th>\n",
       "      <th>cens_os</th>\n",
       "      <th>os_time</th>\n",
       "      <th>cens_dfs</th>\n",
       "      <th>dfs_time</th>\n",
       "      <th>cens_dfs_any_event</th>\n",
       "      <th>dfs_time_any_event</th>\n",
       "      <th>cens_lung_event</th>\n",
       "      <th>lung_event_time</th>\n",
       "      <th>Relapse_cat</th>\n",
       "      <th>Relapse_cat_new</th>\n",
       "      <th>tx100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>217.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>2304</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>69</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>325.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>1323</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>CRUK0010</td>\n",
       "      <td>CRUK0010</td>\n",
       "      <td>CRUK0010</td>\n",
       "      <td>68</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>papillary</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1750.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>2261</td>\n",
       "      <td>1</td>\n",
       "      <td>1750</td>\n",
       "      <td>1</td>\n",
       "      <td>1750</td>\n",
       "      <td>1</td>\n",
       "      <td>1750</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>CRUK0013</td>\n",
       "      <td>CRUK0013</td>\n",
       "      <td>CRUK0013</td>\n",
       "      <td>67</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>25.0</td>\n",
       "      <td>19</td>\n",
       "      <td>23.75</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Segmentectomy or wedge resection</td>\n",
       "      <td>combined LUAD and LCNEC</td>\n",
       "      <td>LCNEC</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo/radiotherapy</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>135.0</td>\n",
       "      <td>162.0</td>\n",
       "      <td>913.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>928</td>\n",
       "      <td>1</td>\n",
       "      <td>913</td>\n",
       "      <td>1</td>\n",
       "      <td>913</td>\n",
       "      <td>1</td>\n",
       "      <td>913</td>\n",
       "      <td>Extrathoracic</td>\n",
       "      <td>Extrathoracic</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>CRUK0021</td>\n",
       "      <td>CRUK0021</td>\n",
       "      <td>CRUK0021</td>\n",
       "      <td>83</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R1</td>\n",
       "      <td>18.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>537.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>1409</td>\n",
       "      <td>1</td>\n",
       "      <td>537</td>\n",
       "      <td>1</td>\n",
       "      <td>537</td>\n",
       "      <td>1</td>\n",
       "      <td>537</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381</th>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>37</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6</td>\n",
       "      <td>0.60</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>papillary</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>2</td>\n",
       "      <td>Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1004.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>1441</td>\n",
       "      <td>1</td>\n",
       "      <td>1004</td>\n",
       "      <td>1</td>\n",
       "      <td>1004</td>\n",
       "      <td>1</td>\n",
       "      <td>1004</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>CRUK0810</td>\n",
       "      <td>CRUK0810</td>\n",
       "      <td>CRUK0810</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>44</td>\n",
       "      <td>22.00</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>2</td>\n",
       "      <td>Cisplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>197.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>493</td>\n",
       "      <td>1</td>\n",
       "      <td>197</td>\n",
       "      <td>1</td>\n",
       "      <td>197</td>\n",
       "      <td>1</td>\n",
       "      <td>197</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>CRUK0817</td>\n",
       "      <td>CRUK0817</td>\n",
       "      <td>CRUK0817</td>\n",
       "      <td>64</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>12.0</td>\n",
       "      <td>27</td>\n",
       "      <td>16.20</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Radiotherapy</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>453</td>\n",
       "      <td>1</td>\n",
       "      <td>203</td>\n",
       "      <td>1</td>\n",
       "      <td>203</td>\n",
       "      <td>1</td>\n",
       "      <td>203</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>66</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>54</td>\n",
       "      <td>54.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>883.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>893</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881_Tumour1;CRUK0881_Tumour2</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>Lesion1: collision tumour</td>\n",
       "      <td>False</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>758</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>142 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      cruk_id tumour_id_muttable_cruk              tumour_id_per_patient  age  \\\n",
       "26   CRUK0003                CRUK0003                           CRUK0003   78   \n",
       "23   CRUK0004                CRUK0004                           CRUK0004   69   \n",
       "78   CRUK0010                CRUK0010                           CRUK0010   68   \n",
       "110  CRUK0013                CRUK0013                           CRUK0013   67   \n",
       "40   CRUK0021                CRUK0021                           CRUK0021   83   \n",
       "..        ...                     ...                                ...  ...   \n",
       "381  CRUK0787                CRUK0787                           CRUK0787   37   \n",
       "419  CRUK0810                CRUK0810                           CRUK0810   59   \n",
       "417  CRUK0817                CRUK0817                           CRUK0817   64   \n",
       "416  CRUK0872                CRUK0872                           CRUK0872   66   \n",
       "420  CRUK0881        CRUK0881_Tumour1  CRUK0881_Tumour1;CRUK0881_Tumour2   84   \n",
       "\n",
       "        sex       ethnicity  cigs_perday  years_smoking  packyears  \\\n",
       "26   Female  White- British          0.0              0       0.00   \n",
       "23   Female  White- British         10.0             27      13.50   \n",
       "78     Male  White- British          0.0              0       0.00   \n",
       "110    Male  White- British         25.0             19      23.75   \n",
       "40   Female  White- British          0.0              0       0.00   \n",
       "..      ...             ...          ...            ...        ...   \n",
       "381  Female  White- British          2.0              6       0.60   \n",
       "419  Female  White- British         10.0             44      22.00   \n",
       "417    Male  White- British         12.0             27      16.20   \n",
       "416    Male  White- British         20.0             54      54.00   \n",
       "420    Male  White- British         20.0             70      70.00   \n",
       "\n",
       "    smoking_status_merged  is.family.lung  ECOG_PS pathologyTNM  \\\n",
       "26           Never Smoked           False      0.0         IIIA   \n",
       "23              Ex-Smoker            True      0.0          IIB   \n",
       "78           Never Smoked            True      0.0         IIIA   \n",
       "110             Ex-Smoker           False      0.0         IIIA   \n",
       "40           Never Smoked           False      0.0           IA   \n",
       "..                    ...             ...      ...          ...   \n",
       "381             Ex-Smoker           False      0.0          IIA   \n",
       "419                Smoker           False      1.0         IIIA   \n",
       "417             Ex-Smoker           False      1.0          IIB   \n",
       "416             Ex-Smoker           False      1.0         IIIA   \n",
       "420                Smoker           False      1.0          IIA   \n",
       "\n",
       "    pT_stage_per_patient  pN_stage_per_patient LVI_per_patient PL_per_patient  \\\n",
       "26                    2b                     2             Yes             No   \n",
       "23                     3                     0              No            Yes   \n",
       "78                    2a                     2              No             No   \n",
       "110                   2b                     2             Yes             No   \n",
       "40                    1a                     0              No             No   \n",
       "..                   ...                   ...             ...            ...   \n",
       "381                   2b                     0              No             No   \n",
       "419                   2a                     2             Yes            Yes   \n",
       "417                    3                     0              No            Yes   \n",
       "416                   1b                     2              No             No   \n",
       "420                   2b                     0             Yes            Yes   \n",
       "\n",
       "    margin_status_per_patient  size_pathology_per_patient  \\\n",
       "26                         R0                        52.0   \n",
       "23                         R0                        47.0   \n",
       "78                         R0                        32.0   \n",
       "110                        R0                        35.0   \n",
       "40                         R1                        18.0   \n",
       "..                        ...                         ...   \n",
       "381                        R0                        52.0   \n",
       "419                        R0                        35.0   \n",
       "417                        R0                       130.0   \n",
       "416                        R0                        25.0   \n",
       "420                        R0                        62.0   \n",
       "\n",
       "                         Surgery_type        histology_lesion1  \\\n",
       "26                          Lobectomy  Invasive adenocarcinoma   \n",
       "23                          Lobectomy  Invasive adenocarcinoma   \n",
       "78                          Lobectomy  Invasive adenocarcinoma   \n",
       "110  Segmentectomy or wedge resection  combined LUAD and LCNEC   \n",
       "40                          Lobectomy  Invasive adenocarcinoma   \n",
       "..                                ...                      ...   \n",
       "381                         Lobectomy  Invasive adenocarcinoma   \n",
       "419                         Lobectomy  Invasive adenocarcinoma   \n",
       "417                         Lobectomy  Invasive adenocarcinoma   \n",
       "416                         Lobectomy  Squamous cell carcinoma   \n",
       "420                         Lobectomy  Invasive adenocarcinoma   \n",
       "\n",
       "    histology_lesion1_merged  lesion1_sampled          histology_lesion2  \\\n",
       "26   Invasive adenocarcinoma             True                        NaN   \n",
       "23   Invasive adenocarcinoma             True                        NaN   \n",
       "78   Invasive adenocarcinoma             True                        NaN   \n",
       "110                    LCNEC             True                        NaN   \n",
       "40   Invasive adenocarcinoma             True                        NaN   \n",
       "..                       ...              ...                        ...   \n",
       "381  Invasive adenocarcinoma             True                        NaN   \n",
       "419  Invasive adenocarcinoma             True                        NaN   \n",
       "417  Invasive adenocarcinoma             True                        NaN   \n",
       "416  Squamous cell carcinoma             True                        NaN   \n",
       "420  Invasive adenocarcinoma             True  Lesion1: collision tumour   \n",
       "\n",
       "    lesion2_sampled histology_multi_full  \\\n",
       "26              NaN                 LUAD   \n",
       "23              NaN                 LUAD   \n",
       "78              NaN                 LUAD   \n",
       "110             NaN                Other   \n",
       "40              NaN                 LUAD   \n",
       "..              ...                  ...   \n",
       "381             NaN                 LUAD   \n",
       "419             NaN                 LUAD   \n",
       "417             NaN                 LUAD   \n",
       "416             NaN                 LUSC   \n",
       "420           False               LUADx2   \n",
       "\n",
       "    histology_multi_full_genomically.confirmed LUAD_pred_subtype  \\\n",
       "26                                        LUAD            acinar   \n",
       "23                                        LUAD            acinar   \n",
       "78                                        LUAD         papillary   \n",
       "110                                      Other               NaN   \n",
       "40                                        LUAD             solid   \n",
       "..                                         ...               ...   \n",
       "381                                       LUAD         papillary   \n",
       "419                                       LUAD            acinar   \n",
       "417                                       LUAD             solid   \n",
       "416                                       LUSC               NaN   \n",
       "420                                     LUADx2            acinar   \n",
       "\n",
       "    adjuvant_treatment_YN     adjuvant_treatment_given num_cycle_na.added  \\\n",
       "26               Adjuvant               Platinum chemo                  4   \n",
       "23               Adjuvant               Platinum chemo                  4   \n",
       "78               Adjuvant               Platinum chemo                  4   \n",
       "110              Adjuvant  Platinum chemo/radiotherapy                  4   \n",
       "40            No adjuvant                          NaN                  0   \n",
       "..                    ...                          ...                ...   \n",
       "381              Adjuvant               Platinum chemo                  2   \n",
       "419              Adjuvant               Platinum chemo                  2   \n",
       "417              Adjuvant                 Radiotherapy                  0   \n",
       "416           No adjuvant                          NaN                  0   \n",
       "420           No adjuvant                          NaN                  0   \n",
       "\n",
       "                               CHMPlatDgName_cleaned  \\\n",
       "26   Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "23           Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "78   Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "110  Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "40                                               NaN   \n",
       "..                                               ...   \n",
       "381                          Carboplatin;Carboplatin   \n",
       "419                            Cisplatin;Carboplatin   \n",
       "417                                              NaN   \n",
       "416                                              NaN   \n",
       "420                                              NaN   \n",
       "\n",
       "                                CHMOthDgName_cleaned  AdjRadStartTime_manual  \\\n",
       "26   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "23   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "78   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "110  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                   135.0   \n",
       "40                                               NaN                     NaN   \n",
       "..                                               ...                     ...   \n",
       "381                          Vinorelbine;Vinorelbine                     NaN   \n",
       "419                          Vinorelbine;Vinorelbine                     NaN   \n",
       "417                                              NaN                    64.0   \n",
       "416                                              NaN                     NaN   \n",
       "420                                              NaN                     NaN   \n",
       "\n",
       "     AdjRadEndTime_manual  Recurrence_time_use newPrim_time_use  \\\n",
       "26                    NaN                217.0              NaN   \n",
       "23                    NaN                325.0              NaN   \n",
       "78                    NaN               1750.0              NaN   \n",
       "110                 162.0                913.0              NaN   \n",
       "40                    NaN                537.0              NaN   \n",
       "..                    ...                  ...              ...   \n",
       "381                   NaN               1004.0              NaN   \n",
       "419                   NaN                197.0              NaN   \n",
       "417                  91.0                203.0              NaN   \n",
       "416                   NaN                883.0              NaN   \n",
       "420                   NaN                748.0              NaN   \n",
       "\n",
       "    first_dfs_any_event_rec.or.new.primary first_event_during_followup  \\\n",
       "26                              recurrence                  recurrence   \n",
       "23                              recurrence                  recurrence   \n",
       "78                              recurrence                  recurrence   \n",
       "110                             recurrence                  recurrence   \n",
       "40                              recurrence                  recurrence   \n",
       "..                                     ...                         ...   \n",
       "381                             recurrence                  recurrence   \n",
       "419                             recurrence                  recurrence   \n",
       "417                             recurrence                  recurrence   \n",
       "416                             recurrence                  recurrence   \n",
       "420                             recurrence                  recurrence   \n",
       "\n",
       "     cens_os  os_time  cens_dfs  dfs_time  cens_dfs_any_event  \\\n",
       "26         0     2304         1       217                   1   \n",
       "23         1     1323         1       325                   1   \n",
       "78         0     2261         1      1750                   1   \n",
       "110        1      928         1       913                   1   \n",
       "40         1     1409         1       537                   1   \n",
       "..       ...      ...       ...       ...                 ...   \n",
       "381        0     1441         1      1004                   1   \n",
       "419        1      493         1       197                   1   \n",
       "417        1      453         1       203                   1   \n",
       "416        1      893         1       883                   1   \n",
       "420        1      758         1       748                   1   \n",
       "\n",
       "     dfs_time_any_event  cens_lung_event  lung_event_time    Relapse_cat  \\\n",
       "26                  217                1              217  Intra & Extra   \n",
       "23                  325                1              325  Intra & Extra   \n",
       "78                 1750                1             1750  Intrathoracic   \n",
       "110                 913                1              913  Extrathoracic   \n",
       "40                  537                1              537  Intrathoracic   \n",
       "..                  ...              ...              ...            ...   \n",
       "381                1004                1             1004  Intrathoracic   \n",
       "419                 197                1              197  Intra & Extra   \n",
       "417                 203                1              203  Intrathoracic   \n",
       "416                 883                1              883  Intra & Extra   \n",
       "420                 748                1              748  Intrathoracic   \n",
       "\n",
       "    Relapse_cat_new  tx100  \n",
       "26    Intra & Extra   True  \n",
       "23    Intra & Extra   True  \n",
       "78    Intrathoracic   True  \n",
       "110   Extrathoracic   True  \n",
       "40    Intrathoracic   True  \n",
       "..              ...    ...  \n",
       "381   Intrathoracic  False  \n",
       "419   Intra & Extra  False  \n",
       "417   Intrathoracic  False  \n",
       "416   Intra & Extra  False  \n",
       "420   Intrathoracic  False  \n",
       "\n",
       "[142 rows x 50 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recurrence_clinical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cruk_id</th>\n",
       "      <th>tumour_id_muttable_cruk</th>\n",
       "      <th>tumour_id_per_patient</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>packyears</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>ECOG_PS</th>\n",
       "      <th>pathologyTNM</th>\n",
       "      <th>pT_stage_per_patient</th>\n",
       "      <th>pN_stage_per_patient</th>\n",
       "      <th>LVI_per_patient</th>\n",
       "      <th>PL_per_patient</th>\n",
       "      <th>margin_status_per_patient</th>\n",
       "      <th>size_pathology_per_patient</th>\n",
       "      <th>Surgery_type</th>\n",
       "      <th>histology_lesion1</th>\n",
       "      <th>histology_lesion1_merged</th>\n",
       "      <th>lesion1_sampled</th>\n",
       "      <th>histology_lesion2</th>\n",
       "      <th>lesion2_sampled</th>\n",
       "      <th>histology_multi_full</th>\n",
       "      <th>histology_multi_full_genomically.confirmed</th>\n",
       "      <th>LUAD_pred_subtype</th>\n",
       "      <th>adjuvant_treatment_YN</th>\n",
       "      <th>adjuvant_treatment_given</th>\n",
       "      <th>num_cycle_na.added</th>\n",
       "      <th>CHMPlatDgName_cleaned</th>\n",
       "      <th>CHMOthDgName_cleaned</th>\n",
       "      <th>AdjRadStartTime_manual</th>\n",
       "      <th>AdjRadEndTime_manual</th>\n",
       "      <th>Recurrence_time_use</th>\n",
       "      <th>newPrim_time_use</th>\n",
       "      <th>first_dfs_any_event_rec.or.new.primary</th>\n",
       "      <th>first_event_during_followup</th>\n",
       "      <th>cens_os</th>\n",
       "      <th>os_time</th>\n",
       "      <th>cens_dfs</th>\n",
       "      <th>dfs_time</th>\n",
       "      <th>cens_dfs_any_event</th>\n",
       "      <th>dfs_time_any_event</th>\n",
       "      <th>cens_lung_event</th>\n",
       "      <th>lung_event_time</th>\n",
       "      <th>Relapse_cat</th>\n",
       "      <th>Relapse_cat_new</th>\n",
       "      <th>tx100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>30</td>\n",
       "      <td>30.000</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>0</td>\n",
       "      <td>1120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>81</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>5.0</td>\n",
       "      <td>35</td>\n",
       "      <td>8.750</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>74</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>56</td>\n",
       "      <td>56.000</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>0</td>\n",
       "      <td>649</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>CRUK0006</td>\n",
       "      <td>CRUK0006</td>\n",
       "      <td>CRUK0006</td>\n",
       "      <td>66</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>17.5</td>\n",
       "      <td>50</td>\n",
       "      <td>43.750</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1811</td>\n",
       "      <td>0</td>\n",
       "      <td>1811</td>\n",
       "      <td>0</td>\n",
       "      <td>1811</td>\n",
       "      <td>0</td>\n",
       "      <td>1811</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>CRUK0007</td>\n",
       "      <td>CRUK0007</td>\n",
       "      <td>CRUK0007</td>\n",
       "      <td>69</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>4.5</td>\n",
       "      <td>3</td>\n",
       "      <td>0.675</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>lepidic</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>2244</td>\n",
       "      <td>0</td>\n",
       "      <td>2244</td>\n",
       "      <td>0</td>\n",
       "      <td>2244</td>\n",
       "      <td>0</td>\n",
       "      <td>2244</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373</th>\n",
       "      <td>CRUK0795</td>\n",
       "      <td>CRUK0795</td>\n",
       "      <td>CRUK0795</td>\n",
       "      <td>72</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>50</td>\n",
       "      <td>50.000</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1b</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1182</td>\n",
       "      <td>new_primary</td>\n",
       "      <td>new_primary</td>\n",
       "      <td>0</td>\n",
       "      <td>1232</td>\n",
       "      <td>0</td>\n",
       "      <td>1232</td>\n",
       "      <td>1</td>\n",
       "      <td>1182</td>\n",
       "      <td>0</td>\n",
       "      <td>1232</td>\n",
       "      <td>Second primary lung</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>CRUK0799</td>\n",
       "      <td>CRUK0799</td>\n",
       "      <td>CRUK0799</td>\n",
       "      <td>85</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.000</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>1016</td>\n",
       "      <td>1</td>\n",
       "      <td>1016</td>\n",
       "      <td>1</td>\n",
       "      <td>1016</td>\n",
       "      <td>0</td>\n",
       "      <td>1016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>65</td>\n",
       "      <td>Female</td>\n",
       "      <td>Middle eastern</td>\n",
       "      <td>10.0</td>\n",
       "      <td>45</td>\n",
       "      <td>22.500</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1b</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>30</td>\n",
       "      <td>15.000</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>15.0</td>\n",
       "      <td>43</td>\n",
       "      <td>32.250</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>279 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      cruk_id tumour_id_muttable_cruk tumour_id_per_patient  age     sex  \\\n",
       "88   CRUK0001                CRUK0001              CRUK0001   84    Male   \n",
       "79   CRUK0002                CRUK0002              CRUK0002   81    Male   \n",
       "7    CRUK0005                CRUK0005              CRUK0005   74  Female   \n",
       "99   CRUK0006                CRUK0006              CRUK0006   66    Male   \n",
       "101  CRUK0007                CRUK0007              CRUK0007   69    Male   \n",
       "..        ...                     ...                   ...  ...     ...   \n",
       "373  CRUK0795                CRUK0795              CRUK0795   72    Male   \n",
       "378  CRUK0799                CRUK0799              CRUK0799   85  Female   \n",
       "414  CRUK0861                CRUK0861              CRUK0861   65  Female   \n",
       "418  CRUK0864                CRUK0864              CRUK0864   78  Female   \n",
       "415  CRUK0893                CRUK0893              CRUK0893   59  Female   \n",
       "\n",
       "          ethnicity  cigs_perday  years_smoking  packyears  \\\n",
       "88   White- British         20.0             30     30.000   \n",
       "79   White- British          5.0             35      8.750   \n",
       "7    White- British         20.0             56     56.000   \n",
       "99   White- British         17.5             50     43.750   \n",
       "101  White- British          4.5              3      0.675   \n",
       "..              ...          ...            ...        ...   \n",
       "373  White- British         20.0             50     50.000   \n",
       "378  White- British         20.0             70     70.000   \n",
       "414  Middle eastern         10.0             45     22.500   \n",
       "418  White- British         10.0             30     15.000   \n",
       "415  White- British         15.0             43     32.250   \n",
       "\n",
       "    smoking_status_merged  is.family.lung  ECOG_PS pathologyTNM  \\\n",
       "88              Ex-Smoker           False      1.0          IIA   \n",
       "79              Ex-Smoker           False      1.0           IB   \n",
       "7               Ex-Smoker           False      1.0         IIIA   \n",
       "99                 Smoker           False      0.0           IA   \n",
       "101             Ex-Smoker           False      0.0           IA   \n",
       "..                    ...             ...      ...          ...   \n",
       "373                Smoker           False      0.0           IA   \n",
       "378                Smoker           False      1.0         IIIA   \n",
       "414                Smoker           False      1.0           IA   \n",
       "418             Ex-Smoker            True      0.0          IIA   \n",
       "415                Smoker           False      1.0           IA   \n",
       "\n",
       "    pT_stage_per_patient  pN_stage_per_patient LVI_per_patient PL_per_patient  \\\n",
       "88                    2a                     1              No            Yes   \n",
       "79                    2a                     0             Yes             No   \n",
       "7                     2a                     2             Yes            Yes   \n",
       "99                    1a                     0             Yes             No   \n",
       "101                   1a                     0              No             No   \n",
       "..                   ...                   ...             ...            ...   \n",
       "373                   1b                     0             Yes             No   \n",
       "378                   2b                     2              No            Yes   \n",
       "414                   1b                     0              No             No   \n",
       "418                   1b                     1              No             No   \n",
       "415                   1a                     0             Yes             No   \n",
       "\n",
       "    margin_status_per_patient  size_pathology_per_patient Surgery_type  \\\n",
       "88                         R0                        26.0    Lobectomy   \n",
       "79                         R0                        35.0    Lobectomy   \n",
       "7                          R0                        43.0    Lobectomy   \n",
       "99                         R0                        18.0    Lobectomy   \n",
       "101                        R0                        18.0    Lobectomy   \n",
       "..                        ...                         ...          ...   \n",
       "373                        R0                        28.0    Lobectomy   \n",
       "378                        R0                        66.0    Lobectomy   \n",
       "414                        R0                        25.0    Lobectomy   \n",
       "418                        R0                        27.0    Lobectomy   \n",
       "415                        R0                        17.0    Lobectomy   \n",
       "\n",
       "           histology_lesion1 histology_lesion1_merged  lesion1_sampled  \\\n",
       "88   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "79   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "7    Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "99   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "101  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "..                       ...                      ...              ...   \n",
       "373  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "378  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "414  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "418  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "415  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "\n",
       "    histology_lesion2 lesion2_sampled histology_multi_full  \\\n",
       "88                NaN             NaN                 LUAD   \n",
       "79                NaN             NaN                 LUAD   \n",
       "7                 NaN             NaN                 LUAD   \n",
       "99                NaN             NaN                 LUAD   \n",
       "101               NaN             NaN                 LUAD   \n",
       "..                ...             ...                  ...   \n",
       "373               NaN             NaN                 LUAD   \n",
       "378               NaN             NaN                 LUAD   \n",
       "414               NaN             NaN                 LUSC   \n",
       "418               NaN             NaN                 LUAD   \n",
       "415               NaN             NaN                 LUAD   \n",
       "\n",
       "    histology_multi_full_genomically.confirmed LUAD_pred_subtype  \\\n",
       "88                                        LUAD            acinar   \n",
       "79                                        LUAD            acinar   \n",
       "7                                         LUAD            acinar   \n",
       "99                                        LUAD            acinar   \n",
       "101                                       LUAD           lepidic   \n",
       "..                                         ...               ...   \n",
       "373                                       LUAD            acinar   \n",
       "378                                       LUAD            acinar   \n",
       "414                                       LUSC               NaN   \n",
       "418                                       LUAD             solid   \n",
       "415                                       LUAD            acinar   \n",
       "\n",
       "    adjuvant_treatment_YN adjuvant_treatment_given num_cycle_na.added  \\\n",
       "88            No adjuvant                      NaN                  0   \n",
       "79            No adjuvant                      NaN                  0   \n",
       "7             No adjuvant                      NaN                  0   \n",
       "99            No adjuvant                      NaN                  0   \n",
       "101           No adjuvant                      NaN                  0   \n",
       "..                    ...                      ...                ...   \n",
       "373           No adjuvant                      NaN                  0   \n",
       "378           No adjuvant                      NaN                  0   \n",
       "414           No adjuvant                      NaN                  0   \n",
       "418              Adjuvant           Platinum chemo                  4   \n",
       "415           No adjuvant                      NaN                  0   \n",
       "\n",
       "                               CHMPlatDgName_cleaned  \\\n",
       "88                                               NaN   \n",
       "79                                               NaN   \n",
       "7                                                NaN   \n",
       "99                                               NaN   \n",
       "101                                              NaN   \n",
       "..                                               ...   \n",
       "373                                              NaN   \n",
       "378                                              NaN   \n",
       "414                                              NaN   \n",
       "418  Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "415                                              NaN   \n",
       "\n",
       "                                CHMOthDgName_cleaned  AdjRadStartTime_manual  \\\n",
       "88                                               NaN                     NaN   \n",
       "79                                               NaN                     NaN   \n",
       "7                                                NaN                     NaN   \n",
       "99                                               NaN                     NaN   \n",
       "101                                              NaN                     NaN   \n",
       "..                                               ...                     ...   \n",
       "373                                              NaN                     NaN   \n",
       "378                                              NaN                     NaN   \n",
       "414                                              NaN                     NaN   \n",
       "418  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "415                                              NaN                     NaN   \n",
       "\n",
       "     AdjRadEndTime_manual  Recurrence_time_use newPrim_time_use  \\\n",
       "88                    NaN                  NaN              NaN   \n",
       "79                    NaN                  NaN              NaN   \n",
       "7                     NaN                  NaN              NaN   \n",
       "99                    NaN                  NaN              NaN   \n",
       "101                   NaN                  NaN              NaN   \n",
       "..                    ...                  ...              ...   \n",
       "373                   NaN                  NaN             1182   \n",
       "378                   NaN                  NaN              NaN   \n",
       "414                   NaN                  NaN              NaN   \n",
       "418                   NaN                  NaN              NaN   \n",
       "415                   NaN                  NaN              NaN   \n",
       "\n",
       "    first_dfs_any_event_rec.or.new.primary first_event_during_followup  \\\n",
       "88                                     NaN                       Death   \n",
       "79                                     NaN                         NaN   \n",
       "7                                      NaN                       Death   \n",
       "99                                     NaN                         NaN   \n",
       "101                                    NaN                         NaN   \n",
       "..                                     ...                         ...   \n",
       "373                            new_primary                 new_primary   \n",
       "378                                    NaN                       Death   \n",
       "414                                    NaN                         NaN   \n",
       "418                                    NaN                         NaN   \n",
       "415                                    NaN                         NaN   \n",
       "\n",
       "     cens_os  os_time  cens_dfs  dfs_time  cens_dfs_any_event  \\\n",
       "88         1     1120         1      1120                   1   \n",
       "79         0     1901         0      1901                   0   \n",
       "7          1      649         1       649                   1   \n",
       "99         0     1811         0      1811                   0   \n",
       "101        0     2244         0      2244                   0   \n",
       "..       ...      ...       ...       ...                 ...   \n",
       "373        0     1232         0      1232                   1   \n",
       "378        1     1016         1      1016                   1   \n",
       "414        0     1084         0      1084                   0   \n",
       "418        0     1434         0      1434                   0   \n",
       "415        0     1001         0      1001                   0   \n",
       "\n",
       "     dfs_time_any_event  cens_lung_event  lung_event_time  \\\n",
       "88                 1120                0             1120   \n",
       "79                 1901                0             1901   \n",
       "7                   649                0              649   \n",
       "99                 1811                0             1811   \n",
       "101                2244                0             2244   \n",
       "..                  ...              ...              ...   \n",
       "373                1182                0             1232   \n",
       "378                1016                0             1016   \n",
       "414                1084                0             1084   \n",
       "418                1434                0             1434   \n",
       "415                1001                0             1001   \n",
       "\n",
       "             Relapse_cat Relapse_cat_new  tx100  \n",
       "88                   NaN          No rec   True  \n",
       "79                   NaN          No rec   True  \n",
       "7                    NaN          No rec   True  \n",
       "99                   NaN          No rec   True  \n",
       "101                  NaN          No rec   True  \n",
       "..                   ...             ...    ...  \n",
       "373  Second primary lung          No rec  False  \n",
       "378                  NaN          No rec  False  \n",
       "414                  NaN          No rec  False  \n",
       "418                  NaN          No rec  False  \n",
       "415                  NaN          No rec  False  \n",
       "\n",
       "[279 rows x 50 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_recurrence_clinical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "recurrence_patient_id = recurrence_clinical_df['cruk_id']\n",
    "no_recurrence_patient_id = no_recurrence_clinical_df['cruk_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# separate mut_table into two dataframes, one for patients where patient_id is in recurrence_patient_id and one for patients where patient_id is in no_recurrence_patient_id\n",
    "recurrence_mut_table = mut_table[mut_table['patient_id'].isin(recurrence_patient_id)]\n",
    "no_recurrence_mut_table = mut_table[mut_table['patient_id'].isin(no_recurrence_patient_id)]\n",
    "\n",
    "# remove rows with DriverMut = False\n",
    "recurrence_mut_table = recurrence_mut_table[recurrence_mut_table['DriverMut'] == True]\n",
    "no_recurrence_mut_table = no_recurrence_mut_table[no_recurrence_mut_table['DriverMut'] == True]\n",
    "\n",
    "# sort by patient_id\n",
    "recurrence_mut_table = recurrence_mut_table.sort_values(by=['patient_id'])\n",
    "no_recurrence_mut_table = no_recurrence_mut_table.sort_values(by=['patient_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>mutation_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>ref</th>\n",
       "      <th>var</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>func</th>\n",
       "      <th>exonic.func</th>\n",
       "      <th>NucleotideChange</th>\n",
       "      <th>AAChange</th>\n",
       "      <th>GL_VAF</th>\n",
       "      <th>GL_nAlt</th>\n",
       "      <th>GL_depth</th>\n",
       "      <th>ITHState</th>\n",
       "      <th>PyCloneClonal_SC</th>\n",
       "      <th>PyCloneCluster_SC</th>\n",
       "      <th>cleanCluster_SC</th>\n",
       "      <th>combTiming_SC</th>\n",
       "      <th>DriverMut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>156560</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003:3:178936094:C</td>\n",
       "      <td>chr3</td>\n",
       "      <td>178936094</td>\n",
       "      <td>178936094</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.C1636A</td>\n",
       "      <td>p.Q546K</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>413</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>late</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156585</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003:7:55242464:A</td>\n",
       "      <td>chr7</td>\n",
       "      <td>55242464</td>\n",
       "      <td>55242464</td>\n",
       "      <td>A</td>\n",
       "      <td>-GGAATTAAGAGAAGC</td>\n",
       "      <td>EGFR</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonframeshift substitution</td>\n",
       "      <td>c.2234_2234delins-GGAATTAAGAGAAGC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.170648464163823</td>\n",
       "      <td>1</td>\n",
       "      <td>586</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156750</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003:3:41266098:A</td>\n",
       "      <td>chr3</td>\n",
       "      <td>41266098</td>\n",
       "      <td>41266098</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.A95T</td>\n",
       "      <td>p.D32V</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>419</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>11.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156829</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003:3:41266125:C</td>\n",
       "      <td>chr3</td>\n",
       "      <td>41266125</td>\n",
       "      <td>41266125</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.C122T</td>\n",
       "      <td>p.T41I</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>418</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>11.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52971</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004:17:7579485:C</td>\n",
       "      <td>chr17</td>\n",
       "      <td>7579485</td>\n",
       "      <td>7579485</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TP53</td>\n",
       "      <td>exonic</td>\n",
       "      <td>stopgain SNV</td>\n",
       "      <td>c.G85T</td>\n",
       "      <td>p.E29X</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>349</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>357267</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour2</td>\n",
       "      <td>CRUK0881_Tumour2:2:225376215:G</td>\n",
       "      <td>chr2</td>\n",
       "      <td>225376215</td>\n",
       "      <td>225376215</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CUL3</td>\n",
       "      <td>exonic</td>\n",
       "      <td>stopgain SNV</td>\n",
       "      <td>c.C541T</td>\n",
       "      <td>p.R181X</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>519</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>4.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355115</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881_Tumour1:3:52651397:C</td>\n",
       "      <td>chr3</td>\n",
       "      <td>52651397</td>\n",
       "      <td>52651397</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>PBRM1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G1699T</td>\n",
       "      <td>p.D567Y</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>499</td>\n",
       "      <td>2</td>\n",
       "      <td>S</td>\n",
       "      <td>21.0</td>\n",
       "      <td>False</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354624</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881_Tumour1:12:25398285:C</td>\n",
       "      <td>chr12</td>\n",
       "      <td>25398285</td>\n",
       "      <td>25398285</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G34T</td>\n",
       "      <td>p.G12C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>369</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>356763</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour2</td>\n",
       "      <td>CRUK0881_Tumour2:13:102235583:G</td>\n",
       "      <td>chr13</td>\n",
       "      <td>102235583</td>\n",
       "      <td>102235583</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ITGBL1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>stopgain SNV</td>\n",
       "      <td>c.G322T</td>\n",
       "      <td>p.E108X</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>296</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355746</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881_Tumour1:16:3778218:G</td>\n",
       "      <td>chr16</td>\n",
       "      <td>3778218</td>\n",
       "      <td>3778218</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.C6716T</td>\n",
       "      <td>p.P2239L</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>386</td>\n",
       "      <td>2</td>\n",
       "      <td>S</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1016 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id         tumour_id                      mutation_id    chr  \\\n",
       "156560   CRUK0003          CRUK0003           CRUK0003:3:178936094:C   chr3   \n",
       "156585   CRUK0003          CRUK0003            CRUK0003:7:55242464:A   chr7   \n",
       "156750   CRUK0003          CRUK0003            CRUK0003:3:41266098:A   chr3   \n",
       "156829   CRUK0003          CRUK0003            CRUK0003:3:41266125:C   chr3   \n",
       "52971    CRUK0004          CRUK0004            CRUK0004:17:7579485:C  chr17   \n",
       "...           ...               ...                              ...    ...   \n",
       "357267   CRUK0881  CRUK0881_Tumour2   CRUK0881_Tumour2:2:225376215:G   chr2   \n",
       "355115   CRUK0881  CRUK0881_Tumour1    CRUK0881_Tumour1:3:52651397:C   chr3   \n",
       "354624   CRUK0881  CRUK0881_Tumour1   CRUK0881_Tumour1:12:25398285:C  chr12   \n",
       "356763   CRUK0881  CRUK0881_Tumour2  CRUK0881_Tumour2:13:102235583:G  chr13   \n",
       "355746   CRUK0881  CRUK0881_Tumour1    CRUK0881_Tumour1:16:3778218:G  chr16   \n",
       "\n",
       "            start       stop ref               var Hugo_Symbol    func  \\\n",
       "156560  178936094  178936094   C                 A      PIK3CA  exonic   \n",
       "156585   55242464   55242464   A  -GGAATTAAGAGAAGC        EGFR  exonic   \n",
       "156750   41266098   41266098   A                 T      CTNNB1  exonic   \n",
       "156829   41266125   41266125   C                 T      CTNNB1  exonic   \n",
       "52971     7579485    7579485   C                 A        TP53  exonic   \n",
       "...           ...        ...  ..               ...         ...     ...   \n",
       "357267  225376215  225376215   G                 A        CUL3  exonic   \n",
       "355115   52651397   52651397   C                 A       PBRM1  exonic   \n",
       "354624   25398285   25398285   C                 A        KRAS  exonic   \n",
       "356763  102235583  102235583   G                 T      ITGBL1  exonic   \n",
       "355746    3778218    3778218   G                 A      CREBBP  exonic   \n",
       "\n",
       "                       exonic.func                   NucleotideChange  \\\n",
       "156560           nonsynonymous SNV                           c.C1636A   \n",
       "156585  nonframeshift substitution  c.2234_2234delins-GGAATTAAGAGAAGC   \n",
       "156750           nonsynonymous SNV                             c.A95T   \n",
       "156829           nonsynonymous SNV                            c.C122T   \n",
       "52971                 stopgain SNV                             c.G85T   \n",
       "...                            ...                                ...   \n",
       "357267                stopgain SNV                            c.C541T   \n",
       "355115           nonsynonymous SNV                           c.G1699T   \n",
       "354624           nonsynonymous SNV                             c.G34T   \n",
       "356763                stopgain SNV                            c.G322T   \n",
       "355746           nonsynonymous SNV                           c.C6716T   \n",
       "\n",
       "        AAChange             GL_VAF GL_nAlt GL_depth ITHState  \\\n",
       "156560   p.Q546K                  0       0      413        1   \n",
       "156585       NaN  0.170648464163823       1      586        1   \n",
       "156750    p.D32V                  0       0      419        3   \n",
       "156829    p.T41I                  0       0      418        3   \n",
       "52971     p.E29X                  0       0      349        1   \n",
       "...          ...                ...     ...      ...      ...   \n",
       "357267   p.R181X                  0       0      519        3   \n",
       "355115   p.D567Y                  0       0      499        2   \n",
       "354624    p.G12C                  0       0      369        1   \n",
       "356763   p.E108X                  0       0      296        1   \n",
       "355746  p.P2239L                  0       0      386        2   \n",
       "\n",
       "       PyCloneClonal_SC  PyCloneCluster_SC cleanCluster_SC combTiming_SC  \\\n",
       "156560                C                1.0            True          late   \n",
       "156585                C                1.0            True         early   \n",
       "156750                S               11.0            True     subclonal   \n",
       "156829                S               11.0            True     subclonal   \n",
       "52971                 C                1.0            True         early   \n",
       "...                 ...                ...             ...           ...   \n",
       "357267                S                4.0            True     subclonal   \n",
       "355115                S               21.0           False     subclonal   \n",
       "354624                C                1.0            True         early   \n",
       "356763                C                1.0            True         early   \n",
       "355746                S                2.0            True     subclonal   \n",
       "\n",
       "        DriverMut  \n",
       "156560       True  \n",
       "156585       True  \n",
       "156750       True  \n",
       "156829       True  \n",
       "52971        True  \n",
       "...           ...  \n",
       "357267       True  \n",
       "355115       True  \n",
       "354624       True  \n",
       "356763       True  \n",
       "355746       True  \n",
       "\n",
       "[1016 rows x 22 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recurrence_mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>mutation_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>ref</th>\n",
       "      <th>var</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>func</th>\n",
       "      <th>exonic.func</th>\n",
       "      <th>NucleotideChange</th>\n",
       "      <th>AAChange</th>\n",
       "      <th>GL_VAF</th>\n",
       "      <th>GL_nAlt</th>\n",
       "      <th>GL_depth</th>\n",
       "      <th>ITHState</th>\n",
       "      <th>PyCloneClonal_SC</th>\n",
       "      <th>PyCloneCluster_SC</th>\n",
       "      <th>cleanCluster_SC</th>\n",
       "      <th>combTiming_SC</th>\n",
       "      <th>DriverMut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>67810</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001:19:47440656:G</td>\n",
       "      <td>chr19</td>\n",
       "      <td>47440656</td>\n",
       "      <td>47440656</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ARHGAP35</td>\n",
       "      <td>exonic</td>\n",
       "      <td>stopgain SNV</td>\n",
       "      <td>c.G3817T</td>\n",
       "      <td>p.E1273X</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>333</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67608</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001:1:51439886:G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>51439886</td>\n",
       "      <td>51439886</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CDKN2C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G451A</td>\n",
       "      <td>p.E151K</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>430</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67527</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001:16:68772312:G</td>\n",
       "      <td>chr16</td>\n",
       "      <td>68772312</td>\n",
       "      <td>68772312</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>CDH1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G161C</td>\n",
       "      <td>p.R54T</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>334</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67414</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001:19:36217187:G</td>\n",
       "      <td>chr19</td>\n",
       "      <td>36217187</td>\n",
       "      <td>36217187</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>KMT2B</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G3936C</td>\n",
       "      <td>p.K1312N</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>202</td>\n",
       "      <td>2</td>\n",
       "      <td>S</td>\n",
       "      <td>10.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67292</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001:2:242078113:C</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242078113</td>\n",
       "      <td>242078113</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PASK</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G697C</td>\n",
       "      <td>p.E233Q</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>680</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361461</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864:23:47035900:A</td>\n",
       "      <td>chrX</td>\n",
       "      <td>47035900</td>\n",
       "      <td>47035900</td>\n",
       "      <td>A</td>\n",
       "      <td>-CTCC</td>\n",
       "      <td>RBM10</td>\n",
       "      <td>exonic</td>\n",
       "      <td>frameshift substitution</td>\n",
       "      <td>c.347_347delins-CTCC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.34965034965035</td>\n",
       "      <td>1</td>\n",
       "      <td>286</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361675</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864:7:101821836:G</td>\n",
       "      <td>chr7</td>\n",
       "      <td>101821836</td>\n",
       "      <td>101821836</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>CUX1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G901A</td>\n",
       "      <td>p.V301M</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>289</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>3.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361724</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864:2:25497907:C</td>\n",
       "      <td>chr2</td>\n",
       "      <td>25497907</td>\n",
       "      <td>25497907</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G542A</td>\n",
       "      <td>p.R181H</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>277</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>6.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276073</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893:12:25398284:C</td>\n",
       "      <td>chr12</td>\n",
       "      <td>25398284</td>\n",
       "      <td>25398284</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G35C</td>\n",
       "      <td>p.G12A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>413</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276116</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893:11:108205739:C</td>\n",
       "      <td>chr11</td>\n",
       "      <td>108205739</td>\n",
       "      <td>108205739</td>\n",
       "      <td>C</td>\n",
       "      <td>-ATTTAAAGCAGA</td>\n",
       "      <td>ATM</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonframeshift substitution</td>\n",
       "      <td>c.8054_8054delins-ATTTAAAGCAGA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>390</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1817 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id tumour_id              mutation_id    chr      start  \\\n",
       "67810    CRUK0001  CRUK0001   CRUK0001:19:47440656:G  chr19   47440656   \n",
       "67608    CRUK0001  CRUK0001    CRUK0001:1:51439886:G   chr1   51439886   \n",
       "67527    CRUK0001  CRUK0001   CRUK0001:16:68772312:G  chr16   68772312   \n",
       "67414    CRUK0001  CRUK0001   CRUK0001:19:36217187:G  chr19   36217187   \n",
       "67292    CRUK0001  CRUK0001   CRUK0001:2:242078113:C   chr2  242078113   \n",
       "...           ...       ...                      ...    ...        ...   \n",
       "361461   CRUK0864  CRUK0864   CRUK0864:23:47035900:A   chrX   47035900   \n",
       "361675   CRUK0864  CRUK0864   CRUK0864:7:101821836:G   chr7  101821836   \n",
       "361724   CRUK0864  CRUK0864    CRUK0864:2:25497907:C   chr2   25497907   \n",
       "276073   CRUK0893  CRUK0893   CRUK0893:12:25398284:C  chr12   25398284   \n",
       "276116   CRUK0893  CRUK0893  CRUK0893:11:108205739:C  chr11  108205739   \n",
       "\n",
       "             stop ref            var Hugo_Symbol    func  \\\n",
       "67810    47440656   G              T    ARHGAP35  exonic   \n",
       "67608    51439886   G              A      CDKN2C  exonic   \n",
       "67527    68772312   G              C        CDH1  exonic   \n",
       "67414    36217187   G              C       KMT2B  exonic   \n",
       "67292   242078113   C              G        PASK  exonic   \n",
       "...           ...  ..            ...         ...     ...   \n",
       "361461   47035900   A          -CTCC       RBM10  exonic   \n",
       "361675  101821836   G              A        CUX1  exonic   \n",
       "361724   25497907   C              T      DNMT3A  exonic   \n",
       "276073   25398284   C              G        KRAS  exonic   \n",
       "276116  108205739   C  -ATTTAAAGCAGA         ATM  exonic   \n",
       "\n",
       "                       exonic.func                NucleotideChange  AAChange  \\\n",
       "67810                 stopgain SNV                        c.G3817T  p.E1273X   \n",
       "67608            nonsynonymous SNV                         c.G451A   p.E151K   \n",
       "67527            nonsynonymous SNV                         c.G161C    p.R54T   \n",
       "67414            nonsynonymous SNV                        c.G3936C  p.K1312N   \n",
       "67292            nonsynonymous SNV                         c.G697C   p.E233Q   \n",
       "...                            ...                             ...       ...   \n",
       "361461     frameshift substitution            c.347_347delins-CTCC       NaN   \n",
       "361675           nonsynonymous SNV                         c.G901A   p.V301M   \n",
       "361724           nonsynonymous SNV                         c.G542A   p.R181H   \n",
       "276073           nonsynonymous SNV                          c.G35C    p.G12A   \n",
       "276116  nonframeshift substitution  c.8054_8054delins-ATTTAAAGCAGA       NaN   \n",
       "\n",
       "                  GL_VAF GL_nAlt GL_depth ITHState PyCloneClonal_SC  \\\n",
       "67810                  0       0      333        3                S   \n",
       "67608                  0       0      430        3                S   \n",
       "67527                  0       0      334        3                S   \n",
       "67414                  0       0      202        2                S   \n",
       "67292                  0       0      680        3                S   \n",
       "...                  ...     ...      ...      ...              ...   \n",
       "361461  0.34965034965035       1      286        1              NaN   \n",
       "361675                 0       0      289        3                S   \n",
       "361724                 0       0      277        3                S   \n",
       "276073                 0       0      413        1                C   \n",
       "276116                 0       0      390        1                C   \n",
       "\n",
       "        PyCloneCluster_SC cleanCluster_SC combTiming_SC  DriverMut  \n",
       "67810                 2.0            True     subclonal       True  \n",
       "67608                 1.0            True     subclonal       True  \n",
       "67527                 1.0            True     subclonal       True  \n",
       "67414                10.0            True     subclonal       True  \n",
       "67292                 1.0            True     subclonal       True  \n",
       "...                   ...             ...           ...        ...  \n",
       "361461                NaN             NaN            NA       True  \n",
       "361675                3.0            True     subclonal       True  \n",
       "361724                6.0            True     subclonal       True  \n",
       "276073                1.0            True         early       True  \n",
       "276116                1.0            True         early       True  \n",
       "\n",
       "[1817 rows x 22 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_recurrence_mut_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "LUAD_patient_id = []\n",
    "LUSC_patient_id = []\n",
    "\n",
    "for index, row in tumor_df.iterrows():\n",
    "    if row['histology_3'] == 'LUAD':\n",
    "        if row['patient_id'] not in LUAD_patient_id:\n",
    "            LUAD_patient_id.append(row['patient_id'])\n",
    "    elif row['histology_3'] == 'LUSC':\n",
    "        if row['patient_id'] not in LUSC_patient_id:\n",
    "            LUSC_patient_id.append(row['patient_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make a dictionary of Hugo_Symbol and the number of times it appears in the recurrence_mut_table\n",
    "recurrence_gene_count = dict(recurrence_mut_table['Hugo_Symbol'].value_counts())\n",
    "recurrence_gene_count = dict(sorted(recurrence_gene_count.items()))\n",
    "\n",
    "# make a dictionary of Hugo_Symbol and the number of times it appears in the no_recurrence_mut_table\n",
    "no_recurrence_gene_count = dict(no_recurrence_mut_table['Hugo_Symbol'].value_counts())\n",
    "no_recurrence_gene_count = dict(sorted(no_recurrence_gene_count.items()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ACVR1B': 3,\n",
       " 'AMER1': 2,\n",
       " 'ANAPC5': 1,\n",
       " 'ANK2': 3,\n",
       " 'APC': 6,\n",
       " 'ARAF': 1,\n",
       " 'ARHGAP35': 8,\n",
       " 'ARHGEF12': 1,\n",
       " 'ARID1A': 11,\n",
       " 'ARID1B': 8,\n",
       " 'ARID2': 13,\n",
       " 'ASXL1': 6,\n",
       " 'ASXL2': 2,\n",
       " 'ATM': 10,\n",
       " 'ATR': 3,\n",
       " 'ATRX': 8,\n",
       " 'AXIN1': 4,\n",
       " 'AXIN2': 2,\n",
       " 'B2M': 9,\n",
       " 'BAP1': 5,\n",
       " 'BCLAF1': 6,\n",
       " 'BCOR': 7,\n",
       " 'BHMT2': 1,\n",
       " 'BIVM-ERCC5': 1,\n",
       " 'BLM': 2,\n",
       " 'BMPR1A': 3,\n",
       " 'BRAF': 4,\n",
       " 'BRCA1': 4,\n",
       " 'BRCA2': 1,\n",
       " 'BRIP1': 1,\n",
       " 'BUB1B': 4,\n",
       " 'CAD': 1,\n",
       " 'CASP8': 2,\n",
       " 'CBL': 3,\n",
       " 'CBLB': 5,\n",
       " 'CDC73': 4,\n",
       " 'CDK12': 4,\n",
       " 'CDKN2A': 14,\n",
       " 'CEP76': 1,\n",
       " 'CHD4': 7,\n",
       " 'CHD8': 4,\n",
       " 'CHEK2': 2,\n",
       " 'CIC': 6,\n",
       " 'CMTR2': 3,\n",
       " 'CNKSR1': 1,\n",
       " 'CNOT3': 3,\n",
       " 'COL2A1': 6,\n",
       " 'COL5A2': 10,\n",
       " 'CREBBP': 13,\n",
       " 'CSMD3': 2,\n",
       " 'CTCF': 2,\n",
       " 'CTNNB1': 7,\n",
       " 'CUL3': 3,\n",
       " 'CUX1': 8,\n",
       " 'CYLD': 4,\n",
       " 'DDB2': 1,\n",
       " 'DDX3X': 5,\n",
       " 'DICER1': 4,\n",
       " 'DNAH12': 4,\n",
       " 'DNM2': 3,\n",
       " 'DNMT3A': 2,\n",
       " 'DOT1L': 5,\n",
       " 'EGFR': 10,\n",
       " 'ELL2': 1,\n",
       " 'EP300': 5,\n",
       " 'EPB41L3': 1,\n",
       " 'ERBB2': 2,\n",
       " 'ERBB4': 1,\n",
       " 'ERCC2': 1,\n",
       " 'ERCC3': 1,\n",
       " 'ERCC4': 1,\n",
       " 'EXT1': 3,\n",
       " 'EZH1': 4,\n",
       " 'FAM166A': 2,\n",
       " 'FAM46C': 1,\n",
       " 'FANCA': 1,\n",
       " 'FANCD2': 1,\n",
       " 'FANCF': 1,\n",
       " 'FANCM': 5,\n",
       " 'FAS': 3,\n",
       " 'FAT1': 18,\n",
       " 'FBXO11': 1,\n",
       " 'FBXW7': 5,\n",
       " 'FGFBP1': 1,\n",
       " 'FGFR1': 1,\n",
       " 'FGFR3': 2,\n",
       " 'FH': 1,\n",
       " 'FLCN': 4,\n",
       " 'FLT4': 5,\n",
       " 'FN1': 2,\n",
       " 'FUBP1': 2,\n",
       " 'GATA3': 3,\n",
       " 'GNPTAB': 2,\n",
       " 'GPS2': 1,\n",
       " 'GRIN2A': 7,\n",
       " 'HCN1': 3,\n",
       " 'HNF1A': 3,\n",
       " 'HOXB3': 1,\n",
       " 'HRAS': 1,\n",
       " 'IKZF1': 4,\n",
       " 'IL21R': 1,\n",
       " 'IQGAP2': 1,\n",
       " 'ITGBL1': 1,\n",
       " 'KDM5C': 8,\n",
       " 'KDM6A': 4,\n",
       " 'KEAP1': 22,\n",
       " 'KLF5': 2,\n",
       " 'KLF6': 1,\n",
       " 'KMT2B': 5,\n",
       " 'KMT2C': 7,\n",
       " 'KMT2D': 31,\n",
       " 'KRAS': 45,\n",
       " 'LAMA4': 1,\n",
       " 'LATS1': 4,\n",
       " 'MAP3K1': 3,\n",
       " 'MAP3K13': 5,\n",
       " 'MAX': 1,\n",
       " 'MBD1': 1,\n",
       " 'MEN1': 2,\n",
       " 'MGA': 7,\n",
       " 'MLH1': 1,\n",
       " 'MSH2': 3,\n",
       " 'MTOR': 9,\n",
       " 'MUTYH': 1,\n",
       " 'MYH2': 1,\n",
       " 'NBN': 5,\n",
       " 'NCOA6': 6,\n",
       " 'NCOR1': 8,\n",
       " 'NEB': 1,\n",
       " 'NF1': 18,\n",
       " 'NF2': 1,\n",
       " 'NFE2L2': 6,\n",
       " 'NFKBIE': 2,\n",
       " 'NIN': 1,\n",
       " 'NOTCH1': 7,\n",
       " 'NOTCH2': 5,\n",
       " 'NRAS': 3,\n",
       " 'NRXN1': 1,\n",
       " 'NSD1': 1,\n",
       " 'NTRK1': 1,\n",
       " 'PALB2': 2,\n",
       " 'PASK': 3,\n",
       " 'PATZ1': 1,\n",
       " 'PBRM1': 7,\n",
       " 'PDYN': 1,\n",
       " 'PHOX2B': 1,\n",
       " 'PIK3CA': 18,\n",
       " 'PIK3CG': 1,\n",
       " 'PIK3R1': 4,\n",
       " 'PLCG2': 1,\n",
       " 'PLXNB2': 9,\n",
       " 'PMS1': 2,\n",
       " 'POLE': 6,\n",
       " 'POT1': 5,\n",
       " 'PPP3CA': 3,\n",
       " 'PRDM1': 3,\n",
       " 'PRF1': 2,\n",
       " 'PRKAR1A': 1,\n",
       " 'PRRX1': 1,\n",
       " 'PTCH1': 1,\n",
       " 'PTEN': 12,\n",
       " 'PTPRB': 12,\n",
       " 'PTPRC': 8,\n",
       " 'PTPRU': 4,\n",
       " 'RAD21': 3,\n",
       " 'RASA1': 8,\n",
       " 'RB1': 6,\n",
       " 'RBM10': 13,\n",
       " 'RIMS2': 1,\n",
       " 'RIT1': 2,\n",
       " 'RNF43': 4,\n",
       " 'RYR2': 4,\n",
       " 'SEPT12': 1,\n",
       " 'SERPINB13': 2,\n",
       " 'SETD2': 11,\n",
       " 'SETDB1': 2,\n",
       " 'SF3B1': 2,\n",
       " 'SGK223': 6,\n",
       " 'SH2B3': 1,\n",
       " 'SMAD4': 7,\n",
       " 'SMARCA4': 25,\n",
       " 'SMARCC2': 1,\n",
       " 'SNX29': 5,\n",
       " 'SPEN': 7,\n",
       " 'STAG2': 3,\n",
       " 'STIM1': 2,\n",
       " 'STK11': 26,\n",
       " 'STX2': 2,\n",
       " 'TBL1XR1': 1,\n",
       " 'TET2': 3,\n",
       " 'TNC': 1,\n",
       " 'TNFAIP3': 2,\n",
       " 'TNFRSF14': 1,\n",
       " 'TNR': 1,\n",
       " 'TP53': 92,\n",
       " 'TP53BP1': 4,\n",
       " 'TSC1': 1,\n",
       " 'TSC2': 2,\n",
       " 'U2AF1': 3,\n",
       " 'UBR5': 8,\n",
       " 'WAS': 3,\n",
       " 'WRN': 1,\n",
       " 'WT1': 6,\n",
       " 'ZFP36L1': 3,\n",
       " 'ZFPM2': 1,\n",
       " 'ZMYM2': 6,\n",
       " 'ZNF180': 1,\n",
       " 'ZNF483': 1}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recurrence_gene_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ACTN2': 1,\n",
       " 'ACVR1B': 7,\n",
       " 'ADCY2': 2,\n",
       " 'AKT1': 1,\n",
       " 'AMER1': 8,\n",
       " 'ANK2': 4,\n",
       " 'APC': 16,\n",
       " 'ARHGAP35': 11,\n",
       " 'ARID1A': 19,\n",
       " 'ARID1B': 9,\n",
       " 'ARID2': 13,\n",
       " 'ASXL1': 3,\n",
       " 'ASXL2': 9,\n",
       " 'ATF7IP': 6,\n",
       " 'ATM': 20,\n",
       " 'ATP5B': 2,\n",
       " 'ATR': 10,\n",
       " 'ATRX': 20,\n",
       " 'AXIN1': 4,\n",
       " 'AXIN2': 3,\n",
       " 'B2M': 6,\n",
       " 'BAP1': 6,\n",
       " 'BCLAF1': 15,\n",
       " 'BCOR': 11,\n",
       " 'BHMT2': 2,\n",
       " 'BIVM-ERCC5': 4,\n",
       " 'BLM': 2,\n",
       " 'BMPR1A': 2,\n",
       " 'BRAF': 8,\n",
       " 'BRCA1': 3,\n",
       " 'BRCA2': 5,\n",
       " 'BRE': 2,\n",
       " 'BRIP1': 5,\n",
       " 'BUB1B': 2,\n",
       " 'CAD': 1,\n",
       " 'CASP8': 4,\n",
       " 'CBL': 6,\n",
       " 'CBLB': 10,\n",
       " 'CDC73': 5,\n",
       " 'CDH1': 4,\n",
       " 'CDH18': 1,\n",
       " 'CDK12': 8,\n",
       " 'CDKN2A': 34,\n",
       " 'CDKN2C': 2,\n",
       " 'CEP76': 2,\n",
       " 'CHD4': 10,\n",
       " 'CHD8': 12,\n",
       " 'CHEK2': 5,\n",
       " 'CIC': 6,\n",
       " 'CMTR2': 12,\n",
       " 'CNBD1': 6,\n",
       " 'CNKSR1': 3,\n",
       " 'CNOT3': 6,\n",
       " 'COL2A1': 13,\n",
       " 'COL5A2': 20,\n",
       " 'CREBBP': 18,\n",
       " 'CSMD3': 3,\n",
       " 'CTCF': 3,\n",
       " 'CTNNB1': 3,\n",
       " 'CUL3': 6,\n",
       " 'CUX1': 15,\n",
       " 'CYLD': 4,\n",
       " 'DARC': 1,\n",
       " 'DDX3X': 6,\n",
       " 'DICER1': 10,\n",
       " 'DNAH12': 12,\n",
       " 'DNM2': 5,\n",
       " 'DNMT3A': 10,\n",
       " 'DOT1L': 7,\n",
       " 'ECT2L': 4,\n",
       " 'EGFR': 18,\n",
       " 'ELL2': 3,\n",
       " 'EP300': 14,\n",
       " 'ERBB2': 1,\n",
       " 'ERBB4': 3,\n",
       " 'ERCC2': 2,\n",
       " 'ERCC3': 5,\n",
       " 'ERCC4': 2,\n",
       " 'EXT1': 5,\n",
       " 'EXT2': 6,\n",
       " 'EZH1': 1,\n",
       " 'FAM166A': 1,\n",
       " 'FAM46C': 1,\n",
       " 'FANCA': 1,\n",
       " 'FANCC': 3,\n",
       " 'FANCD2': 3,\n",
       " 'FANCE': 1,\n",
       " 'FANCG': 2,\n",
       " 'FANCM': 17,\n",
       " 'FAS': 2,\n",
       " 'FAT1': 32,\n",
       " 'FBXO11': 2,\n",
       " 'FBXW7': 6,\n",
       " 'FGFBP1': 1,\n",
       " 'FGFR3': 1,\n",
       " 'FH': 3,\n",
       " 'FLT4': 4,\n",
       " 'FUBP1': 4,\n",
       " 'FYN': 1,\n",
       " 'GATA3': 7,\n",
       " 'GNAS': 2,\n",
       " 'GNPTAB': 1,\n",
       " 'GPS2': 1,\n",
       " 'GRIN2A': 18,\n",
       " 'HCN1': 2,\n",
       " 'HIST1H4E': 8,\n",
       " 'HNF1A': 2,\n",
       " 'HRAS': 1,\n",
       " 'IDH1': 1,\n",
       " 'IKBKB': 1,\n",
       " 'IKZF1': 12,\n",
       " 'IRF6': 1,\n",
       " 'ITGBL1': 4,\n",
       " 'JAK2': 1,\n",
       " 'KDM5C': 8,\n",
       " 'KDM6A': 6,\n",
       " 'KEAP1': 38,\n",
       " 'KLF6': 3,\n",
       " 'KLHL5': 3,\n",
       " 'KMT2B': 8,\n",
       " 'KMT2C': 25,\n",
       " 'KMT2D': 34,\n",
       " 'KRAS': 85,\n",
       " 'LAMA4': 1,\n",
       " 'LATS1': 5,\n",
       " 'LSM14A': 1,\n",
       " 'MAP2K4': 2,\n",
       " 'MAP3K1': 6,\n",
       " 'MAP3K13': 2,\n",
       " 'MBD1': 6,\n",
       " 'MEN1': 5,\n",
       " 'MET': 5,\n",
       " 'MGA': 25,\n",
       " 'MLH1': 3,\n",
       " 'MSH2': 1,\n",
       " 'MSH6': 2,\n",
       " 'MTOR': 8,\n",
       " 'MUTYH': 1,\n",
       " 'MYH2': 3,\n",
       " 'NBN': 5,\n",
       " 'NCOA6': 6,\n",
       " 'NCOR1': 17,\n",
       " 'NF1': 30,\n",
       " 'NF2': 4,\n",
       " 'NFE2L2': 17,\n",
       " 'NOTCH1': 10,\n",
       " 'NOTCH2': 9,\n",
       " 'NRAS': 3,\n",
       " 'NRXN1': 6,\n",
       " 'ODAM': 1,\n",
       " 'PASK': 3,\n",
       " 'PATZ1': 3,\n",
       " 'PBRM1': 4,\n",
       " 'PDGFRB': 1,\n",
       " 'PDYN': 1,\n",
       " 'PHF6': 1,\n",
       " 'PHOX2B': 4,\n",
       " 'PIK3CA': 25,\n",
       " 'PIK3R1': 4,\n",
       " 'PLCG2': 6,\n",
       " 'PLXNB2': 14,\n",
       " 'PMS1': 4,\n",
       " 'PMS2': 3,\n",
       " 'POLE': 8,\n",
       " 'POT1': 9,\n",
       " 'PPP3CA': 6,\n",
       " 'PRDM1': 2,\n",
       " 'PRF1': 3,\n",
       " 'PRKCA': 1,\n",
       " 'PTCH1': 5,\n",
       " 'PTEN': 13,\n",
       " 'PTK2B': 1,\n",
       " 'PTPN11': 1,\n",
       " 'PTPRB': 13,\n",
       " 'PTPRC': 13,\n",
       " 'PTPRU': 3,\n",
       " 'RAD21': 5,\n",
       " 'RASA1': 13,\n",
       " 'RB1': 17,\n",
       " 'RBM10': 26,\n",
       " 'RECQL4': 1,\n",
       " 'RIMS2': 1,\n",
       " 'RIT1': 3,\n",
       " 'RNF43': 4,\n",
       " 'RPS2': 1,\n",
       " 'RYR2': 7,\n",
       " 'SDHAF2': 2,\n",
       " 'SDHB': 1,\n",
       " 'SDHC': 1,\n",
       " 'SEPT12': 2,\n",
       " 'SERPINB13': 4,\n",
       " 'SETD2': 15,\n",
       " 'SETDB1': 9,\n",
       " 'SF3B1': 3,\n",
       " 'SGK223': 6,\n",
       " 'SH2B3': 2,\n",
       " 'SLC4A5': 6,\n",
       " 'SMAD4': 13,\n",
       " 'SMARCA4': 23,\n",
       " 'SMARCB1': 5,\n",
       " 'SMARCC2': 1,\n",
       " 'SMARCD1': 2,\n",
       " 'SMC3': 4,\n",
       " 'SNX29': 1,\n",
       " 'SOCS1': 1,\n",
       " 'SPEN': 10,\n",
       " 'STAG2': 8,\n",
       " 'STAT3': 1,\n",
       " 'STIM1': 1,\n",
       " 'STK11': 39,\n",
       " 'STRN': 1,\n",
       " 'SUFU': 1,\n",
       " 'TBL1XR1': 4,\n",
       " 'TENM4': 1,\n",
       " 'TERT': 1,\n",
       " 'TET2': 13,\n",
       " 'TIAM1': 3,\n",
       " 'TIMM17A': 3,\n",
       " 'TNFAIP3': 2,\n",
       " 'TNKS': 1,\n",
       " 'TNR': 3,\n",
       " 'TP53': 200,\n",
       " 'TP53BP1': 6,\n",
       " 'TSC1': 3,\n",
       " 'TSC2': 10,\n",
       " 'TSHR': 1,\n",
       " 'U2AF1': 3,\n",
       " 'UBR5': 16,\n",
       " 'VHL': 1,\n",
       " 'WAS': 3,\n",
       " 'WRN': 6,\n",
       " 'WT1': 2,\n",
       " 'XPA': 1,\n",
       " 'XPC': 4,\n",
       " 'ZFP36L1': 9,\n",
       " 'ZMYM2': 9,\n",
       " 'ZNF180': 1,\n",
       " 'ZRSR2': 1}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_recurrence_gene_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make a dictionary of Hugo_Symbol and the number of times it appears in the recurrence_mut_table where patient_id is in LUAD_patient_id\n",
    "LUAD_recurrence_gene_count = {}\n",
    "for index, row in recurrence_mut_table.iterrows():\n",
    "    if row['patient_id'] in LUAD_patient_id:\n",
    "        if row['Hugo_Symbol'] in LUAD_recurrence_gene_count:\n",
    "            LUAD_recurrence_gene_count[row['Hugo_Symbol']] += 1\n",
    "        else:\n",
    "            LUAD_recurrence_gene_count[row['Hugo_Symbol']] = 1\n",
    "LUAD_recurrence_gene_count = dict(sorted(LUAD_recurrence_gene_count.items(), key=lambda item: item[1], reverse=True))\n",
    "\n",
    "# make a dictionary of Hugo_Symbol and the number of times it appears in the no_recurrence_mut_table where patient_id is in LUAD_patient_id\n",
    "LUAD_no_recurrence_gene_count = {}\n",
    "for index, row in no_recurrence_mut_table.iterrows():\n",
    "    if row['patient_id'] in LUAD_patient_id:\n",
    "        if row['Hugo_Symbol'] in LUAD_no_recurrence_gene_count:\n",
    "            LUAD_no_recurrence_gene_count[row['Hugo_Symbol']] += 1\n",
    "        else:\n",
    "            LUAD_no_recurrence_gene_count[row['Hugo_Symbol']] = 1\n",
    "LUAD_no_recurrence_gene_count = dict(sorted(LUAD_no_recurrence_gene_count.items(), key=lambda item: item[1], reverse=True))                            "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make a dictionary of Hugo_Symbol and the number of times it appears in the recurrence_mut_table where patient_id is in LUSC_patient_id\n",
    "LUSC_recurrence_gene_count = {}\n",
    "for index, row in recurrence_mut_table.iterrows():\n",
    "    if row['patient_id'] in LUSC_patient_id:\n",
    "        if row['Hugo_Symbol'] in LUSC_recurrence_gene_count:\n",
    "            LUSC_recurrence_gene_count[row['Hugo_Symbol']] += 1\n",
    "        else:\n",
    "            LUSC_recurrence_gene_count[row['Hugo_Symbol']] = 1\n",
    "LUSC_recurrence_gene_count = dict(sorted(LUSC_recurrence_gene_count.items(), key=lambda item: item[1], reverse=True))\n",
    "\n",
    "# make a dictionary of Hugo_Symbol and the number of times it appears in the no_recurrence_mut_table where patient_id is in LUSC_patient_id\n",
    "LUSC_no_recurrence_gene_count = {}\n",
    "for index, row in no_recurrence_mut_table.iterrows():\n",
    "    if row['patient_id'] in LUSC_patient_id:\n",
    "        if row['Hugo_Symbol'] in LUSC_no_recurrence_gene_count:\n",
    "            LUSC_no_recurrence_gene_count[row['Hugo_Symbol']] += 1\n",
    "        else:\n",
    "            LUSC_no_recurrence_gene_count[row['Hugo_Symbol']] = 1\n",
    "LUSC_no_recurrence_gene_count = dict(sorted(LUSC_no_recurrence_gene_count.items(), key=lambda item: item[1], reverse=True))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# normalize the values in the dictionaries by dividing by the number of patients in each group\n",
    "recurrence_gene_count = {k: v / len(recurrence_patient_id) for k, v in recurrence_gene_count.items()}\n",
    "no_recurrence_gene_count = {k: v / len(no_recurrence_patient_id) for k, v in no_recurrence_gene_count.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "recurrence_patient_id=[item for item in recurrence_patient_id.values.tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "no_recurrence_patient_id=[item for item in no_recurrence_patient_id.values.tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "LUAD_recurrence_patient_id = [patient_id for patient_id in LUAD_patient_id if patient_id in recurrence_patient_id]\n",
    "LUAD_no_recurrence_patient_id = [patient_id for patient_id in LUAD_patient_id if patient_id in no_recurrence_patient_id]\n",
    "LUSC_recurrence_patient_id = [patient_id for patient_id in LUSC_patient_id if patient_id in recurrence_patient_id]\n",
    "LUSC_no_recurrence_patient_id = [patient_id for patient_id in LUSC_patient_id if patient_id in no_recurrence_patient_id]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# normalize the values in the dictionaries by dividing by the number of patients in each group\n",
    "LUAD_recurrence_gene_count = {k: v / len(LUAD_recurrence_patient_id) for k, v in LUAD_recurrence_gene_count.items()}\n",
    "LUAD_no_recurrence_gene_count = {k: v / len(LUAD_no_recurrence_patient_id) for k, v in LUAD_no_recurrence_gene_count.items()}\n",
    "LUSC_recurrence_gene_count = {k: v / len(LUSC_recurrence_patient_id) for k, v in LUSC_recurrence_gene_count.items()}\n",
    "LUSC_no_recurrence_gene_count = {k: v / len(LUSC_no_recurrence_patient_id) for k, v in LUSC_no_recurrence_gene_count.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ACVR1B': 0.02112676056338028,\n",
       " 'AMER1': 0.014084507042253521,\n",
       " 'ANAPC5': 0.007042253521126761,\n",
       " 'ANK2': 0.02112676056338028,\n",
       " 'APC': 0.04225352112676056,\n",
       " 'ARAF': 0.007042253521126761,\n",
       " 'ARHGAP35': 0.056338028169014086,\n",
       " 'ARHGEF12': 0.007042253521126761,\n",
       " 'ARID1A': 0.07746478873239436,\n",
       " 'ARID1B': 0.056338028169014086,\n",
       " 'ARID2': 0.09154929577464789,\n",
       " 'ASXL1': 0.04225352112676056,\n",
       " 'ASXL2': 0.014084507042253521,\n",
       " 'ATM': 0.07042253521126761,\n",
       " 'ATR': 0.02112676056338028,\n",
       " 'ATRX': 0.056338028169014086,\n",
       " 'AXIN1': 0.028169014084507043,\n",
       " 'AXIN2': 0.014084507042253521,\n",
       " 'B2M': 0.06338028169014084,\n",
       " 'BAP1': 0.035211267605633804,\n",
       " 'BCLAF1': 0.04225352112676056,\n",
       " 'BCOR': 0.04929577464788732,\n",
       " 'BHMT2': 0.007042253521126761,\n",
       " 'BIVM-ERCC5': 0.007042253521126761,\n",
       " 'BLM': 0.014084507042253521,\n",
       " 'BMPR1A': 0.02112676056338028,\n",
       " 'BRAF': 0.028169014084507043,\n",
       " 'BRCA1': 0.028169014084507043,\n",
       " 'BRCA2': 0.007042253521126761,\n",
       " 'BRIP1': 0.007042253521126761,\n",
       " 'BUB1B': 0.028169014084507043,\n",
       " 'CAD': 0.007042253521126761,\n",
       " 'CASP8': 0.014084507042253521,\n",
       " 'CBL': 0.02112676056338028,\n",
       " 'CBLB': 0.035211267605633804,\n",
       " 'CDC73': 0.028169014084507043,\n",
       " 'CDK12': 0.028169014084507043,\n",
       " 'CDKN2A': 0.09859154929577464,\n",
       " 'CEP76': 0.007042253521126761,\n",
       " 'CHD4': 0.04929577464788732,\n",
       " 'CHD8': 0.028169014084507043,\n",
       " 'CHEK2': 0.014084507042253521,\n",
       " 'CIC': 0.04225352112676056,\n",
       " 'CMTR2': 0.02112676056338028,\n",
       " 'CNKSR1': 0.007042253521126761,\n",
       " 'CNOT3': 0.02112676056338028,\n",
       " 'COL2A1': 0.04225352112676056,\n",
       " 'COL5A2': 0.07042253521126761,\n",
       " 'CREBBP': 0.09154929577464789,\n",
       " 'CSMD3': 0.014084507042253521,\n",
       " 'CTCF': 0.014084507042253521,\n",
       " 'CTNNB1': 0.04929577464788732,\n",
       " 'CUL3': 0.02112676056338028,\n",
       " 'CUX1': 0.056338028169014086,\n",
       " 'CYLD': 0.028169014084507043,\n",
       " 'DDB2': 0.007042253521126761,\n",
       " 'DDX3X': 0.035211267605633804,\n",
       " 'DICER1': 0.028169014084507043,\n",
       " 'DNAH12': 0.028169014084507043,\n",
       " 'DNM2': 0.02112676056338028,\n",
       " 'DNMT3A': 0.014084507042253521,\n",
       " 'DOT1L': 0.035211267605633804,\n",
       " 'EGFR': 0.07042253521126761,\n",
       " 'ELL2': 0.007042253521126761,\n",
       " 'EP300': 0.035211267605633804,\n",
       " 'EPB41L3': 0.007042253521126761,\n",
       " 'ERBB2': 0.014084507042253521,\n",
       " 'ERBB4': 0.007042253521126761,\n",
       " 'ERCC2': 0.007042253521126761,\n",
       " 'ERCC3': 0.007042253521126761,\n",
       " 'ERCC4': 0.007042253521126761,\n",
       " 'EXT1': 0.02112676056338028,\n",
       " 'EZH1': 0.028169014084507043,\n",
       " 'FAM166A': 0.014084507042253521,\n",
       " 'FAM46C': 0.007042253521126761,\n",
       " 'FANCA': 0.007042253521126761,\n",
       " 'FANCD2': 0.007042253521126761,\n",
       " 'FANCF': 0.007042253521126761,\n",
       " 'FANCM': 0.035211267605633804,\n",
       " 'FAS': 0.02112676056338028,\n",
       " 'FAT1': 0.1267605633802817,\n",
       " 'FBXO11': 0.007042253521126761,\n",
       " 'FBXW7': 0.035211267605633804,\n",
       " 'FGFBP1': 0.007042253521126761,\n",
       " 'FGFR1': 0.007042253521126761,\n",
       " 'FGFR3': 0.014084507042253521,\n",
       " 'FH': 0.007042253521126761,\n",
       " 'FLCN': 0.028169014084507043,\n",
       " 'FLT4': 0.035211267605633804,\n",
       " 'FN1': 0.014084507042253521,\n",
       " 'FUBP1': 0.014084507042253521,\n",
       " 'GATA3': 0.02112676056338028,\n",
       " 'GNPTAB': 0.014084507042253521,\n",
       " 'GPS2': 0.007042253521126761,\n",
       " 'GRIN2A': 0.04929577464788732,\n",
       " 'HCN1': 0.02112676056338028,\n",
       " 'HNF1A': 0.02112676056338028,\n",
       " 'HOXB3': 0.007042253521126761,\n",
       " 'HRAS': 0.007042253521126761,\n",
       " 'IKZF1': 0.028169014084507043,\n",
       " 'IL21R': 0.007042253521126761,\n",
       " 'IQGAP2': 0.007042253521126761,\n",
       " 'ITGBL1': 0.007042253521126761,\n",
       " 'KDM5C': 0.056338028169014086,\n",
       " 'KDM6A': 0.028169014084507043,\n",
       " 'KEAP1': 0.15492957746478872,\n",
       " 'KLF5': 0.014084507042253521,\n",
       " 'KLF6': 0.007042253521126761,\n",
       " 'KMT2B': 0.035211267605633804,\n",
       " 'KMT2C': 0.04929577464788732,\n",
       " 'KMT2D': 0.21830985915492956,\n",
       " 'KRAS': 0.31690140845070425,\n",
       " 'LAMA4': 0.007042253521126761,\n",
       " 'LATS1': 0.028169014084507043,\n",
       " 'MAP3K1': 0.02112676056338028,\n",
       " 'MAP3K13': 0.035211267605633804,\n",
       " 'MAX': 0.007042253521126761,\n",
       " 'MBD1': 0.007042253521126761,\n",
       " 'MEN1': 0.014084507042253521,\n",
       " 'MGA': 0.04929577464788732,\n",
       " 'MLH1': 0.007042253521126761,\n",
       " 'MSH2': 0.02112676056338028,\n",
       " 'MTOR': 0.06338028169014084,\n",
       " 'MUTYH': 0.007042253521126761,\n",
       " 'MYH2': 0.007042253521126761,\n",
       " 'NBN': 0.035211267605633804,\n",
       " 'NCOA6': 0.04225352112676056,\n",
       " 'NCOR1': 0.056338028169014086,\n",
       " 'NEB': 0.007042253521126761,\n",
       " 'NF1': 0.1267605633802817,\n",
       " 'NF2': 0.007042253521126761,\n",
       " 'NFE2L2': 0.04225352112676056,\n",
       " 'NFKBIE': 0.014084507042253521,\n",
       " 'NIN': 0.007042253521126761,\n",
       " 'NOTCH1': 0.04929577464788732,\n",
       " 'NOTCH2': 0.035211267605633804,\n",
       " 'NRAS': 0.02112676056338028,\n",
       " 'NRXN1': 0.007042253521126761,\n",
       " 'NSD1': 0.007042253521126761,\n",
       " 'NTRK1': 0.007042253521126761,\n",
       " 'PALB2': 0.014084507042253521,\n",
       " 'PASK': 0.02112676056338028,\n",
       " 'PATZ1': 0.007042253521126761,\n",
       " 'PBRM1': 0.04929577464788732,\n",
       " 'PDYN': 0.007042253521126761,\n",
       " 'PHOX2B': 0.007042253521126761,\n",
       " 'PIK3CA': 0.1267605633802817,\n",
       " 'PIK3CG': 0.007042253521126761,\n",
       " 'PIK3R1': 0.028169014084507043,\n",
       " 'PLCG2': 0.007042253521126761,\n",
       " 'PLXNB2': 0.06338028169014084,\n",
       " 'PMS1': 0.014084507042253521,\n",
       " 'POLE': 0.04225352112676056,\n",
       " 'POT1': 0.035211267605633804,\n",
       " 'PPP3CA': 0.02112676056338028,\n",
       " 'PRDM1': 0.02112676056338028,\n",
       " 'PRF1': 0.014084507042253521,\n",
       " 'PRKAR1A': 0.007042253521126761,\n",
       " 'PRRX1': 0.007042253521126761,\n",
       " 'PTCH1': 0.007042253521126761,\n",
       " 'PTEN': 0.08450704225352113,\n",
       " 'PTPRB': 0.08450704225352113,\n",
       " 'PTPRC': 0.056338028169014086,\n",
       " 'PTPRU': 0.028169014084507043,\n",
       " 'RAD21': 0.02112676056338028,\n",
       " 'RASA1': 0.056338028169014086,\n",
       " 'RB1': 0.04225352112676056,\n",
       " 'RBM10': 0.09154929577464789,\n",
       " 'RIMS2': 0.007042253521126761,\n",
       " 'RIT1': 0.014084507042253521,\n",
       " 'RNF43': 0.028169014084507043,\n",
       " 'RYR2': 0.028169014084507043,\n",
       " 'SEPT12': 0.007042253521126761,\n",
       " 'SERPINB13': 0.014084507042253521,\n",
       " 'SETD2': 0.07746478873239436,\n",
       " 'SETDB1': 0.014084507042253521,\n",
       " 'SF3B1': 0.014084507042253521,\n",
       " 'SGK223': 0.04225352112676056,\n",
       " 'SH2B3': 0.007042253521126761,\n",
       " 'SMAD4': 0.04929577464788732,\n",
       " 'SMARCA4': 0.176056338028169,\n",
       " 'SMARCC2': 0.007042253521126761,\n",
       " 'SNX29': 0.035211267605633804,\n",
       " 'SPEN': 0.04929577464788732,\n",
       " 'STAG2': 0.02112676056338028,\n",
       " 'STIM1': 0.014084507042253521,\n",
       " 'STK11': 0.18309859154929578,\n",
       " 'STX2': 0.014084507042253521,\n",
       " 'TBL1XR1': 0.007042253521126761,\n",
       " 'TET2': 0.02112676056338028,\n",
       " 'TNC': 0.007042253521126761,\n",
       " 'TNFAIP3': 0.014084507042253521,\n",
       " 'TNFRSF14': 0.007042253521126761,\n",
       " 'TNR': 0.007042253521126761,\n",
       " 'TP53': 0.647887323943662,\n",
       " 'TP53BP1': 0.028169014084507043,\n",
       " 'TSC1': 0.007042253521126761,\n",
       " 'TSC2': 0.014084507042253521,\n",
       " 'U2AF1': 0.02112676056338028,\n",
       " 'UBR5': 0.056338028169014086,\n",
       " 'WAS': 0.02112676056338028,\n",
       " 'WRN': 0.007042253521126761,\n",
       " 'WT1': 0.04225352112676056,\n",
       " 'ZFP36L1': 0.02112676056338028,\n",
       " 'ZFPM2': 0.007042253521126761,\n",
       " 'ZMYM2': 0.04225352112676056,\n",
       " 'ZNF180': 0.007042253521126761,\n",
       " 'ZNF483': 0.007042253521126761}"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "recurrence_gene_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ACTN2': 0.0035842293906810036,\n",
       " 'ACVR1B': 0.025089605734767026,\n",
       " 'ADCY2': 0.007168458781362007,\n",
       " 'AKT1': 0.0035842293906810036,\n",
       " 'AMER1': 0.02867383512544803,\n",
       " 'ANK2': 0.014336917562724014,\n",
       " 'APC': 0.05734767025089606,\n",
       " 'ARHGAP35': 0.03942652329749104,\n",
       " 'ARID1A': 0.06810035842293907,\n",
       " 'ARID1B': 0.03225806451612903,\n",
       " 'ARID2': 0.04659498207885305,\n",
       " 'ASXL1': 0.010752688172043012,\n",
       " 'ASXL2': 0.03225806451612903,\n",
       " 'ATF7IP': 0.021505376344086023,\n",
       " 'ATM': 0.07168458781362007,\n",
       " 'ATP5B': 0.007168458781362007,\n",
       " 'ATR': 0.035842293906810034,\n",
       " 'ATRX': 0.07168458781362007,\n",
       " 'AXIN1': 0.014336917562724014,\n",
       " 'AXIN2': 0.010752688172043012,\n",
       " 'B2M': 0.021505376344086023,\n",
       " 'BAP1': 0.021505376344086023,\n",
       " 'BCLAF1': 0.053763440860215055,\n",
       " 'BCOR': 0.03942652329749104,\n",
       " 'BHMT2': 0.007168458781362007,\n",
       " 'BIVM-ERCC5': 0.014336917562724014,\n",
       " 'BLM': 0.007168458781362007,\n",
       " 'BMPR1A': 0.007168458781362007,\n",
       " 'BRAF': 0.02867383512544803,\n",
       " 'BRCA1': 0.010752688172043012,\n",
       " 'BRCA2': 0.017921146953405017,\n",
       " 'BRE': 0.007168458781362007,\n",
       " 'BRIP1': 0.017921146953405017,\n",
       " 'BUB1B': 0.007168458781362007,\n",
       " 'CAD': 0.0035842293906810036,\n",
       " 'CASP8': 0.014336917562724014,\n",
       " 'CBL': 0.021505376344086023,\n",
       " 'CBLB': 0.035842293906810034,\n",
       " 'CDC73': 0.017921146953405017,\n",
       " 'CDH1': 0.014336917562724014,\n",
       " 'CDH18': 0.0035842293906810036,\n",
       " 'CDK12': 0.02867383512544803,\n",
       " 'CDKN2A': 0.12186379928315412,\n",
       " 'CDKN2C': 0.007168458781362007,\n",
       " 'CEP76': 0.007168458781362007,\n",
       " 'CHD4': 0.035842293906810034,\n",
       " 'CHD8': 0.043010752688172046,\n",
       " 'CHEK2': 0.017921146953405017,\n",
       " 'CIC': 0.021505376344086023,\n",
       " 'CMTR2': 0.043010752688172046,\n",
       " 'CNBD1': 0.021505376344086023,\n",
       " 'CNKSR1': 0.010752688172043012,\n",
       " 'CNOT3': 0.021505376344086023,\n",
       " 'COL2A1': 0.04659498207885305,\n",
       " 'COL5A2': 0.07168458781362007,\n",
       " 'CREBBP': 0.06451612903225806,\n",
       " 'CSMD3': 0.010752688172043012,\n",
       " 'CTCF': 0.010752688172043012,\n",
       " 'CTNNB1': 0.010752688172043012,\n",
       " 'CUL3': 0.021505376344086023,\n",
       " 'CUX1': 0.053763440860215055,\n",
       " 'CYLD': 0.014336917562724014,\n",
       " 'DARC': 0.0035842293906810036,\n",
       " 'DDX3X': 0.021505376344086023,\n",
       " 'DICER1': 0.035842293906810034,\n",
       " 'DNAH12': 0.043010752688172046,\n",
       " 'DNM2': 0.017921146953405017,\n",
       " 'DNMT3A': 0.035842293906810034,\n",
       " 'DOT1L': 0.025089605734767026,\n",
       " 'ECT2L': 0.014336917562724014,\n",
       " 'EGFR': 0.06451612903225806,\n",
       " 'ELL2': 0.010752688172043012,\n",
       " 'EP300': 0.05017921146953405,\n",
       " 'ERBB2': 0.0035842293906810036,\n",
       " 'ERBB4': 0.010752688172043012,\n",
       " 'ERCC2': 0.007168458781362007,\n",
       " 'ERCC3': 0.017921146953405017,\n",
       " 'ERCC4': 0.007168458781362007,\n",
       " 'EXT1': 0.017921146953405017,\n",
       " 'EXT2': 0.021505376344086023,\n",
       " 'EZH1': 0.0035842293906810036,\n",
       " 'FAM166A': 0.0035842293906810036,\n",
       " 'FAM46C': 0.0035842293906810036,\n",
       " 'FANCA': 0.0035842293906810036,\n",
       " 'FANCC': 0.010752688172043012,\n",
       " 'FANCD2': 0.010752688172043012,\n",
       " 'FANCE': 0.0035842293906810036,\n",
       " 'FANCG': 0.007168458781362007,\n",
       " 'FANCM': 0.06093189964157706,\n",
       " 'FAS': 0.007168458781362007,\n",
       " 'FAT1': 0.11469534050179211,\n",
       " 'FBXO11': 0.007168458781362007,\n",
       " 'FBXW7': 0.021505376344086023,\n",
       " 'FGFBP1': 0.0035842293906810036,\n",
       " 'FGFR3': 0.0035842293906810036,\n",
       " 'FH': 0.010752688172043012,\n",
       " 'FLT4': 0.014336917562724014,\n",
       " 'FUBP1': 0.014336917562724014,\n",
       " 'FYN': 0.0035842293906810036,\n",
       " 'GATA3': 0.025089605734767026,\n",
       " 'GNAS': 0.007168458781362007,\n",
       " 'GNPTAB': 0.0035842293906810036,\n",
       " 'GPS2': 0.0035842293906810036,\n",
       " 'GRIN2A': 0.06451612903225806,\n",
       " 'HCN1': 0.007168458781362007,\n",
       " 'HIST1H4E': 0.02867383512544803,\n",
       " 'HNF1A': 0.007168458781362007,\n",
       " 'HRAS': 0.0035842293906810036,\n",
       " 'IDH1': 0.0035842293906810036,\n",
       " 'IKBKB': 0.0035842293906810036,\n",
       " 'IKZF1': 0.043010752688172046,\n",
       " 'IRF6': 0.0035842293906810036,\n",
       " 'ITGBL1': 0.014336917562724014,\n",
       " 'JAK2': 0.0035842293906810036,\n",
       " 'KDM5C': 0.02867383512544803,\n",
       " 'KDM6A': 0.021505376344086023,\n",
       " 'KEAP1': 0.13620071684587814,\n",
       " 'KLF6': 0.010752688172043012,\n",
       " 'KLHL5': 0.010752688172043012,\n",
       " 'KMT2B': 0.02867383512544803,\n",
       " 'KMT2C': 0.08960573476702509,\n",
       " 'KMT2D': 0.12186379928315412,\n",
       " 'KRAS': 0.3046594982078853,\n",
       " 'LAMA4': 0.0035842293906810036,\n",
       " 'LATS1': 0.017921146953405017,\n",
       " 'LSM14A': 0.0035842293906810036,\n",
       " 'MAP2K4': 0.007168458781362007,\n",
       " 'MAP3K1': 0.021505376344086023,\n",
       " 'MAP3K13': 0.007168458781362007,\n",
       " 'MBD1': 0.021505376344086023,\n",
       " 'MEN1': 0.017921146953405017,\n",
       " 'MET': 0.017921146953405017,\n",
       " 'MGA': 0.08960573476702509,\n",
       " 'MLH1': 0.010752688172043012,\n",
       " 'MSH2': 0.0035842293906810036,\n",
       " 'MSH6': 0.007168458781362007,\n",
       " 'MTOR': 0.02867383512544803,\n",
       " 'MUTYH': 0.0035842293906810036,\n",
       " 'MYH2': 0.010752688172043012,\n",
       " 'NBN': 0.017921146953405017,\n",
       " 'NCOA6': 0.021505376344086023,\n",
       " 'NCOR1': 0.06093189964157706,\n",
       " 'NF1': 0.10752688172043011,\n",
       " 'NF2': 0.014336917562724014,\n",
       " 'NFE2L2': 0.06093189964157706,\n",
       " 'NOTCH1': 0.035842293906810034,\n",
       " 'NOTCH2': 0.03225806451612903,\n",
       " 'NRAS': 0.010752688172043012,\n",
       " 'NRXN1': 0.021505376344086023,\n",
       " 'ODAM': 0.0035842293906810036,\n",
       " 'PASK': 0.010752688172043012,\n",
       " 'PATZ1': 0.010752688172043012,\n",
       " 'PBRM1': 0.014336917562724014,\n",
       " 'PDGFRB': 0.0035842293906810036,\n",
       " 'PDYN': 0.0035842293906810036,\n",
       " 'PHF6': 0.0035842293906810036,\n",
       " 'PHOX2B': 0.014336917562724014,\n",
       " 'PIK3CA': 0.08960573476702509,\n",
       " 'PIK3R1': 0.014336917562724014,\n",
       " 'PLCG2': 0.021505376344086023,\n",
       " 'PLXNB2': 0.05017921146953405,\n",
       " 'PMS1': 0.014336917562724014,\n",
       " 'PMS2': 0.010752688172043012,\n",
       " 'POLE': 0.02867383512544803,\n",
       " 'POT1': 0.03225806451612903,\n",
       " 'PPP3CA': 0.021505376344086023,\n",
       " 'PRDM1': 0.007168458781362007,\n",
       " 'PRF1': 0.010752688172043012,\n",
       " 'PRKCA': 0.0035842293906810036,\n",
       " 'PTCH1': 0.017921146953405017,\n",
       " 'PTEN': 0.04659498207885305,\n",
       " 'PTK2B': 0.0035842293906810036,\n",
       " 'PTPN11': 0.0035842293906810036,\n",
       " 'PTPRB': 0.04659498207885305,\n",
       " 'PTPRC': 0.04659498207885305,\n",
       " 'PTPRU': 0.010752688172043012,\n",
       " 'RAD21': 0.017921146953405017,\n",
       " 'RASA1': 0.04659498207885305,\n",
       " 'RB1': 0.06093189964157706,\n",
       " 'RBM10': 0.0931899641577061,\n",
       " 'RECQL4': 0.0035842293906810036,\n",
       " 'RIMS2': 0.0035842293906810036,\n",
       " 'RIT1': 0.010752688172043012,\n",
       " 'RNF43': 0.014336917562724014,\n",
       " 'RPS2': 0.0035842293906810036,\n",
       " 'RYR2': 0.025089605734767026,\n",
       " 'SDHAF2': 0.007168458781362007,\n",
       " 'SDHB': 0.0035842293906810036,\n",
       " 'SDHC': 0.0035842293906810036,\n",
       " 'SEPT12': 0.007168458781362007,\n",
       " 'SERPINB13': 0.014336917562724014,\n",
       " 'SETD2': 0.053763440860215055,\n",
       " 'SETDB1': 0.03225806451612903,\n",
       " 'SF3B1': 0.010752688172043012,\n",
       " 'SGK223': 0.021505376344086023,\n",
       " 'SH2B3': 0.007168458781362007,\n",
       " 'SLC4A5': 0.021505376344086023,\n",
       " 'SMAD4': 0.04659498207885305,\n",
       " 'SMARCA4': 0.08243727598566308,\n",
       " 'SMARCB1': 0.017921146953405017,\n",
       " 'SMARCC2': 0.0035842293906810036,\n",
       " 'SMARCD1': 0.007168458781362007,\n",
       " 'SMC3': 0.014336917562724014,\n",
       " 'SNX29': 0.0035842293906810036,\n",
       " 'SOCS1': 0.0035842293906810036,\n",
       " 'SPEN': 0.035842293906810034,\n",
       " 'STAG2': 0.02867383512544803,\n",
       " 'STAT3': 0.0035842293906810036,\n",
       " 'STIM1': 0.0035842293906810036,\n",
       " 'STK11': 0.13978494623655913,\n",
       " 'STRN': 0.0035842293906810036,\n",
       " 'SUFU': 0.0035842293906810036,\n",
       " 'TBL1XR1': 0.014336917562724014,\n",
       " 'TENM4': 0.0035842293906810036,\n",
       " 'TERT': 0.0035842293906810036,\n",
       " 'TET2': 0.04659498207885305,\n",
       " 'TIAM1': 0.010752688172043012,\n",
       " 'TIMM17A': 0.010752688172043012,\n",
       " 'TNFAIP3': 0.007168458781362007,\n",
       " 'TNKS': 0.0035842293906810036,\n",
       " 'TNR': 0.010752688172043012,\n",
       " 'TP53': 0.7168458781362007,\n",
       " 'TP53BP1': 0.021505376344086023,\n",
       " 'TSC1': 0.010752688172043012,\n",
       " 'TSC2': 0.035842293906810034,\n",
       " 'TSHR': 0.0035842293906810036,\n",
       " 'U2AF1': 0.010752688172043012,\n",
       " 'UBR5': 0.05734767025089606,\n",
       " 'VHL': 0.0035842293906810036,\n",
       " 'WAS': 0.010752688172043012,\n",
       " 'WRN': 0.021505376344086023,\n",
       " 'WT1': 0.007168458781362007,\n",
       " 'XPA': 0.0035842293906810036,\n",
       " 'XPC': 0.014336917562724014,\n",
       " 'ZFP36L1': 0.03225806451612903,\n",
       " 'ZMYM2': 0.03225806451612903,\n",
       " 'ZNF180': 0.0035842293906810036,\n",
       " 'ZRSR2': 0.0035842293906810036}"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_recurrence_gene_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'KMT2D': 0.09644605987177544,\n",
       " 'SMARCA4': 0.09361906204250592,\n",
       " 'ARID2': 0.04495431369579484,\n",
       " 'STK11': 0.043313645312736654,\n",
       " 'B2M': 0.04187490534605482,\n",
       " 'CTNNB1': 0.03854308647584431,\n",
       " 'PTEN': 0.037912060174668076,\n",
       " 'PTPRB': 0.037912060174668076,\n",
       " 'PIK3CA': 0.0371548286132566,\n",
       " 'WT1': 0.03508506234539856,\n",
       " 'PBRM1': 0.0349588570851633,\n",
       " 'MTOR': 0.034706446564692815,\n",
       " 'SNX29': 0.0316270382149528,\n",
       " 'ASXL1': 0.03150083295471755,\n",
       " 'FLCN': 0.028169014084507043,\n",
       " 'MAP3K13': 0.0280428088242718,\n",
       " 'KDM5C': 0.027664193043566057,\n",
       " 'CREBBP': 0.027033166742389828,\n",
       " 'EZH1': 0.02458478469382604,\n",
       " 'ARID1B': 0.024079963652885054,\n",
       " 'SETD2': 0.023701347872179306,\n",
       " 'BUB1B': 0.021000555303145034,\n",
       " 'FLT4': 0.02087435004290979,\n",
       " 'CIC': 0.02074814478267454,\n",
       " 'NCOA6': 0.02074814478267454,\n",
       " 'SGK223': 0.02074814478267454,\n",
       " 'NF1': 0.01923368165985158,\n",
       " 'KEAP1': 0.018728860618910576,\n",
       " 'MSH2': 0.01754253117269928,\n",
       " 'BRCA1': 0.01741632591246403,\n",
       " 'PTPRU': 0.01741632591246403,\n",
       " 'NBN': 0.017290120652228787,\n",
       " 'ARHGAP35': 0.01691150487152305,\n",
       " 'FN1': 0.014084507042253521,\n",
       " 'KLF5': 0.014084507042253521,\n",
       " 'NFKBIE': 0.014084507042253521,\n",
       " 'PALB2': 0.014084507042253521,\n",
       " 'STX2': 0.014084507042253521,\n",
       " 'BMPR1A': 0.013958301782018274,\n",
       " 'FAS': 0.013958301782018274,\n",
       " 'HCN1': 0.013958301782018274,\n",
       " 'HNF1A': 0.013958301782018274,\n",
       " 'PRDM1': 0.013958301782018274,\n",
       " 'AXIN1': 0.013832096521783029,\n",
       " 'CYLD': 0.013832096521783029,\n",
       " 'PIK3R1': 0.013832096521783029,\n",
       " 'RNF43': 0.013832096521783029,\n",
       " 'BAP1': 0.013705891261547781,\n",
       " 'DDX3X': 0.013705891261547781,\n",
       " 'FBXW7': 0.013705891261547781,\n",
       " 'POLE': 0.013579686001312534,\n",
       " 'CHD4': 0.013453480741077287,\n",
       " 'NOTCH1': 0.013453480741077287,\n",
       " 'SPEN': 0.013453480741077287,\n",
       " 'PLXNB2': 0.013201070220606792,\n",
       " 'KRAS': 0.012241910242818954,\n",
       " 'FAT1': 0.012065222878489573,\n",
       " 'ERBB2': 0.010500277651572517,\n",
       " 'FAM166A': 0.010500277651572517,\n",
       " 'FGFR3': 0.010500277651572517,\n",
       " 'GNPTAB': 0.010500277651572517,\n",
       " 'STIM1': 0.010500277651572517,\n",
       " 'NRAS': 0.01037407239133727,\n",
       " 'PASK': 0.01037407239133727,\n",
       " 'U2AF1': 0.01037407239133727,\n",
       " 'WAS': 0.01037407239133727,\n",
       " 'CDC73': 0.010247867131102026,\n",
       " 'LATS1': 0.010247867131102026,\n",
       " 'DOT1L': 0.010121661870866779,\n",
       " 'ZMYM2': 0.009995456610631531,\n",
       " 'BCOR': 0.009869251350396284,\n",
       " 'PTPRC': 0.009743046090161037,\n",
       " 'RASA1': 0.009743046090161037,\n",
       " 'ARID1A': 0.009364430309455288,\n",
       " 'ANAPC5': 0.007042253521126761,\n",
       " 'ARAF': 0.007042253521126761,\n",
       " 'ARHGEF12': 0.007042253521126761,\n",
       " 'DDB2': 0.007042253521126761,\n",
       " 'EPB41L3': 0.007042253521126761,\n",
       " 'FANCF': 0.007042253521126761,\n",
       " 'FGFR1': 0.007042253521126761,\n",
       " 'HOXB3': 0.007042253521126761,\n",
       " 'IL21R': 0.007042253521126761,\n",
       " 'IQGAP2': 0.007042253521126761,\n",
       " 'MAX': 0.007042253521126761,\n",
       " 'NEB': 0.007042253521126761,\n",
       " 'NIN': 0.007042253521126761,\n",
       " 'NSD1': 0.007042253521126761,\n",
       " 'NTRK1': 0.007042253521126761,\n",
       " 'PIK3CG': 0.007042253521126761,\n",
       " 'PRKAR1A': 0.007042253521126761,\n",
       " 'PRRX1': 0.007042253521126761,\n",
       " 'TNC': 0.007042253521126761,\n",
       " 'TNFRSF14': 0.007042253521126761,\n",
       " 'ZFPM2': 0.007042253521126761,\n",
       " 'ZNF483': 0.007042253521126761,\n",
       " 'BLM': 0.006916048260891514,\n",
       " 'TNFAIP3': 0.006916048260891514,\n",
       " 'ANK2': 0.006789843000656267,\n",
       " 'KDM6A': 0.00666363774042102,\n",
       " 'TP53BP1': 0.00666363774042102,\n",
       " 'KMT2B': 0.006537432480185776,\n",
       " 'EGFR': 0.005906406179009546,\n",
       " 'CAD': 0.003458024130445757,\n",
       " 'FAM46C': 0.003458024130445757,\n",
       " 'FANCA': 0.003458024130445757,\n",
       " 'FGFBP1': 0.003458024130445757,\n",
       " 'GPS2': 0.003458024130445757,\n",
       " 'HRAS': 0.003458024130445757,\n",
       " 'LAMA4': 0.003458024130445757,\n",
       " 'MUTYH': 0.003458024130445757,\n",
       " 'PDYN': 0.003458024130445757,\n",
       " 'RIMS2': 0.003458024130445757,\n",
       " 'SMARCC2': 0.003458024130445757,\n",
       " 'ZNF180': 0.003458024130445757,\n",
       " 'AXIN2': 0.00333181887021051,\n",
       " 'CSMD3': 0.00333181887021051,\n",
       " 'CTCF': 0.00333181887021051,\n",
       " 'PRF1': 0.00333181887021051,\n",
       " 'RIT1': 0.00333181887021051,\n",
       " 'SF3B1': 0.00333181887021051,\n",
       " 'DNM2': 0.0032056136099752643,\n",
       " 'EXT1': 0.0032056136099752643,\n",
       " 'RAD21': 0.0032056136099752643,\n",
       " 'RYR2': 0.003079408349740017,\n",
       " 'NOTCH2': 0.002953203089504773,\n",
       " 'POT1': 0.002953203089504773,\n",
       " 'SMAD4': 0.0027007925690342716,\n",
       " 'CUX1': 0.002574587308799031,\n",
       " 'BHMT2': -0.00012620526023524644,\n",
       " 'CEP76': -0.00012620526023524644,\n",
       " 'ERCC2': -0.00012620526023524644,\n",
       " 'ERCC4': -0.00012620526023524644,\n",
       " 'FBXO11': -0.00012620526023524644,\n",
       " 'SEPT12': -0.00012620526023524644,\n",
       " 'SH2B3': -0.00012620526023524644,\n",
       " 'CASP8': -0.0002524105204704929,\n",
       " 'FUBP1': -0.0002524105204704929,\n",
       " 'PMS1': -0.0002524105204704929,\n",
       " 'SERPINB13': -0.0002524105204704929,\n",
       " 'CBL': -0.0003786157807057419,\n",
       " 'CNOT3': -0.0003786157807057419,\n",
       " 'CUL3': -0.0003786157807057419,\n",
       " 'MAP3K1': -0.0003786157807057419,\n",
       " 'PPP3CA': -0.0003786157807057419,\n",
       " 'BRAF': -0.0005048210409409858,\n",
       " 'CDK12': -0.0005048210409409858,\n",
       " 'CBLB': -0.0006310263011762296,\n",
       " 'UBR5': -0.0010096420818819715,\n",
       " 'ATM': -0.0012620526023524592,\n",
       " 'COL5A2': -0.0012620526023524592,\n",
       " 'RBM10': -0.001640668383058208,\n",
       " 'CNKSR1': -0.003710434650916251,\n",
       " 'ELL2': -0.003710434650916251,\n",
       " 'ERBB4': -0.003710434650916251,\n",
       " 'FANCD2': -0.003710434650916251,\n",
       " 'FH': -0.003710434650916251,\n",
       " 'KLF6': -0.003710434650916251,\n",
       " 'MLH1': -0.003710434650916251,\n",
       " 'MYH2': -0.003710434650916251,\n",
       " 'PATZ1': -0.003710434650916251,\n",
       " 'TNR': -0.003710434650916251,\n",
       " 'TSC1': -0.003710434650916251,\n",
       " 'CHEK2': -0.0038366399111514956,\n",
       " 'MEN1': -0.0038366399111514956,\n",
       " 'ACVR1B': -0.003962845171386745,\n",
       " 'GATA3': -0.003962845171386745,\n",
       " 'COL2A1': -0.0043414609520924866,\n",
       " 'NCOR1': -0.004593871472562974,\n",
       " 'BIVM-ERCC5': -0.007294664041597254,\n",
       " 'ITGBL1': -0.007294664041597254,\n",
       " 'NF2': -0.007294664041597254,\n",
       " 'PHOX2B': -0.007294664041597254,\n",
       " 'TBL1XR1': -0.007294664041597254,\n",
       " 'STAG2': -0.007547074562067747,\n",
       " 'DICER1': -0.007673279822302991,\n",
       " 'BRCA2': -0.010878893432278255,\n",
       " 'BRIP1': -0.010878893432278255,\n",
       " 'ERCC3': -0.010878893432278255,\n",
       " 'PTCH1': -0.010878893432278255,\n",
       " 'ZFP36L1': -0.01113130395274875,\n",
       " 'BCLAF1': -0.011509919733454492,\n",
       " 'MBD1': -0.014463122822959262,\n",
       " 'NRXN1': -0.014463122822959262,\n",
       " 'PLCG2': -0.014463122822959262,\n",
       " 'WRN': -0.014463122822959262,\n",
       " 'AMER1': -0.014589328083194507,\n",
       " 'ATR': -0.014715533343429753,\n",
       " 'CHD8': -0.014841738603665004,\n",
       " 'DNAH12': -0.014841738603665004,\n",
       " 'IKZF1': -0.014841738603665004,\n",
       " 'EP300': -0.014967943863900247,\n",
       " 'APC': -0.015094149124135495,\n",
       " 'GRIN2A': -0.015220354384370742,\n",
       " 'ATRX': -0.015346559644605982,\n",
       " 'ASXL2': -0.018173557473875508,\n",
       " 'SETDB1': -0.018173557473875508,\n",
       " 'NFE2L2': -0.018678378514816497,\n",
       " 'RB1': -0.018678378514816497,\n",
       " 'DNMT3A': -0.02175778686455651,\n",
       " 'TSC2': -0.02175778686455651,\n",
       " 'CMTR2': -0.021883992124791765,\n",
       " 'CDKN2A': -0.02327224998737948,\n",
       " 'TET2': -0.025468221515472768,\n",
       " 'FANCM': -0.025720632035943256,\n",
       " 'KMT2C': -0.04030996011913777,\n",
       " 'MGA': -0.04030996011913777,\n",
       " 'TP53': -0.06895855419253871}"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make a dictionary of Hugo_Symbol and difference in the number of times it appears in recurrence_mut_table and no_recurrence_mut_table\n",
    "gene_count_diff = {}\n",
    "for gene in recurrence_gene_count:\n",
    "    if gene in no_recurrence_gene_count:\n",
    "        gene_count_diff[gene] = recurrence_gene_count[gene] - no_recurrence_gene_count[gene]\n",
    "    else:\n",
    "        gene_count_diff[gene] = recurrence_gene_count[gene]\n",
    "\n",
    "# sort the dictionary by value in descending order\n",
    "gene_count_diff = dict(sorted(gene_count_diff.items(), key=lambda item: item[1], reverse=True))\n",
    "gene_count_diff\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'SMARCA4': 0.14366151866151866,\n",
       " 'CREBBP': 0.07995495495495497,\n",
       " 'KMT2D': 0.06965894465894466,\n",
       " 'STK11': 0.06628056628056628,\n",
       " 'KDM5C': 0.06483268983268983,\n",
       " 'CTNNB1': 0.06322393822393824,\n",
       " 'SETD2': 0.052927927927927935,\n",
       " 'EZH1': 0.04054054054054054,\n",
       " 'KEAP1': 0.04037966537966539,\n",
       " 'PRDM1': 0.03458815958815959,\n",
       " 'MTOR': 0.03185328185328186,\n",
       " 'ARID1B': 0.03024453024453025,\n",
       " 'PTPRB': 0.02911840411840412,\n",
       " 'WT1': 0.02863577863577864,\n",
       " 'PTPRU': 0.02863577863577864,\n",
       " 'PBRM1': 0.02863577863577864,\n",
       " 'ERBB2': 0.02702702702702703,\n",
       " 'FLCN': 0.02702702702702703,\n",
       " 'STIM1': 0.02702702702702703,\n",
       " 'FN1': 0.02702702702702703,\n",
       " 'EGFR': 0.026383526383526396,\n",
       " 'ATM': 0.026383526383526396,\n",
       " 'NBN': 0.024292149292149297,\n",
       " 'SPEN': 0.024292149292149297,\n",
       " 'NCOR1': 0.023166023166023175,\n",
       " 'TP53BP1': 0.022683397683397687,\n",
       " 'FLT4': 0.022683397683397687,\n",
       " 'SNX29': 0.021074646074646076,\n",
       " 'ASXL1': 0.021074646074646076,\n",
       " 'AXIN1': 0.021074646074646076,\n",
       " 'BAP1': 0.018339768339768345,\n",
       " 'DOT1L': 0.018339768339768345,\n",
       " 'CIC': 0.016731016731016735,\n",
       " 'SGK223': 0.016731016731016735,\n",
       " 'DDX3X': 0.016731016731016735,\n",
       " 'KRAS': 0.01624839124839128,\n",
       " 'ARID2': 0.015604890604890606,\n",
       " 'FAS': 0.015122265122265124,\n",
       " 'LATS1': 0.015122265122265124,\n",
       " 'RIT1': 0.015122265122265124,\n",
       " 'CDC73': 0.015122265122265124,\n",
       " 'BRCA1': 0.015122265122265124,\n",
       " 'TNC': 0.013513513513513514,\n",
       " 'NFKBIE': 0.013513513513513514,\n",
       " 'ERCC4': 0.013513513513513514,\n",
       " 'NTRK1': 0.013513513513513514,\n",
       " 'RIMS2': 0.013513513513513514,\n",
       " 'PRRX1': 0.013513513513513514,\n",
       " 'ZFPM2': 0.013513513513513514,\n",
       " 'ARAF': 0.013513513513513514,\n",
       " 'IQGAP2': 0.013513513513513514,\n",
       " 'EPB41L3': 0.013513513513513514,\n",
       " 'PIK3CG': 0.013513513513513514,\n",
       " 'TNFAIP3': 0.013513513513513514,\n",
       " 'DDB2': 0.013513513513513514,\n",
       " 'NEB': 0.013513513513513514,\n",
       " 'FANCF': 0.013513513513513514,\n",
       " 'MYH2': 0.013513513513513514,\n",
       " 'MAX': 0.013513513513513514,\n",
       " 'BRAF': 0.012387387387387393,\n",
       " 'ZMYM2': 0.010778635778635783,\n",
       " 'NOTCH1': 0.009169884169884172,\n",
       " 'EXT1': 0.009169884169884172,\n",
       " 'NRAS': 0.009169884169884172,\n",
       " 'U2AF1': 0.009169884169884172,\n",
       " 'WAS': 0.009169884169884172,\n",
       " 'DNM2': 0.009169884169884172,\n",
       " 'GATA3': 0.009169884169884172,\n",
       " 'SERPINB13': 0.009169884169884172,\n",
       " 'KDM6A': 0.009169884169884172,\n",
       " 'ATR': 0.009169884169884172,\n",
       " 'SF3B1': 0.009169884169884172,\n",
       " 'CUL3': 0.009169884169884172,\n",
       " 'FBXW7': 0.009169884169884172,\n",
       " 'CHEK2': 0.007561132561132562,\n",
       " 'FANCA': 0.007561132561132562,\n",
       " 'AXIN2': 0.007561132561132562,\n",
       " 'MUTYH': 0.007561132561132562,\n",
       " 'NF2': 0.007561132561132562,\n",
       " 'ZNF180': 0.007561132561132562,\n",
       " 'TNR': 0.007561132561132562,\n",
       " 'PATZ1': 0.007561132561132562,\n",
       " 'ELL2': 0.007561132561132562,\n",
       " 'HCN1': 0.007561132561132562,\n",
       " 'SMARCC2': 0.007561132561132562,\n",
       " 'GPS2': 0.007561132561132562,\n",
       " 'FAT1': 0.006917631917631922,\n",
       " 'CHD8': 0.006435006435006441,\n",
       " 'POT1': 0.006435006435006441,\n",
       " 'RBM10': 0.005791505791505808,\n",
       " 'NOTCH2': 0.0048262548262548305,\n",
       " 'RASA1': 0.0048262548262548305,\n",
       " 'RNF43': 0.0032175032175032203,\n",
       " 'RAD21': 0.0032175032175032203,\n",
       " 'ANK2': 0.0032175032175032203,\n",
       " 'PMS1': 0.0032175032175032203,\n",
       " 'ACVR1B': 0.0032175032175032203,\n",
       " 'PTEN': 0.0032175032175032203,\n",
       " 'WRN': 0.0016087516087516102,\n",
       " 'CSMD3': 0.0016087516087516102,\n",
       " 'SH2B3': 0.0016087516087516102,\n",
       " 'BMPR1A': 0.0016087516087516102,\n",
       " 'TSC1': 0.0016087516087516102,\n",
       " 'BHMT2': 0.0016087516087516102,\n",
       " 'PRF1': 0.0016087516087516102,\n",
       " 'BLM': 0.0016087516087516102,\n",
       " 'CNKSR1': 0.0016087516087516102,\n",
       " 'BRIP1': 0.0016087516087516102,\n",
       " 'ITGBL1': 0.0016087516087516102,\n",
       " 'PTPRC': 0.0004826254826254886,\n",
       " 'BCLAF1': 0.0004826254826254886,\n",
       " 'ARHGAP35': -0.0011261261261261216,\n",
       " 'CMTR2': -0.0011261261261261216,\n",
       " 'POLE': -0.0011261261261261216,\n",
       " 'DICER1': -0.0011261261261261216,\n",
       " 'ATRX': -0.002252252252252243,\n",
       " 'RYR2': -0.0027348777348777317,\n",
       " 'MEN1': -0.0027348777348777317,\n",
       " 'KMT2B': -0.0027348777348777317,\n",
       " 'PHOX2B': -0.004343629343629342,\n",
       " 'PTCH1': -0.004343629343629342,\n",
       " 'CTCF': -0.004343629343629342,\n",
       " 'KLF6': -0.004343629343629342,\n",
       " 'CNOT3': -0.004343629343629342,\n",
       " 'ERCC3': -0.004343629343629342,\n",
       " 'SMAD4': -0.0054697554697554635,\n",
       " 'ASXL2': -0.008687258687258684,\n",
       " 'PPP3CA': -0.008687258687258684,\n",
       " 'NCOA6': -0.010296010296010294,\n",
       " 'FUBP1': -0.010296010296010294,\n",
       " 'CBL': -0.010296010296010294,\n",
       " 'PLCG2': -0.010296010296010294,\n",
       " 'NRXN1': -0.010296010296010294,\n",
       " 'PIK3R1': -0.010296010296010294,\n",
       " 'AMER1': -0.010296010296010294,\n",
       " 'SETDB1': -0.010296010296010294,\n",
       " 'DNAH12': -0.013030888030888026,\n",
       " 'UBR5': -0.013030888030888026,\n",
       " 'CDK12': -0.014639639639639636,\n",
       " 'CHD4': -0.014639639639639636,\n",
       " 'PIK3CA': -0.015765765765765757,\n",
       " 'MAP3K1': -0.016248391248391246,\n",
       " 'CBLB': -0.016248391248391246,\n",
       " 'ARID1A': -0.017374517374517368,\n",
       " 'COL5A2': -0.017374517374517368,\n",
       " 'COL2A1': -0.018983268983268978,\n",
       " 'TET2': -0.020592020592020588,\n",
       " 'IKZF1': -0.020592020592020588,\n",
       " 'GRIN2A': -0.02767052767052766,\n",
       " 'TSC2': -0.02815315315315315,\n",
       " 'PLXNB2': -0.02815315315315315,\n",
       " 'CUX1': -0.02815315315315315,\n",
       " 'BCOR': -0.03249678249678249,\n",
       " 'EP300': -0.03249678249678249,\n",
       " 'STAG2': -0.0341055341055341,\n",
       " 'APC': -0.03684041184041184,\n",
       " 'RB1': -0.040057915057915054,\n",
       " 'ZFP36L1': -0.040057915057915054,\n",
       " 'NF1': -0.04231016731016732,\n",
       " 'FANCM': -0.044401544401544396,\n",
       " 'TP53': -0.050353925353925355,\n",
       " 'MGA': -0.05148005148005147,\n",
       " 'KMT2C': -0.05904118404118404,\n",
       " 'CDKN2A': -0.08172458172458172}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LUAD_gene_count_diff = {}\n",
    "for gene in LUAD_recurrence_gene_count:\n",
    "    if gene in LUAD_no_recurrence_gene_count:\n",
    "        LUAD_gene_count_diff[gene] = LUAD_recurrence_gene_count[gene] - LUAD_no_recurrence_gene_count[gene]\n",
    "    else:\n",
    "        LUAD_gene_count_diff[gene] = LUAD_recurrence_gene_count[gene]\n",
    "\n",
    "# sort the dictionary by value in descending order\n",
    "LUAD_gene_count_diff = dict(sorted(LUAD_gene_count_diff.items(), key=lambda item: item[1], reverse=True))\n",
    "LUAD_gene_count_diff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'MAP3K13': 0.10416666666666667,\n",
       " 'ARID2': 0.10138888888888889,\n",
       " 'KMT2D': 0.09444444444444444,\n",
       " 'BCOR': 0.09305555555555556,\n",
       " 'B2M': 0.09166666666666667,\n",
       " 'NF1': 0.09027777777777779,\n",
       " 'PIK3CA': 0.08611111111111112,\n",
       " 'NCOA6': 0.08194444444444444,\n",
       " 'PBRM1': 0.08194444444444444,\n",
       " 'MTOR': 0.07222222222222222,\n",
       " 'PTPRB': 0.07083333333333333,\n",
       " 'PLXNB2': 0.06944444444444445,\n",
       " 'CUX1': 0.06805555555555556,\n",
       " 'PASK': 0.0625,\n",
       " 'ASXL1': 0.0611111111111111,\n",
       " 'PIK3R1': 0.05138888888888889,\n",
       " 'CHD4': 0.049999999999999996,\n",
       " 'ARHGAP35': 0.049999999999999996,\n",
       " 'KLF5': 0.041666666666666664,\n",
       " 'MSH2': 0.041666666666666664,\n",
       " 'GNPTAB': 0.041666666666666664,\n",
       " 'FLT4': 0.041666666666666664,\n",
       " 'FLCN': 0.041666666666666664,\n",
       " 'BMPR1A': 0.041666666666666664,\n",
       " 'RNF43': 0.041666666666666664,\n",
       " 'SNX29': 0.041666666666666664,\n",
       " 'CDKN2A': 0.04027777777777777,\n",
       " 'SPEN': 0.03888888888888888,\n",
       " 'FGFR3': 0.03055555555555555,\n",
       " 'STAG2': 0.03055555555555555,\n",
       " 'CDK12': 0.03055555555555555,\n",
       " 'BUB1B': 0.03055555555555555,\n",
       " 'HNF1A': 0.03055555555555555,\n",
       " 'HCN1': 0.03055555555555555,\n",
       " 'CYLD': 0.029166666666666667,\n",
       " 'APC': 0.029166666666666667,\n",
       " 'RASA1': 0.027777777777777776,\n",
       " 'ARID1A': 0.026388888888888892,\n",
       " 'FAT1': 0.020833333333333343,\n",
       " 'FAS': 0.020833333333333332,\n",
       " 'BLM': 0.020833333333333332,\n",
       " 'NSD1': 0.020833333333333332,\n",
       " 'EZH1': 0.020833333333333332,\n",
       " 'LAMA4': 0.020833333333333332,\n",
       " 'FAM166A': 0.020833333333333332,\n",
       " 'SEPT12': 0.020833333333333332,\n",
       " 'NIN': 0.020833333333333332,\n",
       " 'STX2': 0.020833333333333332,\n",
       " 'ZNF483': 0.020833333333333332,\n",
       " 'WT1': 0.020833333333333332,\n",
       " 'NRAS': 0.020833333333333332,\n",
       " 'PDYN': 0.020833333333333332,\n",
       " 'ZFP36L1': 0.020833333333333332,\n",
       " 'CAD': 0.020833333333333332,\n",
       " 'FGFBP1': 0.020833333333333332,\n",
       " 'IL21R': 0.020833333333333332,\n",
       " 'PPP3CA': 0.020833333333333332,\n",
       " 'BAP1': 0.020833333333333332,\n",
       " 'PRKAR1A': 0.020833333333333332,\n",
       " 'NFKBIE': 0.020833333333333332,\n",
       " 'LATS1': 0.01944444444444444,\n",
       " 'DDX3X': 0.01944444444444444,\n",
       " 'COL2A1': 0.01944444444444444,\n",
       " 'CIC': 0.01944444444444444,\n",
       " 'AMER1': 0.01944444444444444,\n",
       " 'EP300': 0.018055555555555554,\n",
       " 'CBLB': 0.018055555555555554,\n",
       " 'SMAD4': 0.018055555555555554,\n",
       " 'UBR5': 0.016666666666666663,\n",
       " 'COL5A2': 0.015277777777777779,\n",
       " 'WAS': 0.00972222222222222,\n",
       " 'MAP3K1': 0.00972222222222222,\n",
       " 'CBL': 0.00972222222222222,\n",
       " 'FBXO11': 0.00972222222222222,\n",
       " 'EGFR': 0.00972222222222222,\n",
       " 'EXT1': 0.00972222222222222,\n",
       " 'POLE': 0.00972222222222222,\n",
       " 'CSMD3': 0.00972222222222222,\n",
       " 'ARID1B': 0.00972222222222222,\n",
       " 'DOT1L': 0.00972222222222222,\n",
       " 'PRDM1': 0.00972222222222222,\n",
       " 'ERCC2': 0.00972222222222222,\n",
       " 'FBXW7': 0.008333333333333331,\n",
       " 'PTPRC': 0.008333333333333331,\n",
       " 'KMT2B': 0.008333333333333331,\n",
       " 'ZMYM2': 0.008333333333333331,\n",
       " 'STK11': 0.008333333333333331,\n",
       " 'BRCA1': 0.008333333333333331,\n",
       " 'FANCM': 0.0069444444444444475,\n",
       " 'SETD2': 0.00555555555555555,\n",
       " 'PTEN': 0.004166666666666666,\n",
       " 'IKZF1': -0.001388888888888891,\n",
       " 'KDM6A': -0.001388888888888891,\n",
       " 'CNOT3': -0.001388888888888891,\n",
       " 'SGK223': -0.001388888888888891,\n",
       " 'GATA3': -0.001388888888888891,\n",
       " 'TNFAIP3': -0.001388888888888891,\n",
       " 'RYR2': -0.001388888888888891,\n",
       " 'CEP76': -0.001388888888888891,\n",
       " 'AXIN1': -0.001388888888888891,\n",
       " 'PTPRU': -0.001388888888888891,\n",
       " 'AXIN2': -0.001388888888888891,\n",
       " 'DNM2': -0.001388888888888891,\n",
       " 'ATRX': -0.002777777777777782,\n",
       " 'NOTCH2': -0.002777777777777782,\n",
       " 'SMARCA4': -0.002777777777777782,\n",
       " 'NOTCH1': -0.004166666666666666,\n",
       " 'NFE2L2': -0.008333333333333331,\n",
       " 'MLH1': -0.0125,\n",
       " 'DICER1': -0.0125,\n",
       " 'DNAH12': -0.0125,\n",
       " 'BRCA2': -0.0125,\n",
       " 'CUL3': -0.0125,\n",
       " 'DNMT3A': -0.0125,\n",
       " 'CDC73': -0.0125,\n",
       " 'MBD1': -0.0125,\n",
       " 'BCLAF1': -0.013888888888888888,\n",
       " 'KRAS': -0.013888888888888888,\n",
       " 'MGA': -0.023611111111111114,\n",
       " 'ACVR1B': -0.023611111111111114,\n",
       " 'TP53BP1': -0.023611111111111114,\n",
       " 'SETDB1': -0.023611111111111114,\n",
       " 'CREBBP': -0.026388888888888892,\n",
       " 'KEAP1': -0.027777777777777776,\n",
       " 'RB1': -0.034722222222222224,\n",
       " 'NCOR1': -0.034722222222222224,\n",
       " 'TET2': -0.034722222222222224,\n",
       " 'ATR': -0.05694444444444445,\n",
       " 'KMT2C': -0.05833333333333334,\n",
       " 'TP53': -0.11111111111111105}"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LUSC_gene_count_diff = {}\n",
    "for gene in LUSC_recurrence_gene_count:\n",
    "    if gene in LUSC_no_recurrence_gene_count:\n",
    "        LUSC_gene_count_diff[gene] = LUSC_recurrence_gene_count[gene] - LUSC_no_recurrence_gene_count[gene]\n",
    "    else:\n",
    "        LUSC_gene_count_diff[gene] = LUSC_recurrence_gene_count[gene]\n",
    "\n",
    "# sort the dictionary by value in descending order\n",
    "LUSC_gene_count_diff = dict(sorted(LUSC_gene_count_diff.items(), key=lambda item: item[1], reverse=True))\n",
    "LUSC_gene_count_diff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['KMT2D',\n",
       " 'SMARCA4',\n",
       " 'ARID2',\n",
       " 'STK11',\n",
       " 'B2M',\n",
       " 'CTNNB1',\n",
       " 'PTEN',\n",
       " 'PTPRB',\n",
       " 'PIK3CA',\n",
       " 'WT1',\n",
       " 'PBRM1',\n",
       " 'MTOR',\n",
       " 'SNX29',\n",
       " 'ASXL1',\n",
       " 'FLCN',\n",
       " 'MAP3K13',\n",
       " 'KDM5C',\n",
       " 'CREBBP',\n",
       " 'EZH1',\n",
       " 'ARID1B',\n",
       " 'SETD2',\n",
       " 'BUB1B',\n",
       " 'FLT4',\n",
       " 'CIC',\n",
       " 'NCOA6',\n",
       " 'SGK223',\n",
       " 'NF1',\n",
       " 'KEAP1',\n",
       " 'MSH2',\n",
       " 'BRCA1',\n",
       " 'PTPRU',\n",
       " 'NBN',\n",
       " 'ARHGAP35',\n",
       " 'FN1',\n",
       " 'KLF5',\n",
       " 'NFKBIE',\n",
       " 'PALB2',\n",
       " 'STX2',\n",
       " 'BMPR1A',\n",
       " 'FAS',\n",
       " 'HCN1',\n",
       " 'HNF1A',\n",
       " 'PRDM1',\n",
       " 'AXIN1',\n",
       " 'CYLD',\n",
       " 'PIK3R1',\n",
       " 'RNF43',\n",
       " 'BAP1',\n",
       " 'DDX3X',\n",
       " 'FBXW7',\n",
       " 'POLE',\n",
       " 'CHD4',\n",
       " 'NOTCH1',\n",
       " 'SPEN',\n",
       " 'PLXNB2',\n",
       " 'KRAS',\n",
       " 'FAT1',\n",
       " 'ERBB2',\n",
       " 'FAM166A',\n",
       " 'FGFR3',\n",
       " 'GNPTAB',\n",
       " 'STIM1',\n",
       " 'NRAS',\n",
       " 'PASK',\n",
       " 'U2AF1',\n",
       " 'WAS',\n",
       " 'CDC73',\n",
       " 'LATS1',\n",
       " 'DOT1L',\n",
       " 'ZMYM2',\n",
       " 'BCOR',\n",
       " 'PTPRC',\n",
       " 'RASA1',\n",
       " 'ARID1A',\n",
       " 'ANAPC5',\n",
       " 'ARAF',\n",
       " 'ARHGEF12',\n",
       " 'DDB2',\n",
       " 'EPB41L3',\n",
       " 'FANCF',\n",
       " 'FGFR1',\n",
       " 'HOXB3',\n",
       " 'IL21R',\n",
       " 'IQGAP2',\n",
       " 'MAX',\n",
       " 'NEB',\n",
       " 'NIN',\n",
       " 'NSD1',\n",
       " 'NTRK1',\n",
       " 'PIK3CG',\n",
       " 'PRKAR1A',\n",
       " 'PRRX1',\n",
       " 'TNC',\n",
       " 'TNFRSF14',\n",
       " 'ZFPM2',\n",
       " 'ZNF483',\n",
       " 'BLM',\n",
       " 'TNFAIP3',\n",
       " 'ANK2',\n",
       " 'KDM6A']"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# make a list of the 100 genes with the greatest difference in the number of times they appear in recurrence_mut_table and no_recurrence_mut_table\n",
    "top_100_genes = list(gene_count_diff.keys())[:100]\n",
    "top_100_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "top_50_genes = list(gene_count_diff.keys())[:50]\n",
    "top_20_genes = list(gene_count_diff.keys())[:20]\n",
    "top_10_genes = list(gene_count_diff.keys())[:10]\n",
    "top_5_genes = list(gene_count_diff.keys())[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "LUAD_top_100_genes = list(LUAD_gene_count_diff.keys())[:100]\n",
    "LUAD_top_50_genes = list(LUAD_gene_count_diff.keys())[:50]\n",
    "LUAD_top_20_genes = list(LUAD_gene_count_diff.keys())[:20]\n",
    "LUAD_top_10_genes = list(LUAD_gene_count_diff.keys())[:10]\n",
    "LUAD_top_5_genes = list(LUAD_gene_count_diff.keys())[:5]\n",
    "\n",
    "LUSC_top_100_genes = list(LUSC_gene_count_diff.keys())[:100]\n",
    "LUSC_top_50_genes = list(LUSC_gene_count_diff.keys())[:50]\n",
    "LUSC_top_20_genes = list(LUSC_gene_count_diff.keys())[:20]\n",
    "LUSC_top_10_genes = list(LUSC_gene_count_diff.keys())[:10]\n",
    "LUSC_top_5_genes = list(LUSC_gene_count_diff.keys())[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(421, 100)\n",
      "Hugo_Symbol  ANAPC5  ANK2  ARAF  ARHGAP35  ARHGEF12  ARID1A  ARID1B  ARID2  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001          0     0     0         1         0       0       0      0   \n",
      "CRUK0002          0     0     0         0         0       0       0      0   \n",
      "CRUK0003          0     0     0         0         0       0       0      0   \n",
      "CRUK0004          0     0     0         0         0       0       0      0   \n",
      "CRUK0005          0     0     0         0         0       0       0      0   \n",
      "...             ...   ...   ...       ...       ...     ...     ...    ...   \n",
      "CRUK0861          0     0     0         0         0       0       0      0   \n",
      "CRUK0864          0     0     0         0         0       0       0      0   \n",
      "CRUK0872          0     0     0         0         0       0       0      0   \n",
      "CRUK0881          0     0     0         0         0       0       0      0   \n",
      "CRUK0893          0     0     0         0         0       0       0      0   \n",
      "\n",
      "Hugo_Symbol  ASXL1  AXIN1  B2M  BAP1  BCOR  BLM  BMPR1A  BRCA1  BUB1B  CDC73  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0002         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0003         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0004         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0005         0      0    0     0     0    0       0      0      0      0   \n",
      "...            ...    ...  ...   ...   ...  ...     ...    ...    ...    ...   \n",
      "CRUK0861         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0864         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0872         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0881         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0893         0      0    0     0     0    0       0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  CHD4  CIC  CREBBP  CTNNB1  CYLD  DDB2  DDX3X  DOT1L  EPB41L3  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0002        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0003        0    0       0       1     0     0      0      0        0   \n",
      "CRUK0004        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0005        0    0       0       0     0     0      0      0        0   \n",
      "...           ...  ...     ...     ...   ...   ...    ...    ...      ...   \n",
      "CRUK0861        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0864        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0872        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0881        0    0       1       0     0     0      1      0        0   \n",
      "CRUK0893        0    0       0       0     0     0      0      0        0   \n",
      "\n",
      "Hugo_Symbol  ERBB2  EZH1  FAM166A  FANCF  FAS  FAT1  FBXW7  FGFR1  FGFR3  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0002         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0003         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0004         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0005         0     0        0      0    0     0      0      0      0   \n",
      "...            ...   ...      ...    ...  ...   ...    ...    ...    ...   \n",
      "CRUK0861         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0864         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0872         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0881         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0893         0     0        0      0    0     0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  FLCN  FLT4  FN1  GNPTAB  HCN1  HNF1A  HOXB3  IL21R  IQGAP2  \\\n",
      "patient_id                                                                \n",
      "CRUK0001        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0002        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0003        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0004        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0005        0     0    0       0     0      0      0      0       0   \n",
      "...           ...   ...  ...     ...   ...    ...    ...    ...     ...   \n",
      "CRUK0861        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0864        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0872        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0881        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0893        0     0    0       0     0      0      0      0       0   \n",
      "\n",
      "Hugo_Symbol  KDM5C  KDM6A  KEAP1  KLF5  KMT2D  KRAS  LATS1  MAP3K13  MAX  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0002         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0003         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0004         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0005         0      0      0     0      0     0      0        0    0   \n",
      "...            ...    ...    ...   ...    ...   ...    ...      ...  ...   \n",
      "CRUK0861         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0864         0      0      0     0      0     1      0        0    0   \n",
      "CRUK0872         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0881         0      0      0     0      0     1      0        0    0   \n",
      "CRUK0893         0      0      0     0      0     1      0        0    0   \n",
      "\n",
      "Hugo_Symbol  MSH2  MTOR  NBN  NCOA6  NEB  NF1  NFKBIE  NIN  NOTCH1  NRAS  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0002        0     0    0      0    0    1       0    0       0     0   \n",
      "CRUK0003        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0004        0     0    0      0    0    0       0    0       1     0   \n",
      "CRUK0005        0     0    0      0    0    0       0    0       0     1   \n",
      "...           ...   ...  ...    ...  ...  ...     ...  ...     ...   ...   \n",
      "CRUK0861        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0864        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0872        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0881        0     0    0      0    0    1       0    0       0     0   \n",
      "CRUK0893        0     0    0      0    0    0       0    0       0     0   \n",
      "\n",
      "Hugo_Symbol  NSD1  NTRK1  PALB2  PASK  PBRM1  PIK3CA  PIK3CG  PIK3R1  PLXNB2  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001        0      0      0     1      0       0       0       0       0   \n",
      "CRUK0002        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0003        0      0      0     0      0       1       0       0       0   \n",
      "CRUK0004        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0005        0      0      0     1      0       0       0       0       0   \n",
      "...           ...    ...    ...   ...    ...     ...     ...     ...     ...   \n",
      "CRUK0861        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0864        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0872        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0881        0      0      0     0      1       0       0       0       0   \n",
      "CRUK0893        0      0      0     0      0       0       0       0       0   \n",
      "\n",
      "Hugo_Symbol  POLE  PRDM1  PRKAR1A  PRRX1  PTEN  PTPRB  PTPRC  PTPRU  RASA1  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0002        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0003        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0004        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0005        0      0        0      0     0      0      0      0      0   \n",
      "...           ...    ...      ...    ...   ...    ...    ...    ...    ...   \n",
      "CRUK0861        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0864        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0872        0      0        0      0     0      1      0      0      0   \n",
      "CRUK0881        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0893        0      0        0      0     0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  RNF43  SETD2  SGK223  SMARCA4  SNX29  SPEN  STIM1  STK11  STX2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0002         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0003         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0004         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0005         0      0       0        0      0     0      0      0     0   \n",
      "...            ...    ...     ...      ...    ...   ...    ...    ...   ...   \n",
      "CRUK0861         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0864         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0872         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0881         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0893         0      0       0        0      0     0      0      0     0   \n",
      "\n",
      "Hugo_Symbol  TNC  TNFAIP3  TNFRSF14  U2AF1  WAS  WT1  ZFPM2  ZMYM2  ZNF483  \n",
      "patient_id                                                                  \n",
      "CRUK0001       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0002       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0003       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0004       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0005       0        0         0      0    0    0      0      0       0  \n",
      "...          ...      ...       ...    ...  ...  ...    ...    ...     ...  \n",
      "CRUK0861       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0864       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0872       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0881       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0893       0        0         0      0    0    0      0      0       0  \n",
      "\n",
      "[421 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "driver_table = mut_table[mut_table['DriverMut']]\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(top_100_genes)]\n",
    "\n",
    "hundred_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "hundred_feature_matrix = hundred_feature_matrix.astype(int)\n",
    "\n",
    "# fill in missing patients with 0s\n",
    "missing_patients = list(set(clinical_df['cruk_id']) - set(hundred_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    hundred_feature_matrix.loc[patient] = 0\n",
    "\n",
    "hundred_feature_matrix = hundred_feature_matrix.sort_index()\n",
    "print(hundred_feature_matrix.shape)\n",
    "print(hundred_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(421, 50)\n",
      "Hugo_Symbol  ARHGAP35  ARID1B  ARID2  ASXL1  AXIN1  B2M  BAP1  BMPR1A  BRCA1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001            1       0      0      0      0    0     0       0      0   \n",
      "CRUK0002            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0003            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0004            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0005            0       0      0      0      0    0     0       0      0   \n",
      "...               ...     ...    ...    ...    ...  ...   ...     ...    ...   \n",
      "CRUK0861            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0864            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0872            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0881            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0893            0       0      0      0      0    0     0       0      0   \n",
      "\n",
      "Hugo_Symbol  BUB1B  CIC  CREBBP  CTNNB1  CYLD  DDX3X  EZH1  FAS  FBXW7  FLCN  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0002         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0003         0    0       0       1     0      0     0    0      0     0   \n",
      "CRUK0004         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0005         0    0       0       0     0      0     0    0      0     0   \n",
      "...            ...  ...     ...     ...   ...    ...   ...  ...    ...   ...   \n",
      "CRUK0861         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0864         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0872         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0881         0    0       1       0     0      1     0    0      0     0   \n",
      "CRUK0893         0    0       0       0     0      0     0    0      0     0   \n",
      "\n",
      "Hugo_Symbol  FLT4  FN1  HCN1  HNF1A  KDM5C  KEAP1  KLF5  KMT2D  MAP3K13  MSH2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0002        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0003        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0004        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0005        0    0     0      0      0      0     0      0        0     0   \n",
      "...           ...  ...   ...    ...    ...    ...   ...    ...      ...   ...   \n",
      "CRUK0861        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0864        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0872        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0881        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0893        0    0     0      0      0      0     0      0        0     0   \n",
      "\n",
      "Hugo_Symbol  MTOR  NBN  NCOA6  NF1  NFKBIE  PALB2  PBRM1  PIK3CA  PIK3R1  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0002        0    0      0    1       0      0      0       0       0   \n",
      "CRUK0003        0    0      0    0       0      0      0       1       0   \n",
      "CRUK0004        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0005        0    0      0    0       0      0      0       0       0   \n",
      "...           ...  ...    ...  ...     ...    ...    ...     ...     ...   \n",
      "CRUK0861        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0864        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0872        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0881        0    0      0    1       0      0      1       0       0   \n",
      "CRUK0893        0    0      0    0       0      0      0       0       0   \n",
      "\n",
      "Hugo_Symbol  PRDM1  PTEN  PTPRB  PTPRU  RNF43  SETD2  SGK223  SMARCA4  SNX29  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0002         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0003         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0004         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0005         0     0      0      0      0      0       0        0      0   \n",
      "...            ...   ...    ...    ...    ...    ...     ...      ...    ...   \n",
      "CRUK0861         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0864         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0872         0     0      1      0      0      0       0        0      0   \n",
      "CRUK0881         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0893         0     0      0      0      0      0       0        0      0   \n",
      "\n",
      "Hugo_Symbol  STK11  STX2  WT1  \n",
      "patient_id                     \n",
      "CRUK0001         0     0    0  \n",
      "CRUK0002         0     0    0  \n",
      "CRUK0003         0     0    0  \n",
      "CRUK0004         0     0    0  \n",
      "CRUK0005         0     0    0  \n",
      "...            ...   ...  ...  \n",
      "CRUK0861         0     0    0  \n",
      "CRUK0864         0     0    0  \n",
      "CRUK0872         0     0    0  \n",
      "CRUK0881         0     0    0  \n",
      "CRUK0893         0     0    0  \n",
      "\n",
      "[421 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(top_50_genes)]\n",
    "\n",
    "fifty_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "fifty_feature_matrix = fifty_feature_matrix.astype(int)\n",
    "\n",
    "\n",
    "# fill in the missing patients in the 50 gene matrix\n",
    "missing_patients = list(set(clinical_df['cruk_id']) - set(fifty_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    fifty_feature_matrix.loc[patient] = 0\n",
    "\n",
    "fifty_feature_matrix = fifty_feature_matrix.sort_index()\n",
    "print(fifty_feature_matrix.shape)\n",
    "print(fifty_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(421, 20)\n",
      "Hugo_Symbol  ARID1B  ARID2  ASXL1  B2M  CREBBP  CTNNB1  EZH1  FLCN  KDM5C  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0002          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0003          0      0      0    0       0       1     0     0      0   \n",
      "CRUK0004          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0005          0      0      0    0       0       0     0     0      0   \n",
      "...             ...    ...    ...  ...     ...     ...   ...   ...    ...   \n",
      "CRUK0861          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0864          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0872          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0881          0      0      0    0       1       0     0     0      0   \n",
      "CRUK0893          0      0      0    0       0       0     0     0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  MAP3K13  MTOR  PBRM1  PIK3CA  PTEN  PTPRB  SMARCA4  SNX29  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0002         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0003         0        0     0      0       1     0      0        0      0   \n",
      "CRUK0004         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0005         0        0     0      0       0     0      0        0      0   \n",
      "...            ...      ...   ...    ...     ...   ...    ...      ...    ...   \n",
      "CRUK0861         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0864         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0872         0        0     0      0       0     0      1        0      0   \n",
      "CRUK0881         0        0     0      1       0     0      0        0      0   \n",
      "CRUK0893         0        0     0      0       0     0      0        0      0   \n",
      "\n",
      "Hugo_Symbol  STK11  WT1  \n",
      "patient_id               \n",
      "CRUK0001         0    0  \n",
      "CRUK0002         0    0  \n",
      "CRUK0003         0    0  \n",
      "CRUK0004         0    0  \n",
      "CRUK0005         0    0  \n",
      "...            ...  ...  \n",
      "CRUK0861         0    0  \n",
      "CRUK0864         0    0  \n",
      "CRUK0872         0    0  \n",
      "CRUK0881         0    0  \n",
      "CRUK0893         0    0  \n",
      "\n",
      "[421 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(top_20_genes)]\n",
    "\n",
    "twenty_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "twenty_feature_matrix = twenty_feature_matrix.astype(int)\n",
    "\n",
    "# fill in the missing patients in the 20 gene matrix\n",
    "missing_patients = list(set(clinical_df['cruk_id']) - set(twenty_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    twenty_feature_matrix.loc[patient] = 0\n",
    "\n",
    "twenty_feature_matrix = twenty_feature_matrix.sort_index()\n",
    "print(twenty_feature_matrix.shape)\n",
    "print(twenty_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(421, 10)\n",
      "Hugo_Symbol  ARID2  B2M  CTNNB1  KMT2D  PIK3CA  PTEN  PTPRB  SMARCA4  STK11  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0002         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0003         0    0       1      0       1     0      0        0      0   \n",
      "CRUK0004         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0005         0    0       0      0       0     0      0        0      0   \n",
      "...            ...  ...     ...    ...     ...   ...    ...      ...    ...   \n",
      "CRUK0861         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0864         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0872         0    0       0      0       0     0      1        0      0   \n",
      "CRUK0881         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0893         0    0       0      0       0     0      0        0      0   \n",
      "\n",
      "Hugo_Symbol  WT1  \n",
      "patient_id        \n",
      "CRUK0001       0  \n",
      "CRUK0002       0  \n",
      "CRUK0003       0  \n",
      "CRUK0004       0  \n",
      "CRUK0005       0  \n",
      "...          ...  \n",
      "CRUK0861       0  \n",
      "CRUK0864       0  \n",
      "CRUK0872       0  \n",
      "CRUK0881       0  \n",
      "CRUK0893       0  \n",
      "\n",
      "[421 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(top_10_genes)]\n",
    "\n",
    "ten_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "ten_feature_matrix = ten_feature_matrix.astype(int)\n",
    "\n",
    "# fill in the missing patients in the 10 gene matrix\n",
    "missing_patients = list(set(clinical_df['cruk_id']) - set(ten_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    ten_feature_matrix.loc[patient] = 0\n",
    "\n",
    "ten_feature_matrix = ten_feature_matrix.sort_index()\n",
    "print(ten_feature_matrix.shape)\n",
    "print(ten_feature_matrix)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(421, 5)\n",
      "Hugo_Symbol  ARID2  B2M  KMT2D  SMARCA4  STK11\n",
      "patient_id                                    \n",
      "CRUK0001         0    0      0        0      0\n",
      "CRUK0002         0    0      0        0      0\n",
      "CRUK0003         0    0      0        0      0\n",
      "CRUK0004         0    0      0        0      0\n",
      "CRUK0005         0    0      0        0      0\n",
      "...            ...  ...    ...      ...    ...\n",
      "CRUK0861         0    0      0        0      0\n",
      "CRUK0864         0    0      0        0      0\n",
      "CRUK0872         0    0      0        0      0\n",
      "CRUK0881         0    0      0        0      0\n",
      "CRUK0893         0    0      0        0      0\n",
      "\n",
      "[421 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(top_5_genes)]\n",
    "\n",
    "five_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "five_feature_matrix = five_feature_matrix.astype(int)\n",
    "\n",
    "\n",
    "# fill in the missing patients in the 5 gene matrix\n",
    "missing_patients = list(set(clinical_df['cruk_id']) - set(five_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    five_feature_matrix.loc[patient] = 0\n",
    "\n",
    "five_feature_matrix = five_feature_matrix.sort_index()\n",
    "print(five_feature_matrix.shape)\n",
    "print(five_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0 0 1 1 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 1 0 1 0 0 1 1 1 1\n",
      " 1 0 1 0 0 1 1 0 0 1 0 1 0 0 0 0 0 0 1 0 0 0 0 1 1 0 1 0 0 0 0 0 0 0 1 1 0\n",
      " 0 1 0 0 1 0 0 0 0 0 1 1 0 1 0 1 0 1 0 1 1 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 1\n",
      " 0 0 0 0 0 0 0 1 0 1 0 0 0 0 1 0 0 1 0 0 0 1 1 1 0 1 1 1 1 1 1 0 0 0 0 0 1\n",
      " 1 0 0 0 0 0 0 1 0 1 0 1 0 1 0 1 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 1 0 1 1 0\n",
      " 0 0 0 1 1 0 1 0 1 0 1 0 1 1 0 0 0 0 0 0 1 1 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 1 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 1\n",
      " 0 1 0 1 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 1 1 0 0 0 1 1 0 0 0 1 0 0 1 1 1 0 1\n",
      " 0 0 0 1 0 0 0 1 1 1 0 0 1 0 1 0 0 0 0 0 0 0 1 1 1 0 0 1 0 0 0 0 0 0 0 1 0\n",
      " 1 0 0 1 1 0 0 0 0 1 0 1 0 1 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 1\n",
      " 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0 1 1 0 0 1 0 0 0 1\n",
      " 0 1 0 0 0 0 0 1 1 0 0 1 1 0]\n"
     ]
    }
   ],
   "source": [
    "# Making the training data\n",
    "\n",
    "y_train = []\n",
    "for index, row in clinical_df.iterrows():\n",
    "    if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "        y_train.append(1)\n",
    "    else:\n",
    "        y_train.append(0)\n",
    "\n",
    "y = np.array(y_train)\n",
    "print(y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 100) (169, 100) (252,) (169,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5295900178253119\n",
      "Predictions:  [1 0 0 0 1 0 0 0 1 0 1 0 0 1 0 0 1 1 0 0 0 0 0 0 1 0 1 0 1 0 1 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 1 0 1 1 0 0 1 0 1 1 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 1 1 1 0 0 0 0 1 1 0 1 1 0 0 0 0 1 0 1 0 0 1 0 1 1 0 1 0 0 1 0 1 1 0 0\n",
      " 0 1 0 1 0 1 1 0 0 0 0 0 0 0 1 1 1 1 1 0 1 1 0 0 1 0 0 0 0 1 0 0 0 0 0 1 1\n",
      " 0 1 0 1 0 0 0 1 0 0 0 1 1 0 1 0 1 1 0 0 0]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5675125313283208\n",
      "ROC AUC:  0.4841791979949875\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.71      0.70      0.70       112\n",
      "   Recurrence       0.42      0.44      0.43        57\n",
      "\n",
      "     accuracy                           0.61       169\n",
      "    macro avg       0.57      0.57      0.57       169\n",
      " weighted avg       0.61      0.61      0.61       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.metrics import classification_report\n",
    "from sklearn.metrics import balanced_accuracy_score\n",
    "from sklearn.metrics import roc_curve, roc_auc_score\n",
    "from sklearn.neural_network import MLPClassifier\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(hundred_feature_matrix, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 50) (169, 50) (252,) (169,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.0001, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.5475044563279857\n",
      "Predictions:  [1 0 1 0 0 0 0 0 1 0 0 0 0 1 0 0 1 1 0 0 0 0 0 0 0 0 1 0 1 1 0 0 0 1 0 0 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 1 0 1 0 1 0 1 0 0\n",
      " 0 0 0 1 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1 0 0 1 0 0 0 0 1 0 0 0 0 1 1 0 0\n",
      " 0 1 0 0 0 1 1 0 0 0 0 0 1 1 1 1 1 1 1 0 0 1 0 0 1 0 0 0 0 1 0 1 0 0 1 0 1\n",
      " 0 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0 1 0 0 0 0]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5372023809523809\n",
      "ROC AUC:  0.4515977443609023\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.69      0.74      0.71       112\n",
      "   Recurrence       0.40      0.33      0.36        57\n",
      "\n",
      "     accuracy                           0.60       169\n",
      "    macro avg       0.54      0.54      0.54       169\n",
      " weighted avg       0.59      0.60      0.59       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(fifty_feature_matrix, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 20) (169, 20) (252,) (169,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5482174688057041\n",
      "Predictions:  [0 0 0 0 1 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 1 0 0 1 0 1 0\n",
      " 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 1 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 1 1 0 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 1 1 0 0 0 0 1 0 0 0 1 0 0 0 0]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.6264880952380952\n",
      "ROC AUC:  0.5916353383458646\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.73      0.92      0.81       112\n",
      "   Recurrence       0.68      0.33      0.45        57\n",
      "\n",
      "     accuracy                           0.72       169\n",
      "    macro avg       0.70      0.63      0.63       169\n",
      " weighted avg       0.71      0.72      0.69       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(twenty_feature_matrix, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 10) (169, 10) (252,) (169,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.1, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5548128342245989\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5213815789473684\n",
      "ROC AUC:  0.56296992481203\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.94      0.78       112\n",
      "   Recurrence       0.46      0.11      0.17        57\n",
      "\n",
      "     accuracy                           0.66       169\n",
      "    macro avg       0.57      0.52      0.48       169\n",
      " weighted avg       0.60      0.66      0.58       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(ten_feature_matrix, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 5) (169, 5) (252,) (169,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5136363636363636\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.49968671679197996\n",
      "ROC AUC:  0.4844924812030075\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.66      0.96      0.79       112\n",
      "   Recurrence       0.33      0.04      0.06        57\n",
      "\n",
      "     accuracy                           0.65       169\n",
      "    macro avg       0.50      0.50      0.42       169\n",
      " weighted avg       0.55      0.65      0.54       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(five_feature_matrix, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 100)\n",
      "Hugo_Symbol  ACVR1B  ANK2  ARAF  ARID1B  ARID2  ASXL1  ATM  ATR  AXIN1  AXIN2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001          0     0     0       0      0      0    0    0      0      0   \n",
      "CRUK0002          0     0     0       0      0      0    0    0      0      0   \n",
      "CRUK0003          0     0     0       0      0      0    0    0      0      0   \n",
      "CRUK0004          0     0     0       0      0      0    0    0      0      0   \n",
      "CRUK0005          0     0     0       0      0      0    0    0      0      0   \n",
      "...             ...   ...   ...     ...    ...    ...  ...  ...    ...    ...   \n",
      "CRUK0810          0     0     0       0      0      0    0    0      0      0   \n",
      "CRUK0817          0     0     0       0      0      0    0    0      0      0   \n",
      "CRUK0864          0     0     0       0      0      0    0    0      0      0   \n",
      "CRUK0881          0     0     0       0      0      0    0    0      0      0   \n",
      "CRUK0893          0     0     0       0      0      0    1    0      0      0   \n",
      "\n",
      "Hugo_Symbol  BAP1  BRAF  BRCA1  CDC73  CHD8  CHEK2  CIC  CREBBP  CSMD3  \\\n",
      "patient_id                                                               \n",
      "CRUK0001        0     0      0      0     0      0    0       0      0   \n",
      "CRUK0002        0     0      0      0     0      0    0       0      0   \n",
      "CRUK0003        0     0      0      0     0      0    0       0      0   \n",
      "CRUK0004        0     0      0      0     0      0    0       0      0   \n",
      "CRUK0005        0     1      0      0     0      0    0       0      0   \n",
      "...           ...   ...    ...    ...   ...    ...  ...     ...    ...   \n",
      "CRUK0810        0     0      0      0     0      0    0       0      0   \n",
      "CRUK0817        0     0      0      0     0      0    0       0      0   \n",
      "CRUK0864        0     0      0      0     0      0    0       0      0   \n",
      "CRUK0881        0     1      0      0     0      0    0       1      0   \n",
      "CRUK0893        0     0      0      0     0      0    0       0      0   \n",
      "\n",
      "Hugo_Symbol  CTNNB1  CUL3  DDB2  DDX3X  DNM2  DOT1L  EGFR  ELL2  EPB41L3  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001          0     0     0      0     0      0     0     0        0   \n",
      "CRUK0002          0     0     0      0     0      0     0     0        0   \n",
      "CRUK0003          1     0     0      0     0      0     1     0        0   \n",
      "CRUK0004          0     0     0      0     0      0     1     0        0   \n",
      "CRUK0005          0     0     0      0     0      0     0     0        0   \n",
      "...             ...   ...   ...    ...   ...    ...   ...   ...      ...   \n",
      "CRUK0810          0     0     0      0     0      0     0     0        0   \n",
      "CRUK0817          0     0     0      0     0      0     0     0        0   \n",
      "CRUK0864          0     0     0      0     0      0     0     0        0   \n",
      "CRUK0881          0     1     0      1     0      0     0     0        0   \n",
      "CRUK0893          0     0     0      0     0      0     0     0        0   \n",
      "\n",
      "Hugo_Symbol  ERBB2  ERCC4  EXT1  EZH1  FANCA  FANCF  FAS  FAT1  FBXW7  FLCN  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      0     0     0      0      0    0     0      0     0   \n",
      "CRUK0002         0      0     0     0      0      0    0     0      0     0   \n",
      "CRUK0003         0      0     0     0      0      0    0     0      0     0   \n",
      "CRUK0004         0      0     0     0      0      0    0     0      0     0   \n",
      "CRUK0005         0      0     0     0      0      0    0     0      0     0   \n",
      "...            ...    ...   ...   ...    ...    ...  ...   ...    ...   ...   \n",
      "CRUK0810         0      0     0     0      0      0    0     0      0     0   \n",
      "CRUK0817         0      0     0     0      0      0    0     0      0     0   \n",
      "CRUK0864         0      0     0     0      0      0    0     0      0     0   \n",
      "CRUK0881         0      0     0     0      0      0    0     0      0     0   \n",
      "CRUK0893         0      0     0     0      0      0    0     0      0     0   \n",
      "\n",
      "Hugo_Symbol  FLT4  FN1  GATA3  GPS2  HCN1  IQGAP2  KDM5C  KDM6A  KEAP1  KMT2D  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001        0    0      0     0     0       0      0      0      0      0   \n",
      "CRUK0002        0    0      0     0     0       0      0      0      0      0   \n",
      "CRUK0003        0    0      0     0     0       0      0      0      0      0   \n",
      "CRUK0004        0    0      0     0     0       0      0      0      0      0   \n",
      "CRUK0005        0    0      0     0     0       0      0      0      0      0   \n",
      "...           ...  ...    ...   ...   ...     ...    ...    ...    ...    ...   \n",
      "CRUK0810        0    0      0     1     0       0      0      0      0      0   \n",
      "CRUK0817        0    0      0     0     0       0      0      0      0      0   \n",
      "CRUK0864        0    0      0     0     0       0      0      0      0      0   \n",
      "CRUK0881        0    0      0     0     0       0      0      0      0      0   \n",
      "CRUK0893        0    0      0     0     0       0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  KRAS  LATS1  MAX  MTOR  MUTYH  MYH2  NBN  NCOR1  NEB  NF2  \\\n",
      "patient_id                                                               \n",
      "CRUK0001        0      0    0     0      0     0    0      0    0    0   \n",
      "CRUK0002        0      0    0     0      0     0    0      0    0    0   \n",
      "CRUK0003        0      0    0     0      0     0    0      0    0    0   \n",
      "CRUK0004        0      0    0     0      0     0    0      0    0    0   \n",
      "CRUK0005        0      0    0     0      0     0    0      0    0    0   \n",
      "...           ...    ...  ...   ...    ...   ...  ...    ...  ...  ...   \n",
      "CRUK0810        1      0    0     0      0     0    0      0    0    0   \n",
      "CRUK0817        0      0    0     0      0     0    0      0    0    0   \n",
      "CRUK0864        1      0    0     0      0     0    0      0    0    0   \n",
      "CRUK0881        1      0    0     0      0     0    0      0    0    0   \n",
      "CRUK0893        1      0    0     0      0     0    0      0    0    0   \n",
      "\n",
      "Hugo_Symbol  NFKBIE  NOTCH1  NOTCH2  NRAS  NTRK1  PATZ1  PBRM1  PIK3CG  PMS1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001          0       0       0     0      0      0      0       0     0   \n",
      "CRUK0002          0       0       0     0      0      0      0       0     0   \n",
      "CRUK0003          0       0       0     0      0      0      0       0     0   \n",
      "CRUK0004          0       1       0     0      0      0      0       0     0   \n",
      "CRUK0005          0       0       0     1      0      0      0       0     0   \n",
      "...             ...     ...     ...   ...    ...    ...    ...     ...   ...   \n",
      "CRUK0810          0       0       0     0      0      0      0       0     0   \n",
      "CRUK0817          0       0       0     0      0      0      0       0     0   \n",
      "CRUK0864          0       0       0     0      0      0      0       0     0   \n",
      "CRUK0881          0       0       0     0      0      0      1       0     1   \n",
      "CRUK0893          0       0       0     0      0      0      0       0     0   \n",
      "\n",
      "Hugo_Symbol  POT1  PRDM1  PRRX1  PTEN  PTPRB  PTPRU  RAD21  RASA1  RBM10  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001        0      0      0     0      0      0      0      0      0   \n",
      "CRUK0002        0      0      0     0      0      0      0      0      0   \n",
      "CRUK0003        0      0      0     0      0      0      0      0      0   \n",
      "CRUK0004        0      0      0     0      0      0      0      0      0   \n",
      "CRUK0005        0      0      0     0      0      0      0      0      0   \n",
      "...           ...    ...    ...   ...    ...    ...    ...    ...    ...   \n",
      "CRUK0810        0      0      0     0      0      0      0      0      0   \n",
      "CRUK0817        0      0      0     0      0      0      0      0      1   \n",
      "CRUK0864        0      0      0     0      0      0      0      0      1   \n",
      "CRUK0881        0      0      0     0      0      0      0      0      1   \n",
      "CRUK0893        0      0      0     0      0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  RIMS2  RIT1  RNF43  SERPINB13  SETD2  SF3B1  SGK223  SMARCA4  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001         0     0      0          0      0      0       0        0   \n",
      "CRUK0002         0     0      0          0      0      0       0        0   \n",
      "CRUK0003         0     0      0          0      0      0       0        0   \n",
      "CRUK0004         0     0      0          0      0      0       0        0   \n",
      "CRUK0005         0     0      0          0      0      0       0        0   \n",
      "...            ...   ...    ...        ...    ...    ...     ...      ...   \n",
      "CRUK0810         0     0      0          0      0      0       0        1   \n",
      "CRUK0817         0     0      0          0      0      0       0        0   \n",
      "CRUK0864         0     0      0          0      0      0       0        0   \n",
      "CRUK0881         0     0      0          0      0      0       0        0   \n",
      "CRUK0893         0     0      0          0      0      0       0        0   \n",
      "\n",
      "Hugo_Symbol  SMARCC2  SNX29  SPEN  STIM1  STK11  TNC  TNFAIP3  TNR  TP53BP1  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001           0      0     0      0      0    0        0    0        0   \n",
      "CRUK0002           0      0     0      0      0    0        0    0        0   \n",
      "CRUK0003           0      0     0      0      0    0        0    0        0   \n",
      "CRUK0004           0      0     0      0      0    0        0    0        0   \n",
      "CRUK0005           0      0     0      0      0    0        0    0        0   \n",
      "...              ...    ...   ...    ...    ...  ...      ...  ...      ...   \n",
      "CRUK0810           0      0     0      0      1    0        0    0        0   \n",
      "CRUK0817           0      0     0      0      1    0        0    0        0   \n",
      "CRUK0864           0      0     0      0      0    0        0    0        0   \n",
      "CRUK0881           0      0     0      0      0    0        0    0        0   \n",
      "CRUK0893           0      0     0      0      0    0        0    0        0   \n",
      "\n",
      "Hugo_Symbol  U2AF1  WAS  WRN  WT1  ZFPM2  ZMYM2  ZNF180  \n",
      "patient_id                                               \n",
      "CRUK0001         0    0    1    0      0      0       0  \n",
      "CRUK0002         0    0    0    0      0      0       0  \n",
      "CRUK0003         0    0    0    0      0      0       0  \n",
      "CRUK0004         0    0    0    0      0      0       0  \n",
      "CRUK0005         0    0    0    0      0      0       0  \n",
      "...            ...  ...  ...  ...    ...    ...     ...  \n",
      "CRUK0810         0    0    0    0      0      0       0  \n",
      "CRUK0817         0    0    0    0      0      0       0  \n",
      "CRUK0864         0    0    0    0      0      0       0  \n",
      "CRUK0881         0    0    0    0      0      0       0  \n",
      "CRUK0893         0    0    0    0      0      0       0  \n",
      "\n",
      "[242 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUAD_hundred_feature_matrix with only LUAD_patient_id\n",
    "\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUAD_top_100_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUAD_patient_id)]\n",
    "\n",
    "LUAD_hundred_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUAD_hundred_feature_matrix = LUAD_hundred_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUAD_patient_id) - set(LUAD_hundred_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUAD_hundred_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUAD_hundred_feature_matrix = LUAD_hundred_feature_matrix.sort_index()\n",
    "print(LUAD_hundred_feature_matrix.shape)\n",
    "print(LUAD_hundred_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 50)\n",
      "Hugo_Symbol  ARAF  ARID1B  ARID2  ASXL1  ATM  AXIN1  BAP1  BRCA1  CDC73  CIC  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001        0       0      0      0    0      0     0      0      0    0   \n",
      "CRUK0002        0       0      0      0    0      0     0      0      0    0   \n",
      "CRUK0003        0       0      0      0    0      0     0      0      0    0   \n",
      "CRUK0004        0       0      0      0    0      0     0      0      0    0   \n",
      "CRUK0005        0       0      0      0    0      0     0      0      0    0   \n",
      "...           ...     ...    ...    ...  ...    ...   ...    ...    ...  ...   \n",
      "CRUK0810        0       0      0      0    0      0     0      0      0    0   \n",
      "CRUK0817        0       0      0      0    0      0     0      0      0    0   \n",
      "CRUK0864        0       0      0      0    0      0     0      0      0    0   \n",
      "CRUK0881        0       0      0      0    0      0     0      0      0    0   \n",
      "CRUK0893        0       0      0      0    1      0     0      0      0    0   \n",
      "\n",
      "Hugo_Symbol  CREBBP  CTNNB1  DDX3X  DOT1L  EGFR  ERBB2  ERCC4  EZH1  FAS  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001          0       0      0      0     0      0      0     0    0   \n",
      "CRUK0002          0       0      0      0     0      0      0     0    0   \n",
      "CRUK0003          0       1      0      0     1      0      0     0    0   \n",
      "CRUK0004          0       0      0      0     1      0      0     0    0   \n",
      "CRUK0005          0       0      0      0     0      0      0     0    0   \n",
      "...             ...     ...    ...    ...   ...    ...    ...   ...  ...   \n",
      "CRUK0810          0       0      0      0     0      0      0     0    0   \n",
      "CRUK0817          0       0      0      0     0      0      0     0    0   \n",
      "CRUK0864          0       0      0      0     0      0      0     0    0   \n",
      "CRUK0881          1       0      1      0     0      0      0     0    0   \n",
      "CRUK0893          0       0      0      0     0      0      0     0    0   \n",
      "\n",
      "Hugo_Symbol  FLCN  FLT4  FN1  KDM5C  KEAP1  KMT2D  KRAS  LATS1  MTOR  NBN  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        0     0    0      0      0      0     0      0     0    0   \n",
      "CRUK0002        0     0    0      0      0      0     0      0     0    0   \n",
      "CRUK0003        0     0    0      0      0      0     0      0     0    0   \n",
      "CRUK0004        0     0    0      0      0      0     0      0     0    0   \n",
      "CRUK0005        0     0    0      0      0      0     0      0     0    0   \n",
      "...           ...   ...  ...    ...    ...    ...   ...    ...   ...  ...   \n",
      "CRUK0810        0     0    0      0      0      0     1      0     0    0   \n",
      "CRUK0817        0     0    0      0      0      0     0      0     0    0   \n",
      "CRUK0864        0     0    0      0      0      0     1      0     0    0   \n",
      "CRUK0881        0     0    0      0      0      0     1      0     0    0   \n",
      "CRUK0893        0     0    0      0      0      0     1      0     0    0   \n",
      "\n",
      "Hugo_Symbol  NCOR1  NFKBIE  NTRK1  PBRM1  PRDM1  PRRX1  PTPRB  PTPRU  RIMS2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0       0      0      0      0      0      0      0      0   \n",
      "CRUK0002         0       0      0      0      0      0      0      0      0   \n",
      "CRUK0003         0       0      0      0      0      0      0      0      0   \n",
      "CRUK0004         0       0      0      0      0      0      0      0      0   \n",
      "CRUK0005         0       0      0      0      0      0      0      0      0   \n",
      "...            ...     ...    ...    ...    ...    ...    ...    ...    ...   \n",
      "CRUK0810         0       0      0      0      0      0      0      0      0   \n",
      "CRUK0817         0       0      0      0      0      0      0      0      0   \n",
      "CRUK0864         0       0      0      0      0      0      0      0      0   \n",
      "CRUK0881         0       0      0      1      0      0      0      0      0   \n",
      "CRUK0893         0       0      0      0      0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  RIT1  SETD2  SGK223  SMARCA4  SNX29  SPEN  STIM1  STK11  TNC  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        0      0       0        0      0     0      0      0    0   \n",
      "CRUK0002        0      0       0        0      0     0      0      0    0   \n",
      "CRUK0003        0      0       0        0      0     0      0      0    0   \n",
      "CRUK0004        0      0       0        0      0     0      0      0    0   \n",
      "CRUK0005        0      0       0        0      0     0      0      0    0   \n",
      "...           ...    ...     ...      ...    ...   ...    ...    ...  ...   \n",
      "CRUK0810        0      0       0        1      0     0      0      1    0   \n",
      "CRUK0817        0      0       0        0      0     0      0      1    0   \n",
      "CRUK0864        0      0       0        0      0     0      0      0    0   \n",
      "CRUK0881        0      0       0        0      0     0      0      0    0   \n",
      "CRUK0893        0      0       0        0      0     0      0      0    0   \n",
      "\n",
      "Hugo_Symbol  TP53BP1  WT1  ZFPM2  \n",
      "patient_id                        \n",
      "CRUK0001           0    0      0  \n",
      "CRUK0002           0    0      0  \n",
      "CRUK0003           0    0      0  \n",
      "CRUK0004           0    0      0  \n",
      "CRUK0005           0    0      0  \n",
      "...              ...  ...    ...  \n",
      "CRUK0810           0    0      0  \n",
      "CRUK0817           0    0      0  \n",
      "CRUK0864           0    0      0  \n",
      "CRUK0881           0    0      0  \n",
      "CRUK0893           0    0      0  \n",
      "\n",
      "[242 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUAD_fifty_feature_matrix with only LUAD_patient_id\n",
    "\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUAD_top_50_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUAD_patient_id)]\n",
    "\n",
    "LUAD_fifty_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUAD_fifty_feature_matrix = LUAD_fifty_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUAD_patient_id) - set(LUAD_fifty_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUAD_fifty_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUAD_fifty_feature_matrix = LUAD_fifty_feature_matrix.sort_index()\n",
    "print(LUAD_fifty_feature_matrix.shape)\n",
    "print(LUAD_fifty_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 20)\n",
      "Hugo_Symbol  ARID1B  CREBBP  CTNNB1  ERBB2  EZH1  FLCN  FN1  KDM5C  KEAP1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001          0       0       0      0     0     0    0      0      0   \n",
      "CRUK0002          0       0       0      0     0     0    0      0      0   \n",
      "CRUK0003          0       0       1      0     0     0    0      0      0   \n",
      "CRUK0004          0       0       0      0     0     0    0      0      0   \n",
      "CRUK0005          0       0       0      0     0     0    0      0      0   \n",
      "...             ...     ...     ...    ...   ...   ...  ...    ...    ...   \n",
      "CRUK0810          0       0       0      0     0     0    0      0      0   \n",
      "CRUK0817          0       0       0      0     0     0    0      0      0   \n",
      "CRUK0864          0       0       0      0     0     0    0      0      0   \n",
      "CRUK0881          0       1       0      0     0     0    0      0      0   \n",
      "CRUK0893          0       0       0      0     0     0    0      0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  MTOR  PBRM1  PRDM1  PTPRB  PTPRU  SETD2  SMARCA4  STIM1  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0     0      0      0      0      0      0        0      0   \n",
      "CRUK0002         0     0      0      0      0      0      0        0      0   \n",
      "CRUK0003         0     0      0      0      0      0      0        0      0   \n",
      "CRUK0004         0     0      0      0      0      0      0        0      0   \n",
      "CRUK0005         0     0      0      0      0      0      0        0      0   \n",
      "...            ...   ...    ...    ...    ...    ...    ...      ...    ...   \n",
      "CRUK0810         0     0      0      0      0      0      0        1      0   \n",
      "CRUK0817         0     0      0      0      0      0      0        0      0   \n",
      "CRUK0864         0     0      0      0      0      0      0        0      0   \n",
      "CRUK0881         0     0      1      0      0      0      0        0      0   \n",
      "CRUK0893         0     0      0      0      0      0      0        0      0   \n",
      "\n",
      "Hugo_Symbol  STK11  WT1  \n",
      "patient_id               \n",
      "CRUK0001         0    0  \n",
      "CRUK0002         0    0  \n",
      "CRUK0003         0    0  \n",
      "CRUK0004         0    0  \n",
      "CRUK0005         0    0  \n",
      "...            ...  ...  \n",
      "CRUK0810         1    0  \n",
      "CRUK0817         1    0  \n",
      "CRUK0864         0    0  \n",
      "CRUK0881         0    0  \n",
      "CRUK0893         0    0  \n",
      "\n",
      "[242 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUAD_twenty_feature_matrix with only LUAD_patient_id\n",
    "\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUAD_top_20_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUAD_patient_id)]\n",
    "\n",
    "LUAD_twenty_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUAD_twenty_feature_matrix = LUAD_twenty_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUAD_patient_id) - set(LUAD_twenty_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUAD_twenty_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUAD_twenty_feature_matrix = LUAD_twenty_feature_matrix.sort_index()\n",
    "print(LUAD_twenty_feature_matrix.shape)\n",
    "print(LUAD_twenty_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 10)\n",
      "Hugo_Symbol  CREBBP  CTNNB1  EZH1  KDM5C  KEAP1  KMT2D  PRDM1  SETD2  SMARCA4  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001          0       0     0      0      0      0      0      0        0   \n",
      "CRUK0002          0       0     0      0      0      0      0      0        0   \n",
      "CRUK0003          0       1     0      0      0      0      0      0        0   \n",
      "CRUK0004          0       0     0      0      0      0      0      0        0   \n",
      "CRUK0005          0       0     0      0      0      0      0      0        0   \n",
      "...             ...     ...   ...    ...    ...    ...    ...    ...      ...   \n",
      "CRUK0810          0       0     0      0      0      0      0      0        1   \n",
      "CRUK0817          0       0     0      0      0      0      0      0        0   \n",
      "CRUK0864          0       0     0      0      0      0      0      0        0   \n",
      "CRUK0881          1       0     0      0      0      0      0      0        0   \n",
      "CRUK0893          0       0     0      0      0      0      0      0        0   \n",
      "\n",
      "Hugo_Symbol  STK11  \n",
      "patient_id          \n",
      "CRUK0001         0  \n",
      "CRUK0002         0  \n",
      "CRUK0003         0  \n",
      "CRUK0004         0  \n",
      "CRUK0005         0  \n",
      "...            ...  \n",
      "CRUK0810         1  \n",
      "CRUK0817         1  \n",
      "CRUK0864         0  \n",
      "CRUK0881         0  \n",
      "CRUK0893         0  \n",
      "\n",
      "[242 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUAD_ten_feature_matrix with only LUAD_patient_id\n",
    "\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUAD_top_10_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUAD_patient_id)]\n",
    "\n",
    "LUAD_ten_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUAD_ten_feature_matrix = LUAD_ten_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUAD_patient_id) - set(LUAD_ten_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUAD_ten_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUAD_ten_feature_matrix = LUAD_ten_feature_matrix.sort_index()\n",
    "print(LUAD_ten_feature_matrix.shape)\n",
    "print(LUAD_ten_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 5)\n",
      "Hugo_Symbol  CREBBP  KDM5C  KMT2D  SMARCA4  STK11\n",
      "patient_id                                       \n",
      "CRUK0001          0      0      0        0      0\n",
      "CRUK0002          0      0      0        0      0\n",
      "CRUK0003          0      0      0        0      0\n",
      "CRUK0004          0      0      0        0      0\n",
      "CRUK0005          0      0      0        0      0\n",
      "...             ...    ...    ...      ...    ...\n",
      "CRUK0810          0      0      0        1      1\n",
      "CRUK0817          0      0      0        0      1\n",
      "CRUK0864          0      0      0        0      0\n",
      "CRUK0881          1      0      0        0      0\n",
      "CRUK0893          0      0      0        0      0\n",
      "\n",
      "[242 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUAD_five_feature_matrix with only LUAD_patient_id\n",
    "\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUAD_top_5_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUAD_patient_id)]\n",
    "\n",
    "LUAD_five_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUAD_five_feature_matrix = LUAD_five_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUAD_patient_id) - set(LUAD_five_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUAD_five_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUAD_five_feature_matrix = LUAD_five_feature_matrix.sort_index()\n",
    "print(LUAD_five_feature_matrix.shape)\n",
    "print(LUAD_five_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 100)\n",
      "Hugo_Symbol  AMER1  APC  ARHGAP35  ARID1A  ARID1B  ARID2  ASXL1  AXIN1  B2M  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062         0    0         0       0       0      0      0      0    0   \n",
      "CRUK0063         0    0         0       0       0      0      0      0    0   \n",
      "CRUK0064         0    0         0       0       0      0      0      0    0   \n",
      "CRUK0065         0    0         0       1       0      0      0      0    0   \n",
      "CRUK0066         0    0         0       0       0      0      0      0    0   \n",
      "...            ...  ...       ...     ...     ...    ...    ...    ...  ...   \n",
      "CRUK0772         0    0         0       0       0      0      0      0    0   \n",
      "CRUK0781         0    0         0       0       0      0      0      0    0   \n",
      "CRUK0784         0    0         0       0       0      0      0      0    0   \n",
      "CRUK0861         0    0         0       0       0      0      0      0    0   \n",
      "CRUK0872         0    0         0       0       0      0      0      0    0   \n",
      "\n",
      "Hugo_Symbol  BAP1  BCOR  BLM  BMPR1A  BRCA1  BUB1B  CAD  CBL  CBLB  CDK12  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062        0     0    1       0      0      0    0    0     0      0   \n",
      "CRUK0063        0     0    0       0      0      0    0    0     0      0   \n",
      "CRUK0064        0     0    0       0      0      0    0    0     0      0   \n",
      "CRUK0065        0     0    0       0      0      0    0    0     0      0   \n",
      "CRUK0066        0     0    0       0      0      0    0    0     0      0   \n",
      "...           ...   ...  ...     ...    ...    ...  ...  ...   ...    ...   \n",
      "CRUK0772        0     0    0       0      0      0    0    0     0      0   \n",
      "CRUK0781        0     0    0       0      0      0    0    0     0      0   \n",
      "CRUK0784        0     0    0       0      0      0    0    0     0      0   \n",
      "CRUK0861        0     0    0       0      0      0    0    0     0      0   \n",
      "CRUK0872        0     0    0       0      0      0    0    0     0      0   \n",
      "\n",
      "Hugo_Symbol  CDKN2A  CEP76  CHD4  CIC  CNOT3  COL2A1  COL5A2  CSMD3  CUX1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062          0      0     1    0      0       0       0      0     0   \n",
      "CRUK0063          1      0     0    0      0       0       0      0     0   \n",
      "CRUK0064          0      0     0    0      0       0       0      0     0   \n",
      "CRUK0065          0      0     0    0      0       0       0      0     0   \n",
      "CRUK0066          1      0     0    0      0       0       1      1     0   \n",
      "...             ...    ...   ...  ...    ...     ...     ...    ...   ...   \n",
      "CRUK0772          0      0     0    0      0       0       0      0     0   \n",
      "CRUK0781          0      0     0    0      0       0       0      0     0   \n",
      "CRUK0784          0      0     1    0      0       0       0      0     0   \n",
      "CRUK0861          0      0     0    0      0       0       0      0     0   \n",
      "CRUK0872          0      0     0    0      0       0       0      0     0   \n",
      "\n",
      "Hugo_Symbol  CYLD  DDX3X  DOT1L  EGFR  EP300  ERCC2  EXT1  EZH1  FAM166A  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062        0      0      0     0      0      0     0     0        0   \n",
      "CRUK0063        1      0      0     0      1      0     0     1        0   \n",
      "CRUK0064        0      0      0     0      0      0     0     0        0   \n",
      "CRUK0065        0      0      0     0      0      0     0     0        1   \n",
      "CRUK0066        0      0      0     0      0      0     0     0        0   \n",
      "...           ...    ...    ...   ...    ...    ...   ...   ...      ...   \n",
      "CRUK0772        0      0      0     0      0      0     0     0        0   \n",
      "CRUK0781        0      0      0     0      0      1     0     0        0   \n",
      "CRUK0784        0      0      0     0      0      0     0     0        0   \n",
      "CRUK0861        0      0      0     0      0      0     0     0        0   \n",
      "CRUK0872        0      0      0     0      0      0     0     0        0   \n",
      "\n",
      "Hugo_Symbol  FANCM  FAS  FAT1  FBXO11  FBXW7  FGFBP1  FGFR3  FLCN  FLT4  \\\n",
      "patient_id                                                                \n",
      "CRUK0062         0    1     0       0      0       0      0     0     0   \n",
      "CRUK0063         1    0     0       0      1       0      0     0     0   \n",
      "CRUK0064         0    0     1       0      0       0      0     0     0   \n",
      "CRUK0065         0    0     0       0      0       0      0     0     0   \n",
      "CRUK0066         0    0     0       0      0       0      0     0     0   \n",
      "...            ...  ...   ...     ...    ...     ...    ...   ...   ...   \n",
      "CRUK0772         1    0     0       0      0       0      0     0     0   \n",
      "CRUK0781         0    0     0       0      0       0      0     0     0   \n",
      "CRUK0784         0    0     0       0      0       0      0     0     0   \n",
      "CRUK0861         0    0     0       0      0       0      0     0     0   \n",
      "CRUK0872         0    0     0       0      0       0      0     0     0   \n",
      "\n",
      "Hugo_Symbol  GATA3  GNPTAB  HCN1  HNF1A  IKZF1  IL21R  KDM6A  KLF5  KMT2B  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062         0       0     0      0      0      0      0     0      0   \n",
      "CRUK0063         0       1     0      0      0      0      0     1      0   \n",
      "CRUK0064         0       0     0      0      0      0      0     0      0   \n",
      "CRUK0065         0       0     0      0      0      0      0     0      0   \n",
      "CRUK0066         0       0     0      0      0      0      0     0      0   \n",
      "...            ...     ...   ...    ...    ...    ...    ...   ...    ...   \n",
      "CRUK0772         0       0     0      0      0      0      0     0      0   \n",
      "CRUK0781         0       0     0      0      0      0      0     0      0   \n",
      "CRUK0784         0       0     0      0      0      0      0     0      0   \n",
      "CRUK0861         0       0     0      0      0      0      0     0      0   \n",
      "CRUK0872         0       0     0      0      0      0      0     0      1   \n",
      "\n",
      "Hugo_Symbol  KMT2D  LAMA4  LATS1  MAP3K1  MAP3K13  MSH2  MTOR  NCOA6  NF1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062         0      0      0       0        0     0     0      0    0   \n",
      "CRUK0063         0      0      0       0        0     1     0      0    1   \n",
      "CRUK0064         0      0      0       0        0     0     0      0    0   \n",
      "CRUK0065         0      1      0       0        0     0     0      1    0   \n",
      "CRUK0066         0      0      0       0        0     0     0      0    0   \n",
      "...            ...    ...    ...     ...      ...   ...   ...    ...  ...   \n",
      "CRUK0772         0      0      0       0        0     0     0      0    0   \n",
      "CRUK0781         1      0      0       0        0     0     0      0    0   \n",
      "CRUK0784         0      0      0       0        0     0     0      0    0   \n",
      "CRUK0861         0      0      0       0        0     0     0      0    0   \n",
      "CRUK0872         0      0      0       0        0     0     0      0    0   \n",
      "\n",
      "Hugo_Symbol  NFKBIE  NIN  NRAS  NSD1  PASK  PBRM1  PDYN  PIK3CA  PIK3R1  \\\n",
      "patient_id                                                                \n",
      "CRUK0062          0    0     0     1     0      0     0       0       0   \n",
      "CRUK0063          0    0     0     0     0      0     0       1       0   \n",
      "CRUK0064          0    0     0     0     0      0     0       0       0   \n",
      "CRUK0065          0    1     0     0     0      0     0       1       0   \n",
      "CRUK0066          0    0     0     0     0      0     0       0       0   \n",
      "...             ...  ...   ...   ...   ...    ...   ...     ...     ...   \n",
      "CRUK0772          0    0     0     0     0      0     0       0       0   \n",
      "CRUK0781          0    0     0     0     1      0     0       0       0   \n",
      "CRUK0784          0    0     0     0     0      0     0       0       0   \n",
      "CRUK0861          0    0     0     0     0      0     0       0       0   \n",
      "CRUK0872          0    0     0     0     0      0     0       0       0   \n",
      "\n",
      "Hugo_Symbol  PLXNB2  POLE  PPP3CA  PRDM1  PRKAR1A  PTEN  PTPRB  PTPRC  RASA1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062          1     0       0      0        0     0      0      0      0   \n",
      "CRUK0063          0     0       0      0        0     0      0      0      0   \n",
      "CRUK0064          0     0       0      0        0     0      0      0      0   \n",
      "CRUK0065          0     0       0      0        0     0      0      1      0   \n",
      "CRUK0066          0     0       0      0        0     0      0      0      0   \n",
      "...             ...   ...     ...    ...      ...   ...    ...    ...    ...   \n",
      "CRUK0772          0     0       0      0        0     0      0      0      0   \n",
      "CRUK0781          0     0       0      0        0     1      0      0      0   \n",
      "CRUK0784          0     0       0      0        0     0      0      0      0   \n",
      "CRUK0861          0     0       0      0        0     0      0      0      0   \n",
      "CRUK0872          0     0       0      0        0     0      1      0      0   \n",
      "\n",
      "Hugo_Symbol  RNF43  RYR2  SEPT12  SETD2  SGK223  SMAD4  SNX29  SPEN  STAG2  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062         0     0       0      0       0      0      0     0      0   \n",
      "CRUK0063         0     0       0      0       0      0      0     0      0   \n",
      "CRUK0064         0     0       0      0       0      0      0     0      0   \n",
      "CRUK0065         0     0       1      0       0      0      0     0      0   \n",
      "CRUK0066         0     0       0      0       0      0      0     0      0   \n",
      "...            ...   ...     ...    ...     ...    ...    ...   ...    ...   \n",
      "CRUK0772         0     0       0      0       0      0      0     0      0   \n",
      "CRUK0781         0     0       0      0       0      0      0     0      0   \n",
      "CRUK0784         0     0       0      0       0      0      0     0      0   \n",
      "CRUK0861         0     0       0      0       0      0      0     0      0   \n",
      "CRUK0872         0     0       0      0       0      0      0     0      0   \n",
      "\n",
      "Hugo_Symbol  STK11  STX2  TNFAIP3  UBR5  WAS  WT1  ZFP36L1  ZMYM2  ZNF483  \n",
      "patient_id                                                                 \n",
      "CRUK0062         0     0        0     1    0    0        0      0       0  \n",
      "CRUK0063         0     0        0     0    0    0        0      0       0  \n",
      "CRUK0064         0     0        0     0    0    0        0      0       0  \n",
      "CRUK0065         0     1        0     1    0    0        0      0       0  \n",
      "CRUK0066         0     0        0     0    0    0        0      0       0  \n",
      "...            ...   ...      ...   ...  ...  ...      ...    ...     ...  \n",
      "CRUK0772         0     0        0     0    0    0        0      0       0  \n",
      "CRUK0781         0     0        0     0    0    0        0      0       0  \n",
      "CRUK0784         0     0        0     0    0    0        0      0       0  \n",
      "CRUK0861         0     0        0     0    0    0        0      0       0  \n",
      "CRUK0872         0     0        0     0    0    0        0      0       0  \n",
      "\n",
      "[138 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUSC_hundred_feature_matrix with only LUSC_patient_id\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUSC_top_100_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUSC_patient_id)]\n",
    "\n",
    "LUSC_hundred_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUSC_hundred_feature_matrix = LUSC_hundred_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUSC_patient_id) - set(LUSC_hundred_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUSC_hundred_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUSC_hundred_feature_matrix = LUSC_hundred_feature_matrix.sort_index()\n",
    "print(LUSC_hundred_feature_matrix.shape)\n",
    "print(LUSC_hundred_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 50)\n",
      "Hugo_Symbol  APC  ARHGAP35  ARID1A  ARID2  ASXL1  B2M  BCOR  BLM  BMPR1A  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062       0         0       0      0      0    0     0    1       0   \n",
      "CRUK0063       0         0       0      0      0    0     0    0       0   \n",
      "CRUK0064       0         0       0      0      0    0     0    0       0   \n",
      "CRUK0065       0         0       1      0      0    0     0    0       0   \n",
      "CRUK0066       0         0       0      0      0    0     0    0       0   \n",
      "...          ...       ...     ...    ...    ...  ...   ...  ...     ...   \n",
      "CRUK0772       0         0       0      0      0    0     0    0       0   \n",
      "CRUK0781       0         0       0      0      0    0     0    0       0   \n",
      "CRUK0784       0         0       0      0      0    0     0    0       0   \n",
      "CRUK0861       0         0       0      0      0    0     0    0       0   \n",
      "CRUK0872       0         0       0      0      0    0     0    0       0   \n",
      "\n",
      "Hugo_Symbol  BUB1B  CDK12  CDKN2A  CHD4  CUX1  CYLD  EZH1  FAM166A  FAS  FAT1  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062         0      0       0     1     0     0     0        0    1     0   \n",
      "CRUK0063         0      0       1     0     0     1     1        0    0     0   \n",
      "CRUK0064         0      0       0     0     0     0     0        0    0     1   \n",
      "CRUK0065         0      0       0     0     0     0     0        1    0     0   \n",
      "CRUK0066         0      0       1     0     0     0     0        0    0     0   \n",
      "...            ...    ...     ...   ...   ...   ...   ...      ...  ...   ...   \n",
      "CRUK0772         0      0       0     0     0     0     0        0    0     0   \n",
      "CRUK0781         0      0       0     0     0     0     0        0    0     0   \n",
      "CRUK0784         0      0       0     1     0     0     0        0    0     0   \n",
      "CRUK0861         0      0       0     0     0     0     0        0    0     0   \n",
      "CRUK0872         0      0       0     0     0     0     0        0    0     0   \n",
      "\n",
      "Hugo_Symbol  FGFR3  FLCN  FLT4  GNPTAB  HCN1  HNF1A  KLF5  KMT2D  LAMA4  \\\n",
      "patient_id                                                                \n",
      "CRUK0062         0     0     0       0     0      0     0      0      0   \n",
      "CRUK0063         0     0     0       1     0      0     1      0      0   \n",
      "CRUK0064         0     0     0       0     0      0     0      0      0   \n",
      "CRUK0065         0     0     0       0     0      0     0      0      1   \n",
      "CRUK0066         0     0     0       0     0      0     0      0      0   \n",
      "...            ...   ...   ...     ...   ...    ...   ...    ...    ...   \n",
      "CRUK0772         0     0     0       0     0      0     0      0      0   \n",
      "CRUK0781         0     0     0       0     0      0     0      1      0   \n",
      "CRUK0784         0     0     0       0     0      0     0      0      0   \n",
      "CRUK0861         0     0     0       0     0      0     0      0      0   \n",
      "CRUK0872         0     0     0       0     0      0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  MAP3K13  MSH2  MTOR  NCOA6  NF1  NIN  NSD1  PASK  PBRM1  PIK3CA  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062           0     0     0      0    0    0     1     0      0       0   \n",
      "CRUK0063           0     1     0      0    1    0     0     0      0       1   \n",
      "CRUK0064           0     0     0      0    0    0     0     0      0       0   \n",
      "CRUK0065           0     0     0      1    0    1     0     0      0       1   \n",
      "CRUK0066           0     0     0      0    0    0     0     0      0       0   \n",
      "...              ...   ...   ...    ...  ...  ...   ...   ...    ...     ...   \n",
      "CRUK0772           0     0     0      0    0    0     0     0      0       0   \n",
      "CRUK0781           0     0     0      0    0    0     0     1      0       0   \n",
      "CRUK0784           0     0     0      0    0    0     0     0      0       0   \n",
      "CRUK0861           0     0     0      0    0    0     0     0      0       0   \n",
      "CRUK0872           0     0     0      0    0    0     0     0      0       0   \n",
      "\n",
      "Hugo_Symbol  PIK3R1  PLXNB2  PTPRB  RASA1  RNF43  SEPT12  SNX29  SPEN  STAG2  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062          0       1      0      0      0       0      0     0      0   \n",
      "CRUK0063          0       0      0      0      0       0      0     0      0   \n",
      "CRUK0064          0       0      0      0      0       0      0     0      0   \n",
      "CRUK0065          0       0      0      0      0       1      0     0      0   \n",
      "CRUK0066          0       0      0      0      0       0      0     0      0   \n",
      "...             ...     ...    ...    ...    ...     ...    ...   ...    ...   \n",
      "CRUK0772          0       0      0      0      0       0      0     0      0   \n",
      "CRUK0781          0       0      0      0      0       0      0     0      0   \n",
      "CRUK0784          0       0      0      0      0       0      0     0      0   \n",
      "CRUK0861          0       0      0      0      0       0      0     0      0   \n",
      "CRUK0872          0       0      1      0      0       0      0     0      0   \n",
      "\n",
      "Hugo_Symbol  STX2  WT1  ZNF483  \n",
      "patient_id                      \n",
      "CRUK0062        0    0       0  \n",
      "CRUK0063        0    0       0  \n",
      "CRUK0064        0    0       0  \n",
      "CRUK0065        1    0       0  \n",
      "CRUK0066        0    0       0  \n",
      "...           ...  ...     ...  \n",
      "CRUK0772        0    0       0  \n",
      "CRUK0781        0    0       0  \n",
      "CRUK0784        0    0       0  \n",
      "CRUK0861        0    0       0  \n",
      "CRUK0872        0    0       0  \n",
      "\n",
      "[138 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUSC_fifty_feature_matrix with only LUSC_patient_id\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUSC_top_50_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUSC_patient_id)]\n",
    "\n",
    "LUSC_fifty_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUSC_fifty_feature_matrix = LUSC_fifty_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUSC_patient_id) - set(LUSC_fifty_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUSC_fifty_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUSC_fifty_feature_matrix = LUSC_fifty_feature_matrix.sort_index()\n",
    "print(LUSC_fifty_feature_matrix.shape)\n",
    "print(LUSC_fifty_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 20)\n",
      "Hugo_Symbol  ARHGAP35  ARID2  ASXL1  B2M  BCOR  CHD4  CUX1  KLF5  KMT2D  \\\n",
      "patient_id                                                                \n",
      "CRUK0062            0      0      0    0     0     1     0     0      0   \n",
      "CRUK0063            0      0      0    0     0     0     0     1      0   \n",
      "CRUK0064            0      0      0    0     0     0     0     0      0   \n",
      "CRUK0065            0      0      0    0     0     0     0     0      0   \n",
      "CRUK0066            0      0      0    0     0     0     0     0      0   \n",
      "...               ...    ...    ...  ...   ...   ...   ...   ...    ...   \n",
      "CRUK0772            0      0      0    0     0     0     0     0      0   \n",
      "CRUK0781            0      0      0    0     0     0     0     0      1   \n",
      "CRUK0784            0      0      0    0     0     1     0     0      0   \n",
      "CRUK0861            0      0      0    0     0     0     0     0      0   \n",
      "CRUK0872            0      0      0    0     0     0     0     0      0   \n",
      "\n",
      "Hugo_Symbol  MAP3K13  MSH2  MTOR  NCOA6  NF1  PASK  PBRM1  PIK3CA  PIK3R1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062           0     0     0      0    0     0      0       0       0   \n",
      "CRUK0063           0     1     0      0    1     0      0       1       0   \n",
      "CRUK0064           0     0     0      0    0     0      0       0       0   \n",
      "CRUK0065           0     0     0      1    0     0      0       1       0   \n",
      "CRUK0066           0     0     0      0    0     0      0       0       0   \n",
      "...              ...   ...   ...    ...  ...   ...    ...     ...     ...   \n",
      "CRUK0772           0     0     0      0    0     0      0       0       0   \n",
      "CRUK0781           0     0     0      0    0     1      0       0       0   \n",
      "CRUK0784           0     0     0      0    0     0      0       0       0   \n",
      "CRUK0861           0     0     0      0    0     0      0       0       0   \n",
      "CRUK0872           0     0     0      0    0     0      0       0       0   \n",
      "\n",
      "Hugo_Symbol  PLXNB2  PTPRB  \n",
      "patient_id                  \n",
      "CRUK0062          1      0  \n",
      "CRUK0063          0      0  \n",
      "CRUK0064          0      0  \n",
      "CRUK0065          0      0  \n",
      "CRUK0066          0      0  \n",
      "...             ...    ...  \n",
      "CRUK0772          0      0  \n",
      "CRUK0781          0      0  \n",
      "CRUK0784          0      0  \n",
      "CRUK0861          0      0  \n",
      "CRUK0872          0      1  \n",
      "\n",
      "[138 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUSC_twenty_feature_matrix with only LUSC_patient_id\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUSC_top_20_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUSC_patient_id)]\n",
    "\n",
    "LUSC_twenty_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUSC_twenty_feature_matrix = LUSC_twenty_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUSC_patient_id) - set(LUSC_twenty_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUSC_twenty_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUSC_twenty_feature_matrix = LUSC_twenty_feature_matrix.sort_index()\n",
    "print(LUSC_twenty_feature_matrix.shape)\n",
    "print(LUSC_twenty_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 10)\n",
      "Hugo_Symbol  ARID2  B2M  BCOR  KMT2D  MAP3K13  MTOR  NCOA6  NF1  PBRM1  PIK3CA\n",
      "patient_id                                                                    \n",
      "CRUK0062         0    0     0      0        0     0      0    0      0       0\n",
      "CRUK0063         0    0     0      0        0     0      0    1      0       1\n",
      "CRUK0064         0    0     0      0        0     0      0    0      0       0\n",
      "CRUK0065         0    0     0      0        0     0      1    0      0       1\n",
      "CRUK0066         0    0     0      0        0     0      0    0      0       0\n",
      "...            ...  ...   ...    ...      ...   ...    ...  ...    ...     ...\n",
      "CRUK0772         0    0     0      0        0     0      0    0      0       0\n",
      "CRUK0781         0    0     0      1        0     0      0    0      0       0\n",
      "CRUK0784         0    0     0      0        0     0      0    0      0       0\n",
      "CRUK0861         0    0     0      0        0     0      0    0      0       0\n",
      "CRUK0872         0    0     0      0        0     0      0    0      0       0\n",
      "\n",
      "[138 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUSC_ten_feature_matrix with only LUSC_patient_id\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUSC_top_10_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUSC_patient_id)]\n",
    "\n",
    "LUSC_ten_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUSC_ten_feature_matrix = LUSC_ten_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUSC_patient_id) - set(LUSC_ten_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUSC_ten_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUSC_ten_feature_matrix = LUSC_ten_feature_matrix.sort_index()\n",
    "print(LUSC_ten_feature_matrix.shape)\n",
    "print(LUSC_ten_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 5)\n",
      "Hugo_Symbol  ARID2  B2M  BCOR  KMT2D  MAP3K13\n",
      "patient_id                                   \n",
      "CRUK0062         0    0     0      0        0\n",
      "CRUK0063         0    0     0      0        0\n",
      "CRUK0064         0    0     0      0        0\n",
      "CRUK0065         0    0     0      0        0\n",
      "CRUK0066         0    0     0      0        0\n",
      "...            ...  ...   ...    ...      ...\n",
      "CRUK0772         0    0     0      0        0\n",
      "CRUK0781         0    0     0      1        0\n",
      "CRUK0784         0    0     0      0        0\n",
      "CRUK0861         0    0     0      0        0\n",
      "CRUK0872         0    0     0      0        0\n",
      "\n",
      "[138 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "# Making the LUSC_five_feature_matrix with only LUSC_patient_id\n",
    "filtered_drivers_table = mut_table[mut_table['Hugo_Symbol'].isin(LUSC_top_5_genes)]\n",
    "filtered_drivers_table = filtered_drivers_table[filtered_drivers_table['patient_id'].isin(LUSC_patient_id)]\n",
    "\n",
    "LUSC_five_feature_matrix = pd.crosstab(filtered_drivers_table['patient_id'], filtered_drivers_table['Hugo_Symbol'], values=filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "LUSC_five_feature_matrix = LUSC_five_feature_matrix.astype(int)\n",
    "\n",
    "# Fill in missing patients with 0s\n",
    "missing_patients = list(set(LUSC_patient_id) - set(LUSC_five_feature_matrix.index))\n",
    "for patient in missing_patients:\n",
    "    LUSC_five_feature_matrix.loc[patient] = 0\n",
    "\n",
    "LUSC_five_feature_matrix = LUSC_five_feature_matrix.sort_index()\n",
    "print(LUSC_five_feature_matrix.shape)\n",
    "print(LUSC_five_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cruk_id</th>\n",
       "      <th>tumour_id_muttable_cruk</th>\n",
       "      <th>tumour_id_per_patient</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>packyears</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>ECOG_PS</th>\n",
       "      <th>pathologyTNM</th>\n",
       "      <th>pT_stage_per_patient</th>\n",
       "      <th>pN_stage_per_patient</th>\n",
       "      <th>LVI_per_patient</th>\n",
       "      <th>PL_per_patient</th>\n",
       "      <th>margin_status_per_patient</th>\n",
       "      <th>size_pathology_per_patient</th>\n",
       "      <th>Surgery_type</th>\n",
       "      <th>histology_lesion1</th>\n",
       "      <th>histology_lesion1_merged</th>\n",
       "      <th>lesion1_sampled</th>\n",
       "      <th>histology_lesion2</th>\n",
       "      <th>lesion2_sampled</th>\n",
       "      <th>histology_multi_full</th>\n",
       "      <th>histology_multi_full_genomically.confirmed</th>\n",
       "      <th>LUAD_pred_subtype</th>\n",
       "      <th>adjuvant_treatment_YN</th>\n",
       "      <th>adjuvant_treatment_given</th>\n",
       "      <th>num_cycle_na.added</th>\n",
       "      <th>CHMPlatDgName_cleaned</th>\n",
       "      <th>CHMOthDgName_cleaned</th>\n",
       "      <th>AdjRadStartTime_manual</th>\n",
       "      <th>AdjRadEndTime_manual</th>\n",
       "      <th>Recurrence_time_use</th>\n",
       "      <th>newPrim_time_use</th>\n",
       "      <th>first_dfs_any_event_rec.or.new.primary</th>\n",
       "      <th>first_event_during_followup</th>\n",
       "      <th>cens_os</th>\n",
       "      <th>os_time</th>\n",
       "      <th>cens_dfs</th>\n",
       "      <th>dfs_time</th>\n",
       "      <th>cens_dfs_any_event</th>\n",
       "      <th>dfs_time_any_event</th>\n",
       "      <th>cens_lung_event</th>\n",
       "      <th>lung_event_time</th>\n",
       "      <th>Relapse_cat</th>\n",
       "      <th>Relapse_cat_new</th>\n",
       "      <th>tx100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>30</td>\n",
       "      <td>30.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>0</td>\n",
       "      <td>1120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>81</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>5.0</td>\n",
       "      <td>35</td>\n",
       "      <td>8.75</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>217.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>2304</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>69</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>325.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>1323</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>74</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>56</td>\n",
       "      <td>56.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>0</td>\n",
       "      <td>649</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>65</td>\n",
       "      <td>Female</td>\n",
       "      <td>Middle eastern</td>\n",
       "      <td>10.0</td>\n",
       "      <td>45</td>\n",
       "      <td>22.50</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1b</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>30</td>\n",
       "      <td>15.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>66</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>54</td>\n",
       "      <td>54.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>883.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>893</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881_Tumour1;CRUK0881_Tumour2</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>Lesion1: collision tumour</td>\n",
       "      <td>False</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>758</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>15.0</td>\n",
       "      <td>43</td>\n",
       "      <td>32.25</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>421 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      cruk_id tumour_id_muttable_cruk              tumour_id_per_patient  age  \\\n",
       "88   CRUK0001                CRUK0001                           CRUK0001   84   \n",
       "79   CRUK0002                CRUK0002                           CRUK0002   81   \n",
       "26   CRUK0003                CRUK0003                           CRUK0003   78   \n",
       "23   CRUK0004                CRUK0004                           CRUK0004   69   \n",
       "7    CRUK0005                CRUK0005                           CRUK0005   74   \n",
       "..        ...                     ...                                ...  ...   \n",
       "414  CRUK0861                CRUK0861                           CRUK0861   65   \n",
       "418  CRUK0864                CRUK0864                           CRUK0864   78   \n",
       "416  CRUK0872                CRUK0872                           CRUK0872   66   \n",
       "420  CRUK0881        CRUK0881_Tumour1  CRUK0881_Tumour1;CRUK0881_Tumour2   84   \n",
       "415  CRUK0893                CRUK0893                           CRUK0893   59   \n",
       "\n",
       "        sex       ethnicity  cigs_perday  years_smoking  packyears  \\\n",
       "88     Male  White- British         20.0             30      30.00   \n",
       "79     Male  White- British          5.0             35       8.75   \n",
       "26   Female  White- British          0.0              0       0.00   \n",
       "23   Female  White- British         10.0             27      13.50   \n",
       "7    Female  White- British         20.0             56      56.00   \n",
       "..      ...             ...          ...            ...        ...   \n",
       "414  Female  Middle eastern         10.0             45      22.50   \n",
       "418  Female  White- British         10.0             30      15.00   \n",
       "416    Male  White- British         20.0             54      54.00   \n",
       "420    Male  White- British         20.0             70      70.00   \n",
       "415  Female  White- British         15.0             43      32.25   \n",
       "\n",
       "    smoking_status_merged  is.family.lung  ECOG_PS pathologyTNM  \\\n",
       "88              Ex-Smoker           False      1.0          IIA   \n",
       "79              Ex-Smoker           False      1.0           IB   \n",
       "26           Never Smoked           False      0.0         IIIA   \n",
       "23              Ex-Smoker            True      0.0          IIB   \n",
       "7               Ex-Smoker           False      1.0         IIIA   \n",
       "..                    ...             ...      ...          ...   \n",
       "414                Smoker           False      1.0           IA   \n",
       "418             Ex-Smoker            True      0.0          IIA   \n",
       "416             Ex-Smoker           False      1.0         IIIA   \n",
       "420                Smoker           False      1.0          IIA   \n",
       "415                Smoker           False      1.0           IA   \n",
       "\n",
       "    pT_stage_per_patient  pN_stage_per_patient LVI_per_patient PL_per_patient  \\\n",
       "88                    2a                     1              No            Yes   \n",
       "79                    2a                     0             Yes             No   \n",
       "26                    2b                     2             Yes             No   \n",
       "23                     3                     0              No            Yes   \n",
       "7                     2a                     2             Yes            Yes   \n",
       "..                   ...                   ...             ...            ...   \n",
       "414                   1b                     0              No             No   \n",
       "418                   1b                     1              No             No   \n",
       "416                   1b                     2              No             No   \n",
       "420                   2b                     0             Yes            Yes   \n",
       "415                   1a                     0             Yes             No   \n",
       "\n",
       "    margin_status_per_patient  size_pathology_per_patient Surgery_type  \\\n",
       "88                         R0                        26.0    Lobectomy   \n",
       "79                         R0                        35.0    Lobectomy   \n",
       "26                         R0                        52.0    Lobectomy   \n",
       "23                         R0                        47.0    Lobectomy   \n",
       "7                          R0                        43.0    Lobectomy   \n",
       "..                        ...                         ...          ...   \n",
       "414                        R0                        25.0    Lobectomy   \n",
       "418                        R0                        27.0    Lobectomy   \n",
       "416                        R0                        25.0    Lobectomy   \n",
       "420                        R0                        62.0    Lobectomy   \n",
       "415                        R0                        17.0    Lobectomy   \n",
       "\n",
       "           histology_lesion1 histology_lesion1_merged  lesion1_sampled  \\\n",
       "88   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "79   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "26   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "23   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "7    Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "..                       ...                      ...              ...   \n",
       "414  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "418  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "416  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "420  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "415  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "\n",
       "             histology_lesion2 lesion2_sampled histology_multi_full  \\\n",
       "88                         NaN             NaN                 LUAD   \n",
       "79                         NaN             NaN                 LUAD   \n",
       "26                         NaN             NaN                 LUAD   \n",
       "23                         NaN             NaN                 LUAD   \n",
       "7                          NaN             NaN                 LUAD   \n",
       "..                         ...             ...                  ...   \n",
       "414                        NaN             NaN                 LUSC   \n",
       "418                        NaN             NaN                 LUAD   \n",
       "416                        NaN             NaN                 LUSC   \n",
       "420  Lesion1: collision tumour           False               LUADx2   \n",
       "415                        NaN             NaN                 LUAD   \n",
       "\n",
       "    histology_multi_full_genomically.confirmed LUAD_pred_subtype  \\\n",
       "88                                        LUAD            acinar   \n",
       "79                                        LUAD            acinar   \n",
       "26                                        LUAD            acinar   \n",
       "23                                        LUAD            acinar   \n",
       "7                                         LUAD            acinar   \n",
       "..                                         ...               ...   \n",
       "414                                       LUSC               NaN   \n",
       "418                                       LUAD             solid   \n",
       "416                                       LUSC               NaN   \n",
       "420                                     LUADx2            acinar   \n",
       "415                                       LUAD            acinar   \n",
       "\n",
       "    adjuvant_treatment_YN adjuvant_treatment_given num_cycle_na.added  \\\n",
       "88            No adjuvant                      NaN                  0   \n",
       "79            No adjuvant                      NaN                  0   \n",
       "26               Adjuvant           Platinum chemo                  4   \n",
       "23               Adjuvant           Platinum chemo                  4   \n",
       "7             No adjuvant                      NaN                  0   \n",
       "..                    ...                      ...                ...   \n",
       "414           No adjuvant                      NaN                  0   \n",
       "418              Adjuvant           Platinum chemo                  4   \n",
       "416           No adjuvant                      NaN                  0   \n",
       "420           No adjuvant                      NaN                  0   \n",
       "415           No adjuvant                      NaN                  0   \n",
       "\n",
       "                               CHMPlatDgName_cleaned  \\\n",
       "88                                               NaN   \n",
       "79                                               NaN   \n",
       "26   Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "23           Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "7                                                NaN   \n",
       "..                                               ...   \n",
       "414                                              NaN   \n",
       "418  Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "416                                              NaN   \n",
       "420                                              NaN   \n",
       "415                                              NaN   \n",
       "\n",
       "                                CHMOthDgName_cleaned  AdjRadStartTime_manual  \\\n",
       "88                                               NaN                     NaN   \n",
       "79                                               NaN                     NaN   \n",
       "26   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "23   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "7                                                NaN                     NaN   \n",
       "..                                               ...                     ...   \n",
       "414                                              NaN                     NaN   \n",
       "418  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "416                                              NaN                     NaN   \n",
       "420                                              NaN                     NaN   \n",
       "415                                              NaN                     NaN   \n",
       "\n",
       "     AdjRadEndTime_manual  Recurrence_time_use newPrim_time_use  \\\n",
       "88                    NaN                  NaN              NaN   \n",
       "79                    NaN                  NaN              NaN   \n",
       "26                    NaN                217.0              NaN   \n",
       "23                    NaN                325.0              NaN   \n",
       "7                     NaN                  NaN              NaN   \n",
       "..                    ...                  ...              ...   \n",
       "414                   NaN                  NaN              NaN   \n",
       "418                   NaN                  NaN              NaN   \n",
       "416                   NaN                883.0              NaN   \n",
       "420                   NaN                748.0              NaN   \n",
       "415                   NaN                  NaN              NaN   \n",
       "\n",
       "    first_dfs_any_event_rec.or.new.primary first_event_during_followup  \\\n",
       "88                                     NaN                       Death   \n",
       "79                                     NaN                         NaN   \n",
       "26                              recurrence                  recurrence   \n",
       "23                              recurrence                  recurrence   \n",
       "7                                      NaN                       Death   \n",
       "..                                     ...                         ...   \n",
       "414                                    NaN                         NaN   \n",
       "418                                    NaN                         NaN   \n",
       "416                             recurrence                  recurrence   \n",
       "420                             recurrence                  recurrence   \n",
       "415                                    NaN                         NaN   \n",
       "\n",
       "     cens_os  os_time  cens_dfs  dfs_time  cens_dfs_any_event  \\\n",
       "88         1     1120         1      1120                   1   \n",
       "79         0     1901         0      1901                   0   \n",
       "26         0     2304         1       217                   1   \n",
       "23         1     1323         1       325                   1   \n",
       "7          1      649         1       649                   1   \n",
       "..       ...      ...       ...       ...                 ...   \n",
       "414        0     1084         0      1084                   0   \n",
       "418        0     1434         0      1434                   0   \n",
       "416        1      893         1       883                   1   \n",
       "420        1      758         1       748                   1   \n",
       "415        0     1001         0      1001                   0   \n",
       "\n",
       "     dfs_time_any_event  cens_lung_event  lung_event_time    Relapse_cat  \\\n",
       "88                 1120                0             1120            NaN   \n",
       "79                 1901                0             1901            NaN   \n",
       "26                  217                1              217  Intra & Extra   \n",
       "23                  325                1              325  Intra & Extra   \n",
       "7                   649                0              649            NaN   \n",
       "..                  ...              ...              ...            ...   \n",
       "414                1084                0             1084            NaN   \n",
       "418                1434                0             1434            NaN   \n",
       "416                 883                1              883  Intra & Extra   \n",
       "420                 748                1              748  Intrathoracic   \n",
       "415                1001                0             1001            NaN   \n",
       "\n",
       "    Relapse_cat_new  tx100  \n",
       "88           No rec   True  \n",
       "79           No rec   True  \n",
       "26    Intra & Extra   True  \n",
       "23    Intra & Extra   True  \n",
       "7            No rec   True  \n",
       "..              ...    ...  \n",
       "414          No rec  False  \n",
       "418          No rec  False  \n",
       "416   Intra & Extra  False  \n",
       "420   Intrathoracic  False  \n",
       "415          No rec  False  \n",
       "\n",
       "[421 rows x 50 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 1 0 1 0 0 1 1 1 1 1\n",
      " 0 1 0 0 1 1 0 0 1 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 1 0 1 1 1 0 0 0 0 1 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 1 1 1\n",
      " 0 1 0 1 0 1 1 0 0 0 0 0 1 1 0 0 1 0 1 0 0 0 1 1 0 0 1 1 0 0 0 1 0 0 0 0 0\n",
      " 0 0 1 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 0 1 1 0 1 0 1 0 1 1 0 1 0 0 0\n",
      " 0 1 0 0 1 1 1 0 0 0 0 0 1 0 0 1 0 0 0 0 0 1 0 0 1 0 0 0 0 1 0 1 0 0 1 0 1\n",
      " 0 0 0 0 1 1 0 0 0 1 0 0 0 0 0 1 1 0 1 0]\n",
      "[1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 1 0 0 0 0 0 1 1 0 1 0 1 0 0 0 1 1 0 0\n",
      " 1 0 0 1 0 1 0 0 0 1 1 1 1 0 1 1 0 1 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 1 1 1 1 0 0 1 1 1 1 0 0 1 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 1 1 0 1 0 0\n",
      " 0 0 0 1 1 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1]\n"
     ]
    }
   ],
   "source": [
    "# make y for LUAD\n",
    "y_train_LUAD = []\n",
    "for index, row in clinical_df.iterrows():\n",
    "    if row['cruk_id'] in LUAD_patient_id:\n",
    "        if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "            y_train_LUAD.append(1)\n",
    "        else:\n",
    "            y_train_LUAD.append(0)\n",
    "\n",
    "y_LUAD = np.array(y_train_LUAD)\n",
    "print(y_LUAD)\n",
    "\n",
    "# make y for LUSC\n",
    "y_train_LUSC = []\n",
    "for index, row in clinical_df.iterrows():\n",
    "    if row['cruk_id'] in LUSC_patient_id:\n",
    "        if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "            y_train_LUSC.append(1)\n",
    "        else:\n",
    "            y_train_LUSC.append(0)\n",
    "\n",
    "y_LUSC = np.array(y_train_LUSC)\n",
    "print(y_LUSC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 100) (73, 100) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1, 'hidden_layer_sizes': (2,)}\n",
      "Best score:  0.5100955204216074\n",
      "Predictions:  [0 0 0 1 0 0 1 0 0 0 0 1 0 0 0 0 1 0 0 1 0 1 0 1 1 0 0 0 0 1 0 0 1 0 0 0 0\n",
      " 0 1 0 0 0 0 0 1 0 0 0 0 1 0 0 0 1 0 0 1 0 0 0 0 0 0 1 0 1 0 0 1 0 1 1 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4340463458110517\n",
      "ROC AUC:  0.3253119429590018\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.66      0.69      0.67        51\n",
      "   Recurrence       0.20      0.18      0.19        22\n",
      "\n",
      "     accuracy                           0.53        73\n",
      "    macro avg       0.43      0.43      0.43        73\n",
      " weighted avg       0.52      0.53      0.53        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_hundred_feature_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 50) (73, 50) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.0001, 'hidden_layer_sizes': (2,)}\n",
      "Best score:  0.5700428194993411\n",
      "Predictions:  [0 0 0 0 1 0 1 0 0 0 1 1 1 1 1 1 1 0 0 1 0 1 1 1 1 0 0 0 1 0 0 0 0 0 0 0 0\n",
      " 0 1 1 0 0 0 0 0 1 0 0 1 1 0 0 0 1 0 0 0 1 0 0 0 0 0 1 0 1 0 0 0 0 1 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.482620320855615\n",
      "ROC AUC:  0.3596256684491979\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.69      0.65      0.67        51\n",
      "   Recurrence       0.28      0.32      0.30        22\n",
      "\n",
      "     accuracy                           0.55        73\n",
      "    macro avg       0.48      0.48      0.48        73\n",
      " weighted avg       0.56      0.55      0.56        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_fifty_feature_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 20) (73, 20) (169,) (73,)\n",
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.6413076416337287\n",
      "Predictions:  [0 0 0 0 1 0 0 0 0 0 0 1 0 0 1 1 1 0 0 1 0 1 1 0 1 0 0 1 1 1 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 1 0 1 0 0 0 1 1 0 0 0 0 1 0 1 1 0 0 0 0 0 0 0 1 0 0 1 0 0 1 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5672905525846702\n",
      "ROC AUC:  0.4946524064171123\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.74      0.73      0.73        51\n",
      "   Recurrence       0.39      0.41      0.40        22\n",
      "\n",
      "     accuracy                           0.63        73\n",
      "    macro avg       0.57      0.57      0.57        73\n",
      " weighted avg       0.63      0.63      0.63        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_twenty_feature_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 10) (73, 10) (169,) (73,)\n",
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.63667325428195\n",
      "Predictions:  [0 0 1 0 1 0 0 0 0 0 0 1 0 0 1 0 1 0 0 1 0 1 1 1 1 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 1 1 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 0 0 0 0 0 1 0 1 0 0 1 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4893048128342246\n",
      "ROC AUC:  0.49465240641711233\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.69      0.71      0.70        51\n",
      "   Recurrence       0.29      0.27      0.28        22\n",
      "\n",
      "     accuracy                           0.58        73\n",
      "    macro avg       0.49      0.49      0.49        73\n",
      " weighted avg       0.57      0.58      0.57        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_ten_feature_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 10) (73, 10) (169,) (73,)\n",
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.63667325428195\n",
      "Predictions:  [0 0 1 0 1 0 0 0 0 0 0 1 0 0 1 0 1 0 0 1 0 1 1 1 1 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 1 1 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 0 0 0 0 0 1 0 1 0 0 1 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4893048128342246\n",
      "ROC AUC:  0.49465240641711233\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.69      0.71      0.70        51\n",
      "   Recurrence       0.29      0.27      0.28        22\n",
      "\n",
      "     accuracy                           0.58        73\n",
      "    macro avg       0.49      0.49      0.49        73\n",
      " weighted avg       0.57      0.58      0.57        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_ten_feature_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 5) (73, 5) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.5\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.5\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.70      1.00      0.82        51\n",
      "   Recurrence       0.00      0.00      0.00        22\n",
      "\n",
      "     accuracy                           0.70        73\n",
      "    macro avg       0.35      0.50      0.41        73\n",
      " weighted avg       0.49      0.70      0.57        73\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_five_feature_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 100) (42, 100) (96,) (42,)\n",
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (2,)}\n",
      "Best score:  0.5527472527472528\n",
      "Predictions:  [1 1 0 0 0 1 0 1 0 1 0 0 1 0 0 1 0 0 0 1 0 0 1 0 1 0 0 0 0 0 0 0 0 0 1 1 0\n",
      " 0 0 1 1 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.4814814814814815\n",
      "ROC AUC:  0.45925925925925926\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.63      0.63      0.63        27\n",
      "   Recurrence       0.33      0.33      0.33        15\n",
      "\n",
      "     accuracy                           0.52        42\n",
      "    macro avg       0.48      0.48      0.48        42\n",
      " weighted avg       0.52      0.52      0.52        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_hundred_feature_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 50) (42, 50) (96,) (42,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.6187728937728938\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 1 0\n",
      " 0 0 0 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.5259259259259259\n",
      "ROC AUC:  0.5506172839506173\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.66      0.85      0.74        27\n",
      "   Recurrence       0.43      0.20      0.27        15\n",
      "\n",
      "     accuracy                           0.62        42\n",
      "    macro avg       0.54      0.53      0.51        42\n",
      " weighted avg       0.58      0.62      0.57        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_fifty_feature_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 20) (42, 20) (96,) (42,)\n",
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.6460622710622712\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 1 1 0 1 0 0 1 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0\n",
      " 0 1 1 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.6074074074074074\n",
      "ROC AUC:  0.5456790123456791\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.71      0.81      0.76        27\n",
      "   Recurrence       0.55      0.40      0.46        15\n",
      "\n",
      "     accuracy                           0.67        42\n",
      "    macro avg       0.63      0.61      0.61        42\n",
      " weighted avg       0.65      0.67      0.65        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_twenty_feature_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 10) (42, 10) (96,) (42,)\n",
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.6174908424908425\n",
      "Predictions:  [0 0 0 0 1 0 0 0 0 1 0 0 1 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0\n",
      " 0 1 0 1 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.5592592592592592\n",
      "ROC AUC:  0.5395061728395063\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.68      0.85      0.75        27\n",
      "   Recurrence       0.50      0.27      0.35        15\n",
      "\n",
      "     accuracy                           0.64        42\n",
      "    macro avg       0.59      0.56      0.55        42\n",
      " weighted avg       0.61      0.64      0.61        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_ten_feature_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 5) (42, 5) (96,) (42,)\n",
      "Best parameters:  {'alpha': 0.0001, 'hidden_layer_sizes': (2,)}\n",
      "Best score:  0.617948717948718\n",
      "Predictions:  [0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0\n",
      " 0 0 0 1 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.5111111111111111\n",
      "ROC AUC:  0.5456790123456791\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.65      0.89      0.75        27\n",
      "   Recurrence       0.40      0.13      0.20        15\n",
      "\n",
      "     accuracy                           0.62        42\n",
      "    macro avg       0.52      0.51      0.47        42\n",
      " weighted avg       0.56      0.62      0.55        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_five_feature_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>RegionID</th>\n",
       "      <th>chr</th>\n",
       "      <th>startpos</th>\n",
       "      <th>endpos</th>\n",
       "      <th>n.het</th>\n",
       "      <th>cnTotal</th>\n",
       "      <th>nMajor</th>\n",
       "      <th>nMinor</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>ACF</th>\n",
       "      <th>nAraw</th>\n",
       "      <th>nBraw</th>\n",
       "      <th>cpn_event_vs_ploidy</th>\n",
       "      <th>MSAI</th>\n",
       "      <th>samp_used_to_phase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>2252860</td>\n",
       "      <td>73</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>2.463756</td>\n",
       "      <td>1.184761</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>2252970</td>\n",
       "      <td>2527412</td>\n",
       "      <td>26</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>1.645818</td>\n",
       "      <td>0.284001</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>2527437</td>\n",
       "      <td>2700215</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>2.187291</td>\n",
       "      <td>0.956066</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>2700332</td>\n",
       "      <td>28531815</td>\n",
       "      <td>600</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>1.725844</td>\n",
       "      <td>0.708290</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>28534782</td>\n",
       "      <td>29649999</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>1.711109</td>\n",
       "      <td>0.695610</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419823</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>36962306</td>\n",
       "      <td>39176976</td>\n",
       "      <td>55</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>4.207645</td>\n",
       "      <td>0.208286</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419824</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>39262577</td>\n",
       "      <td>41604353</td>\n",
       "      <td>41</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>2.385038</td>\n",
       "      <td>0.092194</td>\n",
       "      <td>loss</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419825</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>41604399</td>\n",
       "      <td>41612295</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>2.331273</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>loss</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419826</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>41648860</td>\n",
       "      <td>42899151</td>\n",
       "      <td>37</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>3.403717</td>\n",
       "      <td>0.186030</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419827</th>\n",
       "      <td>CRUK0555</td>\n",
       "      <td>CRUK0555_Tumour2</td>\n",
       "      <td>:R2</td>\n",
       "      <td>22</td>\n",
       "      <td>42912377</td>\n",
       "      <td>51219216</td>\n",
       "      <td>226</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3.796640</td>\n",
       "      <td>0.19</td>\n",
       "      <td>1.608646</td>\n",
       "      <td>0.238268</td>\n",
       "      <td>loss</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0555_SU_T1-R1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>419828 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id         tumour_id     RegionID  chr  startpos    endpos  \\\n",
       "0        CRUK0005          CRUK0005  CRUK0005:R1    1    762273   2252860   \n",
       "1        CRUK0005          CRUK0005  CRUK0005:R1    1   2252970   2527412   \n",
       "2        CRUK0005          CRUK0005  CRUK0005:R1    1   2527437   2700215   \n",
       "3        CRUK0005          CRUK0005  CRUK0005:R1    1   2700332  28531815   \n",
       "4        CRUK0005          CRUK0005  CRUK0005:R1    1  28534782  29649999   \n",
       "...           ...               ...          ...  ...       ...       ...   \n",
       "419823   CRUK0555  CRUK0555_Tumour2          :R2   22  36962306  39176976   \n",
       "419824   CRUK0555  CRUK0555_Tumour2          :R2   22  39262577  41604353   \n",
       "419825   CRUK0555  CRUK0555_Tumour2          :R2   22  41604399  41612295   \n",
       "419826   CRUK0555  CRUK0555_Tumour2          :R2   22  41648860  42899151   \n",
       "419827   CRUK0555  CRUK0555_Tumour2          :R2   22  42912377  51219216   \n",
       "\n",
       "        n.het  cnTotal  nMajor  nMinor    Ploidy   ACF     nAraw     nBraw  \\\n",
       "0          73        3       2       1  3.131282  0.15  2.463756  1.184761   \n",
       "1          26        2       2       0  3.131282  0.15  1.645818  0.284001   \n",
       "2          12        3       2       1  3.131282  0.15  2.187291  0.956066   \n",
       "3         600        3       2       1  3.131282  0.15  1.725844  0.708290   \n",
       "4          11        3       2       1  3.131282  0.15  1.711109  0.695610   \n",
       "...       ...      ...     ...     ...       ...   ...       ...       ...   \n",
       "419823     55        4       4       0  3.796640  0.19  4.207645  0.208286   \n",
       "419824     41        2       2       0  3.796640  0.19  2.385038  0.092194   \n",
       "419825      6        2       2       0  3.796640  0.19  2.331273  0.000000   \n",
       "419826     37        3       3       0  3.796640  0.19  3.403717  0.186030   \n",
       "419827    226        2       2       0  3.796640  0.19  1.608646  0.238268   \n",
       "\n",
       "       cpn_event_vs_ploidy   MSAI samp_used_to_phase  \n",
       "0                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "1                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "2                  neutral  False  CRUK0005_SU_T1-R3  \n",
       "3                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "4                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "...                    ...    ...                ...  \n",
       "419823             neutral  False  CRUK0555_SU_T1-R1  \n",
       "419824                loss  False  CRUK0555_SU_T1-R2  \n",
       "419825                loss  False  CRUK0555_SU_T1-R1  \n",
       "419826             neutral  False  CRUK0555_SU_T1-R1  \n",
       "419827                loss  False  CRUK0555_SU_T1-R1  \n",
       "\n",
       "[419828 rows x 17 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cna_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>RegionID</th>\n",
       "      <th>chr</th>\n",
       "      <th>startpos</th>\n",
       "      <th>endpos</th>\n",
       "      <th>n.het</th>\n",
       "      <th>cnTotal</th>\n",
       "      <th>nMajor</th>\n",
       "      <th>nMinor</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>ACF</th>\n",
       "      <th>nAraw</th>\n",
       "      <th>nBraw</th>\n",
       "      <th>cpn_event_vs_ploidy</th>\n",
       "      <th>MSAI</th>\n",
       "      <th>samp_used_to_phase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52958</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>3800149</td>\n",
       "      <td>134</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4.493770</td>\n",
       "      <td>0.22</td>\n",
       "      <td>2.073981</td>\n",
       "      <td>1.625892</td>\n",
       "      <td>neutral</td>\n",
       "      <td>True</td>\n",
       "      <td>CRUK0001_SU_T1-R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288283</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>2537166</td>\n",
       "      <td>72</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.921907</td>\n",
       "      <td>0.38</td>\n",
       "      <td>1.696286</td>\n",
       "      <td>1.044553</td>\n",
       "      <td>gain</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0002_SU_T1-R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141069</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>752894</td>\n",
       "      <td>3427608</td>\n",
       "      <td>110</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.567392</td>\n",
       "      <td>0.54</td>\n",
       "      <td>2.041043</td>\n",
       "      <td>1.081540</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0003_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45121</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762496</td>\n",
       "      <td>3496497</td>\n",
       "      <td>53</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.950603</td>\n",
       "      <td>0.23</td>\n",
       "      <td>1.804428</td>\n",
       "      <td>1.409574</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0004_SU_T1-R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>2252860</td>\n",
       "      <td>73</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>2.463756</td>\n",
       "      <td>1.184761</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368701</th>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>752894</td>\n",
       "      <td>1572735</td>\n",
       "      <td>79</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3.514533</td>\n",
       "      <td>0.76</td>\n",
       "      <td>4.050553</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0861_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375529</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>1448024</td>\n",
       "      <td>90</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.556913</td>\n",
       "      <td>0.31</td>\n",
       "      <td>2.074464</td>\n",
       "      <td>1.035148</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0864_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396701</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>990280</td>\n",
       "      <td>34</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3.919491</td>\n",
       "      <td>0.26</td>\n",
       "      <td>2.102410</td>\n",
       "      <td>1.801966</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0872_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417335</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>65797</td>\n",
       "      <td>1027553</td>\n",
       "      <td>39</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4.186610</td>\n",
       "      <td>0.33</td>\n",
       "      <td>1.992436</td>\n",
       "      <td>0.923056</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0881_SU_T1-R5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261061</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>69511</td>\n",
       "      <td>81969071</td>\n",
       "      <td>1429</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.048684</td>\n",
       "      <td>0.13</td>\n",
       "      <td>1.732114</td>\n",
       "      <td>0.914478</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0893_SU_T1-R2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>394 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id         tumour_id     RegionID  chr  startpos    endpos  \\\n",
       "52958    CRUK0001          CRUK0001  CRUK0001:R1    1    762273   3800149   \n",
       "288283   CRUK0002          CRUK0002  CRUK0002:R1    1    762273   2537166   \n",
       "141069   CRUK0003          CRUK0003  CRUK0003:R1    1    752894   3427608   \n",
       "45121    CRUK0004          CRUK0004  CRUK0004:R1    1    762496   3496497   \n",
       "0        CRUK0005          CRUK0005  CRUK0005:R1    1    762273   2252860   \n",
       "...           ...               ...          ...  ...       ...       ...   \n",
       "368701   CRUK0861          CRUK0861  CRUK0861:R1    1    752894   1572735   \n",
       "375529   CRUK0864          CRUK0864  CRUK0864:R1    1    762273   1448024   \n",
       "396701   CRUK0872          CRUK0872  CRUK0872:R1    1    762273    990280   \n",
       "417335   CRUK0881  CRUK0881_Tumour1          :R1    1     65797   1027553   \n",
       "261061   CRUK0893          CRUK0893  CRUK0893:R1    1     69511  81969071   \n",
       "\n",
       "        n.het  cnTotal  nMajor  nMinor    Ploidy   ACF     nAraw     nBraw  \\\n",
       "52958     134        4       2       2  4.493770  0.22  2.073981  1.625892   \n",
       "288283     72        3       2       1  1.921907  0.38  1.696286  1.044553   \n",
       "141069    110        3       2       1  3.567392  0.54  2.041043  1.081540   \n",
       "45121      53        3       2       1  2.950603  0.23  1.804428  1.409574   \n",
       "0          73        3       2       1  3.131282  0.15  2.463756  1.184761   \n",
       "...       ...      ...     ...     ...       ...   ...       ...       ...   \n",
       "368701     79        4       4       0  3.514533  0.76  4.050553  0.000000   \n",
       "375529     90        3       2       1  3.556913  0.31  2.074464  1.035148   \n",
       "396701     34        4       2       2  3.919491  0.26  2.102410  1.801966   \n",
       "417335     39        3       2       1  4.186610  0.33  1.992436  0.923056   \n",
       "261061   1429        3       2       1  3.048684  0.13  1.732114  0.914478   \n",
       "\n",
       "       cpn_event_vs_ploidy   MSAI samp_used_to_phase  \n",
       "52958              neutral   True  CRUK0001_SU_T1-R3  \n",
       "288283                gain  False  CRUK0002_SU_T1-R3  \n",
       "141069             neutral  False  CRUK0003_SU_T1-R2  \n",
       "45121              neutral  False  CRUK0004_SU_T1-R3  \n",
       "0                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "...                    ...    ...                ...  \n",
       "368701             neutral  False  CRUK0861_SU_T1-R2  \n",
       "375529             neutral  False  CRUK0864_SU_T1-R2  \n",
       "396701             neutral  False  CRUK0872_SU_T1-R2  \n",
       "417335             neutral  False  CRUK0881_SU_T1-R5  \n",
       "261061             neutral  False  CRUK0893_SU_T1-R2  \n",
       "\n",
       "[394 rows x 17 columns]"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cna_table_no_duplicates = cna_table.drop_duplicates(subset='patient_id')\n",
    "cna_table_no_duplicates = cna_table_no_duplicates.sort_values(by=['patient_id'])\n",
    "cna_table_no_duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>combTiming_SC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>A2M</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>A4GALT</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>AARS</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>AASS</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338917</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZNF496</td>\n",
       "      <td>early</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338918</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZNF551</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338919</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZNF750</td>\n",
       "      <td>early</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338920</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZNF831</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338921</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZP4</td>\n",
       "      <td>early</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>338922 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id Hugo_Symbol combTiming_SC\n",
       "0        CRUK0001         A2M     subclonal\n",
       "1        CRUK0001      A4GALT     subclonal\n",
       "2        CRUK0001        AARS     subclonal\n",
       "3        CRUK0001      AARSD1     subclonal\n",
       "4        CRUK0001        AASS     subclonal\n",
       "...           ...         ...           ...\n",
       "338917   CRUK0893      ZNF496         early\n",
       "338918   CRUK0893      ZNF551     subclonal\n",
       "338919   CRUK0893      ZNF750         early\n",
       "338920   CRUK0893      ZNF831     subclonal\n",
       "338921   CRUK0893         ZP4         early\n",
       "\n",
       "[338922 rows x 3 columns]"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Find the most common 'combTiming_SC' for each unique combination of 'patient_id' and 'Hugo_Symbol'\n",
    "most_common_timing = mut_table.groupby(['patient_id', 'Hugo_Symbol'])['combTiming_SC'].agg(lambda x: x.mode().iat[0] if not x.mode().empty else 'Other').reset_index()\n",
    "most_common_timing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed hundred_feature_matrix:\n",
      "Hugo_Symbol  ANAPC5  ANK2  ARAF  ARHGAP35  ARHGEF12  ARID1A  ARID1B  ARID2  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001          0     0     0         3         0       0       0      0   \n",
      "CRUK0002          0     0     0         0         0       0       0      0   \n",
      "CRUK0003          0     0     0         0         0       0       0      0   \n",
      "CRUK0004          0     0     0         0         0       0       0      0   \n",
      "CRUK0005          0     0     0         0         0       0       0      0   \n",
      "...             ...   ...   ...       ...       ...     ...     ...    ...   \n",
      "CRUK0861          0     0     0         0         0       0       0      0   \n",
      "CRUK0864          0     0     0         0         0       0       0      0   \n",
      "CRUK0872          0     0     0         0         0       0       0      0   \n",
      "CRUK0881          0     0     0         0         0       0       0      0   \n",
      "CRUK0893          0     0     0         0         0       0       0      0   \n",
      "\n",
      "Hugo_Symbol  ASXL1  AXIN1  B2M  BAP1  BCOR  BLM  BMPR1A  BRCA1  BUB1B  CDC73  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0002         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0003         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0004         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0005         0      0    0     0     0    0       0      0      0      0   \n",
      "...            ...    ...  ...   ...   ...  ...     ...    ...    ...    ...   \n",
      "CRUK0861         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0864         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0872         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0881         0      0    0     0     0    0       0      0      0      0   \n",
      "CRUK0893         0      0    0     0     0    0       0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  CHD4  CIC  CREBBP  CTNNB1  CYLD  DDB2  DDX3X  DOT1L  EPB41L3  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0002        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0003        0    0       0       3     0     0      0      0        0   \n",
      "CRUK0004        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0005        0    0       0       0     0     0      0      0        0   \n",
      "...           ...  ...     ...     ...   ...   ...    ...    ...      ...   \n",
      "CRUK0861        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0864        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0872        0    0       0       0     0     0      0      0        0   \n",
      "CRUK0881        0    0       1       0     0     0      4      0        0   \n",
      "CRUK0893        0    0       0       0     0     0      0      0        0   \n",
      "\n",
      "Hugo_Symbol  ERBB2  EZH1  FAM166A  FANCF  FAS  FAT1  FBXW7  FGFR1  FGFR3  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0002         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0003         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0004         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0005         0     0        0      0    0     0      0      0      0   \n",
      "...            ...   ...      ...    ...  ...   ...    ...    ...    ...   \n",
      "CRUK0861         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0864         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0872         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0881         0     0        0      0    0     0      0      0      0   \n",
      "CRUK0893         0     0        0      0    0     0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  FLCN  FLT4  FN1  GNPTAB  HCN1  HNF1A  HOXB3  IL21R  IQGAP2  \\\n",
      "patient_id                                                                \n",
      "CRUK0001        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0002        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0003        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0004        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0005        0     0    0       0     0      0      0      0       0   \n",
      "...           ...   ...  ...     ...   ...    ...    ...    ...     ...   \n",
      "CRUK0861        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0864        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0872        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0881        0     0    0       0     0      0      0      0       0   \n",
      "CRUK0893        0     0    0       0     0      0      0      0       0   \n",
      "\n",
      "Hugo_Symbol  KDM5C  KDM6A  KEAP1  KLF5  KMT2D  KRAS  LATS1  MAP3K13  MAX  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0002         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0003         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0004         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0005         0      0      0     0      0     0      0        0    0   \n",
      "...            ...    ...    ...   ...    ...   ...    ...      ...  ...   \n",
      "CRUK0861         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0864         0      0      0     0      0     1      0        0    0   \n",
      "CRUK0872         0      0      0     0      0     0      0        0    0   \n",
      "CRUK0881         0      0      0     0      0     1      0        0    0   \n",
      "CRUK0893         0      0      0     0      0     1      0        0    0   \n",
      "\n",
      "Hugo_Symbol  MSH2  MTOR  NBN  NCOA6  NEB  NF1  NFKBIE  NIN  NOTCH1  NRAS  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0002        0     0    0      0    0    3       0    0       0     0   \n",
      "CRUK0003        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0004        0     0    0      0    0    0       0    0       3     0   \n",
      "CRUK0005        0     0    0      0    0    0       0    0       0     3   \n",
      "...           ...   ...  ...    ...  ...  ...     ...  ...     ...   ...   \n",
      "CRUK0861        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0864        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0872        0     0    0      0    0    0       0    0       0     0   \n",
      "CRUK0881        0     0    0      0    0    1       0    0       0     0   \n",
      "CRUK0893        0     0    0      0    0    0       0    0       0     0   \n",
      "\n",
      "Hugo_Symbol  NSD1  NTRK1  PALB2  PASK  PBRM1  PIK3CA  PIK3CG  PIK3R1  PLXNB2  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001        0      0      0     3      0       0       0       0       0   \n",
      "CRUK0002        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0003        0      0      0     0      0       2       0       0       0   \n",
      "CRUK0004        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0005        0      0      0     3      0       0       0       0       0   \n",
      "...           ...    ...    ...   ...    ...     ...     ...     ...     ...   \n",
      "CRUK0861        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0864        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0872        0      0      0     0      0       0       0       0       0   \n",
      "CRUK0881        0      0      0     0      3       0       0       0       0   \n",
      "CRUK0893        0      0      0     0      0       0       0       0       0   \n",
      "\n",
      "Hugo_Symbol  POLE  PRDM1  PRKAR1A  PRRX1  PTEN  PTPRB  PTPRC  PTPRU  RASA1  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0002        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0003        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0004        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0005        0      0        0      0     0      0      0      0      0   \n",
      "...           ...    ...      ...    ...   ...    ...    ...    ...    ...   \n",
      "CRUK0861        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0864        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0872        0      0        0      0     0      1      0      0      0   \n",
      "CRUK0881        0      0        0      0     0      0      0      0      0   \n",
      "CRUK0893        0      0        0      0     0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  RNF43  SETD2  SGK223  SMARCA4  SNX29  SPEN  STIM1  STK11  STX2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0002         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0003         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0004         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0005         0      0       0        0      0     0      0      0     0   \n",
      "...            ...    ...     ...      ...    ...   ...    ...    ...   ...   \n",
      "CRUK0861         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0864         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0872         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0881         0      0       0        0      0     0      0      0     0   \n",
      "CRUK0893         0      0       0        0      0     0      0      0     0   \n",
      "\n",
      "Hugo_Symbol  TNC  TNFAIP3  TNFRSF14  U2AF1  WAS  WT1  ZFPM2  ZMYM2  ZNF483  \n",
      "patient_id                                                                  \n",
      "CRUK0001       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0002       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0003       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0004       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0005       0        0         0      0    0    0      0      0       0  \n",
      "...          ...      ...       ...    ...  ...  ...    ...    ...     ...  \n",
      "CRUK0861       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0864       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0872       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0881       0        0         0      0    0    0      0      0       0  \n",
      "CRUK0893       0        0         0      0    0    0      0      0       0  \n",
      "\n",
      "[421 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_hundred_feature_matrix = hundred_feature_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_hundred_feature_matrix.columns:\n",
    "            if cumulative_hundred_feature_matrix.loc[cumulative_hundred_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_hundred_feature_matrix.loc[cumulative_hundred_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed hundred_feature_matrix:\")\n",
    "print(cumulative_hundred_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed fifty_feature_matrix:\n",
      "Hugo_Symbol  ARHGAP35  ARID1B  ARID2  ASXL1  AXIN1  B2M  BAP1  BMPR1A  BRCA1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001            3       0      0      0      0    0     0       0      0   \n",
      "CRUK0002            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0003            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0004            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0005            0       0      0      0      0    0     0       0      0   \n",
      "...               ...     ...    ...    ...    ...  ...   ...     ...    ...   \n",
      "CRUK0861            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0864            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0872            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0881            0       0      0      0      0    0     0       0      0   \n",
      "CRUK0893            0       0      0      0      0    0     0       0      0   \n",
      "\n",
      "Hugo_Symbol  BUB1B  CIC  CREBBP  CTNNB1  CYLD  DDX3X  EZH1  FAS  FBXW7  FLCN  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0002         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0003         0    0       0       3     0      0     0    0      0     0   \n",
      "CRUK0004         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0005         0    0       0       0     0      0     0    0      0     0   \n",
      "...            ...  ...     ...     ...   ...    ...   ...  ...    ...   ...   \n",
      "CRUK0861         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0864         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0872         0    0       0       0     0      0     0    0      0     0   \n",
      "CRUK0881         0    0       1       0     0      4     0    0      0     0   \n",
      "CRUK0893         0    0       0       0     0      0     0    0      0     0   \n",
      "\n",
      "Hugo_Symbol  FLT4  FN1  HCN1  HNF1A  KDM5C  KEAP1  KLF5  KMT2D  MAP3K13  MSH2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0002        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0003        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0004        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0005        0    0     0      0      0      0     0      0        0     0   \n",
      "...           ...  ...   ...    ...    ...    ...   ...    ...      ...   ...   \n",
      "CRUK0861        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0864        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0872        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0881        0    0     0      0      0      0     0      0        0     0   \n",
      "CRUK0893        0    0     0      0      0      0     0      0        0     0   \n",
      "\n",
      "Hugo_Symbol  MTOR  NBN  NCOA6  NF1  NFKBIE  PALB2  PBRM1  PIK3CA  PIK3R1  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0002        0    0      0    3       0      0      0       0       0   \n",
      "CRUK0003        0    0      0    0       0      0      0       2       0   \n",
      "CRUK0004        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0005        0    0      0    0       0      0      0       0       0   \n",
      "...           ...  ...    ...  ...     ...    ...    ...     ...     ...   \n",
      "CRUK0861        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0864        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0872        0    0      0    0       0      0      0       0       0   \n",
      "CRUK0881        0    0      0    1       0      0      3       0       0   \n",
      "CRUK0893        0    0      0    0       0      0      0       0       0   \n",
      "\n",
      "Hugo_Symbol  PRDM1  PTEN  PTPRB  PTPRU  RNF43  SETD2  SGK223  SMARCA4  SNX29  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0002         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0003         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0004         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0005         0     0      0      0      0      0       0        0      0   \n",
      "...            ...   ...    ...    ...    ...    ...     ...      ...    ...   \n",
      "CRUK0861         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0864         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0872         0     0      1      0      0      0       0        0      0   \n",
      "CRUK0881         0     0      0      0      0      0       0        0      0   \n",
      "CRUK0893         0     0      0      0      0      0       0        0      0   \n",
      "\n",
      "Hugo_Symbol  STK11  STX2  WT1  \n",
      "patient_id                     \n",
      "CRUK0001         0     0    0  \n",
      "CRUK0002         0     0    0  \n",
      "CRUK0003         0     0    0  \n",
      "CRUK0004         0     0    0  \n",
      "CRUK0005         0     0    0  \n",
      "...            ...   ...  ...  \n",
      "CRUK0861         0     0    0  \n",
      "CRUK0864         0     0    0  \n",
      "CRUK0872         0     0    0  \n",
      "CRUK0881         0     0    0  \n",
      "CRUK0893         0     0    0  \n",
      "\n",
      "[421 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_fifty_feature_matrix = fifty_feature_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_fifty_feature_matrix.columns:\n",
    "            if cumulative_fifty_feature_matrix.loc[cumulative_fifty_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_fifty_feature_matrix.loc[cumulative_fifty_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed fifty_feature_matrix:\")\n",
    "print(cumulative_fifty_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed twenty_feature_matrix:\n",
      "Hugo_Symbol  ARID1B  ARID2  ASXL1  B2M  CREBBP  CTNNB1  EZH1  FLCN  KDM5C  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0002          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0003          0      0      0    0       0       3     0     0      0   \n",
      "CRUK0004          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0005          0      0      0    0       0       0     0     0      0   \n",
      "...             ...    ...    ...  ...     ...     ...   ...   ...    ...   \n",
      "CRUK0861          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0864          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0872          0      0      0    0       0       0     0     0      0   \n",
      "CRUK0881          0      0      0    0       1       0     0     0      0   \n",
      "CRUK0893          0      0      0    0       0       0     0     0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  MAP3K13  MTOR  PBRM1  PIK3CA  PTEN  PTPRB  SMARCA4  SNX29  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0002         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0003         0        0     0      0       2     0      0        0      0   \n",
      "CRUK0004         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0005         0        0     0      0       0     0      0        0      0   \n",
      "...            ...      ...   ...    ...     ...   ...    ...      ...    ...   \n",
      "CRUK0861         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0864         0        0     0      0       0     0      0        0      0   \n",
      "CRUK0872         0        0     0      0       0     0      1        0      0   \n",
      "CRUK0881         0        0     0      3       0     0      0        0      0   \n",
      "CRUK0893         0        0     0      0       0     0      0        0      0   \n",
      "\n",
      "Hugo_Symbol  STK11  WT1  \n",
      "patient_id               \n",
      "CRUK0001         0    0  \n",
      "CRUK0002         0    0  \n",
      "CRUK0003         0    0  \n",
      "CRUK0004         0    0  \n",
      "CRUK0005         0    0  \n",
      "...            ...  ...  \n",
      "CRUK0861         0    0  \n",
      "CRUK0864         0    0  \n",
      "CRUK0872         0    0  \n",
      "CRUK0881         0    0  \n",
      "CRUK0893         0    0  \n",
      "\n",
      "[421 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_twenty_feature_matrix = twenty_feature_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_twenty_feature_matrix.columns:\n",
    "            if cumulative_twenty_feature_matrix.loc[cumulative_twenty_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_twenty_feature_matrix.loc[cumulative_twenty_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed twenty_feature_matrix:\")\n",
    "print(cumulative_twenty_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed ten_feature_matrix:\n",
      "Hugo_Symbol  ARID2  B2M  CTNNB1  KMT2D  PIK3CA  PTEN  PTPRB  SMARCA4  STK11  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0002         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0003         0    0       3      0       2     0      0        0      0   \n",
      "CRUK0004         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0005         0    0       0      0       0     0      0        0      0   \n",
      "...            ...  ...     ...    ...     ...   ...    ...      ...    ...   \n",
      "CRUK0861         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0864         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0872         0    0       0      0       0     0      1        0      0   \n",
      "CRUK0881         0    0       0      0       0     0      0        0      0   \n",
      "CRUK0893         0    0       0      0       0     0      0        0      0   \n",
      "\n",
      "Hugo_Symbol  WT1  \n",
      "patient_id        \n",
      "CRUK0001       0  \n",
      "CRUK0002       0  \n",
      "CRUK0003       0  \n",
      "CRUK0004       0  \n",
      "CRUK0005       0  \n",
      "...          ...  \n",
      "CRUK0861       0  \n",
      "CRUK0864       0  \n",
      "CRUK0872       0  \n",
      "CRUK0881       0  \n",
      "CRUK0893       0  \n",
      "\n",
      "[421 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_ten_feature_matrix = ten_feature_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_ten_feature_matrix.columns:\n",
    "            if cumulative_ten_feature_matrix.loc[cumulative_ten_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_ten_feature_matrix.loc[cumulative_ten_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed ten_feature_matrix:\")\n",
    "print(cumulative_ten_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed five_feature_matrix:\n",
      "Hugo_Symbol  ARID2  B2M  KMT2D  SMARCA4  STK11\n",
      "patient_id                                    \n",
      "CRUK0001         0    0      0        0      0\n",
      "CRUK0002         0    0      0        0      0\n",
      "CRUK0003         0    0      0        0      0\n",
      "CRUK0004         0    0      0        0      0\n",
      "CRUK0005         0    0      0        0      0\n",
      "...            ...  ...    ...      ...    ...\n",
      "CRUK0861         0    0      0        0      0\n",
      "CRUK0864         0    0      0        0      0\n",
      "CRUK0872         0    0      0        0      0\n",
      "CRUK0881         0    0      0        0      0\n",
      "CRUK0893         0    0      0        0      0\n",
      "\n",
      "[421 rows x 5 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_five_feature_matrix = five_feature_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_five_feature_matrix.columns:\n",
    "            if cumulative_five_feature_matrix.loc[cumulative_five_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_five_feature_matrix.loc[cumulative_five_feature_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed five_feature_matrix:\")\n",
    "print(cumulative_five_feature_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>ANAPC5</th>\n",
       "      <th>ANK2</th>\n",
       "      <th>ARAF</th>\n",
       "      <th>ARHGAP35</th>\n",
       "      <th>ARHGEF12</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>ARID1B</th>\n",
       "      <th>ARID2</th>\n",
       "      <th>ASXL1</th>\n",
       "      <th>AXIN1</th>\n",
       "      <th>B2M</th>\n",
       "      <th>BAP1</th>\n",
       "      <th>BCOR</th>\n",
       "      <th>BLM</th>\n",
       "      <th>BMPR1A</th>\n",
       "      <th>BRCA1</th>\n",
       "      <th>BUB1B</th>\n",
       "      <th>CDC73</th>\n",
       "      <th>CHD4</th>\n",
       "      <th>CIC</th>\n",
       "      <th>CREBBP</th>\n",
       "      <th>CTNNB1</th>\n",
       "      <th>CYLD</th>\n",
       "      <th>DDB2</th>\n",
       "      <th>DDX3X</th>\n",
       "      <th>DOT1L</th>\n",
       "      <th>EPB41L3</th>\n",
       "      <th>ERBB2</th>\n",
       "      <th>EZH1</th>\n",
       "      <th>FAM166A</th>\n",
       "      <th>FANCF</th>\n",
       "      <th>FAS</th>\n",
       "      <th>FAT1</th>\n",
       "      <th>FBXW7</th>\n",
       "      <th>FGFR1</th>\n",
       "      <th>FGFR3</th>\n",
       "      <th>FLCN</th>\n",
       "      <th>FLT4</th>\n",
       "      <th>FN1</th>\n",
       "      <th>GNPTAB</th>\n",
       "      <th>HCN1</th>\n",
       "      <th>HNF1A</th>\n",
       "      <th>HOXB3</th>\n",
       "      <th>IL21R</th>\n",
       "      <th>IQGAP2</th>\n",
       "      <th>KDM5C</th>\n",
       "      <th>KDM6A</th>\n",
       "      <th>KEAP1</th>\n",
       "      <th>KLF5</th>\n",
       "      <th>KMT2D</th>\n",
       "      <th>KRAS</th>\n",
       "      <th>LATS1</th>\n",
       "      <th>MAP3K13</th>\n",
       "      <th>MAX</th>\n",
       "      <th>MSH2</th>\n",
       "      <th>MTOR</th>\n",
       "      <th>NBN</th>\n",
       "      <th>NCOA6</th>\n",
       "      <th>NEB</th>\n",
       "      <th>NF1</th>\n",
       "      <th>NFKBIE</th>\n",
       "      <th>NIN</th>\n",
       "      <th>NOTCH1</th>\n",
       "      <th>NRAS</th>\n",
       "      <th>NSD1</th>\n",
       "      <th>NTRK1</th>\n",
       "      <th>PALB2</th>\n",
       "      <th>PASK</th>\n",
       "      <th>PBRM1</th>\n",
       "      <th>PIK3CA</th>\n",
       "      <th>PIK3CG</th>\n",
       "      <th>PIK3R1</th>\n",
       "      <th>PLXNB2</th>\n",
       "      <th>POLE</th>\n",
       "      <th>PRDM1</th>\n",
       "      <th>PRKAR1A</th>\n",
       "      <th>PRRX1</th>\n",
       "      <th>PTEN</th>\n",
       "      <th>PTPRB</th>\n",
       "      <th>PTPRC</th>\n",
       "      <th>PTPRU</th>\n",
       "      <th>RASA1</th>\n",
       "      <th>RNF43</th>\n",
       "      <th>SETD2</th>\n",
       "      <th>SGK223</th>\n",
       "      <th>SMARCA4</th>\n",
       "      <th>SNX29</th>\n",
       "      <th>SPEN</th>\n",
       "      <th>STIM1</th>\n",
       "      <th>STK11</th>\n",
       "      <th>STX2</th>\n",
       "      <th>TNC</th>\n",
       "      <th>TNFAIP3</th>\n",
       "      <th>TNFRSF14</th>\n",
       "      <th>U2AF1</th>\n",
       "      <th>WAS</th>\n",
       "      <th>WT1</th>\n",
       "      <th>ZFPM2</th>\n",
       "      <th>ZMYM2</th>\n",
       "      <th>ZNF483</th>\n",
       "      <th>Ploidy</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CRUK0001</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.493770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.921907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.567392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0004</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.950603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0005</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.131282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0861</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.514533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0864</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.556913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0872</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.919491</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0881</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.186610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0893</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.048684</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>394 rows × 101 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Hugo_Symbol  ANAPC5  ANK2  ARAF  ARHGAP35  ARHGEF12  ARID1A  ARID1B  ARID2  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001          0     0     0         3         0       0       0      0   \n",
       "CRUK0002          0     0     0         0         0       0       0      0   \n",
       "CRUK0003          0     0     0         0         0       0       0      0   \n",
       "CRUK0004          0     0     0         0         0       0       0      0   \n",
       "CRUK0005          0     0     0         0         0       0       0      0   \n",
       "...             ...   ...   ...       ...       ...     ...     ...    ...   \n",
       "CRUK0861          0     0     0         0         0       0       0      0   \n",
       "CRUK0864          0     0     0         0         0       0       0      0   \n",
       "CRUK0872          0     0     0         0         0       0       0      0   \n",
       "CRUK0881          0     0     0         0         0       0       0      0   \n",
       "CRUK0893          0     0     0         0         0       0       0      0   \n",
       "\n",
       "Hugo_Symbol  ASXL1  AXIN1  B2M  BAP1  BCOR  BLM  BMPR1A  BRCA1  BUB1B  CDC73  \\\n",
       "patient_id                                                                     \n",
       "CRUK0001         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0002         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0003         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0004         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0005         0      0    0     0     0    0       0      0      0      0   \n",
       "...            ...    ...  ...   ...   ...  ...     ...    ...    ...    ...   \n",
       "CRUK0861         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0864         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0872         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0881         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0893         0      0    0     0     0    0       0      0      0      0   \n",
       "\n",
       "Hugo_Symbol  CHD4  CIC  CREBBP  CTNNB1  CYLD  DDB2  DDX3X  DOT1L  EPB41L3  \\\n",
       "patient_id                                                                  \n",
       "CRUK0001        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0002        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0003        0    0       0       3     0     0      0      0        0   \n",
       "CRUK0004        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0005        0    0       0       0     0     0      0      0        0   \n",
       "...           ...  ...     ...     ...   ...   ...    ...    ...      ...   \n",
       "CRUK0861        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0864        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0872        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0881        0    0       1       0     0     0      4      0        0   \n",
       "CRUK0893        0    0       0       0     0     0      0      0        0   \n",
       "\n",
       "Hugo_Symbol  ERBB2  EZH1  FAM166A  FANCF  FAS  FAT1  FBXW7  FGFR1  FGFR3  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0002         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0003         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0004         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0005         0     0        0      0    0     0      0      0      0   \n",
       "...            ...   ...      ...    ...  ...   ...    ...    ...    ...   \n",
       "CRUK0861         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0864         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0872         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0881         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0893         0     0        0      0    0     0      0      0      0   \n",
       "\n",
       "Hugo_Symbol  FLCN  FLT4  FN1  GNPTAB  HCN1  HNF1A  HOXB3  IL21R  IQGAP2  \\\n",
       "patient_id                                                                \n",
       "CRUK0001        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0002        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0003        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0004        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0005        0     0    0       0     0      0      0      0       0   \n",
       "...           ...   ...  ...     ...   ...    ...    ...    ...     ...   \n",
       "CRUK0861        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0864        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0872        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0881        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0893        0     0    0       0     0      0      0      0       0   \n",
       "\n",
       "Hugo_Symbol  KDM5C  KDM6A  KEAP1  KLF5  KMT2D  KRAS  LATS1  MAP3K13  MAX  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0002         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0003         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0004         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0005         0      0      0     0      0     0      0        0    0   \n",
       "...            ...    ...    ...   ...    ...   ...    ...      ...  ...   \n",
       "CRUK0861         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0864         0      0      0     0      0     1      0        0    0   \n",
       "CRUK0872         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0881         0      0      0     0      0     1      0        0    0   \n",
       "CRUK0893         0      0      0     0      0     1      0        0    0   \n",
       "\n",
       "Hugo_Symbol  MSH2  MTOR  NBN  NCOA6  NEB  NF1  NFKBIE  NIN  NOTCH1  NRAS  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0002        0     0    0      0    0    3       0    0       0     0   \n",
       "CRUK0003        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0004        0     0    0      0    0    0       0    0       3     0   \n",
       "CRUK0005        0     0    0      0    0    0       0    0       0     3   \n",
       "...           ...   ...  ...    ...  ...  ...     ...  ...     ...   ...   \n",
       "CRUK0861        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0864        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0872        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0881        0     0    0      0    0    1       0    0       0     0   \n",
       "CRUK0893        0     0    0      0    0    0       0    0       0     0   \n",
       "\n",
       "Hugo_Symbol  NSD1  NTRK1  PALB2  PASK  PBRM1  PIK3CA  PIK3CG  PIK3R1  PLXNB2  \\\n",
       "patient_id                                                                     \n",
       "CRUK0001        0      0      0     3      0       0       0       0       0   \n",
       "CRUK0002        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0003        0      0      0     0      0       2       0       0       0   \n",
       "CRUK0004        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0005        0      0      0     3      0       0       0       0       0   \n",
       "...           ...    ...    ...   ...    ...     ...     ...     ...     ...   \n",
       "CRUK0861        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0864        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0872        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0881        0      0      0     0      3       0       0       0       0   \n",
       "CRUK0893        0      0      0     0      0       0       0       0       0   \n",
       "\n",
       "Hugo_Symbol  POLE  PRDM1  PRKAR1A  PRRX1  PTEN  PTPRB  PTPRC  PTPRU  RASA1  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0002        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0003        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0004        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0005        0      0        0      0     0      0      0      0      0   \n",
       "...           ...    ...      ...    ...   ...    ...    ...    ...    ...   \n",
       "CRUK0861        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0864        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0872        0      0        0      0     0      1      0      0      0   \n",
       "CRUK0881        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0893        0      0        0      0     0      0      0      0      0   \n",
       "\n",
       "Hugo_Symbol  RNF43  SETD2  SGK223  SMARCA4  SNX29  SPEN  STIM1  STK11  STX2  \\\n",
       "patient_id                                                                    \n",
       "CRUK0001         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0002         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0003         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0004         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0005         0      0       0        0      0     0      0      0     0   \n",
       "...            ...    ...     ...      ...    ...   ...    ...    ...   ...   \n",
       "CRUK0861         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0864         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0872         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0881         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0893         0      0       0        0      0     0      0      0     0   \n",
       "\n",
       "Hugo_Symbol  TNC  TNFAIP3  TNFRSF14  U2AF1  WAS  WT1  ZFPM2  ZMYM2  ZNF483  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0002       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0003       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0004       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0005       0        0         0      0    0    0      0      0       0   \n",
       "...          ...      ...       ...    ...  ...  ...    ...    ...     ...   \n",
       "CRUK0861       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0864       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0872       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0881       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0893       0        0         0      0    0    0      0      0       0   \n",
       "\n",
       "Hugo_Symbol    Ploidy  \n",
       "patient_id             \n",
       "CRUK0001     4.493770  \n",
       "CRUK0002     1.921907  \n",
       "CRUK0003     3.567392  \n",
       "CRUK0004     2.950603  \n",
       "CRUK0005     3.131282  \n",
       "...               ...  \n",
       "CRUK0861     3.514533  \n",
       "CRUK0864     3.556913  \n",
       "CRUK0872     3.919491  \n",
       "CRUK0881     4.186610  \n",
       "CRUK0893     3.048684  \n",
       "\n",
       "[394 rows x 101 columns]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cumulative_hundred_feature_matrix['Ploidy'] = cumulative_hundred_feature_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_hundred_feature_matrix = cumulative_hundred_feature_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_hundred_feature_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [],
   "source": [
    "cumulative_fifty_feature_matrix['Ploidy'] = cumulative_fifty_feature_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_fifty_feature_matrix = cumulative_fifty_feature_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_twenty_feature_matrix['Ploidy'] = cumulative_twenty_feature_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_twenty_feature_matrix = cumulative_twenty_feature_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_ten_feature_matrix['Ploidy'] = cumulative_ten_feature_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_ten_feature_matrix = cumulative_ten_feature_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_five_feature_matrix['Ploidy'] = cumulative_five_feature_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_five_feature_matrix = cumulative_five_feature_matrix.dropna(subset=['Ploidy'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>ANAPC5</th>\n",
       "      <th>ANK2</th>\n",
       "      <th>ARAF</th>\n",
       "      <th>ARHGAP35</th>\n",
       "      <th>ARHGEF12</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>ARID1B</th>\n",
       "      <th>ARID2</th>\n",
       "      <th>ASXL1</th>\n",
       "      <th>AXIN1</th>\n",
       "      <th>B2M</th>\n",
       "      <th>BAP1</th>\n",
       "      <th>BCOR</th>\n",
       "      <th>BLM</th>\n",
       "      <th>BMPR1A</th>\n",
       "      <th>BRCA1</th>\n",
       "      <th>BUB1B</th>\n",
       "      <th>CDC73</th>\n",
       "      <th>CHD4</th>\n",
       "      <th>CIC</th>\n",
       "      <th>CREBBP</th>\n",
       "      <th>CTNNB1</th>\n",
       "      <th>CYLD</th>\n",
       "      <th>DDB2</th>\n",
       "      <th>DDX3X</th>\n",
       "      <th>DOT1L</th>\n",
       "      <th>EPB41L3</th>\n",
       "      <th>ERBB2</th>\n",
       "      <th>EZH1</th>\n",
       "      <th>FAM166A</th>\n",
       "      <th>FANCF</th>\n",
       "      <th>FAS</th>\n",
       "      <th>FAT1</th>\n",
       "      <th>FBXW7</th>\n",
       "      <th>FGFR1</th>\n",
       "      <th>FGFR3</th>\n",
       "      <th>FLCN</th>\n",
       "      <th>FLT4</th>\n",
       "      <th>FN1</th>\n",
       "      <th>GNPTAB</th>\n",
       "      <th>HCN1</th>\n",
       "      <th>HNF1A</th>\n",
       "      <th>HOXB3</th>\n",
       "      <th>IL21R</th>\n",
       "      <th>IQGAP2</th>\n",
       "      <th>KDM5C</th>\n",
       "      <th>KDM6A</th>\n",
       "      <th>KEAP1</th>\n",
       "      <th>KLF5</th>\n",
       "      <th>KMT2D</th>\n",
       "      <th>KRAS</th>\n",
       "      <th>LATS1</th>\n",
       "      <th>MAP3K13</th>\n",
       "      <th>MAX</th>\n",
       "      <th>MSH2</th>\n",
       "      <th>MTOR</th>\n",
       "      <th>NBN</th>\n",
       "      <th>NCOA6</th>\n",
       "      <th>NEB</th>\n",
       "      <th>NF1</th>\n",
       "      <th>NFKBIE</th>\n",
       "      <th>NIN</th>\n",
       "      <th>NOTCH1</th>\n",
       "      <th>NRAS</th>\n",
       "      <th>NSD1</th>\n",
       "      <th>NTRK1</th>\n",
       "      <th>PALB2</th>\n",
       "      <th>PASK</th>\n",
       "      <th>PBRM1</th>\n",
       "      <th>PIK3CA</th>\n",
       "      <th>PIK3CG</th>\n",
       "      <th>PIK3R1</th>\n",
       "      <th>PLXNB2</th>\n",
       "      <th>POLE</th>\n",
       "      <th>PRDM1</th>\n",
       "      <th>PRKAR1A</th>\n",
       "      <th>PRRX1</th>\n",
       "      <th>PTEN</th>\n",
       "      <th>PTPRB</th>\n",
       "      <th>PTPRC</th>\n",
       "      <th>PTPRU</th>\n",
       "      <th>RASA1</th>\n",
       "      <th>RNF43</th>\n",
       "      <th>SETD2</th>\n",
       "      <th>SGK223</th>\n",
       "      <th>SMARCA4</th>\n",
       "      <th>SNX29</th>\n",
       "      <th>SPEN</th>\n",
       "      <th>STIM1</th>\n",
       "      <th>STK11</th>\n",
       "      <th>STX2</th>\n",
       "      <th>TNC</th>\n",
       "      <th>TNFAIP3</th>\n",
       "      <th>TNFRSF14</th>\n",
       "      <th>U2AF1</th>\n",
       "      <th>WAS</th>\n",
       "      <th>WT1</th>\n",
       "      <th>ZFPM2</th>\n",
       "      <th>ZMYM2</th>\n",
       "      <th>ZNF483</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>LUSC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CRUK0001</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.493770</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1.921907</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.567392</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0004</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2.950603</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0005</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0861</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.514533</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0864</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.556913</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0872</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.919491</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0881</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.186610</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0893</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3.048684</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>394 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Hugo_Symbol  ANAPC5  ANK2  ARAF  ARHGAP35  ARHGEF12  ARID1A  ARID1B  ARID2  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001          0     0     0         3         0       0       0      0   \n",
       "CRUK0002          0     0     0         0         0       0       0      0   \n",
       "CRUK0003          0     0     0         0         0       0       0      0   \n",
       "CRUK0004          0     0     0         0         0       0       0      0   \n",
       "CRUK0005          0     0     0         0         0       0       0      0   \n",
       "...             ...   ...   ...       ...       ...     ...     ...    ...   \n",
       "CRUK0861          0     0     0         0         0       0       0      0   \n",
       "CRUK0864          0     0     0         0         0       0       0      0   \n",
       "CRUK0872          0     0     0         0         0       0       0      0   \n",
       "CRUK0881          0     0     0         0         0       0       0      0   \n",
       "CRUK0893          0     0     0         0         0       0       0      0   \n",
       "\n",
       "Hugo_Symbol  ASXL1  AXIN1  B2M  BAP1  BCOR  BLM  BMPR1A  BRCA1  BUB1B  CDC73  \\\n",
       "patient_id                                                                     \n",
       "CRUK0001         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0002         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0003         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0004         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0005         0      0    0     0     0    0       0      0      0      0   \n",
       "...            ...    ...  ...   ...   ...  ...     ...    ...    ...    ...   \n",
       "CRUK0861         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0864         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0872         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0881         0      0    0     0     0    0       0      0      0      0   \n",
       "CRUK0893         0      0    0     0     0    0       0      0      0      0   \n",
       "\n",
       "Hugo_Symbol  CHD4  CIC  CREBBP  CTNNB1  CYLD  DDB2  DDX3X  DOT1L  EPB41L3  \\\n",
       "patient_id                                                                  \n",
       "CRUK0001        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0002        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0003        0    0       0       3     0     0      0      0        0   \n",
       "CRUK0004        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0005        0    0       0       0     0     0      0      0        0   \n",
       "...           ...  ...     ...     ...   ...   ...    ...    ...      ...   \n",
       "CRUK0861        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0864        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0872        0    0       0       0     0     0      0      0        0   \n",
       "CRUK0881        0    0       1       0     0     0      4      0        0   \n",
       "CRUK0893        0    0       0       0     0     0      0      0        0   \n",
       "\n",
       "Hugo_Symbol  ERBB2  EZH1  FAM166A  FANCF  FAS  FAT1  FBXW7  FGFR1  FGFR3  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0002         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0003         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0004         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0005         0     0        0      0    0     0      0      0      0   \n",
       "...            ...   ...      ...    ...  ...   ...    ...    ...    ...   \n",
       "CRUK0861         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0864         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0872         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0881         0     0        0      0    0     0      0      0      0   \n",
       "CRUK0893         0     0        0      0    0     0      0      0      0   \n",
       "\n",
       "Hugo_Symbol  FLCN  FLT4  FN1  GNPTAB  HCN1  HNF1A  HOXB3  IL21R  IQGAP2  \\\n",
       "patient_id                                                                \n",
       "CRUK0001        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0002        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0003        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0004        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0005        0     0    0       0     0      0      0      0       0   \n",
       "...           ...   ...  ...     ...   ...    ...    ...    ...     ...   \n",
       "CRUK0861        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0864        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0872        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0881        0     0    0       0     0      0      0      0       0   \n",
       "CRUK0893        0     0    0       0     0      0      0      0       0   \n",
       "\n",
       "Hugo_Symbol  KDM5C  KDM6A  KEAP1  KLF5  KMT2D  KRAS  LATS1  MAP3K13  MAX  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0002         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0003         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0004         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0005         0      0      0     0      0     0      0        0    0   \n",
       "...            ...    ...    ...   ...    ...   ...    ...      ...  ...   \n",
       "CRUK0861         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0864         0      0      0     0      0     1      0        0    0   \n",
       "CRUK0872         0      0      0     0      0     0      0        0    0   \n",
       "CRUK0881         0      0      0     0      0     1      0        0    0   \n",
       "CRUK0893         0      0      0     0      0     1      0        0    0   \n",
       "\n",
       "Hugo_Symbol  MSH2  MTOR  NBN  NCOA6  NEB  NF1  NFKBIE  NIN  NOTCH1  NRAS  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0002        0     0    0      0    0    3       0    0       0     0   \n",
       "CRUK0003        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0004        0     0    0      0    0    0       0    0       3     0   \n",
       "CRUK0005        0     0    0      0    0    0       0    0       0     3   \n",
       "...           ...   ...  ...    ...  ...  ...     ...  ...     ...   ...   \n",
       "CRUK0861        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0864        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0872        0     0    0      0    0    0       0    0       0     0   \n",
       "CRUK0881        0     0    0      0    0    1       0    0       0     0   \n",
       "CRUK0893        0     0    0      0    0    0       0    0       0     0   \n",
       "\n",
       "Hugo_Symbol  NSD1  NTRK1  PALB2  PASK  PBRM1  PIK3CA  PIK3CG  PIK3R1  PLXNB2  \\\n",
       "patient_id                                                                     \n",
       "CRUK0001        0      0      0     3      0       0       0       0       0   \n",
       "CRUK0002        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0003        0      0      0     0      0       2       0       0       0   \n",
       "CRUK0004        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0005        0      0      0     3      0       0       0       0       0   \n",
       "...           ...    ...    ...   ...    ...     ...     ...     ...     ...   \n",
       "CRUK0861        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0864        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0872        0      0      0     0      0       0       0       0       0   \n",
       "CRUK0881        0      0      0     0      3       0       0       0       0   \n",
       "CRUK0893        0      0      0     0      0       0       0       0       0   \n",
       "\n",
       "Hugo_Symbol  POLE  PRDM1  PRKAR1A  PRRX1  PTEN  PTPRB  PTPRC  PTPRU  RASA1  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0002        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0003        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0004        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0005        0      0        0      0     0      0      0      0      0   \n",
       "...           ...    ...      ...    ...   ...    ...    ...    ...    ...   \n",
       "CRUK0861        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0864        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0872        0      0        0      0     0      1      0      0      0   \n",
       "CRUK0881        0      0        0      0     0      0      0      0      0   \n",
       "CRUK0893        0      0        0      0     0      0      0      0      0   \n",
       "\n",
       "Hugo_Symbol  RNF43  SETD2  SGK223  SMARCA4  SNX29  SPEN  STIM1  STK11  STX2  \\\n",
       "patient_id                                                                    \n",
       "CRUK0001         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0002         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0003         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0004         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0005         0      0       0        0      0     0      0      0     0   \n",
       "...            ...    ...     ...      ...    ...   ...    ...    ...   ...   \n",
       "CRUK0861         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0864         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0872         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0881         0      0       0        0      0     0      0      0     0   \n",
       "CRUK0893         0      0       0        0      0     0      0      0     0   \n",
       "\n",
       "Hugo_Symbol  TNC  TNFAIP3  TNFRSF14  U2AF1  WAS  WT1  ZFPM2  ZMYM2  ZNF483  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0002       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0003       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0004       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0005       0        0         0      0    0    0      0      0       0   \n",
       "...          ...      ...       ...    ...  ...  ...    ...    ...     ...   \n",
       "CRUK0861       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0864       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0872       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0881       0        0         0      0    0    0      0      0       0   \n",
       "CRUK0893       0        0         0      0    0    0      0      0       0   \n",
       "\n",
       "Hugo_Symbol    Ploidy  LUAD  LUSC  \n",
       "patient_id                         \n",
       "CRUK0001     4.493770     1     0  \n",
       "CRUK0002     1.921907     1     0  \n",
       "CRUK0003     3.567392     1     0  \n",
       "CRUK0004     2.950603     1     0  \n",
       "CRUK0005     3.131282     1     0  \n",
       "...               ...   ...   ...  \n",
       "CRUK0861     3.514533     0     1  \n",
       "CRUK0864     3.556913     1     0  \n",
       "CRUK0872     3.919491     0     1  \n",
       "CRUK0881     4.186610     1     0  \n",
       "CRUK0893     3.048684     1     0  \n",
       "\n",
       "[394 rows x 103 columns]"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cumulative_hundred_feature_matrix['LUAD']= cumulative_hundred_feature_matrix.index.isin(LUAD_patient_id).astype(int)\n",
    "cumulative_hundred_feature_matrix['LUSC']= cumulative_hundred_feature_matrix.index.isin(LUSC_patient_id).astype(int)\n",
    "cumulative_hundred_feature_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [],
   "source": [
    "cumulative_fifty_feature_matrix['LUAD']= cumulative_fifty_feature_matrix.index.isin(LUAD_patient_id).astype(int)\n",
    "cumulative_fifty_feature_matrix['LUSC']= cumulative_fifty_feature_matrix.index.isin(LUSC_patient_id).astype(int)\n",
    "cumulative_twenty_feature_matrix['LUAD']= cumulative_twenty_feature_matrix.index.isin(LUAD_patient_id).astype(int)\n",
    "cumulative_twenty_feature_matrix['LUSC']= cumulative_twenty_feature_matrix.index.isin(LUSC_patient_id).astype(int)\n",
    "cumulative_ten_feature_matrix['LUAD']= cumulative_ten_feature_matrix.index.isin(LUAD_patient_id).astype(int)\n",
    "cumulative_ten_feature_matrix['LUSC']= cumulative_ten_feature_matrix.index.isin(LUSC_patient_id).astype(int)\n",
    "cumulative_five_feature_matrix['LUAD']= cumulative_five_feature_matrix.index.isin(LUAD_patient_id).astype(int)\n",
    "cumulative_five_feature_matrix['LUSC']= cumulative_five_feature_matrix.index.isin(LUSC_patient_id).astype(int) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0 0 1 1 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 1 0 1 0 0 1 1 1 1 1\n",
      " 0 1 0 1 1 0 0 1 0 1 0 0 0 0 0 1 0 0 0 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0\n",
      " 1 0 0 0 0 0 1 1 1 0 1 1 0 1 1 0 0 0 0 0 0 1 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0\n",
      " 0 0 0 0 0 1 0 0 0 1 1 1 0 1 1 1 1 1 1 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 0 1 1\n",
      " 0 1 1 0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 0 0 0 0 1 1 0 1 0 1 0 1 1 1 0 0 0 0 0\n",
      " 0 1 1 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 1 1 0 0 1 1 0 0 0 1 1 0 0 0 0 0 0 0 1\n",
      " 0 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 1 0 1 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 1 1\n",
      " 0 0 0 1 1 0 0 0 1 0 0 1 1 0 1 0 0 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 1\n",
      " 0 0 1 0 0 0 0 0 0 0 1 0 1 0 0 1 1 0 0 0 0 1 0 1 0 0 0 0 0 1 0 1 0 0 0 0 0\n",
      " 0 0 0 0 0 1 0 0 1 0 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 1 0 0 0 1 0 1 0 0 0\n",
      " 0 0 1 1 0 0 1 0 0 0 0 1 0 0 0 0 0 1 1 0 0 1 1 0]\n",
      "length: 394\n"
     ]
    }
   ],
   "source": [
    "ploidy_na = ['CRUK0007','CRUK0042','CRUK0055','CRUK0060','CRUK0089','CRUK0093','CRUK0117','CRUK0128','CRUK0182','CRUK0199','CRUK0202',\n",
    "             'CRUK0227','CRUK0285','CRUK0293','CRUK0301','CRUK0305','CRUK0308','CRUK0325','CRUK0366','CRUK0467','CRUK0471','CRUK0540',\n",
    "             'CRUK0569','CRUK0581','CRUK0584','CRUK0643','CRUK0781']\n",
    "\n",
    "# Making the training data\n",
    "clinical_df_cumulative = clinical_df.copy()\n",
    "clinical_df_cumulative = clinical_df_cumulative.sort_values(by=['cruk_id'])\n",
    "clinical_df_cumulative = clinical_df_cumulative[~clinical_df_cumulative['cruk_id'].isin(ploidy_na)]\n",
    "\n",
    "y_train_cumulative = []\n",
    "for index, row in clinical_df_cumulative.iterrows():\n",
    "    if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "        y_train_cumulative.append(1)\n",
    "    else:\n",
    "        y_train_cumulative.append(0)\n",
    "\n",
    "y_cumulative = np.array(y_train_cumulative)\n",
    "print(y_cumulative)\n",
    "print('length:',len(y_cumulative))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(236, 103) (158, 103) (236,) (158,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5304198292220115\n",
      "Predictions:  [0 1 1 0 1 0 0 1 1 0 1 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 1 1 0 0 1 1 0 1 0 0\n",
      " 1 0 1 1 0 1 0 0 1 0 1 0 1 0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 1\n",
      " 1 0 1 0 0 1 1 0 0 0 0 0 0 1 0 0 0 1 1 0 1 0 0 1 0 1 0 0 1 0 1 0 0 1 0 1 0\n",
      " 0 1 0 1 1 1 1 0 0 0 1 1 1 1 1 0 0 0 1 1 0 0 0 1 1 1 0 1 0 0 1 0 0 0 1 0 1\n",
      " 0 1 0 0 0 1 0 1 0 0]\n",
      "Actual:  [1 1 1 1 0 0 0 1 0 0 0 1 0 1 1 0 0 0 0 0 1 0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 1 0 0 0 1 0 1 1 1 1 0 0 1 1 0\n",
      " 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 1\n",
      " 0 1 1 0 0 1 0 0 1 1 1 0 0 0 0 1 1 0 0 0 0 1 0 0 1 0 1 1 0 0 1 0 1 0 0 0 0\n",
      " 0 0 1 1 1 1 1 1 1 0]\n",
      "Balanced Accuracy:  0.474002849002849\n",
      "ROC AUC:  0.43928062678062685\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.64      0.60      0.62       104\n",
      "   Recurrence       0.31      0.35      0.33        54\n",
      "\n",
      "     accuracy                           0.51       158\n",
      "    macro avg       0.48      0.47      0.47       158\n",
      " weighted avg       0.53      0.51      0.52       158\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_hundred_feature_matrix, y_cumulative, test_size = 0.4, random_state = 0, stratify = y_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(236, 53) (158, 53) (236,) (158,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5526684060721063\n",
      "Predictions:  [0 1 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 0 0 1 1 0 1 0 0 1 0 0 0 0 0 0\n",
      " 1 0 1 0 0 1 0 1 0 0 1 0 1 0 0 1 0 1 0 0 0 1 0 0 0 0 1 0 0 0 1 0 0 0 0 1 0\n",
      " 1 0 0 0 0 1 1 0 1 0 0 0 0 1 1 1 0 1 0 0 1 1 0 0 1 0 0 0 1 0 0 0 0 0 0 1 0\n",
      " 0 1 1 1 1 0 0 0 1 0 0 0 1 1 0 0 1 0 1 1 1 1 0 0 1 1 0 1 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [1 1 1 1 0 0 0 1 0 0 0 1 0 1 1 0 0 0 0 0 1 0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 1 0 0 0 1 0 1 1 1 1 0 0 1 1 0\n",
      " 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 1\n",
      " 0 1 1 0 0 1 0 0 1 1 1 0 0 0 0 1 1 0 0 0 0 1 0 0 1 0 1 1 0 0 1 0 1 0 0 0 0\n",
      " 0 0 1 1 1 1 1 1 1 0]\n",
      "Balanced Accuracy:  0.5268874643874644\n",
      "ROC AUC:  0.4894052706552707\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.68      0.70      0.69       104\n",
      "   Recurrence       0.38      0.35      0.37        54\n",
      "\n",
      "     accuracy                           0.58       158\n",
      "    macro avg       0.53      0.53      0.53       158\n",
      " weighted avg       0.57      0.58      0.58       158\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_fifty_feature_matrix, y_cumulative, test_size = 0.4, random_state = 0, stratify = y_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(236, 23) (158, 23) (236,) (158,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5646584440227704\n",
      "Predictions:  [0 0 0 1 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 1 0 1 0 1 0 0 1 0 0 0 1 0 0 0 0 0 0\n",
      " 1 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 1 1 1 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 1\n",
      " 1 0 0 1 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 1 1\n",
      " 0 0 0 0 0 0 0 0 0 1 0 1 1 1 0 0 1 1 1 0 0 0 0 0 1 1 0 1 0 0 0 0 1 0 0 0 0\n",
      " 0 0 0 0 0 0 0 1 0 0]\n",
      "Actual:  [1 1 1 1 0 0 0 1 0 0 0 1 0 1 1 0 0 0 0 0 1 0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 1 0 0 0 1 0 1 1 1 1 0 0 1 1 0\n",
      " 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 1\n",
      " 0 1 1 0 0 1 0 0 1 1 1 0 0 0 0 1 1 0 0 0 0 1 0 0 1 0 1 1 0 0 1 0 1 0 0 0 0\n",
      " 0 0 1 1 1 1 1 1 1 0]\n",
      "Balanced Accuracy:  0.5657051282051282\n",
      "ROC AUC:  0.5909900284900286\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.70      0.80      0.74       104\n",
      "   Recurrence       0.46      0.33      0.39        54\n",
      "\n",
      "     accuracy                           0.64       158\n",
      "    macro avg       0.58      0.57      0.57       158\n",
      " weighted avg       0.62      0.64      0.62       158\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_twenty_feature_matrix, y_cumulative, test_size = 0.4, random_state = 0, stratify = y_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(236, 13) (158, 13) (236,) (158,)\n",
      "Best parameters:  {'alpha': 1, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.5459084440227705\n",
      "Predictions:  [0 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 1 0\n",
      " 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 1]\n",
      "Actual:  [1 1 1 1 0 0 0 1 0 0 0 1 0 1 1 0 0 0 0 0 1 0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 1 0 0 0 1 0 1 1 1 1 0 0 1 1 0\n",
      " 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 1\n",
      " 0 1 1 0 0 1 0 0 1 1 1 0 0 0 0 1 1 0 0 0 0 1 0 0 1 0 1 1 0 0 1 0 1 0 0 0 0\n",
      " 0 0 1 1 1 1 1 1 1 0]\n",
      "Balanced Accuracy:  0.5256410256410257\n",
      "ROC AUC:  0.5500356125356125\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.88      0.76       104\n",
      "   Recurrence       0.43      0.17      0.24        54\n",
      "\n",
      "     accuracy                           0.64       158\n",
      "    macro avg       0.55      0.53      0.50       158\n",
      " weighted avg       0.59      0.64      0.58       158\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_ten_feature_matrix, y_cumulative, test_size = 0.4, random_state = 0, stratify = y_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(236, 8) (158, 8) (236,) (158,)\n",
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5363021821631879\n",
      "Predictions:  [0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 1 0\n",
      " 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 1 1 1 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [1 1 1 1 0 0 0 1 0 0 0 1 0 1 1 0 0 0 0 0 1 0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 1 0 0 0 1 0 1 1 1 1 0 0 1 1 0\n",
      " 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 1 0 0 1 0 0 0 0 0 0 0 0 1\n",
      " 0 1 1 0 0 1 0 0 1 1 1 0 0 0 0 1 1 0 0 0 0 1 0 0 1 0 1 1 0 0 1 0 1 0 0 0 0\n",
      " 0 0 1 1 1 1 1 1 1 0]\n",
      "Balanced Accuracy:  0.4878917378917379\n",
      "ROC AUC:  0.51130698005698\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.65      0.85      0.74       104\n",
      "   Recurrence       0.30      0.13      0.18        54\n",
      "\n",
      "     accuracy                           0.60       158\n",
      "    macro avg       0.48      0.49      0.46       158\n",
      " weighted avg       0.53      0.60      0.55       158\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_five_feature_matrix, y_cumulative, test_size = 0.4, random_state = 0, stratify = y_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Feature Importances:\n",
      "ANAPC5: 0.012258423365269798\n",
      "ANK2: 0.015140864653004323\n",
      "ARAF: 0.0\n",
      "ARHGAP35: 0.0022594957197331752\n",
      "ARHGEF12: 0.0\n",
      "ARID1A: 0.01733994160512151\n",
      "ARID1B: 0.015835740081064847\n",
      "ARID2: 0.004496232870071674\n",
      "ASXL1: 0.000402399539407259\n",
      "AXIN1: 0.015210702589595806\n",
      "B2M: 0.0\n",
      "BAP1: 0.014643876407624544\n",
      "BCOR: 0.0067801227009057575\n",
      "BLM: 0.0\n",
      "BMPR1A: 0.0035481141996434665\n",
      "BRCA1: 0.03240484548956068\n",
      "BUB1B: 0.018499133344347836\n",
      "CDC73: 0.024848236461552077\n",
      "CHD4: 0.012600768428783873\n",
      "CIC: 0.0\n",
      "CREBBP: 0.03374772595705237\n",
      "CTNNB1: 0.020607408159666402\n",
      "CYLD: 0.010140468393063871\n",
      "DDB2: 0.0\n",
      "DDX3X: 0.013520624524085161\n",
      "DOT1L: 0.033634410732305234\n",
      "EPB41L3: 0.0\n",
      "ERBB2: 0.02706526090577483\n",
      "EZH1: 0.025617809374037407\n",
      "FAM166A: 0.0\n",
      "FANCF: 0.0\n",
      "FAS: 0.029218636647642203\n",
      "FAT1: 0.005106076973622248\n",
      "FBXW7: 0.00045189253088518457\n",
      "FGFR1: 0.0\n",
      "FGFR3: 0.001690078065510646\n",
      "FLCN: 0.0\n",
      "FLT4: 0.012375307699715914\n",
      "FN1: 0.019156842013165965\n",
      "GNPTAB: 0.0\n",
      "HCN1: 0.012318407116878469\n",
      "HNF1A: 0.00042607010054893665\n",
      "HOXB3: 0.0\n",
      "IL21R: 0.0\n",
      "IQGAP2: 0.0\n",
      "KDM5C: 0.013133195263802879\n",
      "KDM6A: 0.010140468393063871\n",
      "KEAP1: 0.02391676169033731\n",
      "KLF5: 0.0\n",
      "KMT2D: 0.02901869038096412\n",
      "KRAS: 0.04737548339355354\n",
      "LATS1: 0.0\n",
      "MAP3K13: 0.018030974368726055\n",
      "MAX: 0.0\n",
      "MSH2: 0.0\n",
      "MTOR: 0.0\n",
      "NBN: 0.010140468393063871\n",
      "NCOA6: 0.0\n",
      "NEB: 0.0\n",
      "NF1: 0.019992344871427137\n",
      "NFKBIE: 0.01882367954337171\n",
      "NIN: 0.0\n",
      "NOTCH1: 0.01525626310227321\n",
      "NRAS: 0.018738071402794\n",
      "NSD1: 0.0\n",
      "NTRK1: 0.012554691443140534\n",
      "PALB2: 0.0\n",
      "PASK: 0.010773752914098366\n",
      "PBRM1: 0.019013378236994758\n",
      "PIK3CA: 0.011313842122465985\n",
      "PIK3CG: 0.0\n",
      "PIK3R1: 0.01736316050321801\n",
      "PLXNB2: 0.004940012771434689\n",
      "POLE: 0.0016265412961305457\n",
      "PRDM1: 0.026260934295561266\n",
      "PRKAR1A: 0.0\n",
      "PRRX1: 0.0\n",
      "PTEN: 0.01406581099683052\n",
      "PTPRB: 0.016717941569103716\n",
      "PTPRC: 0.022827961997460254\n",
      "PTPRU: 0.0\n",
      "RASA1: 0.0033801561310212894\n",
      "RNF43: 0.0010029034674458829\n",
      "SETD2: 0.024548099236140902\n",
      "SGK223: 0.014068396861523398\n",
      "SMARCA4: 0.0058801162168969495\n",
      "SNX29: 0.024998023663375057\n",
      "SPEN: 0.0004801358140655198\n",
      "STIM1: 0.0\n",
      "STK11: 0.041013995052014286\n",
      "STX2: 0.0\n",
      "TNC: 0.0\n",
      "TNFAIP3: 0.0\n",
      "TNFRSF14: 0.01270686952123899\n",
      "U2AF1: 0.024031520986302055\n",
      "WAS: 0.01211420661979604\n",
      "WT1: 0.0\n",
      "ZFPM2: 0.0\n",
      "ZMYM2: 0.0\n",
      "ZNF483: 0.012405230830721863\n",
      "\n",
      "Selected Features: []\n"
     ]
    }
   ],
   "source": [
    "from sklearn.tree import DecisionTreeClassifier\n",
    "\n",
    "\n",
    "# Split data into train and test sets\n",
    "X_train, X_test, y_train, y_test = train_test_split(hundred_feature_matrix, y, test_size=0.4, random_state=0)\n",
    "\n",
    "# Create decision tree classifier\n",
    "dt_classifier = DecisionTreeClassifier(random_state=0)\n",
    "\n",
    "# Fit the classifier to the training data\n",
    "dt_classifier.fit(X_train, y_train)\n",
    "\n",
    "# Print feature importances\n",
    "print(\"Feature Importances:\")\n",
    "for feature, importance in zip(hundred_feature_matrix.columns, dt_classifier.feature_importances_):\n",
    "    print(f\"{feature}: {importance}\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Selected Features: ['ANAPC5', 'ANK2', 'ARID1A', 'ARID1B', 'AXIN1', 'BAP1', 'BRCA1', 'BUB1B', 'CDC73', 'CHD4', 'CREBBP', 'CTNNB1', 'CYLD', 'DDX3X', 'DOT1L', 'ERBB2', 'EZH1', 'FAS', 'FLT4', 'FN1', 'HCN1', 'KDM5C', 'KDM6A', 'KEAP1', 'KMT2D', 'KRAS', 'MAP3K13', 'NBN', 'NF1', 'NFKBIE', 'NOTCH1', 'NRAS', 'NTRK1', 'PASK', 'PBRM1', 'PIK3CA', 'PIK3R1', 'PRDM1', 'PTEN', 'PTPRB', 'PTPRC', 'SETD2', 'SGK223', 'SNX29', 'STK11', 'TNFRSF14', 'U2AF1', 'WAS', 'ZNF483']\n"
     ]
    }
   ],
   "source": [
    "# Select features with importance above a certain threshold\n",
    "threshold = 0.01\n",
    "selected_features = [feature for feature, importance in zip(hundred_feature_matrix.columns, dt_classifier.feature_importances_) if importance > threshold]\n",
    "print(\"\\nSelected Features:\", selected_features)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
