{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import collections\n",
    "import pandas as pd\n",
    "import pyreadr\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "import seaborn as sns\n",
    "from venn import venn\n",
    "import numpy as np\n",
    "import math\n",
    "import scipy\n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cruk_id</th>\n",
       "      <th>tumour_id_muttable_cruk</th>\n",
       "      <th>tumour_id_per_patient</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>packyears</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>ECOG_PS</th>\n",
       "      <th>pathologyTNM</th>\n",
       "      <th>pT_stage_per_patient</th>\n",
       "      <th>pN_stage_per_patient</th>\n",
       "      <th>LVI_per_patient</th>\n",
       "      <th>PL_per_patient</th>\n",
       "      <th>margin_status_per_patient</th>\n",
       "      <th>size_pathology_per_patient</th>\n",
       "      <th>Surgery_type</th>\n",
       "      <th>histology_lesion1</th>\n",
       "      <th>histology_lesion1_merged</th>\n",
       "      <th>lesion1_sampled</th>\n",
       "      <th>histology_lesion2</th>\n",
       "      <th>lesion2_sampled</th>\n",
       "      <th>histology_multi_full</th>\n",
       "      <th>histology_multi_full_genomically.confirmed</th>\n",
       "      <th>LUAD_pred_subtype</th>\n",
       "      <th>adjuvant_treatment_YN</th>\n",
       "      <th>adjuvant_treatment_given</th>\n",
       "      <th>num_cycle_na.added</th>\n",
       "      <th>CHMPlatDgName_cleaned</th>\n",
       "      <th>CHMOthDgName_cleaned</th>\n",
       "      <th>AdjRadStartTime_manual</th>\n",
       "      <th>AdjRadEndTime_manual</th>\n",
       "      <th>Recurrence_time_use</th>\n",
       "      <th>newPrim_time_use</th>\n",
       "      <th>first_dfs_any_event_rec.or.new.primary</th>\n",
       "      <th>first_event_during_followup</th>\n",
       "      <th>cens_os</th>\n",
       "      <th>os_time</th>\n",
       "      <th>cens_dfs</th>\n",
       "      <th>dfs_time</th>\n",
       "      <th>cens_dfs_any_event</th>\n",
       "      <th>dfs_time_any_event</th>\n",
       "      <th>cens_lung_event</th>\n",
       "      <th>lung_event_time</th>\n",
       "      <th>Relapse_cat</th>\n",
       "      <th>Relapse_cat_new</th>\n",
       "      <th>tx100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0034</td>\n",
       "      <td>CRUK0034</td>\n",
       "      <td>CRUK0034</td>\n",
       "      <td>68</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- Irish</td>\n",
       "      <td>20.0</td>\n",
       "      <td>35</td>\n",
       "      <td>35.000</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1849</td>\n",
       "      <td>0</td>\n",
       "      <td>1849</td>\n",
       "      <td>0</td>\n",
       "      <td>1849</td>\n",
       "      <td>0</td>\n",
       "      <td>1849</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0150</td>\n",
       "      <td>CRUK0150</td>\n",
       "      <td>CRUK0150</td>\n",
       "      <td>81</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>44.5</td>\n",
       "      <td>49</td>\n",
       "      <td>109.025</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>LCNEC</td>\n",
       "      <td>LCNEC</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Other</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>1362</td>\n",
       "      <td>1</td>\n",
       "      <td>1362</td>\n",
       "      <td>1</td>\n",
       "      <td>1362</td>\n",
       "      <td>0</td>\n",
       "      <td>1362</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0159</td>\n",
       "      <td>CRUK0159</td>\n",
       "      <td>CRUK0159</td>\n",
       "      <td>60</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>38</td>\n",
       "      <td>38.000</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1935</td>\n",
       "      <td>new_primary</td>\n",
       "      <td>new_primary</td>\n",
       "      <td>1</td>\n",
       "      <td>2224</td>\n",
       "      <td>1</td>\n",
       "      <td>2224</td>\n",
       "      <td>1</td>\n",
       "      <td>1935</td>\n",
       "      <td>0</td>\n",
       "      <td>2224</td>\n",
       "      <td>Second primary lung</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0090</td>\n",
       "      <td>CRUK0090</td>\n",
       "      <td>CRUK0090</td>\n",
       "      <td>65</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>35</td>\n",
       "      <td>17.500</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Segmentectomy or wedge resection</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>3</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1143.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>2365</td>\n",
       "      <td>1</td>\n",
       "      <td>1143</td>\n",
       "      <td>1</td>\n",
       "      <td>1143</td>\n",
       "      <td>1</td>\n",
       "      <td>1143</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0045</td>\n",
       "      <td>CRUK0045</td>\n",
       "      <td>CRUK0045</td>\n",
       "      <td>85</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>25</td>\n",
       "      <td>12.500</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>papillary</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>917.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>986</td>\n",
       "      <td>1</td>\n",
       "      <td>917</td>\n",
       "      <td>1</td>\n",
       "      <td>917</td>\n",
       "      <td>1</td>\n",
       "      <td>917</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    cruk_id tumour_id_muttable_cruk tumour_id_per_patient  age     sex  \\\n",
       "0  CRUK0034                CRUK0034              CRUK0034   68  Female   \n",
       "1  CRUK0150                CRUK0150              CRUK0150   81    Male   \n",
       "2  CRUK0159                CRUK0159              CRUK0159   60    Male   \n",
       "3  CRUK0090                CRUK0090              CRUK0090   65    Male   \n",
       "4  CRUK0045                CRUK0045              CRUK0045   85    Male   \n",
       "\n",
       "        ethnicity  cigs_perday  years_smoking  packyears  \\\n",
       "0    White- Irish         20.0             35     35.000   \n",
       "1  White- British         44.5             49    109.025   \n",
       "2  White- British         20.0             38     38.000   \n",
       "3  White- British         10.0             35     17.500   \n",
       "4  White- British         10.0             25     12.500   \n",
       "\n",
       "  smoking_status_merged  is.family.lung  ECOG_PS pathologyTNM  \\\n",
       "0             Ex-Smoker            True      0.0           IB   \n",
       "1             Ex-Smoker           False      0.0          IIB   \n",
       "2                Smoker           False      0.0           IB   \n",
       "3             Ex-Smoker           False      1.0          IIA   \n",
       "4             Ex-Smoker           False      1.0           IB   \n",
       "\n",
       "  pT_stage_per_patient  pN_stage_per_patient LVI_per_patient PL_per_patient  \\\n",
       "0                   2a                     0             Yes             No   \n",
       "1                    3                     0             Yes             No   \n",
       "2                   2a                     0             Yes            Yes   \n",
       "3                   2a                     1             Yes            Yes   \n",
       "4                   2a                     0             Yes            Yes   \n",
       "\n",
       "  margin_status_per_patient  size_pathology_per_patient  \\\n",
       "0                        R0                        40.0   \n",
       "1                        R0                        75.0   \n",
       "2                        R0                        36.0   \n",
       "3                        R0                        35.0   \n",
       "4                        R0                        48.0   \n",
       "\n",
       "                       Surgery_type        histology_lesion1  \\\n",
       "0                         Lobectomy  Invasive adenocarcinoma   \n",
       "1                         Lobectomy                    LCNEC   \n",
       "2                         Lobectomy  Invasive adenocarcinoma   \n",
       "3  Segmentectomy or wedge resection  Squamous cell carcinoma   \n",
       "4                         Lobectomy  Invasive adenocarcinoma   \n",
       "\n",
       "  histology_lesion1_merged  lesion1_sampled histology_lesion2 lesion2_sampled  \\\n",
       "0  Invasive adenocarcinoma             True               NaN             NaN   \n",
       "1                    LCNEC             True               NaN             NaN   \n",
       "2  Invasive adenocarcinoma             True               NaN             NaN   \n",
       "3  Squamous cell carcinoma             True               NaN             NaN   \n",
       "4  Invasive adenocarcinoma             True               NaN             NaN   \n",
       "\n",
       "  histology_multi_full histology_multi_full_genomically.confirmed  \\\n",
       "0                 LUAD                                       LUAD   \n",
       "1                Other                                      Other   \n",
       "2                 LUAD                                       LUAD   \n",
       "3                 LUSC                                       LUSC   \n",
       "4                 LUAD                                       LUAD   \n",
       "\n",
       "  LUAD_pred_subtype adjuvant_treatment_YN adjuvant_treatment_given  \\\n",
       "0            acinar           No adjuvant                      NaN   \n",
       "1               NaN           No adjuvant                      NaN   \n",
       "2            acinar              Adjuvant           Platinum chemo   \n",
       "3               NaN              Adjuvant           Platinum chemo   \n",
       "4         papillary           No adjuvant                      NaN   \n",
       "\n",
       "  num_cycle_na.added                    CHMPlatDgName_cleaned  \\\n",
       "0                  0                                      NaN   \n",
       "1                  0                                      NaN   \n",
       "2                  4  Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "3                  3            Cisplatin;Cisplatin;Cisplatin   \n",
       "4                  0                                      NaN   \n",
       "\n",
       "                              CHMOthDgName_cleaned  AdjRadStartTime_manual  \\\n",
       "0                                              NaN                     NaN   \n",
       "1                                              NaN                     NaN   \n",
       "2  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "3              Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "4                                              NaN                     NaN   \n",
       "\n",
       "   AdjRadEndTime_manual  Recurrence_time_use newPrim_time_use  \\\n",
       "0                   NaN                  NaN              NaN   \n",
       "1                   NaN                  NaN              NaN   \n",
       "2                   NaN                  NaN             1935   \n",
       "3                   NaN               1143.0              NaN   \n",
       "4                   NaN                917.0              NaN   \n",
       "\n",
       "  first_dfs_any_event_rec.or.new.primary first_event_during_followup  cens_os  \\\n",
       "0                                    NaN                         NaN        0   \n",
       "1                                    NaN                       Death        1   \n",
       "2                            new_primary                 new_primary        1   \n",
       "3                             recurrence                  recurrence        0   \n",
       "4                             recurrence                  recurrence        1   \n",
       "\n",
       "   os_time  cens_dfs  dfs_time  cens_dfs_any_event  dfs_time_any_event  \\\n",
       "0     1849         0      1849                   0                1849   \n",
       "1     1362         1      1362                   1                1362   \n",
       "2     2224         1      2224                   1                1935   \n",
       "3     2365         1      1143                   1                1143   \n",
       "4      986         1       917                   1                 917   \n",
       "\n",
       "   cens_lung_event  lung_event_time          Relapse_cat Relapse_cat_new  \\\n",
       "0                0             1849                  NaN          No rec   \n",
       "1                0             1362                  NaN          No rec   \n",
       "2                0             2224  Second primary lung          No rec   \n",
       "3                1             1143        Intrathoracic   Intrathoracic   \n",
       "4                1              917        Intrathoracic   Intrathoracic   \n",
       "\n",
       "   tx100  \n",
       "0   True  \n",
       "1  False  \n",
       "2  False  \n",
       "3   True  \n",
       "4   True  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/figurecode/data/20221109_TRACERx421_all_patient_df.rds')\n",
    "clinical_df = pd.DataFrame(clinical_df[None])\n",
    "clinical_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>clinical_sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Histology_per_tumour_id_muttable</th>\n",
       "      <th>histology_3</th>\n",
       "      <th>LUAD_pred_subtype_with.IMA_per_tumour</th>\n",
       "      <th>site_per_lesion</th>\n",
       "      <th>pT_stage_per_lesion</th>\n",
       "      <th>pN_stage_per_lesion</th>\n",
       "      <th>size_pathology_per_lesion</th>\n",
       "      <th>vascular_invasion_per_lesion</th>\n",
       "      <th>pleural_invasion_per_lesion</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>pack_years</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>pathologyTNM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0034</td>\n",
       "      <td>CRUK0034</td>\n",
       "      <td>Female</td>\n",
       "      <td>68</td>\n",
       "      <td>White- Irish</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>40</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>35</td>\n",
       "      <td>35.000</td>\n",
       "      <td>True</td>\n",
       "      <td>IB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0150</td>\n",
       "      <td>CRUK0150</td>\n",
       "      <td>Male</td>\n",
       "      <td>81</td>\n",
       "      <td>White- British</td>\n",
       "      <td>LCNEC</td>\n",
       "      <td>Other</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Left Upper Lobe</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>75</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>44.5</td>\n",
       "      <td>49</td>\n",
       "      <td>109.025</td>\n",
       "      <td>False</td>\n",
       "      <td>IIB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0159</td>\n",
       "      <td>CRUK0159</td>\n",
       "      <td>Male</td>\n",
       "      <td>60</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>36</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>38</td>\n",
       "      <td>38.000</td>\n",
       "      <td>False</td>\n",
       "      <td>IB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0090</td>\n",
       "      <td>CRUK0090</td>\n",
       "      <td>Male</td>\n",
       "      <td>65</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>35</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>10.0</td>\n",
       "      <td>35</td>\n",
       "      <td>17.500</td>\n",
       "      <td>False</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0045</td>\n",
       "      <td>CRUK0045</td>\n",
       "      <td>Male</td>\n",
       "      <td>85</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>papillary</td>\n",
       "      <td>Left Lower Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>48</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>10.0</td>\n",
       "      <td>25</td>\n",
       "      <td>12.500</td>\n",
       "      <td>False</td>\n",
       "      <td>IB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  tumour_id patient_id clinical_sex  age       ethnicity  \\\n",
       "0  CRUK0034   CRUK0034       Female   68    White- Irish   \n",
       "1  CRUK0150   CRUK0150         Male   81  White- British   \n",
       "2  CRUK0159   CRUK0159         Male   60  White- British   \n",
       "3  CRUK0090   CRUK0090         Male   65  White- British   \n",
       "4  CRUK0045   CRUK0045         Male   85  White- British   \n",
       "\n",
       "  Histology_per_tumour_id_muttable histology_3  \\\n",
       "0          Invasive adenocarcinoma        LUAD   \n",
       "1                            LCNEC       Other   \n",
       "2          Invasive adenocarcinoma        LUAD   \n",
       "3          Squamous cell carcinoma        LUSC   \n",
       "4          Invasive adenocarcinoma        LUAD   \n",
       "\n",
       "  LUAD_pred_subtype_with.IMA_per_tumour   site_per_lesion pT_stage_per_lesion  \\\n",
       "0                                acinar  Right Upper Lobe                  2a   \n",
       "1                                   NaN   Left Upper Lobe                   3   \n",
       "2                                acinar  Right Upper Lobe                  2a   \n",
       "3                                   NaN  Right Upper Lobe                  2a   \n",
       "4                             papillary   Left Lower Lobe                  2a   \n",
       "\n",
       "  pN_stage_per_lesion size_pathology_per_lesion vascular_invasion_per_lesion  \\\n",
       "0                   0                        40                          Yes   \n",
       "1                   0                        75                          Yes   \n",
       "2                   0                        36                          Yes   \n",
       "3                   1                        35                          Yes   \n",
       "4                   0                        48                          Yes   \n",
       "\n",
       "  pleural_invasion_per_lesion smoking_status_merged  cigs_perday  \\\n",
       "0                          No             Ex-Smoker         20.0   \n",
       "1                          No             Ex-Smoker         44.5   \n",
       "2                         Yes                Smoker         20.0   \n",
       "3                         Yes             Ex-Smoker         10.0   \n",
       "4                         Yes             Ex-Smoker         10.0   \n",
       "\n",
       "   years_smoking  pack_years  is.family.lung pathologyTNM  \n",
       "0             35      35.000            True           IB  \n",
       "1             49     109.025           False          IIB  \n",
       "2             38      38.000           False           IB  \n",
       "3             35      17.500           False          IIA  \n",
       "4             25      12.500           False           IB  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumor_df = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/LCCE_converted_data/20221109_TRACERx421_all_tumour_df.rds')\n",
    "tumor_df = pd.DataFrame(tumor_df[None])\n",
    "tumor_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>mutation_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>ref</th>\n",
       "      <th>var</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>func</th>\n",
       "      <th>exonic.func</th>\n",
       "      <th>NucleotideChange</th>\n",
       "      <th>AAChange</th>\n",
       "      <th>GL_VAF</th>\n",
       "      <th>GL_nAlt</th>\n",
       "      <th>GL_depth</th>\n",
       "      <th>ITHState</th>\n",
       "      <th>PyCloneClonal_SC</th>\n",
       "      <th>PyCloneCluster_SC</th>\n",
       "      <th>cleanCluster_SC</th>\n",
       "      <th>combTiming_SC</th>\n",
       "      <th>DriverMut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:34291428:G</td>\n",
       "      <td>chr19</td>\n",
       "      <td>34291428</td>\n",
       "      <td>34291428</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KCTD15</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>187</td>\n",
       "      <td>1</td>\n",
       "      <td>S</td>\n",
       "      <td>3.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:9084770:C</td>\n",
       "      <td>chr19</td>\n",
       "      <td>9084770</td>\n",
       "      <td>9084770</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>MUC16</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G7045T</td>\n",
       "      <td>p.V2349L</td>\n",
       "      <td>0.39</td>\n",
       "      <td>2</td>\n",
       "      <td>510</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:1:2160418:G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2160418</td>\n",
       "      <td>2160418</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SKI</td>\n",
       "      <td>exonic</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>c.G213T</td>\n",
       "      <td>p.P71P</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:40095981:G</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40095981</td>\n",
       "      <td>40095981</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>LGALS13</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G256T</td>\n",
       "      <td>p.G86C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>659</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:41313703:C</td>\n",
       "      <td>chr19</td>\n",
       "      <td>41313703</td>\n",
       "      <td>41313703</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>RAB4B-EGLN2</td>\n",
       "      <td>ncRNA_intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>894</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>5.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  patient_id tumour_id             mutation_id    chr     start      stop ref  \\\n",
       "0   CRUK0005  CRUK0005  CRUK0005:19:34291428:G  chr19  34291428  34291428   G   \n",
       "1   CRUK0005  CRUK0005   CRUK0005:19:9084770:C  chr19   9084770   9084770   C   \n",
       "2   CRUK0005  CRUK0005    CRUK0005:1:2160418:G   chr1   2160418   2160418   G   \n",
       "3   CRUK0005  CRUK0005  CRUK0005:19:40095981:G  chr19  40095981  40095981   G   \n",
       "4   CRUK0005  CRUK0005  CRUK0005:19:41313703:C  chr19  41313703  41313703   C   \n",
       "\n",
       "  var  Hugo_Symbol            func        exonic.func NucleotideChange  \\\n",
       "0   T       KCTD15        intronic                NaN              NaN   \n",
       "1   A        MUC16          exonic  nonsynonymous SNV         c.G7045T   \n",
       "2   T          SKI          exonic     synonymous SNV          c.G213T   \n",
       "3   T      LGALS13          exonic  nonsynonymous SNV          c.G256T   \n",
       "4   G  RAB4B-EGLN2  ncRNA_intronic                NaN              NaN   \n",
       "\n",
       "   AAChange GL_VAF GL_nAlt GL_depth ITHState PyCloneClonal_SC  \\\n",
       "0       NaN      0       0      187        1                S   \n",
       "1  p.V2349L   0.39       2      510        1                C   \n",
       "2    p.P71P      0       0      114        1                C   \n",
       "3    p.G86C      0       0      659        1                C   \n",
       "4       NaN      0       0      894        3                S   \n",
       "\n",
       "   PyCloneCluster_SC cleanCluster_SC combTiming_SC  DriverMut  \n",
       "0                3.0            True     subclonal      False  \n",
       "1                2.0            True         early      False  \n",
       "2                2.0            True         early      False  \n",
       "3                2.0            True         early      False  \n",
       "4                5.0            True     subclonal      False  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_table = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/LCCE_converted_data/20221109_TRACERx421_mutation_table.rds')\n",
    "mut_table = pd.DataFrame(mut_table[None])\n",
    "mut_table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cruk_id</th>\n",
       "      <th>tumour_id_muttable_cruk</th>\n",
       "      <th>tumour_id_per_patient</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>packyears</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>ECOG_PS</th>\n",
       "      <th>pathologyTNM</th>\n",
       "      <th>pT_stage_per_patient</th>\n",
       "      <th>pN_stage_per_patient</th>\n",
       "      <th>LVI_per_patient</th>\n",
       "      <th>PL_per_patient</th>\n",
       "      <th>margin_status_per_patient</th>\n",
       "      <th>size_pathology_per_patient</th>\n",
       "      <th>Surgery_type</th>\n",
       "      <th>histology_lesion1</th>\n",
       "      <th>histology_lesion1_merged</th>\n",
       "      <th>lesion1_sampled</th>\n",
       "      <th>histology_lesion2</th>\n",
       "      <th>lesion2_sampled</th>\n",
       "      <th>histology_multi_full</th>\n",
       "      <th>histology_multi_full_genomically.confirmed</th>\n",
       "      <th>LUAD_pred_subtype</th>\n",
       "      <th>adjuvant_treatment_YN</th>\n",
       "      <th>adjuvant_treatment_given</th>\n",
       "      <th>num_cycle_na.added</th>\n",
       "      <th>CHMPlatDgName_cleaned</th>\n",
       "      <th>CHMOthDgName_cleaned</th>\n",
       "      <th>AdjRadStartTime_manual</th>\n",
       "      <th>AdjRadEndTime_manual</th>\n",
       "      <th>Recurrence_time_use</th>\n",
       "      <th>newPrim_time_use</th>\n",
       "      <th>first_dfs_any_event_rec.or.new.primary</th>\n",
       "      <th>first_event_during_followup</th>\n",
       "      <th>cens_os</th>\n",
       "      <th>os_time</th>\n",
       "      <th>cens_dfs</th>\n",
       "      <th>dfs_time</th>\n",
       "      <th>cens_dfs_any_event</th>\n",
       "      <th>dfs_time_any_event</th>\n",
       "      <th>cens_lung_event</th>\n",
       "      <th>lung_event_time</th>\n",
       "      <th>Relapse_cat</th>\n",
       "      <th>Relapse_cat_new</th>\n",
       "      <th>tx100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>30</td>\n",
       "      <td>30.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>0</td>\n",
       "      <td>1120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>81</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>5.0</td>\n",
       "      <td>35</td>\n",
       "      <td>8.75</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>217.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>2304</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>69</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>325.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>1323</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>74</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>56</td>\n",
       "      <td>56.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>0</td>\n",
       "      <td>649</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>65</td>\n",
       "      <td>Female</td>\n",
       "      <td>Middle eastern</td>\n",
       "      <td>10.0</td>\n",
       "      <td>45</td>\n",
       "      <td>22.50</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1b</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>30</td>\n",
       "      <td>15.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>66</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>54</td>\n",
       "      <td>54.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>883.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>893</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881_Tumour1;CRUK0881_Tumour2</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>Lesion1: collision tumour</td>\n",
       "      <td>False</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>758</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>15.0</td>\n",
       "      <td>43</td>\n",
       "      <td>32.25</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>421 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      cruk_id tumour_id_muttable_cruk              tumour_id_per_patient  age  \\\n",
       "88   CRUK0001                CRUK0001                           CRUK0001   84   \n",
       "79   CRUK0002                CRUK0002                           CRUK0002   81   \n",
       "26   CRUK0003                CRUK0003                           CRUK0003   78   \n",
       "23   CRUK0004                CRUK0004                           CRUK0004   69   \n",
       "7    CRUK0005                CRUK0005                           CRUK0005   74   \n",
       "..        ...                     ...                                ...  ...   \n",
       "414  CRUK0861                CRUK0861                           CRUK0861   65   \n",
       "418  CRUK0864                CRUK0864                           CRUK0864   78   \n",
       "416  CRUK0872                CRUK0872                           CRUK0872   66   \n",
       "420  CRUK0881        CRUK0881_Tumour1  CRUK0881_Tumour1;CRUK0881_Tumour2   84   \n",
       "415  CRUK0893                CRUK0893                           CRUK0893   59   \n",
       "\n",
       "        sex       ethnicity  cigs_perday  years_smoking  packyears  \\\n",
       "88     Male  White- British         20.0             30      30.00   \n",
       "79     Male  White- British          5.0             35       8.75   \n",
       "26   Female  White- British          0.0              0       0.00   \n",
       "23   Female  White- British         10.0             27      13.50   \n",
       "7    Female  White- British         20.0             56      56.00   \n",
       "..      ...             ...          ...            ...        ...   \n",
       "414  Female  Middle eastern         10.0             45      22.50   \n",
       "418  Female  White- British         10.0             30      15.00   \n",
       "416    Male  White- British         20.0             54      54.00   \n",
       "420    Male  White- British         20.0             70      70.00   \n",
       "415  Female  White- British         15.0             43      32.25   \n",
       "\n",
       "    smoking_status_merged  is.family.lung  ECOG_PS pathologyTNM  \\\n",
       "88              Ex-Smoker           False      1.0          IIA   \n",
       "79              Ex-Smoker           False      1.0           IB   \n",
       "26           Never Smoked           False      0.0         IIIA   \n",
       "23              Ex-Smoker            True      0.0          IIB   \n",
       "7               Ex-Smoker           False      1.0         IIIA   \n",
       "..                    ...             ...      ...          ...   \n",
       "414                Smoker           False      1.0           IA   \n",
       "418             Ex-Smoker            True      0.0          IIA   \n",
       "416             Ex-Smoker           False      1.0         IIIA   \n",
       "420                Smoker           False      1.0          IIA   \n",
       "415                Smoker           False      1.0           IA   \n",
       "\n",
       "    pT_stage_per_patient  pN_stage_per_patient LVI_per_patient PL_per_patient  \\\n",
       "88                    2a                     1              No            Yes   \n",
       "79                    2a                     0             Yes             No   \n",
       "26                    2b                     2             Yes             No   \n",
       "23                     3                     0              No            Yes   \n",
       "7                     2a                     2             Yes            Yes   \n",
       "..                   ...                   ...             ...            ...   \n",
       "414                   1b                     0              No             No   \n",
       "418                   1b                     1              No             No   \n",
       "416                   1b                     2              No             No   \n",
       "420                   2b                     0             Yes            Yes   \n",
       "415                   1a                     0             Yes             No   \n",
       "\n",
       "    margin_status_per_patient  size_pathology_per_patient Surgery_type  \\\n",
       "88                         R0                        26.0    Lobectomy   \n",
       "79                         R0                        35.0    Lobectomy   \n",
       "26                         R0                        52.0    Lobectomy   \n",
       "23                         R0                        47.0    Lobectomy   \n",
       "7                          R0                        43.0    Lobectomy   \n",
       "..                        ...                         ...          ...   \n",
       "414                        R0                        25.0    Lobectomy   \n",
       "418                        R0                        27.0    Lobectomy   \n",
       "416                        R0                        25.0    Lobectomy   \n",
       "420                        R0                        62.0    Lobectomy   \n",
       "415                        R0                        17.0    Lobectomy   \n",
       "\n",
       "           histology_lesion1 histology_lesion1_merged  lesion1_sampled  \\\n",
       "88   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "79   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "26   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "23   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "7    Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "..                       ...                      ...              ...   \n",
       "414  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "418  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "416  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "420  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "415  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "\n",
       "             histology_lesion2 lesion2_sampled histology_multi_full  \\\n",
       "88                         NaN             NaN                 LUAD   \n",
       "79                         NaN             NaN                 LUAD   \n",
       "26                         NaN             NaN                 LUAD   \n",
       "23                         NaN             NaN                 LUAD   \n",
       "7                          NaN             NaN                 LUAD   \n",
       "..                         ...             ...                  ...   \n",
       "414                        NaN             NaN                 LUSC   \n",
       "418                        NaN             NaN                 LUAD   \n",
       "416                        NaN             NaN                 LUSC   \n",
       "420  Lesion1: collision tumour           False               LUADx2   \n",
       "415                        NaN             NaN                 LUAD   \n",
       "\n",
       "    histology_multi_full_genomically.confirmed LUAD_pred_subtype  \\\n",
       "88                                        LUAD            acinar   \n",
       "79                                        LUAD            acinar   \n",
       "26                                        LUAD            acinar   \n",
       "23                                        LUAD            acinar   \n",
       "7                                         LUAD            acinar   \n",
       "..                                         ...               ...   \n",
       "414                                       LUSC               NaN   \n",
       "418                                       LUAD             solid   \n",
       "416                                       LUSC               NaN   \n",
       "420                                     LUADx2            acinar   \n",
       "415                                       LUAD            acinar   \n",
       "\n",
       "    adjuvant_treatment_YN adjuvant_treatment_given num_cycle_na.added  \\\n",
       "88            No adjuvant                      NaN                  0   \n",
       "79            No adjuvant                      NaN                  0   \n",
       "26               Adjuvant           Platinum chemo                  4   \n",
       "23               Adjuvant           Platinum chemo                  4   \n",
       "7             No adjuvant                      NaN                  0   \n",
       "..                    ...                      ...                ...   \n",
       "414           No adjuvant                      NaN                  0   \n",
       "418              Adjuvant           Platinum chemo                  4   \n",
       "416           No adjuvant                      NaN                  0   \n",
       "420           No adjuvant                      NaN                  0   \n",
       "415           No adjuvant                      NaN                  0   \n",
       "\n",
       "                               CHMPlatDgName_cleaned  \\\n",
       "88                                               NaN   \n",
       "79                                               NaN   \n",
       "26   Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "23           Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "7                                                NaN   \n",
       "..                                               ...   \n",
       "414                                              NaN   \n",
       "418  Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "416                                              NaN   \n",
       "420                                              NaN   \n",
       "415                                              NaN   \n",
       "\n",
       "                                CHMOthDgName_cleaned  AdjRadStartTime_manual  \\\n",
       "88                                               NaN                     NaN   \n",
       "79                                               NaN                     NaN   \n",
       "26   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "23   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "7                                                NaN                     NaN   \n",
       "..                                               ...                     ...   \n",
       "414                                              NaN                     NaN   \n",
       "418  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "416                                              NaN                     NaN   \n",
       "420                                              NaN                     NaN   \n",
       "415                                              NaN                     NaN   \n",
       "\n",
       "     AdjRadEndTime_manual  Recurrence_time_use newPrim_time_use  \\\n",
       "88                    NaN                  NaN              NaN   \n",
       "79                    NaN                  NaN              NaN   \n",
       "26                    NaN                217.0              NaN   \n",
       "23                    NaN                325.0              NaN   \n",
       "7                     NaN                  NaN              NaN   \n",
       "..                    ...                  ...              ...   \n",
       "414                   NaN                  NaN              NaN   \n",
       "418                   NaN                  NaN              NaN   \n",
       "416                   NaN                883.0              NaN   \n",
       "420                   NaN                748.0              NaN   \n",
       "415                   NaN                  NaN              NaN   \n",
       "\n",
       "    first_dfs_any_event_rec.or.new.primary first_event_during_followup  \\\n",
       "88                                     NaN                       Death   \n",
       "79                                     NaN                         NaN   \n",
       "26                              recurrence                  recurrence   \n",
       "23                              recurrence                  recurrence   \n",
       "7                                      NaN                       Death   \n",
       "..                                     ...                         ...   \n",
       "414                                    NaN                         NaN   \n",
       "418                                    NaN                         NaN   \n",
       "416                             recurrence                  recurrence   \n",
       "420                             recurrence                  recurrence   \n",
       "415                                    NaN                         NaN   \n",
       "\n",
       "     cens_os  os_time  cens_dfs  dfs_time  cens_dfs_any_event  \\\n",
       "88         1     1120         1      1120                   1   \n",
       "79         0     1901         0      1901                   0   \n",
       "26         0     2304         1       217                   1   \n",
       "23         1     1323         1       325                   1   \n",
       "7          1      649         1       649                   1   \n",
       "..       ...      ...       ...       ...                 ...   \n",
       "414        0     1084         0      1084                   0   \n",
       "418        0     1434         0      1434                   0   \n",
       "416        1      893         1       883                   1   \n",
       "420        1      758         1       748                   1   \n",
       "415        0     1001         0      1001                   0   \n",
       "\n",
       "     dfs_time_any_event  cens_lung_event  lung_event_time    Relapse_cat  \\\n",
       "88                 1120                0             1120            NaN   \n",
       "79                 1901                0             1901            NaN   \n",
       "26                  217                1              217  Intra & Extra   \n",
       "23                  325                1              325  Intra & Extra   \n",
       "7                   649                0              649            NaN   \n",
       "..                  ...              ...              ...            ...   \n",
       "414                1084                0             1084            NaN   \n",
       "418                1434                0             1434            NaN   \n",
       "416                 883                1              883  Intra & Extra   \n",
       "420                 748                1              748  Intrathoracic   \n",
       "415                1001                0             1001            NaN   \n",
       "\n",
       "    Relapse_cat_new  tx100  \n",
       "88           No rec   True  \n",
       "79           No rec   True  \n",
       "26    Intra & Extra   True  \n",
       "23    Intra & Extra   True  \n",
       "7            No rec   True  \n",
       "..              ...    ...  \n",
       "414          No rec  False  \n",
       "418          No rec  False  \n",
       "416   Intra & Extra  False  \n",
       "420   Intrathoracic  False  \n",
       "415          No rec  False  \n",
       "\n",
       "[421 rows x 50 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df = clinical_df.sort_values(by=['cruk_id'])\n",
    "clinical_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>patient_id</th>\n",
       "      <th>clinical_sex</th>\n",
       "      <th>age</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>Histology_per_tumour_id_muttable</th>\n",
       "      <th>histology_3</th>\n",
       "      <th>LUAD_pred_subtype_with.IMA_per_tumour</th>\n",
       "      <th>site_per_lesion</th>\n",
       "      <th>pT_stage_per_lesion</th>\n",
       "      <th>pN_stage_per_lesion</th>\n",
       "      <th>size_pathology_per_lesion</th>\n",
       "      <th>vascular_invasion_per_lesion</th>\n",
       "      <th>pleural_invasion_per_lesion</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>pack_years</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>pathologyTNM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>Male</td>\n",
       "      <td>84</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>30</td>\n",
       "      <td>30.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>Male</td>\n",
       "      <td>81</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Left Lower Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>35</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>5.0</td>\n",
       "      <td>35</td>\n",
       "      <td>8.75</td>\n",
       "      <td>False</td>\n",
       "      <td>IB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>Female</td>\n",
       "      <td>78</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>52</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>Female</td>\n",
       "      <td>69</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Left Upper Lobe</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>10.0</td>\n",
       "      <td>27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>True</td>\n",
       "      <td>IIB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>Female</td>\n",
       "      <td>74</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Lower Lobe</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>43</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>56</td>\n",
       "      <td>56.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>Female</td>\n",
       "      <td>78</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Right Lower Lobe</td>\n",
       "      <td>1b</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>10.0</td>\n",
       "      <td>30</td>\n",
       "      <td>15.00</td>\n",
       "      <td>True</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>Male</td>\n",
       "      <td>66</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>1b</td>\n",
       "      <td>2</td>\n",
       "      <td>25</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>54</td>\n",
       "      <td>54.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>Male</td>\n",
       "      <td>84</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>CRUK0881_Tumour2</td>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>Male</td>\n",
       "      <td>84</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Right Upper Lobe</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>False</td>\n",
       "      <td>IIA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>Female</td>\n",
       "      <td>59</td>\n",
       "      <td>White- British</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Left Upper Lobe</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>15.0</td>\n",
       "      <td>43</td>\n",
       "      <td>32.25</td>\n",
       "      <td>False</td>\n",
       "      <td>IA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>432 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            tumour_id patient_id clinical_sex  age       ethnicity  \\\n",
       "89           CRUK0001   CRUK0001         Male   84  White- British   \n",
       "80           CRUK0002   CRUK0002         Male   81  White- British   \n",
       "26           CRUK0003   CRUK0003       Female   78  White- British   \n",
       "23           CRUK0004   CRUK0004       Female   69  White- British   \n",
       "7            CRUK0005   CRUK0005       Female   74  White- British   \n",
       "..                ...        ...          ...  ...             ...   \n",
       "428          CRUK0864   CRUK0864       Female   78  White- British   \n",
       "426          CRUK0872   CRUK0872         Male   66  White- British   \n",
       "430  CRUK0881_Tumour1   CRUK0881         Male   84  White- British   \n",
       "431  CRUK0881_Tumour2   CRUK0881         Male   84  White- British   \n",
       "425          CRUK0893   CRUK0893       Female   59  White- British   \n",
       "\n",
       "    Histology_per_tumour_id_muttable histology_3  \\\n",
       "89           Invasive adenocarcinoma        LUAD   \n",
       "80           Invasive adenocarcinoma        LUAD   \n",
       "26           Invasive adenocarcinoma        LUAD   \n",
       "23           Invasive adenocarcinoma        LUAD   \n",
       "7            Invasive adenocarcinoma        LUAD   \n",
       "..                               ...         ...   \n",
       "428          Invasive adenocarcinoma        LUAD   \n",
       "426          Squamous cell carcinoma        LUSC   \n",
       "430          Invasive adenocarcinoma        LUAD   \n",
       "431          Invasive adenocarcinoma        LUAD   \n",
       "425          Invasive adenocarcinoma        LUAD   \n",
       "\n",
       "    LUAD_pred_subtype_with.IMA_per_tumour   site_per_lesion  \\\n",
       "89                                 acinar  Right Upper Lobe   \n",
       "80                                 acinar   Left Lower Lobe   \n",
       "26                                 acinar  Right Upper Lobe   \n",
       "23                                 acinar   Left Upper Lobe   \n",
       "7                                  acinar  Right Lower Lobe   \n",
       "..                                    ...               ...   \n",
       "428                                 solid  Right Lower Lobe   \n",
       "426                                   NaN  Right Upper Lobe   \n",
       "430                                acinar  Right Upper Lobe   \n",
       "431                                acinar  Right Upper Lobe   \n",
       "425                                acinar   Left Upper Lobe   \n",
       "\n",
       "    pT_stage_per_lesion pN_stage_per_lesion size_pathology_per_lesion  \\\n",
       "89                   2a                   1                        26   \n",
       "80                   2a                   0                        35   \n",
       "26                   2b                   2                        52   \n",
       "23                    3                   0                        47   \n",
       "7                    2a                   2                        43   \n",
       "..                  ...                 ...                       ...   \n",
       "428                  1b                   1                        27   \n",
       "426                  1b                   2                        25   \n",
       "430                 NaN                   0                       NaN   \n",
       "431                 NaN                   0                       NaN   \n",
       "425                  1a                   0                        17   \n",
       "\n",
       "    vascular_invasion_per_lesion pleural_invasion_per_lesion  \\\n",
       "89                            No                         Yes   \n",
       "80                           Yes                          No   \n",
       "26                           Yes                          No   \n",
       "23                            No                         Yes   \n",
       "7                            Yes                         Yes   \n",
       "..                           ...                         ...   \n",
       "428                           No                          No   \n",
       "426                           No                          No   \n",
       "430                          Yes                         NaN   \n",
       "431                           No                         NaN   \n",
       "425                          Yes                          No   \n",
       "\n",
       "    smoking_status_merged  cigs_perday  years_smoking  pack_years  \\\n",
       "89              Ex-Smoker         20.0             30       30.00   \n",
       "80              Ex-Smoker          5.0             35        8.75   \n",
       "26           Never Smoked          0.0              0        0.00   \n",
       "23              Ex-Smoker         10.0             27       13.50   \n",
       "7               Ex-Smoker         20.0             56       56.00   \n",
       "..                    ...          ...            ...         ...   \n",
       "428             Ex-Smoker         10.0             30       15.00   \n",
       "426             Ex-Smoker         20.0             54       54.00   \n",
       "430                Smoker         20.0             70       70.00   \n",
       "431                Smoker         20.0             70       70.00   \n",
       "425                Smoker         15.0             43       32.25   \n",
       "\n",
       "     is.family.lung pathologyTNM  \n",
       "89            False          IIA  \n",
       "80            False           IB  \n",
       "26            False         IIIA  \n",
       "23             True          IIB  \n",
       "7             False         IIIA  \n",
       "..              ...          ...  \n",
       "428            True          IIA  \n",
       "426           False         IIIA  \n",
       "430           False          IIA  \n",
       "431           False          IIA  \n",
       "425           False           IA  \n",
       "\n",
       "[432 rows x 20 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#sort tumor_df by patient_id\n",
    "tumor_df.sort_values(by=['patient_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0 0 1 1 0 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 1 0 1 0 0 1 1 1 1\n",
      " 1 0 1 0 0 1 1 0 0 1 0 1 0 0 0 0 0 0 1 0 0 0 0 1 1 0 1 0 0 0 0 0 0 0 1 1 0\n",
      " 0 1 0 0 1 0 0 0 0 0 1 1 0 1 0 1 0 1 0 1 1 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 1\n",
      " 0 0 0 0 0 0 0 1 0 1 0 0 0 0 1 0 0 1 0 0 0 1 1 1 0 1 1 1 1 1 1 0 0 0 0 0 1\n",
      " 1 0 0 0 0 0 0 1 0 1 0 1 0 1 0 1 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 1 0 1 1 0\n",
      " 0 0 0 1 1 0 1 0 1 0 1 0 1 1 0 0 0 0 0 0 1 1 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0\n",
      " 1 1 0 0 1 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 1\n",
      " 0 1 0 1 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 1 1 0 0 0 1 1 0 0 0 1 0 0 1 1 1 0 1\n",
      " 0 0 0 1 0 0 0 1 1 1 0 0 1 0 1 0 0 0 0 0 0 0 1 1 1 0 0 1 0 0 0 0 0 0 0 1 0\n",
      " 1 0 0 1 1 0 0 0 0 1 0 1 0 1 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 1\n",
      " 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 0 1 1 0 0 1 0 0 0 1\n",
      " 0 1 0 0 0 0 0 1 1 0 0 1 1 0]\n"
     ]
    }
   ],
   "source": [
    "# Making the training data\n",
    "\n",
    "y_train = []\n",
    "for index, row in clinical_df.iterrows():\n",
    "    if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "        y_train.append(1)\n",
    "    else:\n",
    "        y_train.append(0)\n",
    "\n",
    "\n",
    "y = np.array(y_train)\n",
    "print(y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/100_gene_matrix.csv')\n",
    "driver_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/100_driver_matrix.csv')\n",
    "fifty_gene_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/50_gene_matrix.csv')\n",
    "fifty_driver_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/50_driver_matrix.csv')\n",
    "twenty_gene_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/20_gene_matrix.csv')\n",
    "twenty_driver_matrix = pd.read_csv('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/Matrices/20_driver_matrix.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_matrix = gene_matrix.set_index(gene_matrix.columns[0])\n",
    "driver_matrix = driver_matrix.set_index(driver_matrix.columns[0])\n",
    "fifty_gene_matrix = fifty_gene_matrix.set_index(fifty_gene_matrix.columns[0])\n",
    "fifty_driver_matrix = fifty_driver_matrix.set_index(fifty_driver_matrix.columns[0])\n",
    "twenty_gene_matrix = twenty_gene_matrix.set_index(twenty_gene_matrix.columns[0])\n",
    "twenty_driver_matrix = twenty_driver_matrix.set_index(twenty_driver_matrix.columns[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "LUAD_patient_id = []\n",
    "LUSC_patient_id = []\n",
    "\n",
    "for index, row in tumor_df.iterrows():\n",
    "    if row['histology_3'] == 'LUAD':\n",
    "        LUAD_patient_id.append(row['patient_id'])\n",
    "    elif row['histology_3'] == 'LUSC':\n",
    "        LUSC_patient_id.append(row['patient_id'])\n",
    "\n",
    "# select clinical_df rows with patient_id in LUAD_patient_id\n",
    "clinical_df_LUAD = clinical_df[clinical_df['cruk_id'].isin(LUAD_patient_id)]\n",
    "# select clinical_df rows with patient_id in LUSC_patient_id\n",
    "clinical_df_LUSC = clinical_df[clinical_df['cruk_id'].isin(LUSC_patient_id)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cruk_id</th>\n",
       "      <th>tumour_id_muttable_cruk</th>\n",
       "      <th>tumour_id_per_patient</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>packyears</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>ECOG_PS</th>\n",
       "      <th>pathologyTNM</th>\n",
       "      <th>pT_stage_per_patient</th>\n",
       "      <th>pN_stage_per_patient</th>\n",
       "      <th>LVI_per_patient</th>\n",
       "      <th>PL_per_patient</th>\n",
       "      <th>margin_status_per_patient</th>\n",
       "      <th>size_pathology_per_patient</th>\n",
       "      <th>Surgery_type</th>\n",
       "      <th>histology_lesion1</th>\n",
       "      <th>histology_lesion1_merged</th>\n",
       "      <th>lesion1_sampled</th>\n",
       "      <th>histology_lesion2</th>\n",
       "      <th>lesion2_sampled</th>\n",
       "      <th>histology_multi_full</th>\n",
       "      <th>histology_multi_full_genomically.confirmed</th>\n",
       "      <th>LUAD_pred_subtype</th>\n",
       "      <th>adjuvant_treatment_YN</th>\n",
       "      <th>adjuvant_treatment_given</th>\n",
       "      <th>num_cycle_na.added</th>\n",
       "      <th>CHMPlatDgName_cleaned</th>\n",
       "      <th>CHMOthDgName_cleaned</th>\n",
       "      <th>AdjRadStartTime_manual</th>\n",
       "      <th>AdjRadEndTime_manual</th>\n",
       "      <th>Recurrence_time_use</th>\n",
       "      <th>newPrim_time_use</th>\n",
       "      <th>first_dfs_any_event_rec.or.new.primary</th>\n",
       "      <th>first_event_during_followup</th>\n",
       "      <th>cens_os</th>\n",
       "      <th>os_time</th>\n",
       "      <th>cens_dfs</th>\n",
       "      <th>dfs_time</th>\n",
       "      <th>cens_dfs_any_event</th>\n",
       "      <th>dfs_time_any_event</th>\n",
       "      <th>cens_lung_event</th>\n",
       "      <th>lung_event_time</th>\n",
       "      <th>Relapse_cat</th>\n",
       "      <th>Relapse_cat_new</th>\n",
       "      <th>tx100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>30</td>\n",
       "      <td>30.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1120</td>\n",
       "      <td>0</td>\n",
       "      <td>1120</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>81</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>5.0</td>\n",
       "      <td>35</td>\n",
       "      <td>8.75</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>0</td>\n",
       "      <td>1901</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Never Smoked</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>217.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>2304</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>1</td>\n",
       "      <td>217</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>69</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>325.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>1323</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>1</td>\n",
       "      <td>325</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>74</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>56</td>\n",
       "      <td>56.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Death</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>1</td>\n",
       "      <td>649</td>\n",
       "      <td>0</td>\n",
       "      <td>649</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>CRUK0810</td>\n",
       "      <td>CRUK0810</td>\n",
       "      <td>CRUK0810</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>44</td>\n",
       "      <td>22.00</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>2</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>2</td>\n",
       "      <td>Cisplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>197.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>493</td>\n",
       "      <td>1</td>\n",
       "      <td>197</td>\n",
       "      <td>1</td>\n",
       "      <td>197</td>\n",
       "      <td>1</td>\n",
       "      <td>197</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417</th>\n",
       "      <td>CRUK0817</td>\n",
       "      <td>CRUK0817</td>\n",
       "      <td>CRUK0817</td>\n",
       "      <td>64</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>12.0</td>\n",
       "      <td>27</td>\n",
       "      <td>16.20</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>130.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Radiotherapy</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>64.0</td>\n",
       "      <td>91.0</td>\n",
       "      <td>203.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>453</td>\n",
       "      <td>1</td>\n",
       "      <td>203</td>\n",
       "      <td>1</td>\n",
       "      <td>203</td>\n",
       "      <td>1</td>\n",
       "      <td>203</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>78</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>30</td>\n",
       "      <td>15.00</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>True</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>solid</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Carboplatin;Carboplatin;Carboplatin;Carboplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>0</td>\n",
       "      <td>1434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>CRUK0881_Tumour1;CRUK0881_Tumour2</td>\n",
       "      <td>84</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>70</td>\n",
       "      <td>70.00</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2b</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>Lesion1: collision tumour</td>\n",
       "      <td>False</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>LUADx2</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>748.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>758</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>1</td>\n",
       "      <td>748</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>59</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>15.0</td>\n",
       "      <td>43</td>\n",
       "      <td>32.25</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>Invasive adenocarcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>acinar</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>0</td>\n",
       "      <td>1001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>242 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      cruk_id tumour_id_muttable_cruk              tumour_id_per_patient  age  \\\n",
       "88   CRUK0001                CRUK0001                           CRUK0001   84   \n",
       "79   CRUK0002                CRUK0002                           CRUK0002   81   \n",
       "26   CRUK0003                CRUK0003                           CRUK0003   78   \n",
       "23   CRUK0004                CRUK0004                           CRUK0004   69   \n",
       "7    CRUK0005                CRUK0005                           CRUK0005   74   \n",
       "..        ...                     ...                                ...  ...   \n",
       "419  CRUK0810                CRUK0810                           CRUK0810   59   \n",
       "417  CRUK0817                CRUK0817                           CRUK0817   64   \n",
       "418  CRUK0864                CRUK0864                           CRUK0864   78   \n",
       "420  CRUK0881        CRUK0881_Tumour1  CRUK0881_Tumour1;CRUK0881_Tumour2   84   \n",
       "415  CRUK0893                CRUK0893                           CRUK0893   59   \n",
       "\n",
       "        sex       ethnicity  cigs_perday  years_smoking  packyears  \\\n",
       "88     Male  White- British         20.0             30      30.00   \n",
       "79     Male  White- British          5.0             35       8.75   \n",
       "26   Female  White- British          0.0              0       0.00   \n",
       "23   Female  White- British         10.0             27      13.50   \n",
       "7    Female  White- British         20.0             56      56.00   \n",
       "..      ...             ...          ...            ...        ...   \n",
       "419  Female  White- British         10.0             44      22.00   \n",
       "417    Male  White- British         12.0             27      16.20   \n",
       "418  Female  White- British         10.0             30      15.00   \n",
       "420    Male  White- British         20.0             70      70.00   \n",
       "415  Female  White- British         15.0             43      32.25   \n",
       "\n",
       "    smoking_status_merged  is.family.lung  ECOG_PS pathologyTNM  \\\n",
       "88              Ex-Smoker           False      1.0          IIA   \n",
       "79              Ex-Smoker           False      1.0           IB   \n",
       "26           Never Smoked           False      0.0         IIIA   \n",
       "23              Ex-Smoker            True      0.0          IIB   \n",
       "7               Ex-Smoker           False      1.0         IIIA   \n",
       "..                    ...             ...      ...          ...   \n",
       "419                Smoker           False      1.0         IIIA   \n",
       "417             Ex-Smoker           False      1.0          IIB   \n",
       "418             Ex-Smoker            True      0.0          IIA   \n",
       "420                Smoker           False      1.0          IIA   \n",
       "415                Smoker           False      1.0           IA   \n",
       "\n",
       "    pT_stage_per_patient  pN_stage_per_patient LVI_per_patient PL_per_patient  \\\n",
       "88                    2a                     1              No            Yes   \n",
       "79                    2a                     0             Yes             No   \n",
       "26                    2b                     2             Yes             No   \n",
       "23                     3                     0              No            Yes   \n",
       "7                     2a                     2             Yes            Yes   \n",
       "..                   ...                   ...             ...            ...   \n",
       "419                   2a                     2             Yes            Yes   \n",
       "417                    3                     0              No            Yes   \n",
       "418                   1b                     1              No             No   \n",
       "420                   2b                     0             Yes            Yes   \n",
       "415                   1a                     0             Yes             No   \n",
       "\n",
       "    margin_status_per_patient  size_pathology_per_patient Surgery_type  \\\n",
       "88                         R0                        26.0    Lobectomy   \n",
       "79                         R0                        35.0    Lobectomy   \n",
       "26                         R0                        52.0    Lobectomy   \n",
       "23                         R0                        47.0    Lobectomy   \n",
       "7                          R0                        43.0    Lobectomy   \n",
       "..                        ...                         ...          ...   \n",
       "419                        R0                        35.0    Lobectomy   \n",
       "417                        R0                       130.0    Lobectomy   \n",
       "418                        R0                        27.0    Lobectomy   \n",
       "420                        R0                        62.0    Lobectomy   \n",
       "415                        R0                        17.0    Lobectomy   \n",
       "\n",
       "           histology_lesion1 histology_lesion1_merged  lesion1_sampled  \\\n",
       "88   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "79   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "26   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "23   Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "7    Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "..                       ...                      ...              ...   \n",
       "419  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "417  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "418  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "420  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "415  Invasive adenocarcinoma  Invasive adenocarcinoma             True   \n",
       "\n",
       "             histology_lesion2 lesion2_sampled histology_multi_full  \\\n",
       "88                         NaN             NaN                 LUAD   \n",
       "79                         NaN             NaN                 LUAD   \n",
       "26                         NaN             NaN                 LUAD   \n",
       "23                         NaN             NaN                 LUAD   \n",
       "7                          NaN             NaN                 LUAD   \n",
       "..                         ...             ...                  ...   \n",
       "419                        NaN             NaN                 LUAD   \n",
       "417                        NaN             NaN                 LUAD   \n",
       "418                        NaN             NaN                 LUAD   \n",
       "420  Lesion1: collision tumour           False               LUADx2   \n",
       "415                        NaN             NaN                 LUAD   \n",
       "\n",
       "    histology_multi_full_genomically.confirmed LUAD_pred_subtype  \\\n",
       "88                                        LUAD            acinar   \n",
       "79                                        LUAD            acinar   \n",
       "26                                        LUAD            acinar   \n",
       "23                                        LUAD            acinar   \n",
       "7                                         LUAD            acinar   \n",
       "..                                         ...               ...   \n",
       "419                                       LUAD            acinar   \n",
       "417                                       LUAD             solid   \n",
       "418                                       LUAD             solid   \n",
       "420                                     LUADx2            acinar   \n",
       "415                                       LUAD            acinar   \n",
       "\n",
       "    adjuvant_treatment_YN adjuvant_treatment_given num_cycle_na.added  \\\n",
       "88            No adjuvant                      NaN                  0   \n",
       "79            No adjuvant                      NaN                  0   \n",
       "26               Adjuvant           Platinum chemo                  4   \n",
       "23               Adjuvant           Platinum chemo                  4   \n",
       "7             No adjuvant                      NaN                  0   \n",
       "..                    ...                      ...                ...   \n",
       "419              Adjuvant           Platinum chemo                  2   \n",
       "417              Adjuvant             Radiotherapy                  0   \n",
       "418              Adjuvant           Platinum chemo                  4   \n",
       "420           No adjuvant                      NaN                  0   \n",
       "415           No adjuvant                      NaN                  0   \n",
       "\n",
       "                               CHMPlatDgName_cleaned  \\\n",
       "88                                               NaN   \n",
       "79                                               NaN   \n",
       "26   Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "23           Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "7                                                NaN   \n",
       "..                                               ...   \n",
       "419                            Cisplatin;Carboplatin   \n",
       "417                                              NaN   \n",
       "418  Carboplatin;Carboplatin;Carboplatin;Carboplatin   \n",
       "420                                              NaN   \n",
       "415                                              NaN   \n",
       "\n",
       "                                CHMOthDgName_cleaned  AdjRadStartTime_manual  \\\n",
       "88                                               NaN                     NaN   \n",
       "79                                               NaN                     NaN   \n",
       "26   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "23   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "7                                                NaN                     NaN   \n",
       "..                                               ...                     ...   \n",
       "419                          Vinorelbine;Vinorelbine                     NaN   \n",
       "417                                              NaN                    64.0   \n",
       "418  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "420                                              NaN                     NaN   \n",
       "415                                              NaN                     NaN   \n",
       "\n",
       "     AdjRadEndTime_manual  Recurrence_time_use newPrim_time_use  \\\n",
       "88                    NaN                  NaN              NaN   \n",
       "79                    NaN                  NaN              NaN   \n",
       "26                    NaN                217.0              NaN   \n",
       "23                    NaN                325.0              NaN   \n",
       "7                     NaN                  NaN              NaN   \n",
       "..                    ...                  ...              ...   \n",
       "419                   NaN                197.0              NaN   \n",
       "417                  91.0                203.0              NaN   \n",
       "418                   NaN                  NaN              NaN   \n",
       "420                   NaN                748.0              NaN   \n",
       "415                   NaN                  NaN              NaN   \n",
       "\n",
       "    first_dfs_any_event_rec.or.new.primary first_event_during_followup  \\\n",
       "88                                     NaN                       Death   \n",
       "79                                     NaN                         NaN   \n",
       "26                              recurrence                  recurrence   \n",
       "23                              recurrence                  recurrence   \n",
       "7                                      NaN                       Death   \n",
       "..                                     ...                         ...   \n",
       "419                             recurrence                  recurrence   \n",
       "417                             recurrence                  recurrence   \n",
       "418                                    NaN                         NaN   \n",
       "420                             recurrence                  recurrence   \n",
       "415                                    NaN                         NaN   \n",
       "\n",
       "     cens_os  os_time  cens_dfs  dfs_time  cens_dfs_any_event  \\\n",
       "88         1     1120         1      1120                   1   \n",
       "79         0     1901         0      1901                   0   \n",
       "26         0     2304         1       217                   1   \n",
       "23         1     1323         1       325                   1   \n",
       "7          1      649         1       649                   1   \n",
       "..       ...      ...       ...       ...                 ...   \n",
       "419        1      493         1       197                   1   \n",
       "417        1      453         1       203                   1   \n",
       "418        0     1434         0      1434                   0   \n",
       "420        1      758         1       748                   1   \n",
       "415        0     1001         0      1001                   0   \n",
       "\n",
       "     dfs_time_any_event  cens_lung_event  lung_event_time    Relapse_cat  \\\n",
       "88                 1120                0             1120            NaN   \n",
       "79                 1901                0             1901            NaN   \n",
       "26                  217                1              217  Intra & Extra   \n",
       "23                  325                1              325  Intra & Extra   \n",
       "7                   649                0              649            NaN   \n",
       "..                  ...              ...              ...            ...   \n",
       "419                 197                1              197  Intra & Extra   \n",
       "417                 203                1              203  Intrathoracic   \n",
       "418                1434                0             1434            NaN   \n",
       "420                 748                1              748  Intrathoracic   \n",
       "415                1001                0             1001            NaN   \n",
       "\n",
       "    Relapse_cat_new  tx100  \n",
       "88           No rec   True  \n",
       "79           No rec   True  \n",
       "26    Intra & Extra   True  \n",
       "23    Intra & Extra   True  \n",
       "7            No rec   True  \n",
       "..              ...    ...  \n",
       "419   Intra & Extra  False  \n",
       "417   Intrathoracic  False  \n",
       "418          No rec  False  \n",
       "420   Intrathoracic  False  \n",
       "415          No rec  False  \n",
       "\n",
       "[242 rows x 50 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df_LUAD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cruk_id</th>\n",
       "      <th>tumour_id_muttable_cruk</th>\n",
       "      <th>tumour_id_per_patient</th>\n",
       "      <th>age</th>\n",
       "      <th>sex</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>cigs_perday</th>\n",
       "      <th>years_smoking</th>\n",
       "      <th>packyears</th>\n",
       "      <th>smoking_status_merged</th>\n",
       "      <th>is.family.lung</th>\n",
       "      <th>ECOG_PS</th>\n",
       "      <th>pathologyTNM</th>\n",
       "      <th>pT_stage_per_patient</th>\n",
       "      <th>pN_stage_per_patient</th>\n",
       "      <th>LVI_per_patient</th>\n",
       "      <th>PL_per_patient</th>\n",
       "      <th>margin_status_per_patient</th>\n",
       "      <th>size_pathology_per_patient</th>\n",
       "      <th>Surgery_type</th>\n",
       "      <th>histology_lesion1</th>\n",
       "      <th>histology_lesion1_merged</th>\n",
       "      <th>lesion1_sampled</th>\n",
       "      <th>histology_lesion2</th>\n",
       "      <th>lesion2_sampled</th>\n",
       "      <th>histology_multi_full</th>\n",
       "      <th>histology_multi_full_genomically.confirmed</th>\n",
       "      <th>LUAD_pred_subtype</th>\n",
       "      <th>adjuvant_treatment_YN</th>\n",
       "      <th>adjuvant_treatment_given</th>\n",
       "      <th>num_cycle_na.added</th>\n",
       "      <th>CHMPlatDgName_cleaned</th>\n",
       "      <th>CHMOthDgName_cleaned</th>\n",
       "      <th>AdjRadStartTime_manual</th>\n",
       "      <th>AdjRadEndTime_manual</th>\n",
       "      <th>Recurrence_time_use</th>\n",
       "      <th>newPrim_time_use</th>\n",
       "      <th>first_dfs_any_event_rec.or.new.primary</th>\n",
       "      <th>first_event_during_followup</th>\n",
       "      <th>cens_os</th>\n",
       "      <th>os_time</th>\n",
       "      <th>cens_dfs</th>\n",
       "      <th>dfs_time</th>\n",
       "      <th>cens_dfs_any_event</th>\n",
       "      <th>dfs_time_any_event</th>\n",
       "      <th>cens_lung_event</th>\n",
       "      <th>lung_event_time</th>\n",
       "      <th>Relapse_cat</th>\n",
       "      <th>Relapse_cat_new</th>\n",
       "      <th>tx100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>CRUK0062</td>\n",
       "      <td>CRUK0062</td>\n",
       "      <td>CRUK0062</td>\n",
       "      <td>57</td>\n",
       "      <td>Male</td>\n",
       "      <td>Caribbean</td>\n",
       "      <td>10.0</td>\n",
       "      <td>38</td>\n",
       "      <td>19.0</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>2b</td>\n",
       "      <td>1</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>377.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>0</td>\n",
       "      <td>724</td>\n",
       "      <td>1</td>\n",
       "      <td>377</td>\n",
       "      <td>1</td>\n",
       "      <td>377</td>\n",
       "      <td>1</td>\n",
       "      <td>377</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>Intrathoracic</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CRUK0063</td>\n",
       "      <td>CRUK0063</td>\n",
       "      <td>CRUK0063</td>\n",
       "      <td>65</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>40.0</td>\n",
       "      <td>46</td>\n",
       "      <td>92.0</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIA</td>\n",
       "      <td>2a</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Bilobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>1</td>\n",
       "      <td>Cisplatin</td>\n",
       "      <td>Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>456.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>858</td>\n",
       "      <td>1</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>456</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>CRUK0064</td>\n",
       "      <td>CRUK0064</td>\n",
       "      <td>CRUK0064</td>\n",
       "      <td>68</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>53</td>\n",
       "      <td>26.5</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1820</td>\n",
       "      <td>0</td>\n",
       "      <td>1820</td>\n",
       "      <td>0</td>\n",
       "      <td>1820</td>\n",
       "      <td>0</td>\n",
       "      <td>1820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>CRUK0065</td>\n",
       "      <td>CRUK0065</td>\n",
       "      <td>CRUK0065</td>\n",
       "      <td>76</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>15.0</td>\n",
       "      <td>30</td>\n",
       "      <td>22.5</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>Pneumonectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>265.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>445</td>\n",
       "      <td>1</td>\n",
       "      <td>265</td>\n",
       "      <td>1</td>\n",
       "      <td>265</td>\n",
       "      <td>1</td>\n",
       "      <td>265</td>\n",
       "      <td>Extrathoracic</td>\n",
       "      <td>Extrathoracic</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>CRUK0066</td>\n",
       "      <td>CRUK0066</td>\n",
       "      <td>CRUK0066</td>\n",
       "      <td>78</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>60</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1835</td>\n",
       "      <td>0</td>\n",
       "      <td>1835</td>\n",
       "      <td>0</td>\n",
       "      <td>1835</td>\n",
       "      <td>0</td>\n",
       "      <td>1835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>CRUK0772</td>\n",
       "      <td>CRUK0772</td>\n",
       "      <td>CRUK0772</td>\n",
       "      <td>64</td>\n",
       "      <td>Male</td>\n",
       "      <td>White-European</td>\n",
       "      <td>10.0</td>\n",
       "      <td>40</td>\n",
       "      <td>20.0</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>140.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>726</td>\n",
       "      <td>0</td>\n",
       "      <td>726</td>\n",
       "      <td>0</td>\n",
       "      <td>726</td>\n",
       "      <td>0</td>\n",
       "      <td>726</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>CRUK0781</td>\n",
       "      <td>CRUK0781</td>\n",
       "      <td>CRUK0781</td>\n",
       "      <td>76</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>10.0</td>\n",
       "      <td>61</td>\n",
       "      <td>30.5</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIB</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>R0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Adjuvant</td>\n",
       "      <td>Platinum chemo</td>\n",
       "      <td>4</td>\n",
       "      <td>Cisplatin;Cisplatin;Cisplatin;Cisplatin</td>\n",
       "      <td>Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence (death)</td>\n",
       "      <td>1</td>\n",
       "      <td>313</td>\n",
       "      <td>1</td>\n",
       "      <td>313</td>\n",
       "      <td>1</td>\n",
       "      <td>313</td>\n",
       "      <td>1</td>\n",
       "      <td>313</td>\n",
       "      <td>Missing data</td>\n",
       "      <td>Unknown Site</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>CRUK0784</td>\n",
       "      <td>CRUK0784</td>\n",
       "      <td>CRUK0784</td>\n",
       "      <td>74</td>\n",
       "      <td>Female</td>\n",
       "      <td>White- British</td>\n",
       "      <td>14.0</td>\n",
       "      <td>49</td>\n",
       "      <td>34.3</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>IB</td>\n",
       "      <td>2a</td>\n",
       "      <td>0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1638</td>\n",
       "      <td>0</td>\n",
       "      <td>1638</td>\n",
       "      <td>0</td>\n",
       "      <td>1638</td>\n",
       "      <td>0</td>\n",
       "      <td>1638</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>414</th>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>65</td>\n",
       "      <td>Female</td>\n",
       "      <td>Middle eastern</td>\n",
       "      <td>10.0</td>\n",
       "      <td>45</td>\n",
       "      <td>22.5</td>\n",
       "      <td>Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IA</td>\n",
       "      <td>1b</td>\n",
       "      <td>0</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>0</td>\n",
       "      <td>1084</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No rec</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>416</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>66</td>\n",
       "      <td>Male</td>\n",
       "      <td>White- British</td>\n",
       "      <td>20.0</td>\n",
       "      <td>54</td>\n",
       "      <td>54.0</td>\n",
       "      <td>Ex-Smoker</td>\n",
       "      <td>False</td>\n",
       "      <td>1.0</td>\n",
       "      <td>IIIA</td>\n",
       "      <td>1b</td>\n",
       "      <td>2</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>R0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>Lobectomy</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>LUSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No adjuvant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>883.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>recurrence</td>\n",
       "      <td>1</td>\n",
       "      <td>893</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>1</td>\n",
       "      <td>883</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>Intra &amp; Extra</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>138 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      cruk_id tumour_id_muttable_cruk tumour_id_per_patient  age     sex  \\\n",
       "35   CRUK0062                CRUK0062              CRUK0062   57    Male   \n",
       "6    CRUK0063                CRUK0063              CRUK0063   65    Male   \n",
       "129  CRUK0064                CRUK0064              CRUK0064   68    Male   \n",
       "20   CRUK0065                CRUK0065              CRUK0065   76    Male   \n",
       "14   CRUK0066                CRUK0066              CRUK0066   78    Male   \n",
       "..        ...                     ...                   ...  ...     ...   \n",
       "395  CRUK0772                CRUK0772              CRUK0772   64    Male   \n",
       "371  CRUK0781                CRUK0781              CRUK0781   76    Male   \n",
       "369  CRUK0784                CRUK0784              CRUK0784   74  Female   \n",
       "414  CRUK0861                CRUK0861              CRUK0861   65  Female   \n",
       "416  CRUK0872                CRUK0872              CRUK0872   66    Male   \n",
       "\n",
       "          ethnicity  cigs_perday  years_smoking  packyears  \\\n",
       "35        Caribbean         10.0             38       19.0   \n",
       "6    White- British         40.0             46       92.0   \n",
       "129  White- British         10.0             53       26.5   \n",
       "20   White- British         15.0             30       22.5   \n",
       "14   White- British         20.0             60       60.0   \n",
       "..              ...          ...            ...        ...   \n",
       "395  White-European         10.0             40       20.0   \n",
       "371  White- British         10.0             61       30.5   \n",
       "369  White- British         14.0             49       34.3   \n",
       "414  Middle eastern         10.0             45       22.5   \n",
       "416  White- British         20.0             54       54.0   \n",
       "\n",
       "    smoking_status_merged  is.family.lung  ECOG_PS pathologyTNM  \\\n",
       "35                 Smoker           False      1.0          IIB   \n",
       "6                  Smoker           False      0.0          IIA   \n",
       "129                Smoker           False      0.0           IB   \n",
       "20              Ex-Smoker           False      0.0          IIB   \n",
       "14                 Smoker           False      1.0           IB   \n",
       "..                    ...             ...      ...          ...   \n",
       "395                Smoker           False      0.0          IIB   \n",
       "371                Smoker           False      1.0          IIB   \n",
       "369             Ex-Smoker           False      0.0           IB   \n",
       "414                Smoker           False      1.0           IA   \n",
       "416             Ex-Smoker           False      1.0         IIIA   \n",
       "\n",
       "    pT_stage_per_patient  pN_stage_per_patient LVI_per_patient PL_per_patient  \\\n",
       "35                    2b                     1              No             No   \n",
       "6                     2a                     1             Yes             No   \n",
       "129                   2a                     0              No            Yes   \n",
       "20                     3                     0             Yes             No   \n",
       "14                    2a                     0              No            Yes   \n",
       "..                   ...                   ...             ...            ...   \n",
       "395                    3                     0             Yes            Yes   \n",
       "371                    3                     0             Yes            Yes   \n",
       "369                   2a                     0             Yes             No   \n",
       "414                   1b                     0              No             No   \n",
       "416                   1b                     2              No             No   \n",
       "\n",
       "    margin_status_per_patient  size_pathology_per_patient   Surgery_type  \\\n",
       "35                         R0                        55.0      Lobectomy   \n",
       "6                          R0                        50.0    Bilobectomy   \n",
       "129                        R0                        38.0      Lobectomy   \n",
       "20                         R0                        75.0  Pneumonectomy   \n",
       "14                         R0                        42.0      Lobectomy   \n",
       "..                        ...                         ...            ...   \n",
       "395                        R0                       140.0      Lobectomy   \n",
       "371                        R0                        42.0      Lobectomy   \n",
       "369                        R0                        33.0      Lobectomy   \n",
       "414                        R0                        25.0      Lobectomy   \n",
       "416                        R0                        25.0      Lobectomy   \n",
       "\n",
       "           histology_lesion1 histology_lesion1_merged  lesion1_sampled  \\\n",
       "35   Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "6    Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "129  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "20   Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "14   Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "..                       ...                      ...              ...   \n",
       "395  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "371  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "369  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "414  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "416  Squamous cell carcinoma  Squamous cell carcinoma             True   \n",
       "\n",
       "    histology_lesion2 lesion2_sampled histology_multi_full  \\\n",
       "35                NaN             NaN                 LUSC   \n",
       "6                 NaN             NaN                 LUSC   \n",
       "129               NaN             NaN                 LUSC   \n",
       "20                NaN             NaN                 LUSC   \n",
       "14                NaN             NaN                 LUSC   \n",
       "..                ...             ...                  ...   \n",
       "395               NaN             NaN                 LUSC   \n",
       "371               NaN             NaN                 LUSC   \n",
       "369               NaN             NaN                 LUSC   \n",
       "414               NaN             NaN                 LUSC   \n",
       "416               NaN             NaN                 LUSC   \n",
       "\n",
       "    histology_multi_full_genomically.confirmed LUAD_pred_subtype  \\\n",
       "35                                        LUSC               NaN   \n",
       "6                                         LUSC               NaN   \n",
       "129                                       LUSC               NaN   \n",
       "20                                        LUSC               NaN   \n",
       "14                                        LUSC               NaN   \n",
       "..                                         ...               ...   \n",
       "395                                       LUSC               NaN   \n",
       "371                                       LUSC               NaN   \n",
       "369                                       LUSC               NaN   \n",
       "414                                       LUSC               NaN   \n",
       "416                                       LUSC               NaN   \n",
       "\n",
       "    adjuvant_treatment_YN adjuvant_treatment_given num_cycle_na.added  \\\n",
       "35               Adjuvant           Platinum chemo                  4   \n",
       "6                Adjuvant           Platinum chemo                  1   \n",
       "129           No adjuvant                      NaN                  0   \n",
       "20            No adjuvant                      NaN                  0   \n",
       "14            No adjuvant                      NaN                  0   \n",
       "..                    ...                      ...                ...   \n",
       "395              Adjuvant           Platinum chemo                  4   \n",
       "371              Adjuvant           Platinum chemo                  4   \n",
       "369           No adjuvant                      NaN                  0   \n",
       "414           No adjuvant                      NaN                  0   \n",
       "416           No adjuvant                      NaN                  0   \n",
       "\n",
       "                       CHMPlatDgName_cleaned  \\\n",
       "35   Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "6                                  Cisplatin   \n",
       "129                                      NaN   \n",
       "20                                       NaN   \n",
       "14                                       NaN   \n",
       "..                                       ...   \n",
       "395  Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "371  Cisplatin;Cisplatin;Cisplatin;Cisplatin   \n",
       "369                                      NaN   \n",
       "414                                      NaN   \n",
       "416                                      NaN   \n",
       "\n",
       "                                CHMOthDgName_cleaned  AdjRadStartTime_manual  \\\n",
       "35   Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "6                                        Vinorelbine                     NaN   \n",
       "129                                              NaN                     NaN   \n",
       "20                                               NaN                     NaN   \n",
       "14                                               NaN                     NaN   \n",
       "..                                               ...                     ...   \n",
       "395  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "371  Vinorelbine;Vinorelbine;Vinorelbine;Vinorelbine                     NaN   \n",
       "369                                              NaN                     NaN   \n",
       "414                                              NaN                     NaN   \n",
       "416                                              NaN                     NaN   \n",
       "\n",
       "     AdjRadEndTime_manual  Recurrence_time_use newPrim_time_use  \\\n",
       "35                    NaN                377.0              NaN   \n",
       "6                     NaN                456.0              NaN   \n",
       "129                   NaN                  NaN              NaN   \n",
       "20                    NaN                265.0              NaN   \n",
       "14                    NaN                  NaN              NaN   \n",
       "..                    ...                  ...              ...   \n",
       "395                   NaN                  NaN              NaN   \n",
       "371                   NaN                  NaN              NaN   \n",
       "369                   NaN                  NaN              NaN   \n",
       "414                   NaN                  NaN              NaN   \n",
       "416                   NaN                883.0              NaN   \n",
       "\n",
       "    first_dfs_any_event_rec.or.new.primary first_event_during_followup  \\\n",
       "35                              recurrence                  recurrence   \n",
       "6                               recurrence                  recurrence   \n",
       "129                                    NaN                         NaN   \n",
       "20                              recurrence                  recurrence   \n",
       "14                                     NaN                         NaN   \n",
       "..                                     ...                         ...   \n",
       "395                                    NaN                         NaN   \n",
       "371                             recurrence          recurrence (death)   \n",
       "369                                    NaN                         NaN   \n",
       "414                                    NaN                         NaN   \n",
       "416                             recurrence                  recurrence   \n",
       "\n",
       "     cens_os  os_time  cens_dfs  dfs_time  cens_dfs_any_event  \\\n",
       "35         0      724         1       377                   1   \n",
       "6          1      858         1       456                   1   \n",
       "129        0     1820         0      1820                   0   \n",
       "20         1      445         1       265                   1   \n",
       "14         0     1835         0      1835                   0   \n",
       "..       ...      ...       ...       ...                 ...   \n",
       "395        0      726         0       726                   0   \n",
       "371        1      313         1       313                   1   \n",
       "369        0     1638         0      1638                   0   \n",
       "414        0     1084         0      1084                   0   \n",
       "416        1      893         1       883                   1   \n",
       "\n",
       "     dfs_time_any_event  cens_lung_event  lung_event_time    Relapse_cat  \\\n",
       "35                  377                1              377  Intrathoracic   \n",
       "6                   456                1              456  Intra & Extra   \n",
       "129                1820                0             1820            NaN   \n",
       "20                  265                1              265  Extrathoracic   \n",
       "14                 1835                0             1835            NaN   \n",
       "..                  ...              ...              ...            ...   \n",
       "395                 726                0              726            NaN   \n",
       "371                 313                1              313   Missing data   \n",
       "369                1638                0             1638            NaN   \n",
       "414                1084                0             1084            NaN   \n",
       "416                 883                1              883  Intra & Extra   \n",
       "\n",
       "    Relapse_cat_new  tx100  \n",
       "35    Intrathoracic   True  \n",
       "6     Intra & Extra   True  \n",
       "129          No rec   True  \n",
       "20    Extrathoracic   True  \n",
       "14           No rec   True  \n",
       "..              ...    ...  \n",
       "395          No rec  False  \n",
       "371    Unknown Site  False  \n",
       "369          No rec  False  \n",
       "414          No rec  False  \n",
       "416   Intra & Extra  False  \n",
       "\n",
       "[138 rows x 50 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_df_LUSC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "mut_table_LUAD = mut_table[mut_table['patient_id'].isin(LUAD_patient_id)]\n",
    "mut_table_LUSC = mut_table[mut_table['patient_id'].isin(LUSC_patient_id)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>mutation_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>ref</th>\n",
       "      <th>var</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>func</th>\n",
       "      <th>exonic.func</th>\n",
       "      <th>NucleotideChange</th>\n",
       "      <th>AAChange</th>\n",
       "      <th>GL_VAF</th>\n",
       "      <th>GL_nAlt</th>\n",
       "      <th>GL_depth</th>\n",
       "      <th>ITHState</th>\n",
       "      <th>PyCloneClonal_SC</th>\n",
       "      <th>PyCloneCluster_SC</th>\n",
       "      <th>cleanCluster_SC</th>\n",
       "      <th>combTiming_SC</th>\n",
       "      <th>DriverMut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:34291428:G</td>\n",
       "      <td>chr19</td>\n",
       "      <td>34291428</td>\n",
       "      <td>34291428</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>KCTD15</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>187</td>\n",
       "      <td>1</td>\n",
       "      <td>S</td>\n",
       "      <td>3.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:9084770:C</td>\n",
       "      <td>chr19</td>\n",
       "      <td>9084770</td>\n",
       "      <td>9084770</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>MUC16</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G7045T</td>\n",
       "      <td>p.V2349L</td>\n",
       "      <td>0.39</td>\n",
       "      <td>2</td>\n",
       "      <td>510</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:1:2160418:G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>2160418</td>\n",
       "      <td>2160418</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SKI</td>\n",
       "      <td>exonic</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>c.G213T</td>\n",
       "      <td>p.P71P</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:40095981:G</td>\n",
       "      <td>chr19</td>\n",
       "      <td>40095981</td>\n",
       "      <td>40095981</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>LGALS13</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G256T</td>\n",
       "      <td>p.G86C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>659</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:19:41313703:C</td>\n",
       "      <td>chr19</td>\n",
       "      <td>41313703</td>\n",
       "      <td>41313703</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>RAB4B-EGLN2</td>\n",
       "      <td>ncRNA_intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>894</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>5.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378558</th>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787:1:228362885:G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>228362885</td>\n",
       "      <td>228362885</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>IBA57</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G742T</td>\n",
       "      <td>p.A248S</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>910</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>7.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378559</th>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787:9:140375437:G</td>\n",
       "      <td>chr9</td>\n",
       "      <td>140375437</td>\n",
       "      <td>140375437</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>PNPLA7</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.C2311T</td>\n",
       "      <td>p.R771C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>773</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>3.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378560</th>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787:5:175110459:T</td>\n",
       "      <td>chr5</td>\n",
       "      <td>175110459</td>\n",
       "      <td>175110459</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>HRH2</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.T223G</td>\n",
       "      <td>p.S75A</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1054</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>3.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378561</th>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787:11:6661849:C</td>\n",
       "      <td>chr11</td>\n",
       "      <td>6661849</td>\n",
       "      <td>6661849</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DCHS1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>c.G996A</td>\n",
       "      <td>p.V332V</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1545</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>3.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378562</th>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787</td>\n",
       "      <td>CRUK0787:7:117874743:T</td>\n",
       "      <td>chr7</td>\n",
       "      <td>117874743</td>\n",
       "      <td>117874743</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ANKRD7</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>538</td>\n",
       "      <td>2</td>\n",
       "      <td>S</td>\n",
       "      <td>4.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>226082 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id tumour_id             mutation_id    chr      start  \\\n",
       "0        CRUK0005  CRUK0005  CRUK0005:19:34291428:G  chr19   34291428   \n",
       "1        CRUK0005  CRUK0005   CRUK0005:19:9084770:C  chr19    9084770   \n",
       "2        CRUK0005  CRUK0005    CRUK0005:1:2160418:G   chr1    2160418   \n",
       "3        CRUK0005  CRUK0005  CRUK0005:19:40095981:G  chr19   40095981   \n",
       "4        CRUK0005  CRUK0005  CRUK0005:19:41313703:C  chr19   41313703   \n",
       "...           ...       ...                     ...    ...        ...   \n",
       "378558   CRUK0787  CRUK0787  CRUK0787:1:228362885:G   chr1  228362885   \n",
       "378559   CRUK0787  CRUK0787  CRUK0787:9:140375437:G   chr9  140375437   \n",
       "378560   CRUK0787  CRUK0787  CRUK0787:5:175110459:T   chr5  175110459   \n",
       "378561   CRUK0787  CRUK0787   CRUK0787:11:6661849:C  chr11    6661849   \n",
       "378562   CRUK0787  CRUK0787  CRUK0787:7:117874743:T   chr7  117874743   \n",
       "\n",
       "             stop ref var  Hugo_Symbol            func        exonic.func  \\\n",
       "0        34291428   G   T       KCTD15        intronic                NaN   \n",
       "1         9084770   C   A        MUC16          exonic  nonsynonymous SNV   \n",
       "2         2160418   G   T          SKI          exonic     synonymous SNV   \n",
       "3        40095981   G   T      LGALS13          exonic  nonsynonymous SNV   \n",
       "4        41313703   C   G  RAB4B-EGLN2  ncRNA_intronic                NaN   \n",
       "...           ...  ..  ..          ...             ...                ...   \n",
       "378558  228362885   G   T        IBA57          exonic  nonsynonymous SNV   \n",
       "378559  140375437   G   A       PNPLA7          exonic  nonsynonymous SNV   \n",
       "378560  175110459   T   G         HRH2          exonic  nonsynonymous SNV   \n",
       "378561    6661849   C   T        DCHS1          exonic     synonymous SNV   \n",
       "378562  117874743   T   C       ANKRD7        intronic                NaN   \n",
       "\n",
       "       NucleotideChange  AAChange GL_VAF GL_nAlt GL_depth ITHState  \\\n",
       "0                   NaN       NaN      0       0      187        1   \n",
       "1              c.G7045T  p.V2349L   0.39       2      510        1   \n",
       "2               c.G213T    p.P71P      0       0      114        1   \n",
       "3               c.G256T    p.G86C      0       0      659        1   \n",
       "4                   NaN       NaN      0       0      894        3   \n",
       "...                 ...       ...    ...     ...      ...      ...   \n",
       "378558          c.G742T   p.A248S      0       0      910        3   \n",
       "378559         c.C2311T   p.R771C      0       0      773        3   \n",
       "378560          c.T223G    p.S75A      0       0     1054        3   \n",
       "378561          c.G996A   p.V332V      0       0     1545        3   \n",
       "378562              NaN       NaN      0       0      538        2   \n",
       "\n",
       "       PyCloneClonal_SC  PyCloneCluster_SC cleanCluster_SC combTiming_SC  \\\n",
       "0                     S                3.0            True     subclonal   \n",
       "1                     C                2.0            True         early   \n",
       "2                     C                2.0            True         early   \n",
       "3                     C                2.0            True         early   \n",
       "4                     S                5.0            True     subclonal   \n",
       "...                 ...                ...             ...           ...   \n",
       "378558                S                7.0            True     subclonal   \n",
       "378559                S                3.0            True     subclonal   \n",
       "378560                S                3.0            True     subclonal   \n",
       "378561                S                3.0            True     subclonal   \n",
       "378562                S                4.0            True     subclonal   \n",
       "\n",
       "        DriverMut  \n",
       "0           False  \n",
       "1           False  \n",
       "2           False  \n",
       "3           False  \n",
       "4           False  \n",
       "...           ...  \n",
       "378558      False  \n",
       "378559      False  \n",
       "378560      False  \n",
       "378561      False  \n",
       "378562      False  \n",
       "\n",
       "[226082 rows x 22 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_table_LUAD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>mutation_id</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>ref</th>\n",
       "      <th>var</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>func</th>\n",
       "      <th>exonic.func</th>\n",
       "      <th>NucleotideChange</th>\n",
       "      <th>AAChange</th>\n",
       "      <th>GL_VAF</th>\n",
       "      <th>GL_nAlt</th>\n",
       "      <th>GL_depth</th>\n",
       "      <th>ITHState</th>\n",
       "      <th>PyCloneClonal_SC</th>\n",
       "      <th>PyCloneCluster_SC</th>\n",
       "      <th>cleanCluster_SC</th>\n",
       "      <th>combTiming_SC</th>\n",
       "      <th>DriverMut</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5696</th>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073:8:69351711:T</td>\n",
       "      <td>chr8</td>\n",
       "      <td>69351711</td>\n",
       "      <td>69351711</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>C8orf34</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>342</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5697</th>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073:8:113246500:C</td>\n",
       "      <td>chr8</td>\n",
       "      <td>113246500</td>\n",
       "      <td>113246500</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>CSMD3</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5698</th>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073:8:77765831:G</td>\n",
       "      <td>chr8</td>\n",
       "      <td>77765831</td>\n",
       "      <td>77765831</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G6674T</td>\n",
       "      <td>p.S2225I</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>484</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5699</th>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073:8:145726411:C</td>\n",
       "      <td>chr8</td>\n",
       "      <td>145726411</td>\n",
       "      <td>145726411</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PPP1R16A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5700</th>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073</td>\n",
       "      <td>CRUK0073:8:39537594:A</td>\n",
       "      <td>chr8</td>\n",
       "      <td>39537594</td>\n",
       "      <td>39537594</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ADAM18</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.A1670G</td>\n",
       "      <td>p.H557R</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>190</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>early</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379003</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:10:120513946:C</td>\n",
       "      <td>chr10</td>\n",
       "      <td>120513946</td>\n",
       "      <td>120513946</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CACUL1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.G329A</td>\n",
       "      <td>p.S110N</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>659</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>7.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379004</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:14:99872776:T</td>\n",
       "      <td>chr14</td>\n",
       "      <td>99872776</td>\n",
       "      <td>99872776</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SETD3</td>\n",
       "      <td>intronic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>539</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>7.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379005</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:8:110131566:C</td>\n",
       "      <td>chr8</td>\n",
       "      <td>110131566</td>\n",
       "      <td>110131566</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>TRHR</td>\n",
       "      <td>exonic</td>\n",
       "      <td>stopgain SNV</td>\n",
       "      <td>c.C1079A</td>\n",
       "      <td>p.S360X</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>712</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>7.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379006</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:1:51032853:G</td>\n",
       "      <td>chr1</td>\n",
       "      <td>51032853</td>\n",
       "      <td>51032853</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>FAF1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>c.C1164G</td>\n",
       "      <td>p.N388K</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>398</td>\n",
       "      <td>3</td>\n",
       "      <td>S</td>\n",
       "      <td>9.0</td>\n",
       "      <td>True</td>\n",
       "      <td>subclonal</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379007</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:20:24524120:G</td>\n",
       "      <td>chr20</td>\n",
       "      <td>24524120</td>\n",
       "      <td>24524120</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SYNDIG1</td>\n",
       "      <td>exonic</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>c.G387A</td>\n",
       "      <td>p.P129P</td>\n",
       "      <td>0.13</td>\n",
       "      <td>1</td>\n",
       "      <td>785</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>119912 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id tumour_id              mutation_id    chr      start  \\\n",
       "5696     CRUK0073  CRUK0073    CRUK0073:8:69351711:T   chr8   69351711   \n",
       "5697     CRUK0073  CRUK0073   CRUK0073:8:113246500:C   chr8  113246500   \n",
       "5698     CRUK0073  CRUK0073    CRUK0073:8:77765831:G   chr8   77765831   \n",
       "5699     CRUK0073  CRUK0073   CRUK0073:8:145726411:C   chr8  145726411   \n",
       "5700     CRUK0073  CRUK0073    CRUK0073:8:39537594:A   chr8   39537594   \n",
       "...           ...       ...                      ...    ...        ...   \n",
       "379003   CRUK0872  CRUK0872  CRUK0872:10:120513946:C  chr10  120513946   \n",
       "379004   CRUK0872  CRUK0872   CRUK0872:14:99872776:T  chr14   99872776   \n",
       "379005   CRUK0872  CRUK0872   CRUK0872:8:110131566:C   chr8  110131566   \n",
       "379006   CRUK0872  CRUK0872    CRUK0872:1:51032853:G   chr1   51032853   \n",
       "379007   CRUK0872  CRUK0872   CRUK0872:20:24524120:G  chr20   24524120   \n",
       "\n",
       "             stop ref var Hugo_Symbol      func        exonic.func  \\\n",
       "5696     69351711   T   C     C8orf34  intronic                NaN   \n",
       "5697    113246500   C   A       CSMD3  intronic                NaN   \n",
       "5698     77765831   G   T       ZFHX4    exonic  nonsynonymous SNV   \n",
       "5699    145726411   C   G    PPP1R16A  intronic                NaN   \n",
       "5700     39537594   A   G      ADAM18    exonic  nonsynonymous SNV   \n",
       "...           ...  ..  ..         ...       ...                ...   \n",
       "379003  120513946   C   T      CACUL1    exonic  nonsynonymous SNV   \n",
       "379004   99872776   T   C       SETD3  intronic                NaN   \n",
       "379005  110131566   C   A        TRHR    exonic       stopgain SNV   \n",
       "379006   51032853   G   C        FAF1    exonic  nonsynonymous SNV   \n",
       "379007   24524120   G   A     SYNDIG1    exonic     synonymous SNV   \n",
       "\n",
       "       NucleotideChange  AAChange GL_VAF GL_nAlt GL_depth ITHState  \\\n",
       "5696                NaN       NaN      0       0      342        1   \n",
       "5697                NaN       NaN      0       0       62        1   \n",
       "5698           c.G6674T  p.S2225I      0       0      484        1   \n",
       "5699                NaN       NaN      0       0      337        1   \n",
       "5700           c.A1670G   p.H557R      0       0      190        1   \n",
       "...                 ...       ...    ...     ...      ...      ...   \n",
       "379003          c.G329A   p.S110N      0       0      659        3   \n",
       "379004              NaN       NaN      0       0      539        3   \n",
       "379005         c.C1079A   p.S360X      0       0      712        3   \n",
       "379006         c.C1164G   p.N388K      0       0      398        3   \n",
       "379007          c.G387A   p.P129P   0.13       1      785        3   \n",
       "\n",
       "       PyCloneClonal_SC  PyCloneCluster_SC cleanCluster_SC combTiming_SC  \\\n",
       "5696                  C                1.0            True         early   \n",
       "5697                  C                1.0            True         early   \n",
       "5698                  C                1.0            True         early   \n",
       "5699                  C                1.0            True         early   \n",
       "5700                  C                1.0            True         early   \n",
       "...                 ...                ...             ...           ...   \n",
       "379003                S                7.0            True     subclonal   \n",
       "379004                S                7.0            True     subclonal   \n",
       "379005                S                7.0            True     subclonal   \n",
       "379006                S                9.0            True     subclonal   \n",
       "379007              NaN                NaN             NaN            NA   \n",
       "\n",
       "        DriverMut  \n",
       "5696        False  \n",
       "5697        False  \n",
       "5698        False  \n",
       "5699        False  \n",
       "5700        False  \n",
       "...           ...  \n",
       "379003      False  \n",
       "379004      False  \n",
       "379005      False  \n",
       "379006      False  \n",
       "379007      False  \n",
       "\n",
       "[119912 rows x 22 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mut_table_LUSC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('TTN', 599), ('RYR2', 569), ('CSMD3', 366), ('LRP1B', 349), ('MUC16', 273), ('USH2A', 270), ('OR4E2', 257), ('CSMD1', 250), ('ZFHX4', 223), ('SPTA1', 220), ('OBSCN', 218), ('MUC19', 215), ('COL22A1', 211), ('RYR3', 193), ('NRXN1', 186), ('DMD', 180), ('CACNA1E', 179), ('PCDH15', 178), ('RYR1', 172), ('SSPO', 171), ('FSIP2', 169), ('ANK2', 167), ('UNC80', 165), ('APOB', 157), ('XIRP2', 156), ('DNAH9', 154), ('ABCA13', 153), ('NAV3', 150), ('FAT3', 149), ('TP53', 148), ('RELN', 146), ('DOCK2', 143), ('SYNE1', 141), ('COL11A1', 141), ('CCDC168', 139), ('LPHN3', 137), ('CSMD2', 134), ('NCS1', 134), ('NEB', 133), ('PAPPA2', 132), ('CNTNAP5', 130), ('PCLO', 129), ('PXDNL', 127), ('CACNA1C', 127), ('PTPRD', 125), ('FBN2', 124), ('RIMS2', 124), ('OTOG', 124), ('CTNNA2', 123), ('LAMA2', 123), ('KRAS', 121), ('SI', 120), ('WDFY4', 117), ('FAM135B', 117), ('HMCN1', 117), ('TRY2P', 116), ('CDH23', 114), ('LAMA1', 114), ('ADAMTS12', 114), ('PKHD1', 114), ('BAI3', 113), ('SORCS1', 113), ('PKHD1L1', 113), ('FAT4', 111), ('SORCS3', 110), ('MUC2', 109), ('TTC40', 109), ('LOC93432', 109), ('CACNA1B', 109), ('KIF1A', 108), ('EYS', 108), ('AFF2', 107), ('COL3A1', 106), ('ASTN1', 106), ('NALCN', 106), ('DNAH5', 106), ('DCDC5', 106), ('PTPRT', 106), ('CNTNAP2', 105), ('MYO7B', 104), ('SPEF2', 104), ('LRRC7', 103), ('ELK2AP', 102), ('ZNF536', 102), ('LRP2', 101), ('PLXNA4', 101), ('UNC79', 101), ('DNAH11', 99), ('MYO18B', 99), ('SNTG1', 98), ('SPEG', 97), ('COL4A6', 97), ('GPR112', 97), ('FPGT-TNNI3K', 97), ('ITGA8', 96), ('VCAN', 95), ('HDAC9', 95), ('ADCY2', 94), ('CPS1', 94), ('PIEZO2', 94)]\n",
      "(242, 100)\n",
      "Hugo_Symbol  ABCA13  ADAMTS12  ADCY2  AFF2  ANK2  APOB  ASTN1  BAI3  CACNA1B  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001          1         1      1     1     1     1      0     1        0   \n",
      "CRUK0002          0         0      0     0     0     0      1     0        0   \n",
      "CRUK0003          0         0      0     0     0     0      0     0        1   \n",
      "CRUK0004          1         0      1     0     0     0      1     0        0   \n",
      "CRUK0005          1         1      1     1     1     1      0     0        0   \n",
      "...             ...       ...    ...   ...   ...   ...    ...   ...      ...   \n",
      "CRUK0810          1         0      0     1     0     0      0     0        1   \n",
      "CRUK0817          0         1      0     0     0     0      0     0        0   \n",
      "CRUK0864          0         0      0     0     0     0      0     0        0   \n",
      "CRUK0881          0         1      0     0     1     1      1     0        0   \n",
      "CRUK0893          0         0      0     0     1     1      0     0        0   \n",
      "\n",
      "Hugo_Symbol  CACNA1C  CACNA1E  CCDC168  CDH23  CNTNAP2  CNTNAP5  COL11A1  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001           1        0        0      0        0        1        0   \n",
      "CRUK0002           0        0        0      0        0        0        0   \n",
      "CRUK0003           0        0        0      0        1        0        0   \n",
      "CRUK0004           0        0        1      0        1        0        0   \n",
      "CRUK0005           0        1        1      0        1        0        1   \n",
      "...              ...      ...      ...    ...      ...      ...      ...   \n",
      "CRUK0810           1        1        1      0        0        0        0   \n",
      "CRUK0817           0        1        0      1        0        0        0   \n",
      "CRUK0864           0        0        1      0        0        0        0   \n",
      "CRUK0881           1        1        1      0        1        1        1   \n",
      "CRUK0893           0        1        0      0        1        1        0   \n",
      "\n",
      "Hugo_Symbol  COL22A1  COL3A1  COL4A6  CPS1  CSMD1  CSMD2  CSMD3  CTNNA2  \\\n",
      "patient_id                                                                \n",
      "CRUK0001           1       1       1     0      1      1      1       1   \n",
      "CRUK0002           0       0       0     0      0      1      1       0   \n",
      "CRUK0003           0       0       0     0      0      0      1       0   \n",
      "CRUK0004           0       0       0     0      0      0      1       0   \n",
      "CRUK0005           1       1       0     0      0      1      1       0   \n",
      "...              ...     ...     ...   ...    ...    ...    ...     ...   \n",
      "CRUK0810           1       0       1     0      1      0      1       1   \n",
      "CRUK0817           1       0       0     0      0      0      0       0   \n",
      "CRUK0864           0       0       0     0      0      0      1       0   \n",
      "CRUK0881           0       1       0     1      1      1      1       1   \n",
      "CRUK0893           0       0       0     0      0      0      0       0   \n",
      "\n",
      "Hugo_Symbol  DCDC5  DMD  DNAH11  DNAH5  DNAH9  DOCK2  ELK2AP  EYS  FAM135B  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001         0    1       0      1      0      1       0    1        1   \n",
      "CRUK0002         0    0       0      1      1      1       0    1        0   \n",
      "CRUK0003         0    0       0      0      0      1       1    0        0   \n",
      "CRUK0004         1    0       0      0      0      0       0    0        1   \n",
      "CRUK0005         1    1       0      1      1      1       1    1        1   \n",
      "...            ...  ...     ...    ...    ...    ...     ...  ...      ...   \n",
      "CRUK0810         1    0       0      0      0      0       0    1        0   \n",
      "CRUK0817         1    0       0      0      0      0       0    0        0   \n",
      "CRUK0864         0    1       0      0      1      1       0    0        0   \n",
      "CRUK0881         1    1       1      1      1      1       0    1        1   \n",
      "CRUK0893         0    0       1      0      0      0       0    1        0   \n",
      "\n",
      "Hugo_Symbol  FAT3  FAT4  FBN2  FPGT-TNNI3K  FSIP2  GPR112  HDAC9  HMCN1  \\\n",
      "patient_id                                                                \n",
      "CRUK0001        0     1     1            1      1       0      1      1   \n",
      "CRUK0002        0     0     0            0      0       0      0      0   \n",
      "CRUK0003        0     0     0            0      0       0      0      0   \n",
      "CRUK0004        0     0     0            1      0       0      0      0   \n",
      "CRUK0005        1     1     1            1      1       1      0      1   \n",
      "...           ...   ...   ...          ...    ...     ...    ...    ...   \n",
      "CRUK0810        1     0     1            1      0       0      0      0   \n",
      "CRUK0817        1     0     0            0      1       0      0      0   \n",
      "CRUK0864        0     0     1            0      1       0      0      0   \n",
      "CRUK0881        1     0     1            1      1       1      0      1   \n",
      "CRUK0893        0     0     0            0      1       0      0      0   \n",
      "\n",
      "Hugo_Symbol  ITGA8  KIF1A  KRAS  LAMA1  LAMA2  LOC93432  LPHN3  LRP1B  LRP2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      1     0      1      1         0      0      0     1   \n",
      "CRUK0002         0      0     0      0      0         0      0      0     0   \n",
      "CRUK0003         0      0     0      0      0         1      0      1     1   \n",
      "CRUK0004         0      0     0      0      0         0      1      1     0   \n",
      "CRUK0005         0      1     1      1      0         1      1      1     1   \n",
      "...            ...    ...   ...    ...    ...       ...    ...    ...   ...   \n",
      "CRUK0810         0      0     1      0      0         1      0      1     0   \n",
      "CRUK0817         0      0     0      0      0         1      0      0     1   \n",
      "CRUK0864         0      0     1      0      1         0      0      1     0   \n",
      "CRUK0881         1      1     1      1      1         1      1      1     1   \n",
      "CRUK0893         1      0     1      0      1         0      0      0     0   \n",
      "\n",
      "Hugo_Symbol  LRRC7  MUC16  MUC19  MUC2  MYO18B  MYO7B  NALCN  NAV3  NCS1  NEB  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001         1      0      1     1       1      0      0     1     1    0   \n",
      "CRUK0002         0      1      0     0       0      0      0     0     0    0   \n",
      "CRUK0003         0      1      0     0       1      0      0     1     0    0   \n",
      "CRUK0004         0      1      0     0       0      0      0     0     0    0   \n",
      "CRUK0005         0      1      0     1       0      1      1     0     0    1   \n",
      "...            ...    ...    ...   ...     ...    ...    ...   ...   ...  ...   \n",
      "CRUK0810         0      1      1     0       0      0      1     1     0    0   \n",
      "CRUK0817         0      0      1     0       0      0      0     0     0    0   \n",
      "CRUK0864         0      1      0     0       0      0      0     0     0    0   \n",
      "CRUK0881         1      1      1     1       0      1      1     1     1    1   \n",
      "CRUK0893         0      0      0     1       0      0      0     0     0    0   \n",
      "\n",
      "Hugo_Symbol  NRXN1  OBSCN  OR4E2  OTOG  PAPPA2  PCDH15  PCLO  PIEZO2  PKHD1  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      1      1     0       0       0     1       0      1   \n",
      "CRUK0002         0      0      0     0       0       0     1       0      0   \n",
      "CRUK0003         0      1      0     0       0       0     1       0      0   \n",
      "CRUK0004         0      1      0     0       0       0     0       0      0   \n",
      "CRUK0005         1      1      1     1       1       1     0       1      1   \n",
      "...            ...    ...    ...   ...     ...     ...   ...     ...    ...   \n",
      "CRUK0810         1      0      0     0       0       0     0       1      0   \n",
      "CRUK0817         1      0      1     0       0       0     1       1      0   \n",
      "CRUK0864         0      0      0     1       0       0     0       0      0   \n",
      "CRUK0881         1      1      1     1       1       1     1       1      1   \n",
      "CRUK0893         1      0      1     1       0       0     0       0      0   \n",
      "\n",
      "Hugo_Symbol  PKHD1L1  PLXNA4  PTPRD  PTPRT  PXDNL  RELN  RIMS2  RYR1  RYR2  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001           0       1      0      0      0     1      1     1     1   \n",
      "CRUK0002           0       1      0      0      0     0      0     0     0   \n",
      "CRUK0003           0       0      1      1      0     1      0     0     1   \n",
      "CRUK0004           0       0      0      0      0     0      0     0     1   \n",
      "CRUK0005           1       1      1      1      1     1      1     0     1   \n",
      "...              ...     ...    ...    ...    ...   ...    ...   ...   ...   \n",
      "CRUK0810           0       0      1      0      0     1      0     0     1   \n",
      "CRUK0817           0       0      0      0      0     0      0     0     1   \n",
      "CRUK0864           0       0      0      0      0     0      0     0     0   \n",
      "CRUK0881           1       1      1      0      1     1      1     1     1   \n",
      "CRUK0893           0       0      0      0      0     0      1     1     1   \n",
      "\n",
      "Hugo_Symbol  RYR3  SI  SNTG1  SORCS1  SORCS3  SPEF2  SPEG  SPTA1  SSPO  SYNE1  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001        1   1      0       1       0      0     0      1     1      1   \n",
      "CRUK0002        1   0      0       0       0      0     0      0     0      0   \n",
      "CRUK0003        0   0      0       0       0      0     0      0     0      0   \n",
      "CRUK0004        0   0      0       0       0      0     0      1     0      0   \n",
      "CRUK0005        0   1      1       1       1      1     0      0     1      0   \n",
      "...           ...  ..    ...     ...     ...    ...   ...    ...   ...    ...   \n",
      "CRUK0810        1   0      0       0       0      0     1      0     0      1   \n",
      "CRUK0817        0   0      0       0       0      0     1      1     0      0   \n",
      "CRUK0864        1   0      0       0       0      0     0      0     0      0   \n",
      "CRUK0881        1   1      1       1       1      1     1      1     1      1   \n",
      "CRUK0893        1   0      0       1       1      0     0      0     1      0   \n",
      "\n",
      "Hugo_Symbol  TP53  TRY2P  TTC40  TTN  UNC79  UNC80  USH2A  VCAN  WDFY4  XIRP2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001        1      1      0    1      0      1      1     0      1      0   \n",
      "CRUK0002        0      0      0    1      0      0      0     0      1      0   \n",
      "CRUK0003        0      0      0    1      1      0      1     1      0      0   \n",
      "CRUK0004        1      1      0    0      0      0      0     0      0      0   \n",
      "CRUK0005        1      1      0    1      0      1      1     0      1      1   \n",
      "...           ...    ...    ...  ...    ...    ...    ...   ...    ...    ...   \n",
      "CRUK0810        0      0      1    1      0      0      1     0      1      1   \n",
      "CRUK0817        1      0      0    0      0      0      0     0      0      0   \n",
      "CRUK0864        0      0      0    0      1      0      0     0      0      0   \n",
      "CRUK0881        1      0      1    1      0      1      1     1      1      1   \n",
      "CRUK0893        0      0      0    1      0      0      0     0      0      1   \n",
      "\n",
      "Hugo_Symbol  ZFHX4  ZNF536  \n",
      "patient_id                  \n",
      "CRUK0001         0       1  \n",
      "CRUK0002         0       0  \n",
      "CRUK0003         0       0  \n",
      "CRUK0004         0       0  \n",
      "CRUK0005         1       1  \n",
      "...            ...     ...  \n",
      "CRUK0810         1       0  \n",
      "CRUK0817         1       1  \n",
      "CRUK0864         0       0  \n",
      "CRUK0881         1       0  \n",
      "CRUK0893         0       0  \n",
      "\n",
      "[242 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# LUAD analysis\n",
    "\n",
    "# Use value_counts to get the frequency of each gene\n",
    "LUAD_gene_counts = mut_table_LUAD['Hugo_Symbol'].value_counts()\n",
    "\n",
    "# Get the top 100 most common genes\n",
    "LUAD_top_100_genes = LUAD_gene_counts.head(100)\n",
    "\n",
    "# Convert the result to a list of tuples (gene, frequency)\n",
    "LUAD_gene_frequency_list = list(zip(LUAD_top_100_genes.index, LUAD_top_100_genes.values))\n",
    "\n",
    "print(LUAD_gene_frequency_list)\n",
    "\n",
    "# Extract the gene names from the gene_frequency_list\n",
    "LUAD_hundred_genes = [gene for gene, _ in LUAD_gene_frequency_list]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUAD_filtered_mut_table = mut_table_LUAD[mut_table_LUAD['Hugo_Symbol'].isin(LUAD_hundred_genes)]\n",
    "\n",
    "# Create a binary matrix (presence or absence of genes for each patient)\n",
    "LUAD_gene_matrix = pd.crosstab(LUAD_filtered_mut_table['patient_id'], LUAD_filtered_mut_table['Hugo_Symbol']).clip(upper=1)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUAD_gene_matrix.shape)\n",
    "print(LUAD_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('TTN', 505), ('RYR2', 250), ('MUC19', 205), ('CSMD3', 204), ('LRP1B', 177), ('USH2A', 171), ('SYNE1', 167), ('OR4E2', 163), ('MUC16', 150), ('TP53', 135), ('COL22A1', 127), ('ZFHX4', 121), ('OBSCN', 108), ('RYR3', 107), ('SPTA1', 106), ('DMD', 103), ('COL11A1', 102), ('RELN', 98), ('SI', 93), ('SSPO', 90), ('CSMD1', 89), ('NAV3', 89), ('APOB', 89), ('PKHD1L1', 88), ('MROH2B', 88), ('PKHD1', 83), ('UNC80', 82), ('DNAH5', 81), ('NRXN1', 81), ('MIR548N', 79), ('FAM135B', 79), ('CTNNA2', 78), ('CACNA1E', 78), ('FSIP2', 77), ('RYR1', 75), ('CSMD2', 74), ('ANK2', 74), ('DNAH9', 73), ('EYS', 73), ('PCDH15', 72), ('PLEC', 72), ('NEB', 71), ('PCLO', 69), ('LRP2', 69), ('LPHN3', 68), ('HCN1', 67), ('DNAH6', 64), ('XIRP2', 64), ('HMCN1', 64), ('DPP6', 64), ('LAMA2', 64), ('FBN2', 63), ('OTOG', 61), ('CCDC168', 61), ('DNAH11', 60), ('DNAH8', 60), ('CDH10', 60), ('SPEF2', 59), ('CNTNAP5', 58), ('TNR', 58), ('MYO16', 57), ('ABCA13', 57), ('COL5A2', 56), ('DPP10', 56), ('SPHKAP', 56), ('THSD7B', 55), ('OTOGL', 55), ('BAI3', 55), ('PLXNA4', 55), ('ROS1', 55), ('CPS1', 55), ('MALRD1', 55), ('LRRC7', 54), ('UNC13C', 54), ('ADGB', 54), ('TRDN', 53), ('ANK1', 52), ('CACNA1C', 52), ('PPFIA2', 52), ('LOC93432', 52), ('MORC1', 52), ('SORCS1', 52), ('PAPPA2', 51), ('FAT4', 51), ('CUBN', 51), ('RIMS2', 51), ('ADAMTS12', 51), ('STAB2', 51), ('UTRN', 51), ('TRIO', 50), ('FAT3', 50), ('TENM3', 50), ('WDFY4', 50), ('COL6A5', 50), ('TNN', 50), ('BRINP3', 49), ('ANKS1B', 49), ('NCS1', 49), ('ADAMTS16', 49), ('FPGT-TNNI3K', 49)]\n",
      "(138, 100)\n",
      "Hugo_Symbol  ABCA13  ADAMTS12  ADAMTS16  ADGB  ANK1  ANK2  ANKS1B  APOB  BAI3  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062          0         0         1     0     1     1       1     0     0   \n",
      "CRUK0063          0         1         0     0     0     0       0     0     0   \n",
      "CRUK0064          0         0         1     0     0     0       1     1     0   \n",
      "CRUK0065          1         0         0     0     1     1       0     1     1   \n",
      "CRUK0066          1         1         1     1     1     0       0     0     0   \n",
      "...             ...       ...       ...   ...   ...   ...     ...   ...   ...   \n",
      "CRUK0772          1         0         0     1     0     0       0     1     0   \n",
      "CRUK0781          1         1         0     0     0     0       0     0     0   \n",
      "CRUK0784          0         1         0     1     0     0       0     0     0   \n",
      "CRUK0861          0         0         0     0     0     1       0     1     1   \n",
      "CRUK0872          0         0         0     1     0     0       1     1     0   \n",
      "\n",
      "Hugo_Symbol  BRINP3  CACNA1C  CACNA1E  CCDC168  CDH10  CNTNAP5  COL11A1  \\\n",
      "patient_id                                                                \n",
      "CRUK0062          0        1        1        0      1        0        0   \n",
      "CRUK0063          0        0        0        0      0        0        0   \n",
      "CRUK0064          0        0        0        0      0        0        0   \n",
      "CRUK0065          0        0        1        0      1        0        0   \n",
      "CRUK0066          1        0        1        0      1        1        1   \n",
      "...             ...      ...      ...      ...    ...      ...      ...   \n",
      "CRUK0772          0        0        0        0      0        0        1   \n",
      "CRUK0781          1        0        1        0      0        1        1   \n",
      "CRUK0784          0        1        0        1      0        1        1   \n",
      "CRUK0861          0        0        0        0      0        1        0   \n",
      "CRUK0872          0        0        0        0      0        0        1   \n",
      "\n",
      "Hugo_Symbol  COL22A1  COL5A2  COL6A5  CPS1  CSMD1  CSMD2  CSMD3  CTNNA2  CUBN  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062           1       0       0     0      0      1      1       1     0   \n",
      "CRUK0063           0       0       0     0      0      0      0       0     0   \n",
      "CRUK0064           1       0       0     1      0      1      0       0     0   \n",
      "CRUK0065           1       0       0     0      0      0      1       0     0   \n",
      "CRUK0066           1       1       1     0      0      0      1       0     0   \n",
      "...              ...     ...     ...   ...    ...    ...    ...     ...   ...   \n",
      "CRUK0772           1       0       0     0      0      0      0       0     1   \n",
      "CRUK0781           1       0       0     0      0      1      1       0     0   \n",
      "CRUK0784           0       0       1     1      0      1      1       0     0   \n",
      "CRUK0861           0       0       0     0      0      0      1       0     0   \n",
      "CRUK0872           1       1       1     0      0      0      1       0     1   \n",
      "\n",
      "Hugo_Symbol  DMD  DNAH11  DNAH5  DNAH6  DNAH8  DNAH9  DPP10  DPP6  EYS  \\\n",
      "patient_id                                                               \n",
      "CRUK0062       0       0      1      0      1      0      1     0    1   \n",
      "CRUK0063       0       0      1      0      1      1      0     0    0   \n",
      "CRUK0064       0       0      0      1      0      0      0     1    0   \n",
      "CRUK0065       1       0      1      0      0      1      0     1    1   \n",
      "CRUK0066       1       0      1      0      0      0      0     1    1   \n",
      "...          ...     ...    ...    ...    ...    ...    ...   ...  ...   \n",
      "CRUK0772       0       1      0      0      1      0      0     0    0   \n",
      "CRUK0781       0       0      1      1      0      0      0     0    1   \n",
      "CRUK0784       0       1      0      0      0      0      1     0    1   \n",
      "CRUK0861       0       0      0      1      0      1      0     0    1   \n",
      "CRUK0872       0       0      1      0      0      0      0     0    0   \n",
      "\n",
      "Hugo_Symbol  FAM135B  FAT3  FAT4  FBN2  FPGT-TNNI3K  FSIP2  HCN1  HMCN1  \\\n",
      "patient_id                                                                \n",
      "CRUK0062           1     0     0     1            1      1     1      0   \n",
      "CRUK0063           0     0     0     0            1      1     0      1   \n",
      "CRUK0064           0     1     0     0            0      0     1      1   \n",
      "CRUK0065           0     0     1     0            0      1     1      1   \n",
      "CRUK0066           0     0     0     0            0      0     1      0   \n",
      "...              ...   ...   ...   ...          ...    ...   ...    ...   \n",
      "CRUK0772           1     0     0     0            1      0     0      1   \n",
      "CRUK0781           1     0     0     1            0      0     0      0   \n",
      "CRUK0784           0     1     0     1            0      1     1      0   \n",
      "CRUK0861           0     0     0     1            0      0     0      0   \n",
      "CRUK0872           0     0     0     0            1      0     1      0   \n",
      "\n",
      "Hugo_Symbol  LAMA2  LOC93432  LPHN3  LRP1B  LRP2  LRRC7  MALRD1  MIR548N  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062         1         1      0      1     0      1       0        0   \n",
      "CRUK0063         0         0      1      1     0      0       0        0   \n",
      "CRUK0064         1         0      1      1     1      0       0        1   \n",
      "CRUK0065         0         0      0      0     1      0       0        1   \n",
      "CRUK0066         0         0      0      1     0      0       0        0   \n",
      "...            ...       ...    ...    ...   ...    ...     ...      ...   \n",
      "CRUK0772         1         0      0      1     1      0       1        0   \n",
      "CRUK0781         0         0      1      1     1      1       1        1   \n",
      "CRUK0784         1         0      1      1     0      0       1        1   \n",
      "CRUK0861         0         0      0      0     1      0       0        0   \n",
      "CRUK0872         0         0      1      1     1      0       0        0   \n",
      "\n",
      "Hugo_Symbol  MORC1  MROH2B  MUC16  MUC19  MYO16  NAV3  NCS1  NEB  NRXN1  \\\n",
      "patient_id                                                                \n",
      "CRUK0062         0       1      1      0      1     0     1    1      0   \n",
      "CRUK0063         0       0      0      0      0     0     0    0      0   \n",
      "CRUK0064         0       0      1      1      0     0     0    0      0   \n",
      "CRUK0065         0       0      1      1      1     1     1    1      0   \n",
      "CRUK0066         1       1      1      1      0     0     0    0      1   \n",
      "...            ...     ...    ...    ...    ...   ...   ...  ...    ...   \n",
      "CRUK0772         1       0      1      1      0     0     1    0      0   \n",
      "CRUK0781         0       0      1      0      0     0     0    0      0   \n",
      "CRUK0784         0       0      1      0      1     0     0    0      1   \n",
      "CRUK0861         0       0      1      0      0     0     0    1      0   \n",
      "CRUK0872         0       1      1      1      1     0     0    0      0   \n",
      "\n",
      "Hugo_Symbol  OBSCN  OR4E2  OTOG  OTOGL  PAPPA2  PCDH15  PCLO  PKHD1  PKHD1L1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062         1      1     0      0       1       1     0      1        1   \n",
      "CRUK0063         1      1     0      0       0       0     0      0        0   \n",
      "CRUK0064         0      1     0      0       0       0     0      1        0   \n",
      "CRUK0065         1      0     0      0       1       0     0      1        0   \n",
      "CRUK0066         0      0     0      0       0       1     0      1        1   \n",
      "...            ...    ...   ...    ...     ...     ...   ...    ...      ...   \n",
      "CRUK0772         0      1     0      0       0       0     0      1        0   \n",
      "CRUK0781         0      0     0      0       1       1     0      0        1   \n",
      "CRUK0784         1      0     0      0       0       0     0      1        1   \n",
      "CRUK0861         0      0     0      0       0       0     0      0        1   \n",
      "CRUK0872         0      0     0      1       0       1     0      0        0   \n",
      "\n",
      "Hugo_Symbol  PLEC  PLXNA4  PPFIA2  RELN  RIMS2  ROS1  RYR1  RYR2  RYR3  SI  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062        1       0       0     1      0     1     0     1     1   1   \n",
      "CRUK0063        0       0       1     0      0     0     1     0     0   1   \n",
      "CRUK0064        1       0       0     1      1     0     0     1     0   0   \n",
      "CRUK0065        1       1       1     1      0     1     1     1     1   1   \n",
      "CRUK0066        0       0       0     0      0     1     1     1     0   0   \n",
      "...           ...     ...     ...   ...    ...   ...   ...   ...   ...  ..   \n",
      "CRUK0772        1       0       0     0      0     0     0     1     0   0   \n",
      "CRUK0781        0       1       1     0      1     0     0     1     1   0   \n",
      "CRUK0784        0       1       0     1      0     1     1     1     0   1   \n",
      "CRUK0861        0       0       0     1      0     0     0     1     1   0   \n",
      "CRUK0872        0       0       0     1      0     1     0     1     0   1   \n",
      "\n",
      "Hugo_Symbol  SORCS1  SPEF2  SPHKAP  SPTA1  SSPO  STAB2  SYNE1  TENM3  THSD7B  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062          0      0       0      0     0      0      1      0       0   \n",
      "CRUK0063          0      0       0      0     0      0      0      1       0   \n",
      "CRUK0064          0      0       0      1     0      0      1      0       0   \n",
      "CRUK0065          0      1       0      1     0      0      1      0       0   \n",
      "CRUK0066          1      1       0      1     0      0      1      0       0   \n",
      "...             ...    ...     ...    ...   ...    ...    ...    ...     ...   \n",
      "CRUK0772          0      1       0      0     0      0      0      0       0   \n",
      "CRUK0781          0      0       0      1     1      0      0      0       1   \n",
      "CRUK0784          1      0       0      0     1      0      0      0       0   \n",
      "CRUK0861          0      1       0      1     1      0      1      0       0   \n",
      "CRUK0872          0      0       0      0     1      0      0      0       0   \n",
      "\n",
      "Hugo_Symbol  TNN  TNR  TP53  TRDN  TRIO  TTN  UNC13C  UNC80  USH2A  UTRN  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062       1    0     1     1     1    1       0      1      1     0   \n",
      "CRUK0063       0    1     1     0     0    1       0      0      0     0   \n",
      "CRUK0064       0    0     1     0     0    1       0      1      0     0   \n",
      "CRUK0065       0    1     1     0     1    1       0      1      1     1   \n",
      "CRUK0066       0    1     1     0     0    1       0      0      0     0   \n",
      "...          ...  ...   ...   ...   ...  ...     ...    ...    ...   ...   \n",
      "CRUK0772       0    0     1     0     0    1       0      1      1     0   \n",
      "CRUK0781       1    0     1     0     0    1       1      0      1     0   \n",
      "CRUK0784       0    1     1     1     0    1       0      1      1     0   \n",
      "CRUK0861       0    1     1     0     0    1       0      0      0     0   \n",
      "CRUK0872       0    1     1     0     0    1       0      1      1     0   \n",
      "\n",
      "Hugo_Symbol  WDFY4  XIRP2  ZFHX4  \n",
      "patient_id                        \n",
      "CRUK0062         0      0      1  \n",
      "CRUK0063         0      0      0  \n",
      "CRUK0064         1      0      1  \n",
      "CRUK0065         0      1      1  \n",
      "CRUK0066         0      1      1  \n",
      "...            ...    ...    ...  \n",
      "CRUK0772         0      1      0  \n",
      "CRUK0781         0      0      1  \n",
      "CRUK0784         0      0      1  \n",
      "CRUK0861         0      0      0  \n",
      "CRUK0872         0      0      1  \n",
      "\n",
      "[138 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# LUSC analysis\n",
    "\n",
    "# Use value_counts to get the frequency of each gene\n",
    "LUSC_gene_counts = mut_table_LUSC['Hugo_Symbol'].value_counts()\n",
    "\n",
    "# Get the top 100 most common genes\n",
    "LUSC_top_100_genes = LUSC_gene_counts.head(100)\n",
    "\n",
    "# Convert the result to a list of tuples (gene, frequency)\n",
    "LUSC_gene_frequency_list = list(zip(LUSC_top_100_genes.index, LUSC_top_100_genes.values))\n",
    "\n",
    "print(LUSC_gene_frequency_list)\n",
    "\n",
    "# Extract the gene names from the gene_frequency_list\n",
    "LUSC_hundred_genes = [gene for gene, _ in LUSC_gene_frequency_list]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUSC_filtered_mut_table = mut_table_LUSC[mut_table_LUSC['Hugo_Symbol'].isin(LUSC_hundred_genes)]\n",
    "\n",
    "# Create a binary matrix (presence or absence of genes for each patient)\n",
    "LUSC_gene_matrix = pd.crosstab(LUSC_filtered_mut_table['patient_id'], LUSC_filtered_mut_table['Hugo_Symbol']).clip(upper=1)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUSC_gene_matrix.shape)\n",
    "print(LUSC_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0 0 1 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 1 0 1 0 0 1 1 1 1 1\n",
      " 0 1 0 0 1 1 0 0 1 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 1 0 1 1 1 0 0 0 0 1 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 1 1 1\n",
      " 0 1 0 1 0 1 1 0 0 0 0 0 1 1 0 0 1 0 1 0 0 0 1 1 0 0 1 1 0 0 0 1 0 0 0 0 0\n",
      " 0 0 1 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 0 1 1 0 1 0 1 0 1 1 0 1 0 0 0\n",
      " 0 1 0 0 1 1 1 0 0 0 0 0 1 0 0 1 0 0 0 0 0 1 0 0 1 0 0 0 0 1 0 1 0 0 1 0 1\n",
      " 0 0 0 0 1 1 0 0 0 1 0 0 0 0 0 1 1 0 1 0]\n",
      "[1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 1 0 0 0 0 0 1 1 0 1 0 1 0 0 0 1 1 0 0\n",
      " 1 0 0 1 0 1 0 0 0 1 1 1 1 0 1 1 0 1 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 1 1 1 1 0 0 1 1 1 1 0 0 1 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 1 1 0 1 0 0\n",
      " 0 0 0 1 1 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 1]\n"
     ]
    }
   ],
   "source": [
    "# make y for LUAD\n",
    "y_train_LUAD = []\n",
    "for index, row in clinical_df_LUAD.iterrows():\n",
    "    if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "        y_train_LUAD.append(1)\n",
    "    else:\n",
    "        y_train_LUAD.append(0)\n",
    "\n",
    "y_LUAD = np.array(y_train_LUAD)\n",
    "print(y_LUAD)\n",
    "\n",
    "# make y for LUSC\n",
    "y_train_LUSC = []\n",
    "for index, row in clinical_df_LUSC.iterrows():\n",
    "    if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "        y_train_LUSC.append(1)\n",
    "    else:\n",
    "        y_train_LUSC.append(0)\n",
    "\n",
    "y_LUSC = np.array(y_train_LUSC)\n",
    "print(y_LUSC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 100) (73, 100) (169,) (73,)\n",
      "Best parameters:  {'C': 1e-05}\n",
      "Best score:  0.5\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.5409982174688057\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.70      1.00      0.82        51\n",
      "   Recurrence       0.00      0.00      0.00        22\n",
      "\n",
      "     accuracy                           0.70        73\n",
      "    macro avg       0.35      0.50      0.41        73\n",
      " weighted avg       0.49      0.70      0.57        73\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 100 most commonly mutated genes in LUAD\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.metrics import classification_report\n",
    "from sklearn.metrics import balanced_accuracy_score\n",
    "from sklearn.metrics import roc_curve, roc_auc_score\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_gene_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 100) (42, 100) (96,) (42,)\n",
      "Best parameters:  {'C': 1}\n",
      "Best score:  0.5511904761904761\n",
      "Predictions:  [0 0 0 0 0 0 0 1 0 0 0 1 0 0 1 0 1 1 1 0 1 0 1 1 0 1 0 0 1 1 0 0 0 0 0 0 0\n",
      " 1 0 0 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.4395061728395062\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.64      0.67      0.65        27\n",
      "   Recurrence       0.36      0.33      0.34        15\n",
      "\n",
      "     accuracy                           0.55        42\n",
      "    macro avg       0.50      0.50      0.50        42\n",
      " weighted avg       0.54      0.55      0.54        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 100 most commonly mutated genes in LUSC\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_gene_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 100) (73, 100) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5153030303030303\n",
      "Predictions:  [0 1 0 0 0 0 1 1 0 0 0 0 0 0 0 0 1 1 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 1\n",
      " 0 0 0 0 1 0 0 0 0 1 0 1 0 0 1 1 1 1 0 1 0 0 1 1 0 1 1 0 0 0 0 1 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5347593582887701\n",
      "ROC AUC:  0.5730837789661319\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.72      0.71      0.71        51\n",
      "   Recurrence       0.35      0.36      0.36        22\n",
      "\n",
      "     accuracy                           0.60        73\n",
      "    macro avg       0.53      0.53      0.53        73\n",
      " weighted avg       0.61      0.60      0.61        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm\n",
    "\n",
    "from sklearn.neural_network import MLPClassifier\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_gene_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 100) (42, 100) (96,) (42,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.5522893772893773\n",
      "Predictions:  [0 1 0 0 0 0 0 1 0 1 0 1 0 0 1 0 1 0 1 1 1 0 1 1 0 1 0 0 1 1 0 0 0 1 0 0 1\n",
      " 1 0 0 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.37407407407407406\n",
      "ROC AUC:  0.41728395061728396\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.54      0.48      0.51        27\n",
      "   Recurrence       0.22      0.27      0.24        15\n",
      "\n",
      "     accuracy                           0.40        42\n",
      "    macro avg       0.38      0.37      0.38        42\n",
      " weighted avg       0.43      0.40      0.41        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_gene_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('TP53', 136), ('KRAS', 115), ('STK11', 51), ('KEAP1', 39), ('SMARCA4', 38), ('RBM10', 35), ('NF1', 30), ('MGA', 25), ('FAT1', 25), ('EGFR', 25), ('ATM', 25), ('KMT2C', 23), ('GRIN2A', 21), ('KMT2D', 21), ('ATRX', 20), ('PIK3CA', 19), ('NCOR1', 19), ('PTPRB', 18), ('ARID2', 17), ('CDKN2A', 17), ('ARID1A', 16), ('COL5A2', 16), ('CREBBP', 16), ('APC', 16), ('SETD2', 14), ('SMAD4', 14), ('FANCM', 14), ('PTPRC', 13), ('COL2A1', 13), ('BCLAF1', 13), ('POT1', 12), ('EP300', 12), ('DNAH12', 12), ('BCOR', 12), ('UBR5', 12), ('CHD8', 12), ('KDM5C', 12), ('BRAF', 11), ('MTOR', 11), ('TET2', 10), ('ARHGAP35', 10), ('RB1', 10), ('CMTR2', 10), ('DICER1', 10), ('BAP1', 10), ('IKZF1', 10), ('DOT1L', 10), ('ZFP36L1', 10), ('POLE', 10), ('SPEN', 9), ('CDK12', 9), ('NBN', 9), ('RASA1', 9), ('CTNNB1', 9), ('STAG2', 9), ('CHD4', 9), ('NOTCH2', 9), ('CUX1', 8), ('ARID1B', 8), ('ASXL2', 8), ('PLXNB2', 8), ('TSC2', 8), ('ZMYM2', 8), ('PPP3CA', 8), ('SGK223', 7), ('CIC', 7), ('MEN1', 7), ('KMT2B', 7), ('DNMT3A', 7), ('RYR2', 7), ('DDX3X', 7), ('ACVR1B', 6), ('RAD21', 6), ('CBLB', 6), ('PTEN', 6), ('ANK2', 6), ('RNF43', 6), ('MAP3K1', 6), ('FLT4', 6), ('PMS1', 6), ('TP53BP1', 6), ('PTPRU', 5), ('PIK3R1', 5), ('CUL3', 5), ('DNM2', 5), ('SERPINB13', 5), ('AMER1', 5), ('WAS', 5), ('NCOA6', 5), ('PLCG2', 5), ('WT1', 5), ('NRAS', 5), ('NRXN1', 5), ('PBRM1', 5), ('U2AF1', 5), ('SETDB1', 5), ('SF3B1', 5), ('MET', 5), ('GATA3', 5), ('NFE2L2', 5)]\n",
      "(242, 100)\n",
      "Hugo_Symbol  ACVR1B  AMER1  ANK2  APC  ARHGAP35  ARID1A  ARID1B  ARID2  ASXL2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001          0      0     0    0         1       0       0      0      0   \n",
      "CRUK0002          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0003          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0004          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0005          0      0     0    0         0       0       0      0      0   \n",
      "...             ...    ...   ...  ...       ...     ...     ...    ...    ...   \n",
      "CRUK0810          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0817          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0864          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0881          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0893          0      0     0    0         0       0       0      0      0   \n",
      "\n",
      "Hugo_Symbol  ATM  ATRX  BAP1  BCLAF1  BCOR  BRAF  CBLB  CDK12  CDKN2A  CHD4  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0002       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0003       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0004       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0005       0     1     0       0     0     1     0      0       0     0   \n",
      "...          ...   ...   ...     ...   ...   ...   ...    ...     ...   ...   \n",
      "CRUK0810       0     1     0       0     0     0     0      0       0     0   \n",
      "CRUK0817       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0864       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0881       0     0     0       0     0     1     0      0       0     0   \n",
      "CRUK0893       1     0     0       0     0     0     0      0       0     0   \n",
      "\n",
      "Hugo_Symbol  CHD8  CIC  CMTR2  COL2A1  COL5A2  CREBBP  CTNNB1  CUL3  CUX1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0002        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0003        0    0      0       0       0       0       1     0     0   \n",
      "CRUK0004        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0005        0    0      1       0       0       0       0     0     0   \n",
      "...           ...  ...    ...     ...     ...     ...     ...   ...   ...   \n",
      "CRUK0810        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0817        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0864        0    0      0       0       0       0       0     0     1   \n",
      "CRUK0881        0    0      0       0       0       1       0     1     0   \n",
      "CRUK0893        0    0      0       0       0       0       0     0     0   \n",
      "\n",
      "Hugo_Symbol  DDX3X  DICER1  DNAH12  DNM2  DNMT3A  DOT1L  EGFR  EP300  FANCM  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0       0       0     0       0      0     0      0      0   \n",
      "CRUK0002         0       0       0     0       0      0     0      1      0   \n",
      "CRUK0003         0       0       0     0       0      0     1      0      0   \n",
      "CRUK0004         0       0       0     0       0      0     1      0      0   \n",
      "CRUK0005         0       0       0     0       0      0     0      0      0   \n",
      "...            ...     ...     ...   ...     ...    ...   ...    ...    ...   \n",
      "CRUK0810         0       0       0     0       0      0     0      0      0   \n",
      "CRUK0817         0       0       0     0       0      0     0      1      0   \n",
      "CRUK0864         0       0       0     0       1      0     0      0      0   \n",
      "CRUK0881         1       0       1     0       0      0     0      0      0   \n",
      "CRUK0893         0       0       0     0       0      0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  FAT1  FLT4  GATA3  GRIN2A  IKZF1  KDM5C  KEAP1  KMT2B  KMT2C  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        0     0      0       0      0      0      0      1      0   \n",
      "CRUK0002        0     0      0       0      1      0      0      0      0   \n",
      "CRUK0003        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0004        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0005        0     0      0       0      0      0      0      0      0   \n",
      "...           ...   ...    ...     ...    ...    ...    ...    ...    ...   \n",
      "CRUK0810        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0817        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0864        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0881        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0893        0     0      0       0      0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  KRAS  MAP3K1  MEN1  MET  MGA  MTOR  NBN  NCOA6  NCOR1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001         0     0       0     0    0    1     0    0      0      0   \n",
      "CRUK0002         0     0       0     0    1    0     0    0      0      0   \n",
      "CRUK0003         0     0       0     0    0    0     0    0      0      0   \n",
      "CRUK0004         0     0       0     0    0    0     0    0      0      0   \n",
      "CRUK0005         0     0       0     0    0    0     0    0      0      0   \n",
      "...            ...   ...     ...   ...  ...  ...   ...  ...    ...    ...   \n",
      "CRUK0810         0     1       0     0    0    0     0    0      0      0   \n",
      "CRUK0817         0     0       0     0    0    0     0    0      0      0   \n",
      "CRUK0864         0     1       0     0    0    0     0    0      0      0   \n",
      "CRUK0881         0     1       0     0    0    0     0    0      0      0   \n",
      "CRUK0893         0     1       0     0    0    0     0    0      0      0   \n",
      "\n",
      "Hugo_Symbol  NF1  NFE2L2  NOTCH2  NRAS  NRXN1  PBRM1  PIK3CA  PIK3R1  PLCG2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001       0       0       0     0      0      0       0       0      0   \n",
      "CRUK0002       1       0       0     0      0      0       0       0      0   \n",
      "CRUK0003       0       0       0     0      0      0       1       0      0   \n",
      "CRUK0004       0       0       0     0      0      0       0       0      0   \n",
      "CRUK0005       0       0       0     1      0      0       0       0      0   \n",
      "...          ...     ...     ...   ...    ...    ...     ...     ...    ...   \n",
      "CRUK0810       0       0       0     0      0      0       0       0      0   \n",
      "CRUK0817       1       0       0     0      0      0       0       0      0   \n",
      "CRUK0864       0       0       0     0      0      0       0       0      0   \n",
      "CRUK0881       1       0       0     0      0      1       0       0      0   \n",
      "CRUK0893       0       0       0     0      0      0       0       0      0   \n",
      "\n",
      "Hugo_Symbol  PLXNB2  PMS1  POLE  POT1  PPP3CA  PTEN  PTPRB  PTPRC  PTPRU  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0002          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0003          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0004          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0005          0     0     0     0       0     0      0      0      0   \n",
      "...             ...   ...   ...   ...     ...   ...    ...    ...    ...   \n",
      "CRUK0810          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0817          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0864          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0881          0     1     0     0       1     0      0      0      0   \n",
      "CRUK0893          0     0     0     0       0     0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  RAD21  RASA1  RB1  RBM10  RNF43  RYR2  SERPINB13  SETD2  SETDB1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         0      0    0      0      0     0          0      0       0   \n",
      "CRUK0002         0      0    1      0      0     0          0      0       0   \n",
      "CRUK0003         0      0    0      0      0     0          0      0       0   \n",
      "CRUK0004         0      0    0      0      0     0          0      0       0   \n",
      "CRUK0005         0      0    0      0      0     0          0      0       0   \n",
      "...            ...    ...  ...    ...    ...   ...        ...    ...     ...   \n",
      "CRUK0810         0      0    0      0      0     0          0      0       0   \n",
      "CRUK0817         0      0    0      1      0     0          0      0       0   \n",
      "CRUK0864         0      0    0      1      0     0          0      0       0   \n",
      "CRUK0881         0      0    0      1      0     0          0      0       1   \n",
      "CRUK0893         0      0    0      0      0     0          0      0       0   \n",
      "\n",
      "Hugo_Symbol  SF3B1  SGK223  SMAD4  SMARCA4  SPEN  STAG2  STK11  TET2  TP53  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001         0       0      0        0     0      0      0     0     1   \n",
      "CRUK0002         0       0      0        0     0      0      0     0     0   \n",
      "CRUK0003         0       0      0        0     0      0      0     0     0   \n",
      "CRUK0004         0       0      1        0     0      0      0     0     1   \n",
      "CRUK0005         0       0      0        0     0      0      0     0     1   \n",
      "...            ...     ...    ...      ...   ...    ...    ...   ...   ...   \n",
      "CRUK0810         0       0      0        1     0      0      1     0     0   \n",
      "CRUK0817         0       0      0        0     0      0      1     0     1   \n",
      "CRUK0864         0       0      0        0     0      0      0     0     0   \n",
      "CRUK0881         0       0      0        0     0      0      0     1     1   \n",
      "CRUK0893         0       0      0        0     0      0      0     0     0   \n",
      "\n",
      "Hugo_Symbol  TP53BP1  TSC2  U2AF1  UBR5  WAS  WT1  ZFP36L1  ZMYM2  \n",
      "patient_id                                                         \n",
      "CRUK0001           0     0      0     0    0    0        0      0  \n",
      "CRUK0002           0     0      0     0    0    0        0      0  \n",
      "CRUK0003           0     0      0     0    0    0        0      0  \n",
      "CRUK0004           0     0      0     0    0    0        0      0  \n",
      "CRUK0005           0     0      0     0    0    0        0      0  \n",
      "...              ...   ...    ...   ...  ...  ...      ...    ...  \n",
      "CRUK0810           0     0      0     0    0    0        0      0  \n",
      "CRUK0817           0     0      0     0    0    0        0      0  \n",
      "CRUK0864           0     0      0     0    0    0        0      0  \n",
      "CRUK0881           0     0      0     0    0    0        0      0  \n",
      "CRUK0893           0     0      0     0    0    0        0      0  \n",
      "\n",
      "[242 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# Use boolean indexing to filter rows where 'DriverMut' is True\n",
    "LUAD_driver_table = mut_table_LUAD[mut_table_LUAD['DriverMut']]\n",
    "\n",
    "# Count the occurrences of each gene in the filtered DataFrame\n",
    "LUAD_driver_counts = LUAD_driver_table['Hugo_Symbol'].value_counts()\n",
    "\n",
    "# Get the top 100 most common genes\n",
    "LUAD_top_100_drivers = LUAD_driver_counts.head(100)\n",
    "\n",
    "# Convert the result to a list of tuples (gene, frequency)\n",
    "LUAD_driver_frequency_list = list(zip(LUAD_top_100_drivers.index, LUAD_top_100_drivers.values))\n",
    "\n",
    "print(LUAD_driver_frequency_list)\n",
    "\n",
    "# Extract the gene names from the driver_frequency_list\n",
    "LUAD_hundred_drivers = [gene for gene, _ in LUAD_driver_frequency_list]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUAD_filtered_drivers_table = mut_table_LUAD[mut_table_LUAD['Hugo_Symbol'].isin(LUAD_hundred_drivers)]\n",
    "\n",
    "# Create a binary matrix based on both gene presence/absence and 'DriverMut' values\n",
    "LUAD_driver_matrix = pd.crosstab(LUAD_filtered_drivers_table['patient_id'], LUAD_filtered_drivers_table['Hugo_Symbol'], values=LUAD_filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "\n",
    "LUAD_driver_matrix = LUAD_driver_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUAD_driver_matrix.shape)\n",
    "print(LUAD_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('TP53', 125), ('CDKN2A', 28), ('KMT2D', 26), ('FAT1', 24), ('PIK3CA', 21), ('NFE2L2', 18), ('KEAP1', 14), ('CUX1', 14), ('PTEN', 14), ('COL5A2', 13), ('NF1', 12), ('ARID1A', 12), ('ARID2', 11), ('SETD2', 11), ('PLXNB2', 11), ('CREBBP', 11), ('KMT2C', 11), ('UBR5', 10), ('RASA1', 9), ('NOTCH1', 9), ('B2M', 9), ('ATR', 8), ('FANCM', 8), ('SPEN', 8), ('PTPRB', 8), ('CHD4', 7), ('SMAD4', 7), ('CBLB', 7), ('ARHGAP35', 7), ('BCLAF1', 7), ('NCOA6', 7), ('KRAS', 7), ('EP300', 7), ('PBRM1', 7), ('ASXL1', 6), ('CYLD', 6), ('NCOR1', 6), ('ATRX', 6), ('TET2', 6), ('NOTCH2', 6), ('SMARCA4', 6), ('RB1', 6), ('APC', 6), ('BCOR', 6), ('CHD8', 5), ('PTPRC', 5), ('SETDB1', 5), ('STK11', 5), ('KMT2B', 5), ('ACVR1B', 5), ('TP53BP1', 5), ('ZMYM2', 5), ('MGA', 5), ('MAP3K13', 5), ('MTOR', 5), ('FBXW7', 5), ('BRCA1', 5), ('DDX3X', 4), ('TSC2', 4), ('LATS1', 4), ('DICER1', 4), ('CIC', 4), ('PIK3R1', 4), ('MLH1', 4), ('CASP8', 4), ('AMER1', 4), ('CMTR2', 4), ('CDC73', 4), ('WRN', 4), ('COL2A1', 4), ('MBD1', 4), ('DNAH12', 4), ('BRCA2', 4), ('DNMT3A', 4), ('CUL3', 4), ('ASXL2', 3), ('PASK', 3), ('KDM6A', 3), ('AXIN1', 3), ('GATA3', 3), ('FGFR3', 3), ('TNFAIP3', 3), ('HIST1H4E', 3), ('BRIP1', 3), ('AXIN2', 3), ('CDK12', 3), ('DNM2', 3), ('ATM', 3), ('SGK223', 3), ('CHEK2', 3), ('CEP76', 3), ('HCN1', 3), ('NF2', 3), ('CNOT3', 3), ('RYR2', 3), ('STAG2', 3), ('KDM5C', 3), ('IKZF1', 3), ('EXT2', 3), ('PTPRU', 3)]\n",
      "(138, 100)\n",
      "Hugo_Symbol  ACVR1B  AMER1  APC  ARHGAP35  ARID1A  ARID2  ASXL1  ASXL2  ATM  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0063          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0064          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0065          0      0    0         0       1      0      0      0    0   \n",
      "CRUK0066          0      0    0         0       0      0      0      0    0   \n",
      "...             ...    ...  ...       ...     ...    ...    ...    ...  ...   \n",
      "CRUK0772          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0781          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0784          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0861          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0872          0      0    0         0       0      0      0      0    0   \n",
      "\n",
      "Hugo_Symbol  ATR  ATRX  AXIN1  AXIN2  B2M  BCLAF1  BCOR  BRCA1  BRCA2  BRIP1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062       0     1      0      0    0       0     0      0      0      0   \n",
      "CRUK0063       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0064       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0065       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0066       0     0      0      0    0       0     0      0      0      1   \n",
      "...          ...   ...    ...    ...  ...     ...   ...    ...    ...    ...   \n",
      "CRUK0772       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0781       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0784       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0861       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0872       0     0      0      0    0       0     0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  CASP8  CBLB  CDC73  CDK12  CDKN2A  CEP76  CHD4  CHD8  CHEK2  CIC  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062         0     0      0      0       0      0     1     0      0    0   \n",
      "CRUK0063         0     0      0      0       1      0     0     0      0    0   \n",
      "CRUK0064         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0065         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0066         0     0      0      0       1      0     0     0      0    0   \n",
      "...            ...   ...    ...    ...     ...    ...   ...   ...    ...  ...   \n",
      "CRUK0772         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0781         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0784         0     0      0      0       0      0     1     0      0    0   \n",
      "CRUK0861         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0872         0     0      0      0       0      0     0     0      0    0   \n",
      "\n",
      "Hugo_Symbol  CMTR2  CNOT3  COL2A1  COL5A2  CREBBP  CUL3  CUX1  CYLD  DDX3X  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0063         0      0       0       0       0     0     0     1      0   \n",
      "CRUK0064         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0065         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0066         0      0       0       1       0     0     0     0      0   \n",
      "...            ...    ...     ...     ...     ...   ...   ...   ...    ...   \n",
      "CRUK0772         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0781         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0784         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0861         0      0       0       0       0     1     0     0      0   \n",
      "CRUK0872         0      0       0       0       0     0     0     0      0   \n",
      "\n",
      "Hugo_Symbol  DICER1  DNAH12  DNM2  DNMT3A  EP300  EXT2  FANCM  FAT1  FBXW7  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0063          0       0     0       0      1     0      1     0      1   \n",
      "CRUK0064          0       0     0       0      0     0      0     1      0   \n",
      "CRUK0065          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0066          0       0     0       0      0     0      0     0      0   \n",
      "...             ...     ...   ...     ...    ...   ...    ...   ...    ...   \n",
      "CRUK0772          0       0     0       0      0     0      1     0      0   \n",
      "CRUK0781          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0784          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0861          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0872          0       0     0       0      0     0      0     0      0   \n",
      "\n",
      "Hugo_Symbol  FGFR3  GATA3  HCN1  HIST1H4E  IKZF1  KDM5C  KDM6A  KEAP1  KMT2B  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0063         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0064         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0065         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0066         0      0     0         0      0      0      0      0      0   \n",
      "...            ...    ...   ...       ...    ...    ...    ...    ...    ...   \n",
      "CRUK0772         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0781         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0784         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0861         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0872         0      0     0         0      0      0      0      0      1   \n",
      "\n",
      "Hugo_Symbol  KMT2C  KMT2D  KRAS  LATS1  MAP3K13  MBD1  MGA  MLH1  MTOR  NCOA6  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0063         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0064         0      0     0      0        0     0    0     1     0      0   \n",
      "CRUK0065         0      0     0      0        0     0    0     1     0      1   \n",
      "CRUK0066         0      0     0      0        0     0    0     0     0      0   \n",
      "...            ...    ...   ...    ...      ...   ...  ...   ...   ...    ...   \n",
      "CRUK0772         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0781         0      1     0      0        0     0    0     0     0      0   \n",
      "CRUK0784         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0861         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0872         0      0     0      0        0     0    0     0     0      0   \n",
      "\n",
      "Hugo_Symbol  NCOR1  NF1  NF2  NFE2L2  NOTCH1  NOTCH2  PASK  PBRM1  PIK3CA  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062         0    0    0       0       0       0     0      0       0   \n",
      "CRUK0063         0    1    0       0       0       0     0      0       1   \n",
      "CRUK0064         0    0    0       0       0       0     0      0       0   \n",
      "CRUK0065         0    0    0       1       1       0     0      0       1   \n",
      "CRUK0066         0    0    0       0       1       0     0      0       0   \n",
      "...            ...  ...  ...     ...     ...     ...   ...    ...     ...   \n",
      "CRUK0772         0    0    0       1       0       0     0      0       0   \n",
      "CRUK0781         0    0    0       0       0       0     1      0       0   \n",
      "CRUK0784         0    0    0       0       0       0     0      0       0   \n",
      "CRUK0861         0    0    0       0       0       0     0      0       0   \n",
      "CRUK0872         0    0    0       0       0       0     0      0       0   \n",
      "\n",
      "Hugo_Symbol  PIK3R1  PLXNB2  PTEN  PTPRB  PTPRC  PTPRU  RASA1  RB1  RYR2  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062          0       1     0      0      0      0      0    0     0   \n",
      "CRUK0063          0       0     0      0      0      0      0    0     0   \n",
      "CRUK0064          0       0     0      0      0      0      0    0     0   \n",
      "CRUK0065          0       0     0      0      1      0      0    0     0   \n",
      "CRUK0066          0       0     0      0      0      0      0    0     0   \n",
      "...             ...     ...   ...    ...    ...    ...    ...  ...   ...   \n",
      "CRUK0772          0       0     0      0      0      0      0    1     0   \n",
      "CRUK0781          0       0     1      0      0      0      0    0     0   \n",
      "CRUK0784          0       0     0      0      0      0      0    0     0   \n",
      "CRUK0861          0       0     0      0      0      0      0    0     0   \n",
      "CRUK0872          0       0     0      1      0      0      0    0     0   \n",
      "\n",
      "Hugo_Symbol  SETD2  SETDB1  SGK223  SMAD4  SMARCA4  SPEN  STAG2  STK11  TET2  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0063         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0064         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0065         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0066         0       0       0      0        0     0      0      0     0   \n",
      "...            ...     ...     ...    ...      ...   ...    ...    ...   ...   \n",
      "CRUK0772         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0781         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0784         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0861         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0872         0       0       0      0        0     0      0      0     0   \n",
      "\n",
      "Hugo_Symbol  TNFAIP3  TP53  TP53BP1  TSC2  UBR5  WRN  ZMYM2  \n",
      "patient_id                                                   \n",
      "CRUK0062           0     1        0     0     1    0      0  \n",
      "CRUK0063           0     1        0     0     0    0      0  \n",
      "CRUK0064           0     1        0     0     0    0      0  \n",
      "CRUK0065           0     1        0     0     1    0      0  \n",
      "CRUK0066           0     1        0     0     0    1      0  \n",
      "...              ...   ...      ...   ...   ...  ...    ...  \n",
      "CRUK0772           0     1        1     0     0    0      0  \n",
      "CRUK0781           0     1        0     0     0    0      0  \n",
      "CRUK0784           0     1        0     0     0    0      0  \n",
      "CRUK0861           0     1        0     0     0    0      0  \n",
      "CRUK0872           0     1        0     0     0    0      0  \n",
      "\n",
      "[138 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# Use boolean indexing to filter rows where 'DriverMut' is True\n",
    "LUSC_driver_table = mut_table_LUSC[mut_table_LUSC['DriverMut']]\n",
    "\n",
    "# Count the occurrences of each gene in the filtered DataFrame\n",
    "LUSC_driver_counts = LUSC_driver_table['Hugo_Symbol'].value_counts()\n",
    "\n",
    "# Get the top 100 most common genes\n",
    "LUSC_top_100_drivers = LUSC_driver_counts.head(100)\n",
    "\n",
    "# Convert the result to a list of tuples (gene, frequency)\n",
    "LUSC_driver_frequency_list = list(zip(LUSC_top_100_drivers.index, LUSC_top_100_drivers.values))\n",
    "\n",
    "print(LUSC_driver_frequency_list)\n",
    "\n",
    "# Extract the gene names from the driver_frequency_list\n",
    "LUSC_hundred_drivers = [gene for gene, _ in LUSC_driver_frequency_list]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUSC_filtered_drivers_table = mut_table_LUSC[mut_table_LUSC['Hugo_Symbol'].isin(LUSC_hundred_drivers)]\n",
    "\n",
    "# Create a binary matrix based on both gene presence/absence and 'DriverMut' values\n",
    "LUSC_driver_matrix = pd.crosstab(LUSC_filtered_drivers_table['patient_id'], LUSC_filtered_drivers_table['Hugo_Symbol'], values=LUSC_filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "\n",
    "LUSC_driver_matrix = LUSC_driver_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUSC_driver_matrix.shape)\n",
    "print(LUSC_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 100) (73, 100) (169,) (73,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'C': 100}\n",
      "Best score:  0.5451152832674572\n",
      "Predictions:  [0 0 1 1 1 0 1 0 0 0 0 1 0 1 0 0 0 1 1 1 0 1 0 1 1 0 1 1 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 1 1 1 0 0 0 1 0 0 0 0 1 0 0 1 0 1 0 0 1 0 0 1 0 0 0 1 1 1]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4206773618538324\n",
      "ROC AUC:  0.4598930481283422\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.64      0.57      0.60        51\n",
      "   Recurrence       0.21      0.27      0.24        22\n",
      "\n",
      "     accuracy                           0.48        73\n",
      "    macro avg       0.43      0.42      0.42        73\n",
      " weighted avg       0.51      0.48      0.49        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 100 most common driver mutations in LUAD\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_driver_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 100) (42, 100) (96,) (42,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'C': 1}\n",
      "Best score:  0.5245421245421246\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0\n",
      " 0 0 0 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.5259259259259259\n",
      "ROC AUC:  0.4345679012345679\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.66      0.85      0.74        27\n",
      "   Recurrence       0.43      0.20      0.27        15\n",
      "\n",
      "     accuracy                           0.62        42\n",
      "    macro avg       0.54      0.53      0.51        42\n",
      " weighted avg       0.58      0.62      0.57        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 100 most common driver mutations in LUSC\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_driver_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 50)\n",
      "Hugo_Symbol  ABCA13  ANK2  APOB  CACNA1C  CACNA1E  CCDC168  CNTNAP5  COL11A1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001          1     1     1        1        0        0        1        0   \n",
      "CRUK0002          0     0     0        0        0        0        0        0   \n",
      "CRUK0003          0     0     0        0        0        0        0        0   \n",
      "CRUK0004          1     0     0        0        0        1        0        0   \n",
      "CRUK0005          1     1     1        0        1        1        0        1   \n",
      "...             ...   ...   ...      ...      ...      ...      ...      ...   \n",
      "CRUK0864          0     0     0        0        0        1        0        0   \n",
      "CRUK0881          0     1     1        1        1        1        1        1   \n",
      "CRUK0893          0     1     1        0        1        0        1        0   \n",
      "CRUK0059          0     0     0        0        0        0        0        0   \n",
      "CRUK0525          0     0     0        0        0        0        0        0   \n",
      "\n",
      "Hugo_Symbol  COL22A1  CSMD1  CSMD2  CSMD3  CTNNA2  DMD  DNAH9  DOCK2  FAT3  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001           1      1      1      1       1    1      0      1     0   \n",
      "CRUK0002           0      0      1      1       0    0      1      1     0   \n",
      "CRUK0003           0      0      0      1       0    0      0      1     0   \n",
      "CRUK0004           0      0      0      1       0    0      0      0     0   \n",
      "CRUK0005           1      0      1      1       0    1      1      1     1   \n",
      "...              ...    ...    ...    ...     ...  ...    ...    ...   ...   \n",
      "CRUK0864           0      0      0      1       0    1      1      1     0   \n",
      "CRUK0881           0      1      1      1       1    1      1      1     1   \n",
      "CRUK0893           0      0      0      0       0    0      0      0     0   \n",
      "CRUK0059           0      0      0      0       0    0      0      0     0   \n",
      "CRUK0525           0      0      0      0       0    0      0      0     0   \n",
      "\n",
      "Hugo_Symbol  FBN2  FSIP2  LAMA2  LPHN3  LRP1B  MUC16  MUC19  NAV3  NCS1  NEB  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001        1      1      1      0      0      0      1     1     1    0   \n",
      "CRUK0002        0      0      0      0      0      1      0     0     0    0   \n",
      "CRUK0003        0      0      0      0      1      1      0     1     0    0   \n",
      "CRUK0004        0      0      0      1      1      1      0     0     0    0   \n",
      "CRUK0005        1      1      0      1      1      1      0     0     0    1   \n",
      "...           ...    ...    ...    ...    ...    ...    ...   ...   ...  ...   \n",
      "CRUK0864        1      1      1      0      1      1      0     0     0    0   \n",
      "CRUK0881        1      1      1      1      1      1      1     1     1    1   \n",
      "CRUK0893        0      1      1      0      0      0      0     0     0    0   \n",
      "CRUK0059        0      0      0      0      0      0      0     0     0    0   \n",
      "CRUK0525        0      0      0      0      0      0      0     0     0    0   \n",
      "\n",
      "Hugo_Symbol  NRXN1  OBSCN  OR4E2  OTOG  PAPPA2  PCDH15  PCLO  PTPRD  PXDNL  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001         0      1      1     0       0       0     1      0      0   \n",
      "CRUK0002         0      0      0     0       0       0     1      0      0   \n",
      "CRUK0003         0      1      0     0       0       0     1      1      0   \n",
      "CRUK0004         0      1      0     0       0       0     0      0      0   \n",
      "CRUK0005         1      1      1     1       1       1     0      1      1   \n",
      "...            ...    ...    ...   ...     ...     ...   ...    ...    ...   \n",
      "CRUK0864         0      0      0     1       0       0     0      0      0   \n",
      "CRUK0881         1      1      1     1       1       1     1      1      1   \n",
      "CRUK0893         1      0      1     1       0       0     0      0      0   \n",
      "CRUK0059         0      0      0     0       0       0     0      0      0   \n",
      "CRUK0525         0      0      0     0       0       0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  RELN  RIMS2  RYR1  RYR2  RYR3  SPTA1  SSPO  SYNE1  TP53  TTN  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        1      1     1     1     1      1     1      1     1    1   \n",
      "CRUK0002        0      0     0     0     1      0     0      0     0    1   \n",
      "CRUK0003        1      0     0     1     0      0     0      0     0    1   \n",
      "CRUK0004        0      0     0     1     0      1     0      0     1    0   \n",
      "CRUK0005        1      1     0     1     0      0     1      0     1    1   \n",
      "...           ...    ...   ...   ...   ...    ...   ...    ...   ...  ...   \n",
      "CRUK0864        0      0     0     0     1      0     0      0     0    0   \n",
      "CRUK0881        1      1     1     1     1      1     1      1     1    1   \n",
      "CRUK0893        0      1     1     1     1      0     1      0     0    1   \n",
      "CRUK0059        0      0     0     0     0      0     0      0     0    0   \n",
      "CRUK0525        0      0     0     0     0      0     0      0     0    0   \n",
      "\n",
      "Hugo_Symbol  UNC80  USH2A  XIRP2  ZFHX4  \n",
      "patient_id                               \n",
      "CRUK0001         1      1      0      0  \n",
      "CRUK0002         0      0      0      0  \n",
      "CRUK0003         0      1      0      0  \n",
      "CRUK0004         0      0      0      0  \n",
      "CRUK0005         1      1      1      1  \n",
      "...            ...    ...    ...    ...  \n",
      "CRUK0864         0      0      0      0  \n",
      "CRUK0881         1      1      1      1  \n",
      "CRUK0893         0      0      1      0  \n",
      "CRUK0059         0      0      0      0  \n",
      "CRUK0525         0      0      0      0  \n",
      "\n",
      "[242 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUAD_gene_frequency_list\n",
    "LUAD_fifty_genes = [gene for gene, _ in LUAD_gene_frequency_list[:50]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUAD_filtered_mut_table = mut_table_LUAD[mut_table_LUAD['Hugo_Symbol'].isin(LUAD_fifty_genes)]\n",
    "\n",
    "# Create a binary matrix (presence or absence of genes for each patient)\n",
    "LUAD_fifty_gene_matrix = pd.crosstab(LUAD_filtered_mut_table['patient_id'], LUAD_filtered_mut_table['Hugo_Symbol']).clip(upper=1)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUAD['patient_id'])\n",
    "patients_with_genes = set(LUAD_filtered_mut_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUAD_fifty_gene_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUAD_fifty_gene_matrix = LUAD_fifty_gene_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUAD_fifty_gene_matrix.shape)\n",
    "print(LUAD_fifty_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 50)\n",
      "Hugo_Symbol  ANK2  APOB  CACNA1E  COL11A1  COL22A1  CSMD1  CSMD2  CSMD3  \\\n",
      "patient_id                                                                \n",
      "CRUK0062        1     0        1        0        1      0      1      1   \n",
      "CRUK0063        0     0        0        0        0      0      0      0   \n",
      "CRUK0064        0     1        0        0        1      0      1      0   \n",
      "CRUK0065        1     1        1        0        1      0      0      1   \n",
      "CRUK0066        0     0        1        1        1      0      0      1   \n",
      "...           ...   ...      ...      ...      ...    ...    ...    ...   \n",
      "CRUK0772        0     1        0        1        1      0      0      0   \n",
      "CRUK0781        0     0        1        1        1      0      1      1   \n",
      "CRUK0784        0     0        0        1        0      0      1      1   \n",
      "CRUK0861        1     1        0        0        0      0      0      1   \n",
      "CRUK0872        0     1        0        1        1      0      0      1   \n",
      "\n",
      "Hugo_Symbol  CTNNA2  DMD  DNAH5  DNAH6  DNAH9  DPP6  EYS  FAM135B  FSIP2  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062          1    0      1      0      0     0    1        1      1   \n",
      "CRUK0063          0    0      1      0      1     0    0        0      1   \n",
      "CRUK0064          0    0      0      1      0     1    0        0      0   \n",
      "CRUK0065          0    1      1      0      1     1    1        0      1   \n",
      "CRUK0066          0    1      1      0      0     1    1        0      0   \n",
      "...             ...  ...    ...    ...    ...   ...  ...      ...    ...   \n",
      "CRUK0772          0    0      0      0      0     0    0        1      0   \n",
      "CRUK0781          0    0      1      1      0     0    1        1      0   \n",
      "CRUK0784          0    0      0      0      0     0    1        0      1   \n",
      "CRUK0861          0    0      0      1      1     0    1        0      0   \n",
      "CRUK0872          0    0      1      0      0     0    0        0      0   \n",
      "\n",
      "Hugo_Symbol  HCN1  HMCN1  LPHN3  LRP1B  LRP2  MIR548N  MROH2B  MUC16  MUC19  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062        1      0      0      1     0        0       1      1      0   \n",
      "CRUK0063        0      1      1      1     0        0       0      0      0   \n",
      "CRUK0064        1      1      1      1     1        1       0      1      1   \n",
      "CRUK0065        1      1      0      0     1        1       0      1      1   \n",
      "CRUK0066        1      0      0      1     0        0       1      1      1   \n",
      "...           ...    ...    ...    ...   ...      ...     ...    ...    ...   \n",
      "CRUK0772        0      1      0      1     1        0       0      1      1   \n",
      "CRUK0781        0      0      1      1     1        1       0      1      0   \n",
      "CRUK0784        1      0      1      1     0        1       0      1      0   \n",
      "CRUK0861        0      0      0      0     1        0       0      1      0   \n",
      "CRUK0872        1      0      1      1     1        0       1      1      1   \n",
      "\n",
      "Hugo_Symbol  NAV3  NEB  NRXN1  OBSCN  OR4E2  PCDH15  PCLO  PKHD1  PKHD1L1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062        0    1      0      1      1       1     0      1        1   \n",
      "CRUK0063        0    0      0      1      1       0     0      0        0   \n",
      "CRUK0064        0    0      0      0      1       0     0      1        0   \n",
      "CRUK0065        1    1      0      1      0       0     0      1        0   \n",
      "CRUK0066        0    0      1      0      0       1     0      1        1   \n",
      "...           ...  ...    ...    ...    ...     ...   ...    ...      ...   \n",
      "CRUK0772        0    0      0      0      1       0     0      1        0   \n",
      "CRUK0781        0    0      0      0      0       1     0      0        1   \n",
      "CRUK0784        0    0      1      1      0       0     0      1        1   \n",
      "CRUK0861        0    1      0      0      0       0     0      0        1   \n",
      "CRUK0872        0    0      0      0      0       1     0      0        0   \n",
      "\n",
      "Hugo_Symbol  PLEC  RELN  RYR1  RYR2  RYR3  SI  SPTA1  SSPO  SYNE1  TP53  TTN  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062        1     1     0     1     1   1      0     0      1     1    1   \n",
      "CRUK0063        0     0     1     0     0   1      0     0      0     1    1   \n",
      "CRUK0064        1     1     0     1     0   0      1     0      1     1    1   \n",
      "CRUK0065        1     1     1     1     1   1      1     0      1     1    1   \n",
      "CRUK0066        0     0     1     1     0   0      1     0      1     1    1   \n",
      "...           ...   ...   ...   ...   ...  ..    ...   ...    ...   ...  ...   \n",
      "CRUK0772        1     0     0     1     0   0      0     0      0     1    1   \n",
      "CRUK0781        0     0     0     1     1   0      1     1      0     1    1   \n",
      "CRUK0784        0     1     1     1     0   1      0     1      0     1    1   \n",
      "CRUK0861        0     1     0     1     1   0      1     1      1     1    1   \n",
      "CRUK0872        0     1     0     1     0   1      0     1      0     1    1   \n",
      "\n",
      "Hugo_Symbol  UNC80  USH2A  XIRP2  ZFHX4  \n",
      "patient_id                               \n",
      "CRUK0062         1      1      0      1  \n",
      "CRUK0063         0      0      0      0  \n",
      "CRUK0064         1      0      0      1  \n",
      "CRUK0065         1      1      1      1  \n",
      "CRUK0066         0      0      1      1  \n",
      "...            ...    ...    ...    ...  \n",
      "CRUK0772         1      1      1      0  \n",
      "CRUK0781         0      1      0      1  \n",
      "CRUK0784         1      1      0      1  \n",
      "CRUK0861         0      0      0      0  \n",
      "CRUK0872         1      1      0      1  \n",
      "\n",
      "[138 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUSC_gene_frequency_list\n",
    "LUSC_fifty_genes = [gene for gene, _ in LUSC_gene_frequency_list[:50]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUSC_filtered_mut_table = mut_table_LUSC[mut_table_LUSC['Hugo_Symbol'].isin(LUSC_fifty_genes)]\n",
    "\n",
    "# Create a binary matrix (presence or absence of genes for each patient)\n",
    "LUSC_fifty_gene_matrix = pd.crosstab(LUSC_filtered_mut_table['patient_id'], LUSC_filtered_mut_table['Hugo_Symbol']).clip(upper=1)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUSC['patient_id'])\n",
    "patients_with_genes = set(LUSC_filtered_mut_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUSC_fifty_gene_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUSC_fifty_gene_matrix = LUSC_fifty_gene_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUSC_fifty_gene_matrix.shape)\n",
    "print(LUSC_fifty_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 20)\n",
      "Hugo_Symbol  CACNA1E  COL22A1  CSMD1  CSMD3  DMD  LRP1B  MUC16  MUC19  NRXN1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001           0        1      1      1    1      0      0      1      0   \n",
      "CRUK0002           0        0      0      1    0      0      1      0      0   \n",
      "CRUK0003           0        0      0      1    0      1      1      0      0   \n",
      "CRUK0004           0        0      0      1    0      1      1      0      0   \n",
      "CRUK0005           1        1      0      1    1      1      1      0      1   \n",
      "...              ...      ...    ...    ...  ...    ...    ...    ...    ...   \n",
      "CRUK0010           0        0      0      0    0      0      0      0      0   \n",
      "CRUK0038           0        0      0      0    0      0      0      0      0   \n",
      "CRUK0059           0        0      0      0    0      0      0      0      0   \n",
      "CRUK0525           0        0      0      0    0      0      0      0      0   \n",
      "CRUK0386           0        0      0      0    0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  OBSCN  OR4E2  PCDH15  RYR1  RYR2  RYR3  SPTA1  SSPO  TTN  USH2A  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         1      1       0     1     1     1      1     1    1      1   \n",
      "CRUK0002         0      0       0     0     0     1      0     0    1      0   \n",
      "CRUK0003         1      0       0     0     1     0      0     0    1      1   \n",
      "CRUK0004         1      0       0     0     1     0      1     0    0      0   \n",
      "CRUK0005         1      1       1     0     1     0      0     1    1      1   \n",
      "...            ...    ...     ...   ...   ...   ...    ...   ...  ...    ...   \n",
      "CRUK0010         0      0       0     0     0     0      0     0    0      0   \n",
      "CRUK0038         0      0       0     0     0     0      0     0    0      0   \n",
      "CRUK0059         0      0       0     0     0     0      0     0    0      0   \n",
      "CRUK0525         0      0       0     0     0     0      0     0    0      0   \n",
      "CRUK0386         0      0       0     0     0     0      0     0    0      0   \n",
      "\n",
      "Hugo_Symbol  ZFHX4  \n",
      "patient_id          \n",
      "CRUK0001         0  \n",
      "CRUK0002         0  \n",
      "CRUK0003         0  \n",
      "CRUK0004         0  \n",
      "CRUK0005         1  \n",
      "...            ...  \n",
      "CRUK0010         0  \n",
      "CRUK0038         0  \n",
      "CRUK0059         0  \n",
      "CRUK0525         0  \n",
      "CRUK0386         0  \n",
      "\n",
      "[242 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUAD_gene_frequency_list\n",
    "LUAD_twenty_genes = [gene for gene, _ in LUAD_gene_frequency_list[:20]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUAD_filtered_mut_table = mut_table_LUAD[mut_table_LUAD['Hugo_Symbol'].isin(LUAD_twenty_genes)]\n",
    "\n",
    "# Create a binary matrix (presence or absence of genes for each patient)\n",
    "LUAD_twenty_gene_matrix = pd.crosstab(LUAD_filtered_mut_table['patient_id'], LUAD_filtered_mut_table['Hugo_Symbol']).clip(upper=1)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUAD['patient_id'])\n",
    "patients_with_genes = set(LUAD_filtered_mut_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUAD_twenty_gene_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUAD_twenty_gene_matrix = LUAD_twenty_gene_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUAD_twenty_gene_matrix.shape)\n",
    "print(LUAD_twenty_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 20)\n",
      "Hugo_Symbol  COL11A1  COL22A1  CSMD3  DMD  LRP1B  MUC16  MUC19  OBSCN  OR4E2  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062           0        1      1    0      1      1      0      1      1   \n",
      "CRUK0063           0        0      0    0      1      0      0      1      1   \n",
      "CRUK0064           0        1      0    0      1      1      1      0      1   \n",
      "CRUK0065           0        1      1    1      0      1      1      1      0   \n",
      "CRUK0066           1        1      1    1      1      1      1      0      0   \n",
      "...              ...      ...    ...  ...    ...    ...    ...    ...    ...   \n",
      "CRUK0772           1        1      0    0      1      1      1      0      1   \n",
      "CRUK0781           1        1      1    0      1      1      0      0      0   \n",
      "CRUK0784           1        0      1    0      1      1      0      1      0   \n",
      "CRUK0861           0        0      1    0      0      1      0      0      0   \n",
      "CRUK0872           1        1      1    0      1      1      1      0      0   \n",
      "\n",
      "Hugo_Symbol  RELN  RYR2  RYR3  SI  SPTA1  SSPO  SYNE1  TP53  TTN  USH2A  ZFHX4  \n",
      "patient_id                                                                      \n",
      "CRUK0062        1     1     1   1      0     0      1     1    1      1      1  \n",
      "CRUK0063        0     0     0   1      0     0      0     1    1      0      0  \n",
      "CRUK0064        1     1     0   0      1     0      1     1    1      0      1  \n",
      "CRUK0065        1     1     1   1      1     0      1     1    1      1      1  \n",
      "CRUK0066        0     1     0   0      1     0      1     1    1      0      1  \n",
      "...           ...   ...   ...  ..    ...   ...    ...   ...  ...    ...    ...  \n",
      "CRUK0772        0     1     0   0      0     0      0     1    1      1      0  \n",
      "CRUK0781        0     1     1   0      1     1      0     1    1      1      1  \n",
      "CRUK0784        1     1     0   1      0     1      0     1    1      1      1  \n",
      "CRUK0861        1     1     1   0      1     1      1     1    1      0      0  \n",
      "CRUK0872        1     1     0   1      0     1      0     1    1      1      1  \n",
      "\n",
      "[138 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUSC_gene_frequency_list\n",
    "LUSC_twenty_genes = [gene for gene, _ in LUSC_gene_frequency_list[:20]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUSC_filtered_mut_table = mut_table_LUSC[mut_table_LUSC['Hugo_Symbol'].isin(LUSC_twenty_genes)]\n",
    "\n",
    "# Create a binary matrix (presence or absence of genes for each patient)\n",
    "LUSC_twenty_gene_matrix = pd.crosstab(LUSC_filtered_mut_table['patient_id'], LUSC_filtered_mut_table['Hugo_Symbol']).clip(upper=1)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUSC['patient_id'])\n",
    "patients_with_genes = set(LUSC_filtered_mut_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUSC_twenty_gene_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUSC_twenty_gene_matrix = LUSC_twenty_gene_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUSC_twenty_gene_matrix.shape)\n",
    "print(LUSC_twenty_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 50)\n",
      "Hugo_Symbol  APC  ARHGAP35  ARID1A  ARID2  ATM  ATRX  BAP1  BCLAF1  BCOR  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001       0         1       0      0    0     0     0       0     0   \n",
      "CRUK0002       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0003       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0004       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0005       0         0       0      0    0     1     0       0     0   \n",
      "...          ...       ...     ...    ...  ...   ...   ...     ...   ...   \n",
      "CRUK0817       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0864       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0881       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0893       0         0       0      0    1     0     0       0     0   \n",
      "CRUK0764       0         0       0      0    0     0     0       0     0   \n",
      "\n",
      "Hugo_Symbol  BRAF  CDKN2A  CHD8  CMTR2  COL2A1  COL5A2  CREBBP  DICER1  \\\n",
      "patient_id                                                               \n",
      "CRUK0001        0       0     0      0       0       0       0       0   \n",
      "CRUK0002        0       0     0      0       0       0       0       0   \n",
      "CRUK0003        0       0     0      0       0       0       0       0   \n",
      "CRUK0004        0       0     0      0       0       0       0       0   \n",
      "CRUK0005        1       0     0      1       0       0       0       0   \n",
      "...           ...     ...   ...    ...     ...     ...     ...     ...   \n",
      "CRUK0817        0       0     0      0       0       0       0       0   \n",
      "CRUK0864        0       0     0      0       0       0       0       0   \n",
      "CRUK0881        1       0     0      0       0       0       1       0   \n",
      "CRUK0893        0       0     0      0       0       0       0       0   \n",
      "CRUK0764        0       0     0      0       0       0       0       0   \n",
      "\n",
      "Hugo_Symbol  DNAH12  DOT1L  EGFR  EP300  FANCM  FAT1  GRIN2A  IKZF1  KDM5C  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001          0      0     0      0      0     0       0      0      0   \n",
      "CRUK0002          0      0     0      1      0     0       0      1      0   \n",
      "CRUK0003          0      0     1      0      0     0       0      0      0   \n",
      "CRUK0004          0      0     1      0      0     0       0      0      0   \n",
      "CRUK0005          0      0     0      0      0     0       0      0      0   \n",
      "...             ...    ...   ...    ...    ...   ...     ...    ...    ...   \n",
      "CRUK0817          0      0     0      1      0     0       0      0      0   \n",
      "CRUK0864          0      0     0      0      0     0       0      0      0   \n",
      "CRUK0881          1      0     0      0      0     0       0      0      0   \n",
      "CRUK0893          0      0     0      0      0     0       0      0      0   \n",
      "CRUK0764          0      0     0      0      0     0       0      0      0   \n",
      "\n",
      "Hugo_Symbol  KEAP1  KMT2C  KMT2D  KRAS  MGA  MTOR  NCOR1  NF1  PIK3CA  POLE  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      0      0     0    1     0      0    0       0     0   \n",
      "CRUK0002         0      0      0     0    0     0      0    1       0     0   \n",
      "CRUK0003         0      0      0     0    0     0      0    0       1     0   \n",
      "CRUK0004         0      0      0     0    0     0      0    0       0     0   \n",
      "CRUK0005         0      0      0     0    0     0      0    0       0     0   \n",
      "...            ...    ...    ...   ...  ...   ...    ...  ...     ...   ...   \n",
      "CRUK0817         0      0      0     0    0     0      0    1       0     0   \n",
      "CRUK0864         0      0      0     1    0     0      0    0       0     0   \n",
      "CRUK0881         0      0      0     1    0     0      0    1       0     0   \n",
      "CRUK0893         0      0      0     1    0     0      0    0       0     0   \n",
      "CRUK0764         0      0      0     0    0     0      0    0       0     0   \n",
      "\n",
      "Hugo_Symbol  POT1  PTPRB  PTPRC  RB1  RBM10  SETD2  SMAD4  SMARCA4  SPEN  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001        0      0      0    0      0      0      0        0     0   \n",
      "CRUK0002        0      0      0    1      0      0      0        0     0   \n",
      "CRUK0003        0      0      0    0      0      0      0        0     0   \n",
      "CRUK0004        0      0      0    0      0      0      1        0     0   \n",
      "CRUK0005        0      0      0    0      0      0      0        0     0   \n",
      "...           ...    ...    ...  ...    ...    ...    ...      ...   ...   \n",
      "CRUK0817        0      0      0    0      1      0      0        0     0   \n",
      "CRUK0864        0      0      0    0      1      0      0        0     0   \n",
      "CRUK0881        0      0      0    0      1      0      0        0     0   \n",
      "CRUK0893        0      0      0    0      0      0      0        0     0   \n",
      "CRUK0764        0      0      0    0      0      0      0        0     0   \n",
      "\n",
      "Hugo_Symbol  STK11  TET2  TP53  UBR5  ZFP36L1  \n",
      "patient_id                                     \n",
      "CRUK0001         0     0     1     0        0  \n",
      "CRUK0002         0     0     0     0        0  \n",
      "CRUK0003         0     0     0     0        0  \n",
      "CRUK0004         0     0     1     0        0  \n",
      "CRUK0005         0     0     1     0        0  \n",
      "...            ...   ...   ...   ...      ...  \n",
      "CRUK0817         1     0     1     0        0  \n",
      "CRUK0864         0     0     0     0        0  \n",
      "CRUK0881         0     1     1     0        0  \n",
      "CRUK0893         0     0     0     0        0  \n",
      "CRUK0764         0     0     0     0        0  \n",
      "\n",
      "[242 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUAD_driver_frequency_list\n",
    "LUAD_fifty_drivers = [gene for gene, _ in LUAD_driver_frequency_list[:50]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUAD_filtered_drivers_table = mut_table_LUAD[mut_table_LUAD['Hugo_Symbol'].isin(LUAD_fifty_drivers)]\n",
    "\n",
    "# Create a binary matrix based on both gene presence/absence and 'DriverMut' values\n",
    "LUAD_fifty_driver_matrix = pd.crosstab(LUAD_filtered_drivers_table['patient_id'], LUAD_filtered_drivers_table['Hugo_Symbol'], values=LUAD_filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUAD['patient_id'])\n",
    "patients_with_genes = set(LUAD_filtered_drivers_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUAD_fifty_driver_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUAD_fifty_driver_matrix = LUAD_fifty_driver_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUAD_fifty_driver_matrix.shape)\n",
    "print(LUAD_fifty_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 50)\n",
      "Hugo_Symbol  ACVR1B  APC  ARHGAP35  ARID1A  ARID2  ASXL1  ATR  ATRX  B2M  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062          0    0         0       0      0      0    0     1    0   \n",
      "CRUK0063          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0064          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0065          0    0         0       1      0      0    0     0    0   \n",
      "CRUK0066          0    0         0       0      0      0    0     0    0   \n",
      "...             ...  ...       ...     ...    ...    ...  ...   ...  ...   \n",
      "CRUK0772          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0781          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0784          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0861          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0872          0    0         0       0      0      0    0     0    0   \n",
      "\n",
      "Hugo_Symbol  BCLAF1  BCOR  CBLB  CDKN2A  CHD4  CHD8  COL5A2  CREBBP  CUX1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062          0     0     0       0     1     0       0       0     0   \n",
      "CRUK0063          0     0     0       1     0     0       0       0     0   \n",
      "CRUK0064          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0065          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0066          0     0     0       1     0     0       1       0     0   \n",
      "...             ...   ...   ...     ...   ...   ...     ...     ...   ...   \n",
      "CRUK0772          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0781          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0784          0     0     0       0     1     0       0       0     0   \n",
      "CRUK0861          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0872          0     0     0       0     0     0       0       0     0   \n",
      "\n",
      "Hugo_Symbol  CYLD  EP300  FANCM  FAT1  KEAP1  KMT2B  KMT2C  KMT2D  KRAS  \\\n",
      "patient_id                                                                \n",
      "CRUK0062        0      0      0     0      0      0      0      0     0   \n",
      "CRUK0063        1      1      1     0      0      0      0      0     0   \n",
      "CRUK0064        0      0      0     1      0      0      0      0     0   \n",
      "CRUK0065        0      0      0     0      0      0      0      0     0   \n",
      "CRUK0066        0      0      0     0      0      0      0      0     0   \n",
      "...           ...    ...    ...   ...    ...    ...    ...    ...   ...   \n",
      "CRUK0772        0      0      1     0      0      0      0      0     0   \n",
      "CRUK0781        0      0      0     0      0      0      0      1     0   \n",
      "CRUK0784        0      0      0     0      0      0      0      0     0   \n",
      "CRUK0861        0      0      0     0      0      0      0      0     0   \n",
      "CRUK0872        0      0      0     0      0      1      0      0     0   \n",
      "\n",
      "Hugo_Symbol  NCOA6  NCOR1  NF1  NFE2L2  NOTCH1  NOTCH2  PBRM1  PIK3CA  PLXNB2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062         0      0    0       0       0       0      0       0       1   \n",
      "CRUK0063         0      0    1       0       0       0      0       1       0   \n",
      "CRUK0064         0      0    0       0       0       0      0       0       0   \n",
      "CRUK0065         1      0    0       1       1       0      0       1       0   \n",
      "CRUK0066         0      0    0       0       1       0      0       0       0   \n",
      "...            ...    ...  ...     ...     ...     ...    ...     ...     ...   \n",
      "CRUK0772         0      0    0       1       0       0      0       0       0   \n",
      "CRUK0781         0      0    0       0       0       0      0       0       0   \n",
      "CRUK0784         0      0    0       0       0       0      0       0       0   \n",
      "CRUK0861         0      0    0       0       0       0      0       0       0   \n",
      "CRUK0872         0      0    0       0       0       0      0       0       0   \n",
      "\n",
      "Hugo_Symbol  PTEN  PTPRB  PTPRC  RASA1  RB1  SETD2  SETDB1  SMAD4  SMARCA4  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0063        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0064        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0065        0      0      1      0    0      0       0      0        0   \n",
      "CRUK0066        0      0      0      0    0      0       0      0        0   \n",
      "...           ...    ...    ...    ...  ...    ...     ...    ...      ...   \n",
      "CRUK0772        0      0      0      0    1      0       0      0        0   \n",
      "CRUK0781        1      0      0      0    0      0       0      0        0   \n",
      "CRUK0784        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0861        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0872        0      1      0      0    0      0       0      0        0   \n",
      "\n",
      "Hugo_Symbol  SPEN  STK11  TET2  TP53  UBR5  \n",
      "patient_id                                  \n",
      "CRUK0062        0      0     0     1     1  \n",
      "CRUK0063        0      0     0     1     0  \n",
      "CRUK0064        0      0     0     1     0  \n",
      "CRUK0065        0      0     0     1     1  \n",
      "CRUK0066        0      0     0     1     0  \n",
      "...           ...    ...   ...   ...   ...  \n",
      "CRUK0772        0      0     0     1     0  \n",
      "CRUK0781        0      0     0     1     0  \n",
      "CRUK0784        0      0     0     1     0  \n",
      "CRUK0861        0      0     0     1     0  \n",
      "CRUK0872        0      0     0     1     0  \n",
      "\n",
      "[138 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUSC_driver_frequency_list\n",
    "LUSC_fifty_drivers = [gene for gene, _ in LUSC_driver_frequency_list[:50]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUSC_filtered_drivers_table = mut_table_LUSC[mut_table_LUSC['Hugo_Symbol'].isin(LUSC_fifty_drivers)]\n",
    "\n",
    "# Create a binary matrix based on both gene presence/absence and 'DriverMut' values\n",
    "LUSC_fifty_driver_matrix = pd.crosstab(LUSC_filtered_drivers_table['patient_id'], LUSC_filtered_drivers_table['Hugo_Symbol'], values=LUSC_filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUSC['patient_id'])\n",
    "patients_with_genes = set(LUSC_filtered_drivers_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUSC_fifty_driver_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUSC_fifty_driver_matrix = LUSC_fifty_driver_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUSC_fifty_driver_matrix.shape)\n",
    "print(LUSC_fifty_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 20)\n",
      "Hugo_Symbol  ARID2  ATM  ATRX  CDKN2A  EGFR  FAT1  GRIN2A  KEAP1  KMT2C  \\\n",
      "patient_id                                                                \n",
      "CRUK0001         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0002         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0003         0    0     0       0     1     0       0      0      0   \n",
      "CRUK0004         0    0     0       0     1     0       0      0      0   \n",
      "CRUK0005         0    0     1       0     0     0       0      0      0   \n",
      "...            ...  ...   ...     ...   ...   ...     ...    ...    ...   \n",
      "CRUK0045         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0041         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0742         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0764         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0043         0    0     0       0     0     0       0      0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  KRAS  MGA  NCOR1  NF1  PIK3CA  PTPRB  RBM10  SMARCA4  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001         0     0    1      0    0       0      0      0        0   \n",
      "CRUK0002         0     0    0      0    1       0      0      0        0   \n",
      "CRUK0003         0     0    0      0    0       1      0      0        0   \n",
      "CRUK0004         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0005         0     0    0      0    0       0      0      0        0   \n",
      "...            ...   ...  ...    ...  ...     ...    ...    ...      ...   \n",
      "CRUK0045         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0041         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0742         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0764         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0043         0     0    0      0    0       0      0      0        0   \n",
      "\n",
      "Hugo_Symbol  STK11  TP53  \n",
      "patient_id                \n",
      "CRUK0001         0     1  \n",
      "CRUK0002         0     0  \n",
      "CRUK0003         0     0  \n",
      "CRUK0004         0     1  \n",
      "CRUK0005         0     1  \n",
      "...            ...   ...  \n",
      "CRUK0045         0     0  \n",
      "CRUK0041         0     0  \n",
      "CRUK0742         0     0  \n",
      "CRUK0764         0     0  \n",
      "CRUK0043         0     0  \n",
      "\n",
      "[242 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUAD_driver_frequency_list\n",
    "LUAD_twenty_drivers = [gene for gene, _ in LUAD_driver_frequency_list[:20]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUAD_filtered_drivers_table = mut_table_LUAD[mut_table_LUAD['Hugo_Symbol'].isin(LUAD_twenty_drivers)]\n",
    "\n",
    "# Create a binary matrix based on both gene presence/absence and 'DriverMut' values\n",
    "LUAD_twenty_driver_matrix = pd.crosstab(LUAD_filtered_drivers_table['patient_id'], LUAD_filtered_drivers_table['Hugo_Symbol'], values=LUAD_filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUAD['patient_id'])\n",
    "patients_with_genes = set(LUAD_filtered_drivers_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUAD_twenty_driver_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUAD_twenty_driver_matrix = LUAD_twenty_driver_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUAD_twenty_driver_matrix.shape)\n",
    "print(LUAD_twenty_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 20)\n",
      "Hugo_Symbol  ARID1A  ARID2  CDKN2A  COL5A2  CREBBP  CUX1  FAT1  KEAP1  KMT2C  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0063          0      0       1       0       0     0     0      0      0   \n",
      "CRUK0064          0      0       0       0       0     0     1      0      0   \n",
      "CRUK0065          1      0       0       0       0     0     0      0      0   \n",
      "CRUK0066          0      0       1       1       0     0     0      0      0   \n",
      "...             ...    ...     ...     ...     ...   ...   ...    ...    ...   \n",
      "CRUK0784          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0861          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0872          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0331          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0118          0      0       0       0       0     0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  NF1  NFE2L2  NOTCH1  PIK3CA  PLXNB2  PTEN  RASA1  SETD2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062         0    0       0       0       0       1     0      0      0   \n",
      "CRUK0063         0    1       0       0       1       0     0      0      0   \n",
      "CRUK0064         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0065         0    0       1       1       1       0     0      0      0   \n",
      "CRUK0066         0    0       0       1       0       0     0      0      0   \n",
      "...            ...  ...     ...     ...     ...     ...   ...    ...    ...   \n",
      "CRUK0784         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0861         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0872         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0331         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0118         0    0       0       0       0       0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  TP53  UBR5  \n",
      "patient_id               \n",
      "CRUK0062        1     1  \n",
      "CRUK0063        1     0  \n",
      "CRUK0064        1     0  \n",
      "CRUK0065        1     1  \n",
      "CRUK0066        1     0  \n",
      "...           ...   ...  \n",
      "CRUK0784        1     0  \n",
      "CRUK0861        1     0  \n",
      "CRUK0872        1     0  \n",
      "CRUK0331        0     0  \n",
      "CRUK0118        0     0  \n",
      "\n",
      "[138 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUSC_driver_frequency_list\n",
    "LUSC_twenty_drivers = [gene for gene, _ in LUSC_driver_frequency_list[:20]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUSC_filtered_drivers_table = mut_table_LUSC[mut_table_LUSC['Hugo_Symbol'].isin(LUSC_twenty_drivers)]\n",
    "\n",
    "# Create a binary matrix based on both gene presence/absence and 'DriverMut' values\n",
    "LUSC_twenty_driver_matrix = pd.crosstab(LUSC_filtered_drivers_table['patient_id'], LUSC_filtered_drivers_table['Hugo_Symbol'], values=LUSC_filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUSC['patient_id'])\n",
    "patients_with_genes = set(LUSC_filtered_drivers_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUSC_twenty_driver_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUSC_twenty_driver_matrix = LUSC_twenty_driver_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUSC_twenty_driver_matrix.shape)\n",
    "print(LUSC_twenty_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 50) (73, 50) (169,) (73,)\n",
      "Best parameters:  {'C': 1}\n",
      "Best score:  0.5081752305665349\n",
      "Predictions:  [0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 0 1 0 1\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 1 0 0 0 1 0 0 0 0 0 1 0 1 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4830659536541889\n",
      "ROC AUC:  0.4973262032085562\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.69      0.78      0.73        51\n",
      "   Recurrence       0.27      0.18      0.22        22\n",
      "\n",
      "     accuracy                           0.60        73\n",
      "    macro avg       0.48      0.48      0.48        73\n",
      " weighted avg       0.56      0.60      0.58        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 50 most commonly mutated genes\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_fifty_gene_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 50) (42, 50) (96,) (42,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'C': 10}\n",
      "Best score:  0.5452380952380952\n",
      "Predictions:  [0 0 0 0 0 0 0 1 0 1 0 1 0 1 1 0 0 0 0 0 1 0 1 0 0 1 0 0 1 1 0 0 0 0 0 0 1\n",
      " 1 1 1 0 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.4481481481481482\n",
      "ROC AUC:  0.454320987654321\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.61      0.63      0.62        27\n",
      "   Recurrence       0.29      0.27      0.28        15\n",
      "\n",
      "     accuracy                           0.50        42\n",
      "    macro avg       0.45      0.45      0.45        42\n",
      " weighted avg       0.49      0.50      0.50        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 50 most commonly mutated genes\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_fifty_gene_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 20) (73, 20) (169,) (73,)\n",
      "Best parameters:  {'C': 1e-05}\n",
      "Best score:  0.5\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.5155971479500892\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.70      1.00      0.82        51\n",
      "   Recurrence       0.00      0.00      0.00        22\n",
      "\n",
      "     accuracy                           0.70        73\n",
      "    macro avg       0.35      0.50      0.41        73\n",
      " weighted avg       0.49      0.70      0.57        73\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 20 most commonly mutated genes\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_twenty_gene_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 20) (42, 20) (96,) (42,)\n",
      "Best parameters:  {'C': 10}\n",
      "Best score:  0.5606227106227106\n",
      "Predictions:  [0 0 0 0 0 0 0 1 1 0 0 1 0 1 0 0 1 0 0 0 1 0 1 1 0 0 1 0 1 1 0 0 0 0 0 0 1\n",
      " 1 0 1 0 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.4481481481481482\n",
      "ROC AUC:  0.5209876543209877\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.61      0.63      0.62        27\n",
      "   Recurrence       0.29      0.27      0.28        15\n",
      "\n",
      "     accuracy                           0.50        42\n",
      "    macro avg       0.45      0.45      0.45        42\n",
      " weighted avg       0.49      0.50      0.50        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 20 most commonly mutated genes\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_twenty_gene_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 50) (73, 50) (169,) (73,)\n",
      "Best parameters:  {'C': 100}\n",
      "Best score:  0.5366040843214757\n",
      "Predictions:  [0 0 1 0 1 0 1 1 0 0 0 1 0 0 0 0 1 0 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1\n",
      " 0 0 1 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 1 0 0 0 0 1 0 0 1 0 0 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4407308377896613\n",
      "ROC AUC:  0.4563279857397504\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.75      0.70        51\n",
      "   Recurrence       0.19      0.14      0.16        22\n",
      "\n",
      "     accuracy                           0.56        73\n",
      "    macro avg       0.43      0.44      0.43        73\n",
      " weighted avg       0.52      0.56      0.54        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 50 most common driver mutations\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_fifty_driver_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 50) (42, 50) (96,) (42,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'C': 100}\n",
      "Best score:  0.5171245421245422\n",
      "Predictions:  [0 0 0 0 1 1 0 0 0 1 0 0 1 0 0 1 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 1\n",
      " 0 0 1 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.3814814814814815\n",
      "ROC AUC:  0.26419753086419756\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.57      0.63      0.60        27\n",
      "   Recurrence       0.17      0.13      0.15        15\n",
      "\n",
      "     accuracy                           0.45        42\n",
      "    macro avg       0.37      0.38      0.37        42\n",
      " weighted avg       0.42      0.45      0.44        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 50 most common driver mutations\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_fifty_driver_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 20) (73, 20) (169,) (73,)\n",
      "Best parameters:  {'C': 1e-05}\n",
      "Best score:  0.5\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.5423351158645275\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.70      1.00      0.82        51\n",
      "   Recurrence       0.00      0.00      0.00        22\n",
      "\n",
      "     accuracy                           0.70        73\n",
      "    macro avg       0.35      0.50      0.41        73\n",
      " weighted avg       0.49      0.70      0.57        73\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 20 most common driver mutations\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_twenty_driver_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 20) (42, 20) (96,) (42,)\n",
      "Best parameters:  {'C': 1}\n",
      "Best score:  0.552014652014652\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 0 0 0 1 0\n",
      " 0 0 0 0 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.562962962962963\n",
      "ROC AUC:  0.6567901234567901\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.68      0.93      0.78        27\n",
      "   Recurrence       0.60      0.20      0.30        15\n",
      "\n",
      "     accuracy                           0.67        42\n",
      "    macro avg       0.64      0.56      0.54        42\n",
      " weighted avg       0.65      0.67      0.61        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 20 most common driver mutations\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_twenty_driver_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 50) (73, 50) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.5852239789196311\n",
      "Predictions:  [1 0 0 0 0 0 0 0 1 0 1 0 1 1 0 0 0 0 0 0 0 1 0 1 1 0 0 0 1 1 1 1 0 0 1 0 1\n",
      " 0 0 0 0 1 0 1 1 0 1 1 0 0 0 0 0 0 1 1 0 0 1 1 0 0 1 0 0 0 0 0 1 0 1 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4728163992869875\n",
      "ROC AUC:  0.49554367201426025\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.68      0.63      0.65        51\n",
      "   Recurrence       0.27      0.32      0.29        22\n",
      "\n",
      "     accuracy                           0.53        73\n",
      "    macro avg       0.48      0.47      0.47        73\n",
      " weighted avg       0.56      0.53      0.54        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_fifty_gene_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 50) (42, 50) (96,) (42,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.6064102564102564\n",
      "Predictions:  [0 0 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 1 0 1 0 1 0 0 1 0 0 1 1 0 0 0 0 0 0 1\n",
      " 1 0 1 0 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.43333333333333335\n",
      "ROC AUC:  0.38518518518518524\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.60      0.67      0.63        27\n",
      "   Recurrence       0.25      0.20      0.22        15\n",
      "\n",
      "     accuracy                           0.50        42\n",
      "    macro avg       0.42      0.43      0.43        42\n",
      " weighted avg       0.47      0.50      0.49        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_fifty_gene_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 20) (73, 20) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.0001, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.5384848484848485\n",
      "Predictions:  [0 0 0 1 0 0 0 0 0 1 0 1 1 0 1 1 1 0 0 0 1 0 0 0 1 0 0 1 0 0 0 1 1 0 0 1 1\n",
      " 0 1 0 0 0 0 0 1 0 0 1 1 0 1 1 0 0 1 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 1 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4500891265597148\n",
      "ROC AUC:  0.4696969696969697\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.63      0.65        51\n",
      "   Recurrence       0.24      0.27      0.26        22\n",
      "\n",
      "     accuracy                           0.52        73\n",
      "    macro avg       0.45      0.45      0.45        73\n",
      " weighted avg       0.54      0.52      0.53        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_twenty_gene_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 20) (42, 20) (96,) (42,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.5858058608058607\n",
      "Predictions:  [1 1 0 0 1 1 0 1 1 0 1 1 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.4481481481481482\n",
      "ROC AUC:  0.5111111111111112\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.61      0.63      0.62        27\n",
      "   Recurrence       0.29      0.27      0.28        15\n",
      "\n",
      "     accuracy                           0.50        42\n",
      "    macro avg       0.45      0.45      0.45        42\n",
      " weighted avg       0.49      0.50      0.50        42\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_twenty_gene_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 100) (73, 100) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.571024374176548\n",
      "Predictions:  [0 0 1 1 0 0 1 0 0 0 0 1 1 1 0 0 1 0 1 1 0 1 0 1 1 0 0 1 0 0 0 0 0 0 0 0 0\n",
      " 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 0 1 0 0 1 0 0 0 1 1 1]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.3819073083778966\n",
      "ROC AUC:  0.4028520499108735\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.63      0.63      0.63        51\n",
      "   Recurrence       0.14      0.14      0.14        22\n",
      "\n",
      "     accuracy                           0.48        73\n",
      "    macro avg       0.38      0.38      0.38        73\n",
      " weighted avg       0.48      0.48      0.48        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_driver_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 100) (42, 100) (96,) (42,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.0001, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.5460622710622711\n",
      "Predictions:  [0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.48148148148148145\n",
      "ROC AUC:  0.4814814814814815\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.63      0.96      0.76        27\n",
      "   Recurrence       0.00      0.00      0.00        15\n",
      "\n",
      "     accuracy                           0.62        42\n",
      "    macro avg       0.32      0.48      0.38        42\n",
      " weighted avg       0.41      0.62      0.49        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_driver_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 50) (73, 50) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.586709486166008\n",
      "Predictions:  [0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 1 1 0 1 0 0 1 1 0 0 1 0 0 0 0 0 0 0 1\n",
      " 0 0 1 1 1 0 0 0 0 0 0 0 1 0 1 0 0 0 1 1 0 1 0 1 0 0 1 1 1 1 0 0 0 0 0 1]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.3850267379679144\n",
      "ROC AUC:  0.3306595365418895\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.62      0.59      0.61        51\n",
      "   Recurrence       0.16      0.18      0.17        22\n",
      "\n",
      "     accuracy                           0.47        73\n",
      "    macro avg       0.39      0.39      0.39        73\n",
      " weighted avg       0.48      0.47      0.47        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_fifty_driver_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 50) (42, 50) (96,) (42,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.1, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.559981684981685\n",
      "Predictions:  [0 0 0 0 0 1 0 0 1 0 0 0 1 0 0 1 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 0\n",
      " 0 0 1 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.3851851851851852\n",
      "ROC AUC:  0.43950617283950616\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.58      0.70      0.63        27\n",
      "   Recurrence       0.11      0.07      0.08        15\n",
      "\n",
      "     accuracy                           0.48        42\n",
      "    macro avg       0.34      0.39      0.36        42\n",
      " weighted avg       0.41      0.48      0.44        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_fifty_driver_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 20) (73, 20) (169,) (73,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 10, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.5\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.5\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.70      1.00      0.82        51\n",
      "   Recurrence       0.00      0.00      0.00        22\n",
      "\n",
      "     accuracy                           0.70        73\n",
      "    macro avg       0.35      0.50      0.41        73\n",
      " weighted avg       0.49      0.70      0.57        73\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_twenty_driver_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 20) (42, 20) (96,) (42,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.6088827838827839\n",
      "Predictions:  [1 0 0 0 1 1 0 0 0 0 0 1 1 1 1 0 0 1 1 0 0 1 1 1 1 1 0 1 1 0 0 0 0 0 0 1 0\n",
      " 0 0 1 0 0]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.5296296296296297\n",
      "ROC AUC:  0.5629629629629629\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.59      0.63        27\n",
      "   Recurrence       0.39      0.47      0.42        15\n",
      "\n",
      "     accuracy                           0.55        42\n",
      "    macro avg       0.53      0.53      0.53        42\n",
      " weighted avg       0.57      0.55      0.55        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_twenty_driver_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(242, 10)\n",
      "Hugo_Symbol  EGFR  FAT1  KEAP1  KRAS  MGA  NF1  RBM10  SMARCA4  STK11  TP53\n",
      "patient_id                                                                 \n",
      "CRUK0001        0     0      0     0    1    0      0        0      0     1\n",
      "CRUK0002        0     0      0     0    0    1      0        0      0     0\n",
      "CRUK0003        1     0      0     0    0    0      0        0      0     0\n",
      "CRUK0004        1     0      0     0    0    0      0        0      0     1\n",
      "CRUK0005        0     0      0     0    0    0      0        0      0     1\n",
      "...           ...   ...    ...   ...  ...  ...    ...      ...    ...   ...\n",
      "CRUK0764        0     0      0     0    0    0      0        0      0     0\n",
      "CRUK0193        0     0      0     0    0    0      0        0      0     0\n",
      "CRUK0043        0     0      0     0    0    0      0        0      0     0\n",
      "CRUK0730        0     0      0     0    0    0      0        0      0     0\n",
      "CRUK0253        0     0      0     0    0    0      0        0      0     0\n",
      "\n",
      "[242 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUAD_driver_frequency_list\n",
    "LUAD_ten_drivers = [gene for gene, _ in LUAD_driver_frequency_list[:10]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUAD_filtered_drivers_table = mut_table_LUAD[mut_table_LUAD['Hugo_Symbol'].isin(LUAD_ten_drivers)]\n",
    "\n",
    "# Create a binary matrix based on both gene presence/absence and 'DriverMut' values\n",
    "LUAD_ten_driver_matrix = pd.crosstab(LUAD_filtered_drivers_table['patient_id'], LUAD_filtered_drivers_table['Hugo_Symbol'], values=LUAD_filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUAD['patient_id'])\n",
    "patients_with_genes = set(LUAD_filtered_drivers_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUAD_ten_driver_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUAD_ten_driver_matrix = LUAD_ten_driver_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUAD_ten_driver_matrix.shape)\n",
    "print(LUAD_ten_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(138, 10)\n",
      "Hugo_Symbol  CDKN2A  COL5A2  CUX1  FAT1  KEAP1  KMT2D  NFE2L2  PIK3CA  PTEN  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0063          1       0     0     0      0      0       0       1     0   \n",
      "CRUK0064          0       0     0     1      0      0       0       0     0   \n",
      "CRUK0065          0       0     0     0      0      0       1       1     0   \n",
      "CRUK0066          1       1     0     0      0      0       0       0     0   \n",
      "...             ...     ...   ...   ...    ...    ...     ...     ...   ...   \n",
      "CRUK0861          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0872          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0331          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0118          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0084          0       0     0     0      0      0       0       0     0   \n",
      "\n",
      "Hugo_Symbol  TP53  \n",
      "patient_id         \n",
      "CRUK0062        1  \n",
      "CRUK0063        1  \n",
      "CRUK0064        1  \n",
      "CRUK0065        1  \n",
      "CRUK0066        1  \n",
      "...           ...  \n",
      "CRUK0861        1  \n",
      "CRUK0872        1  \n",
      "CRUK0331        0  \n",
      "CRUK0118        0  \n",
      "CRUK0084        0  \n",
      "\n",
      "[138 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "# Extract the gene names from the LUSC_driver_frequency_list\n",
    "LUSC_ten_drivers = [gene for gene, _ in LUSC_driver_frequency_list[:10]]\n",
    "\n",
    "# Filter the original DataFrame based on the selected genes\n",
    "LUSC_filtered_drivers_table = mut_table_LUSC[mut_table_LUSC['Hugo_Symbol'].isin(LUSC_ten_drivers)]\n",
    "\n",
    "# Create a binary matrix based on both gene presence/absence and 'DriverMut' values\n",
    "LUSC_ten_driver_matrix = pd.crosstab(LUSC_filtered_drivers_table['patient_id'], LUSC_filtered_drivers_table['Hugo_Symbol'], values=LUSC_filtered_drivers_table['DriverMut'].astype(int), aggfunc='max').clip(upper=1).fillna(0)\n",
    "\n",
    "# Identify missing patients\n",
    "all_patients = set(mut_table_LUSC['patient_id'])\n",
    "patients_with_genes = set(LUSC_filtered_drivers_table['patient_id'])\n",
    "missing_patients = all_patients - patients_with_genes\n",
    "\n",
    "# Add missing patients to the binary matrix with all 0s\n",
    "for missing_patient in missing_patients:\n",
    "    LUSC_ten_driver_matrix.loc[missing_patient, :] = 0\n",
    "\n",
    "# Explicitly set the data type of the entire DataFrame to integers\n",
    "LUSC_ten_driver_matrix = LUSC_ten_driver_matrix.astype(int)\n",
    "\n",
    "# Display the resulting gene matrix\n",
    "print(LUSC_ten_driver_matrix.shape)\n",
    "print(LUSC_ten_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 10) (73, 10) (169,) (73,)\n",
      "Best parameters:  {'C': 10}\n",
      "Best score:  0.5026383399209486\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.49019607843137253\n",
      "ROC AUC:  0.44340463458110513\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.69      0.98      0.81        51\n",
      "   Recurrence       0.00      0.00      0.00        22\n",
      "\n",
      "     accuracy                           0.68        73\n",
      "    macro avg       0.35      0.49      0.41        73\n",
      " weighted avg       0.49      0.68      0.57        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 10 most common driver mutations\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_ten_driver_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 10) (42, 10) (96,) (42,)\n",
      "Best parameters:  {'C': 1}\n",
      "Best score:  0.5085164835164836\n",
      "Predictions:  [1 0 0 1 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.5259259259259259\n",
      "ROC AUC:  0.5061728395061729\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.66      0.85      0.74        27\n",
      "   Recurrence       0.43      0.20      0.27        15\n",
      "\n",
      "     accuracy                           0.62        42\n",
      "    macro avg       0.54      0.53      0.51        42\n",
      " weighted avg       0.58      0.62      0.57        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# predicting recurrence with absence/presence of 10 most common driver mutations\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_ten_driver_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(169, 10) (73, 10) (169,) (73,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5528722002635046\n",
      "Predictions:  [1 0 0 0 0 1 0 0 0 1 0 0 0 0 1 1 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 1 0 1 1 1 0\n",
      " 0 0 0 1 1 0 0 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0 0]\n",
      "Actual:  [0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 0 1\n",
      " 1 0 0 0 0 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 1 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5187165775401069\n",
      "ROC AUC:  0.44429590017825316\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.71      0.76      0.74        51\n",
      "   Recurrence       0.33      0.27      0.30        22\n",
      "\n",
      "     accuracy                           0.62        73\n",
      "    macro avg       0.52      0.52      0.52        73\n",
      " weighted avg       0.60      0.62      0.60        73\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUAD_ten_driver_matrix, y_LUAD, test_size = 0.3, random_state = 0, stratify = y_LUAD)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96, 10) (42, 10) (96,) (42,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5379120879120879\n",
      "Predictions:  [1 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 1 0 0 0 1 0 0 0\n",
      " 1 0 0 0 1]\n",
      "Actual:  [1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 1 1 0 0 0 1 1 0 0 1 1 0 1 0 0 1 0 0 1 1 0\n",
      " 0 1 0 1 0]\n",
      "Balanced Accuracy:  0.4703703703703703\n",
      "ROC AUC:  0.4098765432098766\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.62      0.74      0.68        27\n",
      "   Recurrence       0.30      0.20      0.24        15\n",
      "\n",
      "     accuracy                           0.55        42\n",
      "    macro avg       0.46      0.47      0.46        42\n",
      " weighted avg       0.51      0.55      0.52        42\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm - layer sizes are optimized\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(LUSC_ten_driver_matrix, y_LUSC, test_size = 0.3, random_state = 0, stratify = y_LUSC)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_matrix_w_hist = gene_matrix.copy()\n",
    "gene_matrix_w_hist['LUAD']= gene_matrix_w_hist.index.isin(LUAD_patient_id).astype(int)\n",
    "gene_matrix_w_hist['LUSC']= gene_matrix_w_hist.index.isin(LUSC_patient_id).astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ABCA13</th>\n",
       "      <th>ADAMTS12</th>\n",
       "      <th>ADAMTS16</th>\n",
       "      <th>ADAMTS20</th>\n",
       "      <th>AFF2</th>\n",
       "      <th>ANK2</th>\n",
       "      <th>APOB</th>\n",
       "      <th>ASTN1</th>\n",
       "      <th>BAI3</th>\n",
       "      <th>CACNA1B</th>\n",
       "      <th>CACNA1C</th>\n",
       "      <th>CACNA1E</th>\n",
       "      <th>CCDC168</th>\n",
       "      <th>CDH10</th>\n",
       "      <th>CDH23</th>\n",
       "      <th>CNTNAP2</th>\n",
       "      <th>CNTNAP5</th>\n",
       "      <th>COL11A1</th>\n",
       "      <th>COL22A1</th>\n",
       "      <th>CPS1</th>\n",
       "      <th>CSMD1</th>\n",
       "      <th>CSMD2</th>\n",
       "      <th>CSMD3</th>\n",
       "      <th>CTNNA2</th>\n",
       "      <th>CTNND2</th>\n",
       "      <th>DMD</th>\n",
       "      <th>DNAH11</th>\n",
       "      <th>DNAH5</th>\n",
       "      <th>DNAH6</th>\n",
       "      <th>DNAH9</th>\n",
       "      <th>DOCK2</th>\n",
       "      <th>DPP6</th>\n",
       "      <th>DYSF</th>\n",
       "      <th>ELK2AP</th>\n",
       "      <th>EYS</th>\n",
       "      <th>FAM135B</th>\n",
       "      <th>FAT3</th>\n",
       "      <th>FAT4</th>\n",
       "      <th>FBN2</th>\n",
       "      <th>FPGT-TNNI3K</th>\n",
       "      <th>FSIP2</th>\n",
       "      <th>HMCN1</th>\n",
       "      <th>KIF1A</th>\n",
       "      <th>LAMA1</th>\n",
       "      <th>LAMA2</th>\n",
       "      <th>LOC93432</th>\n",
       "      <th>LPHN3</th>\n",
       "      <th>LRP1B</th>\n",
       "      <th>LRP2</th>\n",
       "      <th>LRRC7</th>\n",
       "      <th>MIR548N</th>\n",
       "      <th>MROH2B</th>\n",
       "      <th>MUC16</th>\n",
       "      <th>MUC19</th>\n",
       "      <th>MUC2</th>\n",
       "      <th>MYO18B</th>\n",
       "      <th>MYO7B</th>\n",
       "      <th>NAV3</th>\n",
       "      <th>NCS1</th>\n",
       "      <th>NEB</th>\n",
       "      <th>NRXN1</th>\n",
       "      <th>OBSCN</th>\n",
       "      <th>OR4E2</th>\n",
       "      <th>OTOG</th>\n",
       "      <th>OTOGL</th>\n",
       "      <th>PAPPA2</th>\n",
       "      <th>PCDH15</th>\n",
       "      <th>PCLO</th>\n",
       "      <th>PIEZO2</th>\n",
       "      <th>PKHD1</th>\n",
       "      <th>PKHD1L1</th>\n",
       "      <th>PLEC</th>\n",
       "      <th>PLXNA4</th>\n",
       "      <th>PTPRD</th>\n",
       "      <th>PTPRT</th>\n",
       "      <th>PXDNL</th>\n",
       "      <th>RELN</th>\n",
       "      <th>RIMS2</th>\n",
       "      <th>RYR1</th>\n",
       "      <th>RYR2</th>\n",
       "      <th>RYR3</th>\n",
       "      <th>SI</th>\n",
       "      <th>SORCS1</th>\n",
       "      <th>SORCS3</th>\n",
       "      <th>SPEF2</th>\n",
       "      <th>SPTA1</th>\n",
       "      <th>SSPO</th>\n",
       "      <th>SYNE1</th>\n",
       "      <th>TNR</th>\n",
       "      <th>TP53</th>\n",
       "      <th>TRY2P</th>\n",
       "      <th>TTC40</th>\n",
       "      <th>TTN</th>\n",
       "      <th>UNC79</th>\n",
       "      <th>UNC80</th>\n",
       "      <th>USH2A</th>\n",
       "      <th>WDFY4</th>\n",
       "      <th>XIRP2</th>\n",
       "      <th>ZFHX4</th>\n",
       "      <th>ZNF536</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CRUK0001</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0004</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0005</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0861</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0864</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0872</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0881</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0893</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>421 rows × 100 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            ABCA13  ADAMTS12  ADAMTS16  ADAMTS20  AFF2  ANK2  APOB  ASTN1  \\\n",
       "patient_id                                                                  \n",
       "CRUK0001         1         1         0         1     1     1     1      0   \n",
       "CRUK0002         0         0         0         1     0     0     0      1   \n",
       "CRUK0003         0         0         1         0     0     0     0      0   \n",
       "CRUK0004         1         0         0         0     0     0     0      1   \n",
       "CRUK0005         1         1         0         1     1     1     1      0   \n",
       "...            ...       ...       ...       ...   ...   ...   ...    ...   \n",
       "CRUK0861         0         0         0         0     1     1     1      0   \n",
       "CRUK0864         0         0         0         0     0     0     0      0   \n",
       "CRUK0872         0         0         0         0     0     0     1      1   \n",
       "CRUK0881         0         1         1         1     0     1     1      1   \n",
       "CRUK0893         0         0         0         0     0     1     1      0   \n",
       "\n",
       "            BAI3  CACNA1B  CACNA1C  CACNA1E  CCDC168  CDH10  CDH23  CNTNAP2  \\\n",
       "patient_id                                                                    \n",
       "CRUK0001       1        0        1        0        0      0      0        0   \n",
       "CRUK0002       0        0        0        0        0      0      0        0   \n",
       "CRUK0003       0        1        0        0        0      0      0        1   \n",
       "CRUK0004       0        0        0        0        1      0      0        1   \n",
       "CRUK0005       0        0        0        1        1      1      0        1   \n",
       "...          ...      ...      ...      ...      ...    ...    ...      ...   \n",
       "CRUK0861       1        0        0        0        0      0      0        1   \n",
       "CRUK0864       0        0        0        0        1      0      0        0   \n",
       "CRUK0872       0        0        0        0        0      0      1        0   \n",
       "CRUK0881       0        0        1        1        1      1      0        1   \n",
       "CRUK0893       0        0        0        1        0      0      0        1   \n",
       "\n",
       "            CNTNAP5  COL11A1  COL22A1  CPS1  CSMD1  CSMD2  CSMD3  CTNNA2  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001          1        0        1     0      1      1      1       1   \n",
       "CRUK0002          0        0        0     0      0      1      1       0   \n",
       "CRUK0003          0        0        0     0      0      0      1       0   \n",
       "CRUK0004          0        0        0     0      0      0      1       0   \n",
       "CRUK0005          0        1        1     0      0      1      1       0   \n",
       "...             ...      ...      ...   ...    ...    ...    ...     ...   \n",
       "CRUK0861          1        0        0     0      0      0      1       0   \n",
       "CRUK0864          0        0        0     0      0      0      1       0   \n",
       "CRUK0872          0        1        1     0      0      0      1       0   \n",
       "CRUK0881          1        1        0     1      1      1      1       1   \n",
       "CRUK0893          1        0        0     0      0      0      0       0   \n",
       "\n",
       "            CTNND2  DMD  DNAH11  DNAH5  DNAH6  DNAH9  DOCK2  DPP6  DYSF  \\\n",
       "patient_id                                                                \n",
       "CRUK0001         0    1       0      1      0      0      1     0     1   \n",
       "CRUK0002         1    0       0      1      1      1      1     0     0   \n",
       "CRUK0003         0    0       0      0      0      0      1     0     0   \n",
       "CRUK0004         0    0       0      0      0      0      0     0     0   \n",
       "CRUK0005         1    1       0      1      0      1      1     0     1   \n",
       "...            ...  ...     ...    ...    ...    ...    ...   ...   ...   \n",
       "CRUK0861         0    0       0      0      1      1      0     0     0   \n",
       "CRUK0864         0    1       0      0      0      1      1     1     0   \n",
       "CRUK0872         0    0       0      1      0      0      1     0     0   \n",
       "CRUK0881         1    1       1      1      1      1      1     1     0   \n",
       "CRUK0893         0    0       1      0      0      0      0     0     0   \n",
       "\n",
       "            ELK2AP  EYS  FAM135B  FAT3  FAT4  FBN2  FPGT-TNNI3K  FSIP2  HMCN1  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001         0    1        1     0     1     1            1      1      1   \n",
       "CRUK0002         0    1        0     0     0     0            0      0      0   \n",
       "CRUK0003         1    0        0     0     0     0            0      0      0   \n",
       "CRUK0004         0    0        1     0     0     0            1      0      0   \n",
       "CRUK0005         1    1        1     1     1     1            1      1      1   \n",
       "...            ...  ...      ...   ...   ...   ...          ...    ...    ...   \n",
       "CRUK0861         0    1        0     0     0     1            0      0      0   \n",
       "CRUK0864         0    0        0     0     0     1            0      1      0   \n",
       "CRUK0872         0    0        0     0     0     0            1      0      0   \n",
       "CRUK0881         0    1        1     1     0     1            1      1      1   \n",
       "CRUK0893         0    1        0     0     0     0            0      1      0   \n",
       "\n",
       "            KIF1A  LAMA1  LAMA2  LOC93432  LPHN3  LRP1B  LRP2  LRRC7  MIR548N  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001        1      1      1         0      0      0     1      1        0   \n",
       "CRUK0002        0      0      0         0      0      0     0      0        0   \n",
       "CRUK0003        0      0      0         1      0      1     1      0        0   \n",
       "CRUK0004        0      0      0         0      1      1     0      0        0   \n",
       "CRUK0005        1      1      0         1      1      1     1      0        0   \n",
       "...           ...    ...    ...       ...    ...    ...   ...    ...      ...   \n",
       "CRUK0861        0      1      0         0      0      0     1      0        0   \n",
       "CRUK0864        0      0      1         0      0      1     0      0        0   \n",
       "CRUK0872        0      0      0         0      1      1     1      0        0   \n",
       "CRUK0881        1      1      1         1      1      1     1      1        1   \n",
       "CRUK0893        0      0      1         0      0      0     0      0        0   \n",
       "\n",
       "            MROH2B  MUC16  MUC19  MUC2  MYO18B  MYO7B  NAV3  NCS1  NEB  NRXN1  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001         0      0      1     1       1      0     1     1    0      0   \n",
       "CRUK0002         0      1      0     0       0      0     0     0    0      0   \n",
       "CRUK0003         1      1      0     0       1      0     1     0    0      0   \n",
       "CRUK0004         0      1      0     0       0      0     0     0    0      0   \n",
       "CRUK0005         1      1      0     1       0      1     0     0    1      1   \n",
       "...            ...    ...    ...   ...     ...    ...   ...   ...  ...    ...   \n",
       "CRUK0861         0      1      0     0       0      0     0     0    1      0   \n",
       "CRUK0864         0      1      0     0       0      0     0     0    0      0   \n",
       "CRUK0872         1      1      1     0       0      0     0     0    0      0   \n",
       "CRUK0881         1      1      1     1       0      1     1     1    1      1   \n",
       "CRUK0893         0      0      0     1       0      0     0     0    0      1   \n",
       "\n",
       "            OBSCN  OR4E2  OTOG  OTOGL  PAPPA2  PCDH15  PCLO  PIEZO2  PKHD1  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001        1      1     0      0       0       0     1       0      1   \n",
       "CRUK0002        0      0     0      0       0       0     1       0      0   \n",
       "CRUK0003        1      0     0      0       0       0     1       0      0   \n",
       "CRUK0004        1      0     0      0       0       0     0       0      0   \n",
       "CRUK0005        1      1     1      1       1       1     0       1      1   \n",
       "...           ...    ...   ...    ...     ...     ...   ...     ...    ...   \n",
       "CRUK0861        0      0     0      0       0       0     0       0      0   \n",
       "CRUK0864        0      0     1      0       0       0     0       0      0   \n",
       "CRUK0872        0      0     0      1       0       1     0       0      0   \n",
       "CRUK0881        1      1     1      0       1       1     1       1      1   \n",
       "CRUK0893        0      1     1      0       0       0     0       0      0   \n",
       "\n",
       "            PKHD1L1  PLEC  PLXNA4  PTPRD  PTPRT  PXDNL  RELN  RIMS2  RYR1  \\\n",
       "patient_id                                                                  \n",
       "CRUK0001          0     0       1      0      0      0     1      1     1   \n",
       "CRUK0002          0     1       1      0      0      0     0      0     0   \n",
       "CRUK0003          0     0       0      1      1      0     1      0     0   \n",
       "CRUK0004          0     0       0      0      0      0     0      0     0   \n",
       "CRUK0005          1     0       1      1      1      1     1      1     0   \n",
       "...             ...   ...     ...    ...    ...    ...   ...    ...   ...   \n",
       "CRUK0861          1     0       0      0      0      0     1      0     0   \n",
       "CRUK0864          0     0       0      0      0      0     0      0     0   \n",
       "CRUK0872          0     0       0      0      0      0     1      0     0   \n",
       "CRUK0881          1     1       1      1      0      1     1      1     1   \n",
       "CRUK0893          0     0       0      0      0      0     0      1     1   \n",
       "\n",
       "            RYR2  RYR3  SI  SORCS1  SORCS3  SPEF2  SPTA1  SSPO  SYNE1  TNR  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001       1     1   1       1       0      0      1     1      1    0   \n",
       "CRUK0002       0     1   0       0       0      0      0     0      0    0   \n",
       "CRUK0003       1     0   0       0       0      0      0     0      0    1   \n",
       "CRUK0004       1     0   0       0       0      0      1     0      0    0   \n",
       "CRUK0005       1     0   1       1       1      1      0     1      0    1   \n",
       "...          ...   ...  ..     ...     ...    ...    ...   ...    ...  ...   \n",
       "CRUK0861       1     1   0       0       0      1      1     1      1    1   \n",
       "CRUK0864       0     1   0       0       0      0      0     0      0    1   \n",
       "CRUK0872       1     0   1       0       0      0      0     1      0    1   \n",
       "CRUK0881       1     1   1       1       1      1      1     1      1    1   \n",
       "CRUK0893       1     1   0       1       1      0      0     1      0    0   \n",
       "\n",
       "            TP53  TRY2P  TTC40  TTN  UNC79  UNC80  USH2A  WDFY4  XIRP2  ZFHX4  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001       1      1      0    1      0      1      1      1      0      0   \n",
       "CRUK0002       0      0      0    1      0      0      0      1      0      0   \n",
       "CRUK0003       0      0      0    1      1      0      1      0      0      0   \n",
       "CRUK0004       1      1      0    0      0      0      0      0      0      0   \n",
       "CRUK0005       1      1      0    1      0      1      1      1      1      1   \n",
       "...          ...    ...    ...  ...    ...    ...    ...    ...    ...    ...   \n",
       "CRUK0861       1      0      0    1      0      0      0      0      0      0   \n",
       "CRUK0864       0      0      0    0      1      0      0      0      0      0   \n",
       "CRUK0872       1      0      0    1      0      1      1      0      0      1   \n",
       "CRUK0881       1      0      1    1      0      1      1      1      1      1   \n",
       "CRUK0893       0      0      0    1      0      0      0      0      1      0   \n",
       "\n",
       "            ZNF536  \n",
       "patient_id          \n",
       "CRUK0001         1  \n",
       "CRUK0002         0  \n",
       "CRUK0003         0  \n",
       "CRUK0004         0  \n",
       "CRUK0005         1  \n",
       "...            ...  \n",
       "CRUK0861         0  \n",
       "CRUK0864         0  \n",
       "CRUK0872         0  \n",
       "CRUK0881         0  \n",
       "CRUK0893         0  \n",
       "\n",
       "[421 rows x 100 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ABCA13</th>\n",
       "      <th>ADAMTS12</th>\n",
       "      <th>ADAMTS16</th>\n",
       "      <th>ADAMTS20</th>\n",
       "      <th>AFF2</th>\n",
       "      <th>ANK2</th>\n",
       "      <th>APOB</th>\n",
       "      <th>ASTN1</th>\n",
       "      <th>BAI3</th>\n",
       "      <th>CACNA1B</th>\n",
       "      <th>CACNA1C</th>\n",
       "      <th>CACNA1E</th>\n",
       "      <th>CCDC168</th>\n",
       "      <th>CDH10</th>\n",
       "      <th>CDH23</th>\n",
       "      <th>CNTNAP2</th>\n",
       "      <th>CNTNAP5</th>\n",
       "      <th>COL11A1</th>\n",
       "      <th>COL22A1</th>\n",
       "      <th>CPS1</th>\n",
       "      <th>CSMD1</th>\n",
       "      <th>CSMD2</th>\n",
       "      <th>CSMD3</th>\n",
       "      <th>CTNNA2</th>\n",
       "      <th>CTNND2</th>\n",
       "      <th>DMD</th>\n",
       "      <th>DNAH11</th>\n",
       "      <th>DNAH5</th>\n",
       "      <th>DNAH6</th>\n",
       "      <th>DNAH9</th>\n",
       "      <th>DOCK2</th>\n",
       "      <th>DPP6</th>\n",
       "      <th>DYSF</th>\n",
       "      <th>ELK2AP</th>\n",
       "      <th>EYS</th>\n",
       "      <th>FAM135B</th>\n",
       "      <th>FAT3</th>\n",
       "      <th>FAT4</th>\n",
       "      <th>FBN2</th>\n",
       "      <th>FPGT-TNNI3K</th>\n",
       "      <th>FSIP2</th>\n",
       "      <th>HMCN1</th>\n",
       "      <th>KIF1A</th>\n",
       "      <th>LAMA1</th>\n",
       "      <th>LAMA2</th>\n",
       "      <th>LOC93432</th>\n",
       "      <th>LPHN3</th>\n",
       "      <th>LRP1B</th>\n",
       "      <th>LRP2</th>\n",
       "      <th>LRRC7</th>\n",
       "      <th>MIR548N</th>\n",
       "      <th>MROH2B</th>\n",
       "      <th>MUC16</th>\n",
       "      <th>MUC19</th>\n",
       "      <th>MUC2</th>\n",
       "      <th>MYO18B</th>\n",
       "      <th>MYO7B</th>\n",
       "      <th>NAV3</th>\n",
       "      <th>NCS1</th>\n",
       "      <th>NEB</th>\n",
       "      <th>NRXN1</th>\n",
       "      <th>OBSCN</th>\n",
       "      <th>OR4E2</th>\n",
       "      <th>OTOG</th>\n",
       "      <th>OTOGL</th>\n",
       "      <th>PAPPA2</th>\n",
       "      <th>PCDH15</th>\n",
       "      <th>PCLO</th>\n",
       "      <th>PIEZO2</th>\n",
       "      <th>PKHD1</th>\n",
       "      <th>PKHD1L1</th>\n",
       "      <th>PLEC</th>\n",
       "      <th>PLXNA4</th>\n",
       "      <th>PTPRD</th>\n",
       "      <th>PTPRT</th>\n",
       "      <th>PXDNL</th>\n",
       "      <th>RELN</th>\n",
       "      <th>RIMS2</th>\n",
       "      <th>RYR1</th>\n",
       "      <th>RYR2</th>\n",
       "      <th>RYR3</th>\n",
       "      <th>SI</th>\n",
       "      <th>SORCS1</th>\n",
       "      <th>SORCS3</th>\n",
       "      <th>SPEF2</th>\n",
       "      <th>SPTA1</th>\n",
       "      <th>SSPO</th>\n",
       "      <th>SYNE1</th>\n",
       "      <th>TNR</th>\n",
       "      <th>TP53</th>\n",
       "      <th>TRY2P</th>\n",
       "      <th>TTC40</th>\n",
       "      <th>TTN</th>\n",
       "      <th>UNC79</th>\n",
       "      <th>UNC80</th>\n",
       "      <th>USH2A</th>\n",
       "      <th>WDFY4</th>\n",
       "      <th>XIRP2</th>\n",
       "      <th>ZFHX4</th>\n",
       "      <th>ZNF536</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>LUSC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CRUK0001</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0004</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0005</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0861</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0864</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0872</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0881</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0893</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>421 rows × 102 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            ABCA13  ADAMTS12  ADAMTS16  ADAMTS20  AFF2  ANK2  APOB  ASTN1  \\\n",
       "patient_id                                                                  \n",
       "CRUK0001         1         1         0         1     1     1     1      0   \n",
       "CRUK0002         0         0         0         1     0     0     0      1   \n",
       "CRUK0003         0         0         1         0     0     0     0      0   \n",
       "CRUK0004         1         0         0         0     0     0     0      1   \n",
       "CRUK0005         1         1         0         1     1     1     1      0   \n",
       "...            ...       ...       ...       ...   ...   ...   ...    ...   \n",
       "CRUK0861         0         0         0         0     1     1     1      0   \n",
       "CRUK0864         0         0         0         0     0     0     0      0   \n",
       "CRUK0872         0         0         0         0     0     0     1      1   \n",
       "CRUK0881         0         1         1         1     0     1     1      1   \n",
       "CRUK0893         0         0         0         0     0     1     1      0   \n",
       "\n",
       "            BAI3  CACNA1B  CACNA1C  CACNA1E  CCDC168  CDH10  CDH23  CNTNAP2  \\\n",
       "patient_id                                                                    \n",
       "CRUK0001       1        0        1        0        0      0      0        0   \n",
       "CRUK0002       0        0        0        0        0      0      0        0   \n",
       "CRUK0003       0        1        0        0        0      0      0        1   \n",
       "CRUK0004       0        0        0        0        1      0      0        1   \n",
       "CRUK0005       0        0        0        1        1      1      0        1   \n",
       "...          ...      ...      ...      ...      ...    ...    ...      ...   \n",
       "CRUK0861       1        0        0        0        0      0      0        1   \n",
       "CRUK0864       0        0        0        0        1      0      0        0   \n",
       "CRUK0872       0        0        0        0        0      0      1        0   \n",
       "CRUK0881       0        0        1        1        1      1      0        1   \n",
       "CRUK0893       0        0        0        1        0      0      0        1   \n",
       "\n",
       "            CNTNAP5  COL11A1  COL22A1  CPS1  CSMD1  CSMD2  CSMD3  CTNNA2  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001          1        0        1     0      1      1      1       1   \n",
       "CRUK0002          0        0        0     0      0      1      1       0   \n",
       "CRUK0003          0        0        0     0      0      0      1       0   \n",
       "CRUK0004          0        0        0     0      0      0      1       0   \n",
       "CRUK0005          0        1        1     0      0      1      1       0   \n",
       "...             ...      ...      ...   ...    ...    ...    ...     ...   \n",
       "CRUK0861          1        0        0     0      0      0      1       0   \n",
       "CRUK0864          0        0        0     0      0      0      1       0   \n",
       "CRUK0872          0        1        1     0      0      0      1       0   \n",
       "CRUK0881          1        1        0     1      1      1      1       1   \n",
       "CRUK0893          1        0        0     0      0      0      0       0   \n",
       "\n",
       "            CTNND2  DMD  DNAH11  DNAH5  DNAH6  DNAH9  DOCK2  DPP6  DYSF  \\\n",
       "patient_id                                                                \n",
       "CRUK0001         0    1       0      1      0      0      1     0     1   \n",
       "CRUK0002         1    0       0      1      1      1      1     0     0   \n",
       "CRUK0003         0    0       0      0      0      0      1     0     0   \n",
       "CRUK0004         0    0       0      0      0      0      0     0     0   \n",
       "CRUK0005         1    1       0      1      0      1      1     0     1   \n",
       "...            ...  ...     ...    ...    ...    ...    ...   ...   ...   \n",
       "CRUK0861         0    0       0      0      1      1      0     0     0   \n",
       "CRUK0864         0    1       0      0      0      1      1     1     0   \n",
       "CRUK0872         0    0       0      1      0      0      1     0     0   \n",
       "CRUK0881         1    1       1      1      1      1      1     1     0   \n",
       "CRUK0893         0    0       1      0      0      0      0     0     0   \n",
       "\n",
       "            ELK2AP  EYS  FAM135B  FAT3  FAT4  FBN2  FPGT-TNNI3K  FSIP2  HMCN1  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001         0    1        1     0     1     1            1      1      1   \n",
       "CRUK0002         0    1        0     0     0     0            0      0      0   \n",
       "CRUK0003         1    0        0     0     0     0            0      0      0   \n",
       "CRUK0004         0    0        1     0     0     0            1      0      0   \n",
       "CRUK0005         1    1        1     1     1     1            1      1      1   \n",
       "...            ...  ...      ...   ...   ...   ...          ...    ...    ...   \n",
       "CRUK0861         0    1        0     0     0     1            0      0      0   \n",
       "CRUK0864         0    0        0     0     0     1            0      1      0   \n",
       "CRUK0872         0    0        0     0     0     0            1      0      0   \n",
       "CRUK0881         0    1        1     1     0     1            1      1      1   \n",
       "CRUK0893         0    1        0     0     0     0            0      1      0   \n",
       "\n",
       "            KIF1A  LAMA1  LAMA2  LOC93432  LPHN3  LRP1B  LRP2  LRRC7  MIR548N  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001        1      1      1         0      0      0     1      1        0   \n",
       "CRUK0002        0      0      0         0      0      0     0      0        0   \n",
       "CRUK0003        0      0      0         1      0      1     1      0        0   \n",
       "CRUK0004        0      0      0         0      1      1     0      0        0   \n",
       "CRUK0005        1      1      0         1      1      1     1      0        0   \n",
       "...           ...    ...    ...       ...    ...    ...   ...    ...      ...   \n",
       "CRUK0861        0      1      0         0      0      0     1      0        0   \n",
       "CRUK0864        0      0      1         0      0      1     0      0        0   \n",
       "CRUK0872        0      0      0         0      1      1     1      0        0   \n",
       "CRUK0881        1      1      1         1      1      1     1      1        1   \n",
       "CRUK0893        0      0      1         0      0      0     0      0        0   \n",
       "\n",
       "            MROH2B  MUC16  MUC19  MUC2  MYO18B  MYO7B  NAV3  NCS1  NEB  NRXN1  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001         0      0      1     1       1      0     1     1    0      0   \n",
       "CRUK0002         0      1      0     0       0      0     0     0    0      0   \n",
       "CRUK0003         1      1      0     0       1      0     1     0    0      0   \n",
       "CRUK0004         0      1      0     0       0      0     0     0    0      0   \n",
       "CRUK0005         1      1      0     1       0      1     0     0    1      1   \n",
       "...            ...    ...    ...   ...     ...    ...   ...   ...  ...    ...   \n",
       "CRUK0861         0      1      0     0       0      0     0     0    1      0   \n",
       "CRUK0864         0      1      0     0       0      0     0     0    0      0   \n",
       "CRUK0872         1      1      1     0       0      0     0     0    0      0   \n",
       "CRUK0881         1      1      1     1       0      1     1     1    1      1   \n",
       "CRUK0893         0      0      0     1       0      0     0     0    0      1   \n",
       "\n",
       "            OBSCN  OR4E2  OTOG  OTOGL  PAPPA2  PCDH15  PCLO  PIEZO2  PKHD1  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001        1      1     0      0       0       0     1       0      1   \n",
       "CRUK0002        0      0     0      0       0       0     1       0      0   \n",
       "CRUK0003        1      0     0      0       0       0     1       0      0   \n",
       "CRUK0004        1      0     0      0       0       0     0       0      0   \n",
       "CRUK0005        1      1     1      1       1       1     0       1      1   \n",
       "...           ...    ...   ...    ...     ...     ...   ...     ...    ...   \n",
       "CRUK0861        0      0     0      0       0       0     0       0      0   \n",
       "CRUK0864        0      0     1      0       0       0     0       0      0   \n",
       "CRUK0872        0      0     0      1       0       1     0       0      0   \n",
       "CRUK0881        1      1     1      0       1       1     1       1      1   \n",
       "CRUK0893        0      1     1      0       0       0     0       0      0   \n",
       "\n",
       "            PKHD1L1  PLEC  PLXNA4  PTPRD  PTPRT  PXDNL  RELN  RIMS2  RYR1  \\\n",
       "patient_id                                                                  \n",
       "CRUK0001          0     0       1      0      0      0     1      1     1   \n",
       "CRUK0002          0     1       1      0      0      0     0      0     0   \n",
       "CRUK0003          0     0       0      1      1      0     1      0     0   \n",
       "CRUK0004          0     0       0      0      0      0     0      0     0   \n",
       "CRUK0005          1     0       1      1      1      1     1      1     0   \n",
       "...             ...   ...     ...    ...    ...    ...   ...    ...   ...   \n",
       "CRUK0861          1     0       0      0      0      0     1      0     0   \n",
       "CRUK0864          0     0       0      0      0      0     0      0     0   \n",
       "CRUK0872          0     0       0      0      0      0     1      0     0   \n",
       "CRUK0881          1     1       1      1      0      1     1      1     1   \n",
       "CRUK0893          0     0       0      0      0      0     0      1     1   \n",
       "\n",
       "            RYR2  RYR3  SI  SORCS1  SORCS3  SPEF2  SPTA1  SSPO  SYNE1  TNR  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001       1     1   1       1       0      0      1     1      1    0   \n",
       "CRUK0002       0     1   0       0       0      0      0     0      0    0   \n",
       "CRUK0003       1     0   0       0       0      0      0     0      0    1   \n",
       "CRUK0004       1     0   0       0       0      0      1     0      0    0   \n",
       "CRUK0005       1     0   1       1       1      1      0     1      0    1   \n",
       "...          ...   ...  ..     ...     ...    ...    ...   ...    ...  ...   \n",
       "CRUK0861       1     1   0       0       0      1      1     1      1    1   \n",
       "CRUK0864       0     1   0       0       0      0      0     0      0    1   \n",
       "CRUK0872       1     0   1       0       0      0      0     1      0    1   \n",
       "CRUK0881       1     1   1       1       1      1      1     1      1    1   \n",
       "CRUK0893       1     1   0       1       1      0      0     1      0    0   \n",
       "\n",
       "            TP53  TRY2P  TTC40  TTN  UNC79  UNC80  USH2A  WDFY4  XIRP2  ZFHX4  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001       1      1      0    1      0      1      1      1      0      0   \n",
       "CRUK0002       0      0      0    1      0      0      0      1      0      0   \n",
       "CRUK0003       0      0      0    1      1      0      1      0      0      0   \n",
       "CRUK0004       1      1      0    0      0      0      0      0      0      0   \n",
       "CRUK0005       1      1      0    1      0      1      1      1      1      1   \n",
       "...          ...    ...    ...  ...    ...    ...    ...    ...    ...    ...   \n",
       "CRUK0861       1      0      0    1      0      0      0      0      0      0   \n",
       "CRUK0864       0      0      0    0      1      0      0      0      0      0   \n",
       "CRUK0872       1      0      0    1      0      1      1      0      0      1   \n",
       "CRUK0881       1      0      1    1      0      1      1      1      1      1   \n",
       "CRUK0893       0      0      0    1      0      0      0      0      1      0   \n",
       "\n",
       "            ZNF536  LUAD  LUSC  \n",
       "patient_id                      \n",
       "CRUK0001         1     1     0  \n",
       "CRUK0002         0     1     0  \n",
       "CRUK0003         0     1     0  \n",
       "CRUK0004         0     1     0  \n",
       "CRUK0005         1     1     0  \n",
       "...            ...   ...   ...  \n",
       "CRUK0861         0     0     1  \n",
       "CRUK0864         0     1     0  \n",
       "CRUK0872         0     0     1  \n",
       "CRUK0881         0     1     0  \n",
       "CRUK0893         0     1     0  \n",
       "\n",
       "[421 rows x 102 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_matrix_w_hist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACVR1B</th>\n",
       "      <th>AMER1</th>\n",
       "      <th>APC</th>\n",
       "      <th>ARHGAP35</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>ARID1B</th>\n",
       "      <th>ARID2</th>\n",
       "      <th>ASXL1</th>\n",
       "      <th>ASXL2</th>\n",
       "      <th>ATM</th>\n",
       "      <th>ATR</th>\n",
       "      <th>ATRX</th>\n",
       "      <th>AXIN1</th>\n",
       "      <th>B2M</th>\n",
       "      <th>BAP1</th>\n",
       "      <th>BCLAF1</th>\n",
       "      <th>BCOR</th>\n",
       "      <th>BRAF</th>\n",
       "      <th>CBL</th>\n",
       "      <th>CBLB</th>\n",
       "      <th>CDC73</th>\n",
       "      <th>CDK12</th>\n",
       "      <th>CDKN2A</th>\n",
       "      <th>CHD4</th>\n",
       "      <th>CHD8</th>\n",
       "      <th>CIC</th>\n",
       "      <th>CMTR2</th>\n",
       "      <th>CNOT3</th>\n",
       "      <th>COL2A1</th>\n",
       "      <th>COL5A2</th>\n",
       "      <th>CREBBP</th>\n",
       "      <th>CTNNB1</th>\n",
       "      <th>CUL3</th>\n",
       "      <th>CUX1</th>\n",
       "      <th>CYLD</th>\n",
       "      <th>DDX3X</th>\n",
       "      <th>DICER1</th>\n",
       "      <th>DNAH12</th>\n",
       "      <th>DNMT3A</th>\n",
       "      <th>DOT1L</th>\n",
       "      <th>EGFR</th>\n",
       "      <th>EP300</th>\n",
       "      <th>FANCM</th>\n",
       "      <th>FAT1</th>\n",
       "      <th>FBXW7</th>\n",
       "      <th>FLT4</th>\n",
       "      <th>GATA3</th>\n",
       "      <th>GRIN2A</th>\n",
       "      <th>HIST1H4E</th>\n",
       "      <th>IKZF1</th>\n",
       "      <th>KDM5C</th>\n",
       "      <th>KDM6A</th>\n",
       "      <th>KEAP1</th>\n",
       "      <th>KMT2B</th>\n",
       "      <th>KMT2C</th>\n",
       "      <th>KMT2D</th>\n",
       "      <th>KRAS</th>\n",
       "      <th>LATS1</th>\n",
       "      <th>MAP3K1</th>\n",
       "      <th>MGA</th>\n",
       "      <th>MTOR</th>\n",
       "      <th>NBN</th>\n",
       "      <th>NCOA6</th>\n",
       "      <th>NCOR1</th>\n",
       "      <th>NF1</th>\n",
       "      <th>NFE2L2</th>\n",
       "      <th>NOTCH1</th>\n",
       "      <th>NOTCH2</th>\n",
       "      <th>PBRM1</th>\n",
       "      <th>PIK3CA</th>\n",
       "      <th>PIK3R1</th>\n",
       "      <th>PLXNB2</th>\n",
       "      <th>POLE</th>\n",
       "      <th>POT1</th>\n",
       "      <th>PPP3CA</th>\n",
       "      <th>PTEN</th>\n",
       "      <th>PTPRB</th>\n",
       "      <th>PTPRC</th>\n",
       "      <th>RAD21</th>\n",
       "      <th>RASA1</th>\n",
       "      <th>RB1</th>\n",
       "      <th>RBM10</th>\n",
       "      <th>RNF43</th>\n",
       "      <th>RYR2</th>\n",
       "      <th>SETD2</th>\n",
       "      <th>SETDB1</th>\n",
       "      <th>SGK223</th>\n",
       "      <th>SMAD4</th>\n",
       "      <th>SMARCA4</th>\n",
       "      <th>SPEN</th>\n",
       "      <th>STAG2</th>\n",
       "      <th>STK11</th>\n",
       "      <th>TET2</th>\n",
       "      <th>TP53</th>\n",
       "      <th>TP53BP1</th>\n",
       "      <th>TSC2</th>\n",
       "      <th>UBR5</th>\n",
       "      <th>WT1</th>\n",
       "      <th>ZFP36L1</th>\n",
       "      <th>ZMYM2</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>LUSC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CRUK0001</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0004</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0005</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0861</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0864</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0872</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0881</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0893</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>421 rows × 102 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            ACVR1B  AMER1  APC  ARHGAP35  ARID1A  ARID1B  ARID2  ASXL1  ASXL2  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001         0      0    0         1       0       0      0      0      0   \n",
       "CRUK0002         0      0    0         0       0       0      0      0      0   \n",
       "CRUK0003         0      0    0         0       0       0      0      0      0   \n",
       "CRUK0004         0      0    0         0       0       0      0      0      0   \n",
       "CRUK0005         0      0    0         0       0       0      0      0      0   \n",
       "...            ...    ...  ...       ...     ...     ...    ...    ...    ...   \n",
       "CRUK0861         0      0    0         0       0       0      0      0      0   \n",
       "CRUK0864         0      0    0         0       0       0      0      0      0   \n",
       "CRUK0872         0      0    0         0       0       0      0      0      0   \n",
       "CRUK0881         0      0    0         0       0       0      0      0      0   \n",
       "CRUK0893         0      0    0         0       0       0      0      0      0   \n",
       "\n",
       "            ATM  ATR  ATRX  AXIN1  B2M  BAP1  BCLAF1  BCOR  BRAF  CBL  CBLB  \\\n",
       "patient_id                                                                    \n",
       "CRUK0001      0    0     0      0    0     0       0     0     0    0     0   \n",
       "CRUK0002      0    0     0      0    0     0       0     0     0    0     0   \n",
       "CRUK0003      0    0     0      0    0     0       0     0     0    0     0   \n",
       "CRUK0004      0    0     0      0    0     0       0     0     0    0     0   \n",
       "CRUK0005      0    0     1      0    0     0       0     0     1    0     0   \n",
       "...         ...  ...   ...    ...  ...   ...     ...   ...   ...  ...   ...   \n",
       "CRUK0861      0    0     0      0    0     0       0     0     0    0     0   \n",
       "CRUK0864      0    0     0      0    0     0       0     0     0    0     0   \n",
       "CRUK0872      0    0     0      0    0     0       0     0     0    0     0   \n",
       "CRUK0881      0    0     0      0    0     0       0     0     1    0     0   \n",
       "CRUK0893      1    0     0      0    0     0       0     0     0    0     0   \n",
       "\n",
       "            CDC73  CDK12  CDKN2A  CHD4  CHD8  CIC  CMTR2  CNOT3  COL2A1  \\\n",
       "patient_id                                                                \n",
       "CRUK0001        0      0       0     0     0    0      0      0       0   \n",
       "CRUK0002        0      0       0     0     0    0      0      0       0   \n",
       "CRUK0003        0      0       0     0     0    0      0      0       0   \n",
       "CRUK0004        0      0       0     0     0    0      0      0       0   \n",
       "CRUK0005        0      0       0     0     0    0      1      0       0   \n",
       "...           ...    ...     ...   ...   ...  ...    ...    ...     ...   \n",
       "CRUK0861        0      0       0     0     0    0      0      0       0   \n",
       "CRUK0864        0      0       0     0     0    0      0      0       0   \n",
       "CRUK0872        0      0       0     0     0    0      0      0       0   \n",
       "CRUK0881        0      0       0     0     0    0      0      0       0   \n",
       "CRUK0893        0      0       0     0     0    0      0      0       0   \n",
       "\n",
       "            COL5A2  CREBBP  CTNNB1  CUL3  CUX1  CYLD  DDX3X  DICER1  DNAH12  \\\n",
       "patient_id                                                                    \n",
       "CRUK0001         0       0       0     0     0     0      0       0       0   \n",
       "CRUK0002         0       0       0     0     0     0      0       0       0   \n",
       "CRUK0003         0       0       1     0     0     0      0       0       0   \n",
       "CRUK0004         0       0       0     0     0     0      0       0       0   \n",
       "CRUK0005         0       0       0     0     0     0      0       0       0   \n",
       "...            ...     ...     ...   ...   ...   ...    ...     ...     ...   \n",
       "CRUK0861         0       0       0     1     0     0      0       0       0   \n",
       "CRUK0864         0       0       0     0     1     0      0       0       0   \n",
       "CRUK0872         0       0       0     0     0     0      0       0       0   \n",
       "CRUK0881         0       1       0     1     0     0      1       0       1   \n",
       "CRUK0893         0       0       0     0     0     0      0       0       0   \n",
       "\n",
       "            DNMT3A  DOT1L  EGFR  EP300  FANCM  FAT1  FBXW7  FLT4  GATA3  \\\n",
       "patient_id                                                                \n",
       "CRUK0001         0      0     0      0      0     0      0     0      0   \n",
       "CRUK0002         0      0     0      1      0     0      0     0      0   \n",
       "CRUK0003         0      0     1      0      0     0      0     0      0   \n",
       "CRUK0004         0      0     1      0      0     0      0     0      0   \n",
       "CRUK0005         0      0     0      0      0     0      0     0      0   \n",
       "...            ...    ...   ...    ...    ...   ...    ...   ...    ...   \n",
       "CRUK0861         0      0     0      0      0     0      0     0      0   \n",
       "CRUK0864         1      0     0      0      0     0      0     0      0   \n",
       "CRUK0872         0      0     0      0      0     0      0     0      0   \n",
       "CRUK0881         0      0     0      0      0     0      0     0      0   \n",
       "CRUK0893         0      0     0      0      0     0      0     0      0   \n",
       "\n",
       "            GRIN2A  HIST1H4E  IKZF1  KDM5C  KDM6A  KEAP1  KMT2B  KMT2C  KMT2D  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001         0         0      0      0      0      0      1      0      0   \n",
       "CRUK0002         0         0      1      0      0      0      0      0      0   \n",
       "CRUK0003         0         0      0      0      0      0      0      0      0   \n",
       "CRUK0004         0         0      0      0      0      0      0      0      0   \n",
       "CRUK0005         0         0      0      0      0      0      0      0      0   \n",
       "...            ...       ...    ...    ...    ...    ...    ...    ...    ...   \n",
       "CRUK0861         0         0      0      0      0      0      0      0      0   \n",
       "CRUK0864         0         0      0      0      0      0      0      0      0   \n",
       "CRUK0872         0         0      0      0      0      0      1      0      0   \n",
       "CRUK0881         0         0      0      0      0      0      0      0      0   \n",
       "CRUK0893         0         0      0      0      0      0      0      0      0   \n",
       "\n",
       "            KRAS  LATS1  MAP3K1  MGA  MTOR  NBN  NCOA6  NCOR1  NF1  NFE2L2  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001       0      0       0    1     0    0      0      0    0       0   \n",
       "CRUK0002       0      0       0    0     0    0      0      0    1       0   \n",
       "CRUK0003       0      0       0    0     0    0      0      0    0       0   \n",
       "CRUK0004       0      0       0    0     0    0      0      0    0       0   \n",
       "CRUK0005       0      0       0    0     0    0      0      0    0       0   \n",
       "...          ...    ...     ...  ...   ...  ...    ...    ...  ...     ...   \n",
       "CRUK0861       0      0       0    0     0    0      0      0    0       0   \n",
       "CRUK0864       1      0       0    0     0    0      0      0    0       0   \n",
       "CRUK0872       0      0       0    0     0    0      0      0    0       0   \n",
       "CRUK0881       1      0       0    0     0    0      0      0    1       0   \n",
       "CRUK0893       1      0       0    0     0    0      0      0    0       0   \n",
       "\n",
       "            NOTCH1  NOTCH2  PBRM1  PIK3CA  PIK3R1  PLXNB2  POLE  POT1  PPP3CA  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001         0       0      0       0       0       0     0     0       0   \n",
       "CRUK0002         0       0      0       0       0       0     0     0       0   \n",
       "CRUK0003         0       0      0       1       0       0     0     0       0   \n",
       "CRUK0004         1       0      0       0       0       0     0     0       0   \n",
       "CRUK0005         0       0      0       0       0       0     0     0       0   \n",
       "...            ...     ...    ...     ...     ...     ...   ...   ...     ...   \n",
       "CRUK0861         0       0      0       0       0       0     0     0       0   \n",
       "CRUK0864         0       0      0       0       0       0     0     0       0   \n",
       "CRUK0872         0       0      0       0       0       0     0     0       0   \n",
       "CRUK0881         0       0      1       0       0       0     0     0       1   \n",
       "CRUK0893         0       0      0       0       0       0     0     0       0   \n",
       "\n",
       "            PTEN  PTPRB  PTPRC  RAD21  RASA1  RB1  RBM10  RNF43  RYR2  SETD2  \\\n",
       "patient_id                                                                     \n",
       "CRUK0001       0      0      0      0      0    0      0      0     0      0   \n",
       "CRUK0002       0      0      0      0      0    1      0      0     0      0   \n",
       "CRUK0003       0      0      0      0      0    0      0      0     0      0   \n",
       "CRUK0004       0      0      0      0      0    0      0      0     0      0   \n",
       "CRUK0005       0      0      0      0      0    0      0      0     0      0   \n",
       "...          ...    ...    ...    ...    ...  ...    ...    ...   ...    ...   \n",
       "CRUK0861       0      0      0      0      0    0      0      0     0      0   \n",
       "CRUK0864       0      0      0      0      0    0      1      0     0      0   \n",
       "CRUK0872       0      1      0      0      0    0      0      0     0      0   \n",
       "CRUK0881       0      0      0      0      0    0      1      0     0      0   \n",
       "CRUK0893       0      0      0      0      0    0      0      0     0      0   \n",
       "\n",
       "            SETDB1  SGK223  SMAD4  SMARCA4  SPEN  STAG2  STK11  TET2  TP53  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001         0       0      0        0     0      0      0     0     1   \n",
       "CRUK0002         0       0      0        0     0      0      0     0     0   \n",
       "CRUK0003         0       0      0        0     0      0      0     0     0   \n",
       "CRUK0004         0       0      1        0     0      0      0     0     1   \n",
       "CRUK0005         0       0      0        0     0      0      0     0     1   \n",
       "...            ...     ...    ...      ...   ...    ...    ...   ...   ...   \n",
       "CRUK0861         0       0      0        0     0      0      0     0     1   \n",
       "CRUK0864         0       0      0        0     0      0      0     0     0   \n",
       "CRUK0872         0       0      0        0     0      0      0     0     1   \n",
       "CRUK0881         1       0      0        0     0      0      0     1     1   \n",
       "CRUK0893         0       0      0        0     0      0      0     0     0   \n",
       "\n",
       "            TP53BP1  TSC2  UBR5  WT1  ZFP36L1  ZMYM2  LUAD  LUSC  \n",
       "patient_id                                                        \n",
       "CRUK0001          0     0     0    0        0      0     1     0  \n",
       "CRUK0002          0     0     0    0        0      0     1     0  \n",
       "CRUK0003          0     0     0    0        0      0     1     0  \n",
       "CRUK0004          0     0     0    0        0      0     1     0  \n",
       "CRUK0005          0     0     0    0        0      0     1     0  \n",
       "...             ...   ...   ...  ...      ...    ...   ...   ...  \n",
       "CRUK0861          0     0     0    0        0      0     0     1  \n",
       "CRUK0864          0     0     0    0        0      0     1     0  \n",
       "CRUK0872          0     0     0    0        0      0     0     1  \n",
       "CRUK0881          0     0     0    0        0      0     1     0  \n",
       "CRUK0893          0     0     0    0        0      0     1     0  \n",
       "\n",
       "[421 rows x 102 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "driver_matrix_w_hist = driver_matrix.copy()\n",
    "driver_matrix_w_hist['LUAD']= driver_matrix_w_hist.index.isin(LUAD_patient_id).astype(int)\n",
    "driver_matrix_w_hist['LUSC']= driver_matrix_w_hist.index.isin(LUSC_patient_id).astype(int)\n",
    "driver_matrix_w_hist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 102) (169, 102) (252,) (169,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:469: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  n_iter_i = _check_optimize_result(\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:469: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  n_iter_i = _check_optimize_result(\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:469: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "Please also refer to the documentation for alternative solver options:\n",
      "    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression\n",
      "  n_iter_i = _check_optimize_result(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'C': 1}\n",
      "Best score:  0.5172905525846703\n",
      "Predictions:  [1 1 1 0 1 0 1 0 0 1 1 0 0 0 1 1 1 0 0 1 0 0 0 0 0 0 0 1 1 0 0 0 1 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 0 1 0 1 0 0 0 0 1 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0\n",
      " 1 1 1 0 0 0 0 1 1 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 1 1\n",
      " 1 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 0 0 1\n",
      " 1 1 0 0 1 0 0 0 0 1 0 0 0 1 0 1 0 0 0 0 1]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.4974937343358396\n",
      "ROC AUC:  0.5148809523809524\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.66      0.71      0.69       112\n",
      "   Recurrence       0.33      0.28      0.30        57\n",
      "\n",
      "     accuracy                           0.57       169\n",
      "    macro avg       0.50      0.50      0.50       169\n",
      " weighted avg       0.55      0.57      0.56       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(gene_matrix_w_hist, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 102) (169, 102) (252,) (169,)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'C': 1}\n",
      "Best score:  0.5068627450980392\n",
      "Predictions:  [0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0\n",
      " 0 0 1 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0\n",
      " 0 0 0 0 0 0 1 1 0 0 0 0 0 1 1 0 1 0 1 0 1]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5205983709273183\n",
      "ROC AUC:  0.4760338345864662\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.85      0.75       112\n",
      "   Recurrence       0.39      0.19      0.26        57\n",
      "\n",
      "     accuracy                           0.63       169\n",
      "    macro avg       0.53      0.52      0.50       169\n",
      " weighted avg       0.58      0.63      0.58       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(driver_matrix_w_hist, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'C': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "lr_model = LogisticRegression(penalty='l2')\n",
    "\n",
    "grid_search = GridSearchCV(lr_model, param_grid, scoring='balanced_accuracy')\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 102) (169, 102) (252,) (169,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.544741532976827\n",
      "Predictions:  [0 1 1 0 1 0 0 0 0 0 1 0 1 0 0 1 1 0 1 1 0 0 0 0 1 0 0 1 1 0 0 0 0 0 0 0 0\n",
      " 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 1 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0\n",
      " 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 1 0 0 0 1 0 1 1\n",
      " 0 0 0 0 1 1 0 0 0 0 0 1 1 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 0\n",
      " 1 0 0 0 0 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 1]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.5199718045112782\n",
      "ROC AUC:  0.5090852130325815\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.78      0.72       112\n",
      "   Recurrence       0.38      0.26      0.31        57\n",
      "\n",
      "     accuracy                           0.60       169\n",
      "    macro avg       0.52      0.52      0.52       169\n",
      " weighted avg       0.57      0.60      0.58       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(gene_matrix_w_hist, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(252, 102) (169, 102) (252,) (169,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.1, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5116755793226382\n",
      "Predictions:  [0 0 0 0 1 0 1 1 0 0 0 1 0 0 0 0 1 1 0 1 0 0 1 0 1 1 1 1 1 0 0 0 0 1 0 0 0\n",
      " 1 1 0 0 0 0 1 0 1 1 0 0 1 0 1 0 1 0 0 1 0 0 0 0 0 0 1 1 1 0 1 0 0 0 1 0 0\n",
      " 1 0 0 1 1 0 0 0 0 1 0 1 0 1 0 0 1 0 1 1 1 0 1 0 0 0 0 1 0 0 0 0 0 1 0 0 0\n",
      " 0 1 1 1 0 0 0 0 0 0 1 0 0 0 0 1 1 1 1 0 0 1 0 1 0 0 0 0 0 0 1 0 0 0 1 1 1\n",
      " 0 1 0 1 1 0 1 1 0 0 0 0 0 0 1 0 1 0 1 0 1]\n",
      "Actual:  [0 1 1 0 1 0 0 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 1 0 0 1 1 0\n",
      " 1 0 0 1 0 1 1 1 0 0 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0\n",
      " 0 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 1\n",
      " 0 1 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 1 0 0 0 1 0 0 0 0 1 0 0 0\n",
      " 0 0 0 0 0 0 1 1 0 1 0 1 0 0 1 0 1 0 0 0 0]\n",
      "Balanced Accuracy:  0.46577380952380953\n",
      "ROC AUC:  0.4694548872180451\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.64      0.60      0.62       112\n",
      "   Recurrence       0.30      0.33      0.31        57\n",
      "\n",
      "     accuracy                           0.51       169\n",
      "    macro avg       0.47      0.47      0.47       169\n",
      " weighted avg       0.52      0.51      0.52       169\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Neural network/Multi-layer perceptron (MLP) algorithm\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(driver_matrix_w_hist, y, test_size = 0.4, random_state = 0, stratify = y)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>tumour_id</th>\n",
       "      <th>RegionID</th>\n",
       "      <th>chr</th>\n",
       "      <th>startpos</th>\n",
       "      <th>endpos</th>\n",
       "      <th>n.het</th>\n",
       "      <th>cnTotal</th>\n",
       "      <th>nMajor</th>\n",
       "      <th>nMinor</th>\n",
       "      <th>Ploidy</th>\n",
       "      <th>ACF</th>\n",
       "      <th>nAraw</th>\n",
       "      <th>nBraw</th>\n",
       "      <th>cpn_event_vs_ploidy</th>\n",
       "      <th>MSAI</th>\n",
       "      <th>samp_used_to_phase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52958</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>CRUK0001:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>3800149</td>\n",
       "      <td>134</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4.493770</td>\n",
       "      <td>0.22</td>\n",
       "      <td>2.073981</td>\n",
       "      <td>1.625892</td>\n",
       "      <td>neutral</td>\n",
       "      <td>True</td>\n",
       "      <td>CRUK0001_SU_T1-R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>288283</th>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002</td>\n",
       "      <td>CRUK0002:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>2537166</td>\n",
       "      <td>72</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.921907</td>\n",
       "      <td>0.38</td>\n",
       "      <td>1.696286</td>\n",
       "      <td>1.044553</td>\n",
       "      <td>gain</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0002_SU_T1-R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141069</th>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003</td>\n",
       "      <td>CRUK0003:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>752894</td>\n",
       "      <td>3427608</td>\n",
       "      <td>110</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.567392</td>\n",
       "      <td>0.54</td>\n",
       "      <td>2.041043</td>\n",
       "      <td>1.081540</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0003_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45121</th>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004</td>\n",
       "      <td>CRUK0004:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762496</td>\n",
       "      <td>3496497</td>\n",
       "      <td>53</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>2.950603</td>\n",
       "      <td>0.23</td>\n",
       "      <td>1.804428</td>\n",
       "      <td>1.409574</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0004_SU_T1-R3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005</td>\n",
       "      <td>CRUK0005:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>2252860</td>\n",
       "      <td>73</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.131282</td>\n",
       "      <td>0.15</td>\n",
       "      <td>2.463756</td>\n",
       "      <td>1.184761</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0005_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368701</th>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861</td>\n",
       "      <td>CRUK0861:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>752894</td>\n",
       "      <td>1572735</td>\n",
       "      <td>79</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3.514533</td>\n",
       "      <td>0.76</td>\n",
       "      <td>4.050553</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0861_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>375529</th>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864</td>\n",
       "      <td>CRUK0864:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>1448024</td>\n",
       "      <td>90</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.556913</td>\n",
       "      <td>0.31</td>\n",
       "      <td>2.074464</td>\n",
       "      <td>1.035148</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0864_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396701</th>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872</td>\n",
       "      <td>CRUK0872:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>762273</td>\n",
       "      <td>990280</td>\n",
       "      <td>34</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>3.919491</td>\n",
       "      <td>0.26</td>\n",
       "      <td>2.102410</td>\n",
       "      <td>1.801966</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0872_SU_T1-R2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>417335</th>\n",
       "      <td>CRUK0881</td>\n",
       "      <td>CRUK0881_Tumour1</td>\n",
       "      <td>:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>65797</td>\n",
       "      <td>1027553</td>\n",
       "      <td>39</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4.186610</td>\n",
       "      <td>0.33</td>\n",
       "      <td>1.992436</td>\n",
       "      <td>0.923056</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0881_SU_T1-R5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261061</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>CRUK0893:R1</td>\n",
       "      <td>1</td>\n",
       "      <td>69511</td>\n",
       "      <td>81969071</td>\n",
       "      <td>1429</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>3.048684</td>\n",
       "      <td>0.13</td>\n",
       "      <td>1.732114</td>\n",
       "      <td>0.914478</td>\n",
       "      <td>neutral</td>\n",
       "      <td>False</td>\n",
       "      <td>CRUK0893_SU_T1-R2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>394 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id         tumour_id     RegionID  chr  startpos    endpos  \\\n",
       "52958    CRUK0001          CRUK0001  CRUK0001:R1    1    762273   3800149   \n",
       "288283   CRUK0002          CRUK0002  CRUK0002:R1    1    762273   2537166   \n",
       "141069   CRUK0003          CRUK0003  CRUK0003:R1    1    752894   3427608   \n",
       "45121    CRUK0004          CRUK0004  CRUK0004:R1    1    762496   3496497   \n",
       "0        CRUK0005          CRUK0005  CRUK0005:R1    1    762273   2252860   \n",
       "...           ...               ...          ...  ...       ...       ...   \n",
       "368701   CRUK0861          CRUK0861  CRUK0861:R1    1    752894   1572735   \n",
       "375529   CRUK0864          CRUK0864  CRUK0864:R1    1    762273   1448024   \n",
       "396701   CRUK0872          CRUK0872  CRUK0872:R1    1    762273    990280   \n",
       "417335   CRUK0881  CRUK0881_Tumour1          :R1    1     65797   1027553   \n",
       "261061   CRUK0893          CRUK0893  CRUK0893:R1    1     69511  81969071   \n",
       "\n",
       "        n.het  cnTotal  nMajor  nMinor    Ploidy   ACF     nAraw     nBraw  \\\n",
       "52958     134        4       2       2  4.493770  0.22  2.073981  1.625892   \n",
       "288283     72        3       2       1  1.921907  0.38  1.696286  1.044553   \n",
       "141069    110        3       2       1  3.567392  0.54  2.041043  1.081540   \n",
       "45121      53        3       2       1  2.950603  0.23  1.804428  1.409574   \n",
       "0          73        3       2       1  3.131282  0.15  2.463756  1.184761   \n",
       "...       ...      ...     ...     ...       ...   ...       ...       ...   \n",
       "368701     79        4       4       0  3.514533  0.76  4.050553  0.000000   \n",
       "375529     90        3       2       1  3.556913  0.31  2.074464  1.035148   \n",
       "396701     34        4       2       2  3.919491  0.26  2.102410  1.801966   \n",
       "417335     39        3       2       1  4.186610  0.33  1.992436  0.923056   \n",
       "261061   1429        3       2       1  3.048684  0.13  1.732114  0.914478   \n",
       "\n",
       "       cpn_event_vs_ploidy   MSAI samp_used_to_phase  \n",
       "52958              neutral   True  CRUK0001_SU_T1-R3  \n",
       "288283                gain  False  CRUK0002_SU_T1-R3  \n",
       "141069             neutral  False  CRUK0003_SU_T1-R2  \n",
       "45121              neutral  False  CRUK0004_SU_T1-R3  \n",
       "0                  neutral  False  CRUK0005_SU_T1-R2  \n",
       "...                    ...    ...                ...  \n",
       "368701             neutral  False  CRUK0861_SU_T1-R2  \n",
       "375529             neutral  False  CRUK0864_SU_T1-R2  \n",
       "396701             neutral  False  CRUK0872_SU_T1-R2  \n",
       "417335             neutral  False  CRUK0881_SU_T1-R5  \n",
       "261061             neutral  False  CRUK0893_SU_T1-R2  \n",
       "\n",
       "[394 rows x 17 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Cumulative LUAD\n",
    "\n",
    "cna_table = pyreadr.read_r('/Users/jasminechau/PycharmProjects/pythonProject1/INIM0038/LCCE_converted_data/20221109_TRACERx421_scna_table.rds')\n",
    "# Obtain dataframe and rename column sample to RegionID\n",
    "cna_table = pd.DataFrame(cna_table[None])\n",
    "cna_table_no_duplicates = cna_table.drop_duplicates(subset='patient_id')\n",
    "cna_table_no_duplicates = cna_table_no_duplicates.sort_values(by=['patient_id'])\n",
    "cna_table_no_duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Find the most common 'combTiming_SC' for each unique combination of 'patient_id' and 'Hugo_Symbol'\n",
    "most_common_timing = mut_table.groupby(['patient_id', 'Hugo_Symbol'])['combTiming_SC'].agg(lambda x: x.mode().iat[0] if not x.mode().empty else 'Other').reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient_id</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>combTiming_SC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>A2M</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>A4GALT</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>AARS</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRUK0001</td>\n",
       "      <td>AASS</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338917</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZNF496</td>\n",
       "      <td>early</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338918</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZNF551</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338919</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZNF750</td>\n",
       "      <td>early</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338920</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZNF831</td>\n",
       "      <td>subclonal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338921</th>\n",
       "      <td>CRUK0893</td>\n",
       "      <td>ZP4</td>\n",
       "      <td>early</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>338922 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       patient_id Hugo_Symbol combTiming_SC\n",
       "0        CRUK0001         A2M     subclonal\n",
       "1        CRUK0001      A4GALT     subclonal\n",
       "2        CRUK0001        AARS     subclonal\n",
       "3        CRUK0001      AARSD1     subclonal\n",
       "4        CRUK0001        AASS     subclonal\n",
       "...           ...         ...           ...\n",
       "338917   CRUK0893      ZNF496         early\n",
       "338918   CRUK0893      ZNF551     subclonal\n",
       "338919   CRUK0893      ZNF750         early\n",
       "338920   CRUK0893      ZNF831     subclonal\n",
       "338921   CRUK0893         ZP4         early\n",
       "\n",
       "[338922 rows x 3 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "most_common_timing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>ABCA13</th>\n",
       "      <th>ADAMTS12</th>\n",
       "      <th>ADCY2</th>\n",
       "      <th>AFF2</th>\n",
       "      <th>ANK2</th>\n",
       "      <th>APOB</th>\n",
       "      <th>ASTN1</th>\n",
       "      <th>BAI3</th>\n",
       "      <th>CACNA1B</th>\n",
       "      <th>CACNA1C</th>\n",
       "      <th>CACNA1E</th>\n",
       "      <th>CCDC168</th>\n",
       "      <th>CDH23</th>\n",
       "      <th>CNTNAP2</th>\n",
       "      <th>CNTNAP5</th>\n",
       "      <th>COL11A1</th>\n",
       "      <th>COL22A1</th>\n",
       "      <th>COL3A1</th>\n",
       "      <th>COL4A6</th>\n",
       "      <th>CPS1</th>\n",
       "      <th>CSMD1</th>\n",
       "      <th>CSMD2</th>\n",
       "      <th>CSMD3</th>\n",
       "      <th>CTNNA2</th>\n",
       "      <th>DCDC5</th>\n",
       "      <th>DMD</th>\n",
       "      <th>DNAH11</th>\n",
       "      <th>DNAH5</th>\n",
       "      <th>DNAH9</th>\n",
       "      <th>DOCK2</th>\n",
       "      <th>ELK2AP</th>\n",
       "      <th>EYS</th>\n",
       "      <th>FAM135B</th>\n",
       "      <th>FAT3</th>\n",
       "      <th>FAT4</th>\n",
       "      <th>FBN2</th>\n",
       "      <th>FPGT-TNNI3K</th>\n",
       "      <th>FSIP2</th>\n",
       "      <th>GPR112</th>\n",
       "      <th>HDAC9</th>\n",
       "      <th>HMCN1</th>\n",
       "      <th>ITGA8</th>\n",
       "      <th>KIF1A</th>\n",
       "      <th>KRAS</th>\n",
       "      <th>LAMA1</th>\n",
       "      <th>LAMA2</th>\n",
       "      <th>LOC93432</th>\n",
       "      <th>LPHN3</th>\n",
       "      <th>LRP1B</th>\n",
       "      <th>LRP2</th>\n",
       "      <th>LRRC7</th>\n",
       "      <th>MUC16</th>\n",
       "      <th>MUC19</th>\n",
       "      <th>MUC2</th>\n",
       "      <th>MYO18B</th>\n",
       "      <th>MYO7B</th>\n",
       "      <th>NALCN</th>\n",
       "      <th>NAV3</th>\n",
       "      <th>NCS1</th>\n",
       "      <th>NEB</th>\n",
       "      <th>NRXN1</th>\n",
       "      <th>OBSCN</th>\n",
       "      <th>OR4E2</th>\n",
       "      <th>OTOG</th>\n",
       "      <th>PAPPA2</th>\n",
       "      <th>PCDH15</th>\n",
       "      <th>PCLO</th>\n",
       "      <th>PIEZO2</th>\n",
       "      <th>PKHD1</th>\n",
       "      <th>PKHD1L1</th>\n",
       "      <th>PLXNA4</th>\n",
       "      <th>PTPRD</th>\n",
       "      <th>PTPRT</th>\n",
       "      <th>PXDNL</th>\n",
       "      <th>RELN</th>\n",
       "      <th>RIMS2</th>\n",
       "      <th>RYR1</th>\n",
       "      <th>RYR2</th>\n",
       "      <th>RYR3</th>\n",
       "      <th>SI</th>\n",
       "      <th>SNTG1</th>\n",
       "      <th>SORCS1</th>\n",
       "      <th>SORCS3</th>\n",
       "      <th>SPEF2</th>\n",
       "      <th>SPEG</th>\n",
       "      <th>SPTA1</th>\n",
       "      <th>SSPO</th>\n",
       "      <th>SYNE1</th>\n",
       "      <th>TP53</th>\n",
       "      <th>TRY2P</th>\n",
       "      <th>TTC40</th>\n",
       "      <th>TTN</th>\n",
       "      <th>UNC79</th>\n",
       "      <th>UNC80</th>\n",
       "      <th>USH2A</th>\n",
       "      <th>VCAN</th>\n",
       "      <th>WDFY4</th>\n",
       "      <th>XIRP2</th>\n",
       "      <th>ZFHX4</th>\n",
       "      <th>ZNF536</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CRUK0001</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0004</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0005</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0810</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0817</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0864</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0881</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0893</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>242 rows × 100 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Hugo_Symbol  ABCA13  ADAMTS12  ADCY2  AFF2  ANK2  APOB  ASTN1  BAI3  CACNA1B  \\\n",
       "patient_id                                                                     \n",
       "CRUK0001          1         1      1     1     1     1      0     1        0   \n",
       "CRUK0002          0         0      0     0     0     0      1     0        0   \n",
       "CRUK0003          0         0      0     0     0     0      0     0        1   \n",
       "CRUK0004          1         0      1     0     0     0      1     0        0   \n",
       "CRUK0005          1         1      1     1     1     1      0     0        0   \n",
       "...             ...       ...    ...   ...   ...   ...    ...   ...      ...   \n",
       "CRUK0810          1         0      0     1     0     0      0     0        1   \n",
       "CRUK0817          0         1      0     0     0     0      0     0        0   \n",
       "CRUK0864          0         0      0     0     0     0      0     0        0   \n",
       "CRUK0881          0         1      0     0     1     1      1     0        0   \n",
       "CRUK0893          0         0      0     0     1     1      0     0        0   \n",
       "\n",
       "Hugo_Symbol  CACNA1C  CACNA1E  CCDC168  CDH23  CNTNAP2  CNTNAP5  COL11A1  \\\n",
       "patient_id                                                                 \n",
       "CRUK0001           1        0        0      0        0        1        0   \n",
       "CRUK0002           0        0        0      0        0        0        0   \n",
       "CRUK0003           0        0        0      0        1        0        0   \n",
       "CRUK0004           0        0        1      0        1        0        0   \n",
       "CRUK0005           0        1        1      0        1        0        1   \n",
       "...              ...      ...      ...    ...      ...      ...      ...   \n",
       "CRUK0810           1        1        1      0        0        0        0   \n",
       "CRUK0817           0        1        0      1        0        0        0   \n",
       "CRUK0864           0        0        1      0        0        0        0   \n",
       "CRUK0881           1        1        1      0        1        1        1   \n",
       "CRUK0893           0        1        0      0        1        1        0   \n",
       "\n",
       "Hugo_Symbol  COL22A1  COL3A1  COL4A6  CPS1  CSMD1  CSMD2  CSMD3  CTNNA2  \\\n",
       "patient_id                                                                \n",
       "CRUK0001           1       1       1     0      1      1      1       1   \n",
       "CRUK0002           0       0       0     0      0      1      1       0   \n",
       "CRUK0003           0       0       0     0      0      0      1       0   \n",
       "CRUK0004           0       0       0     0      0      0      1       0   \n",
       "CRUK0005           1       1       0     0      0      1      1       0   \n",
       "...              ...     ...     ...   ...    ...    ...    ...     ...   \n",
       "CRUK0810           1       0       1     0      1      0      1       1   \n",
       "CRUK0817           1       0       0     0      0      0      0       0   \n",
       "CRUK0864           0       0       0     0      0      0      1       0   \n",
       "CRUK0881           0       1       0     1      1      1      1       1   \n",
       "CRUK0893           0       0       0     0      0      0      0       0   \n",
       "\n",
       "Hugo_Symbol  DCDC5  DMD  DNAH11  DNAH5  DNAH9  DOCK2  ELK2AP  EYS  FAM135B  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001         0    1       0      1      0      1       0    1        1   \n",
       "CRUK0002         0    0       0      1      1      1       0    1        0   \n",
       "CRUK0003         0    0       0      0      0      1       1    0        0   \n",
       "CRUK0004         1    0       0      0      0      0       0    0        1   \n",
       "CRUK0005         1    1       0      1      1      1       1    1        1   \n",
       "...            ...  ...     ...    ...    ...    ...     ...  ...      ...   \n",
       "CRUK0810         1    0       0      0      0      0       0    1        0   \n",
       "CRUK0817         1    0       0      0      0      0       0    0        0   \n",
       "CRUK0864         0    1       0      0      1      1       0    0        0   \n",
       "CRUK0881         1    1       1      1      1      1       0    1        1   \n",
       "CRUK0893         0    0       1      0      0      0       0    1        0   \n",
       "\n",
       "Hugo_Symbol  FAT3  FAT4  FBN2  FPGT-TNNI3K  FSIP2  GPR112  HDAC9  HMCN1  \\\n",
       "patient_id                                                                \n",
       "CRUK0001        0     1     1            1      1       0      1      1   \n",
       "CRUK0002        0     0     0            0      0       0      0      0   \n",
       "CRUK0003        0     0     0            0      0       0      0      0   \n",
       "CRUK0004        0     0     0            1      0       0      0      0   \n",
       "CRUK0005        1     1     1            1      1       1      0      1   \n",
       "...           ...   ...   ...          ...    ...     ...    ...    ...   \n",
       "CRUK0810        1     0     1            1      0       0      0      0   \n",
       "CRUK0817        1     0     0            0      1       0      0      0   \n",
       "CRUK0864        0     0     1            0      1       0      0      0   \n",
       "CRUK0881        1     0     1            1      1       1      0      1   \n",
       "CRUK0893        0     0     0            0      1       0      0      0   \n",
       "\n",
       "Hugo_Symbol  ITGA8  KIF1A  KRAS  LAMA1  LAMA2  LOC93432  LPHN3  LRP1B  LRP2  \\\n",
       "patient_id                                                                    \n",
       "CRUK0001         0      1     0      1      1         0      0      0     1   \n",
       "CRUK0002         0      0     0      0      0         0      0      0     0   \n",
       "CRUK0003         0      0     0      0      0         1      0      1     1   \n",
       "CRUK0004         0      0     0      0      0         0      1      1     0   \n",
       "CRUK0005         0      1     1      1      0         1      1      1     1   \n",
       "...            ...    ...   ...    ...    ...       ...    ...    ...   ...   \n",
       "CRUK0810         0      0     1      0      0         1      0      1     0   \n",
       "CRUK0817         0      0     0      0      0         1      0      0     1   \n",
       "CRUK0864         0      0     1      0      1         0      0      1     0   \n",
       "CRUK0881         1      1     1      1      1         1      1      1     1   \n",
       "CRUK0893         1      0     1      0      1         0      0      0     0   \n",
       "\n",
       "Hugo_Symbol  LRRC7  MUC16  MUC19  MUC2  MYO18B  MYO7B  NALCN  NAV3  NCS1  NEB  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001         1      0      1     1       1      0      0     1     1    0   \n",
       "CRUK0002         0      1      0     0       0      0      0     0     0    0   \n",
       "CRUK0003         0      1      0     0       1      0      0     1     0    0   \n",
       "CRUK0004         0      1      0     0       0      0      0     0     0    0   \n",
       "CRUK0005         0      1      0     1       0      1      1     0     0    1   \n",
       "...            ...    ...    ...   ...     ...    ...    ...   ...   ...  ...   \n",
       "CRUK0810         0      1      1     0       0      0      1     1     0    0   \n",
       "CRUK0817         0      0      1     0       0      0      0     0     0    0   \n",
       "CRUK0864         0      1      0     0       0      0      0     0     0    0   \n",
       "CRUK0881         1      1      1     1       0      1      1     1     1    1   \n",
       "CRUK0893         0      0      0     1       0      0      0     0     0    0   \n",
       "\n",
       "Hugo_Symbol  NRXN1  OBSCN  OR4E2  OTOG  PAPPA2  PCDH15  PCLO  PIEZO2  PKHD1  \\\n",
       "patient_id                                                                    \n",
       "CRUK0001         0      1      1     0       0       0     1       0      1   \n",
       "CRUK0002         0      0      0     0       0       0     1       0      0   \n",
       "CRUK0003         0      1      0     0       0       0     1       0      0   \n",
       "CRUK0004         0      1      0     0       0       0     0       0      0   \n",
       "CRUK0005         1      1      1     1       1       1     0       1      1   \n",
       "...            ...    ...    ...   ...     ...     ...   ...     ...    ...   \n",
       "CRUK0810         1      0      0     0       0       0     0       1      0   \n",
       "CRUK0817         1      0      1     0       0       0     1       1      0   \n",
       "CRUK0864         0      0      0     1       0       0     0       0      0   \n",
       "CRUK0881         1      1      1     1       1       1     1       1      1   \n",
       "CRUK0893         1      0      1     1       0       0     0       0      0   \n",
       "\n",
       "Hugo_Symbol  PKHD1L1  PLXNA4  PTPRD  PTPRT  PXDNL  RELN  RIMS2  RYR1  RYR2  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001           0       1      0      0      0     1      1     1     1   \n",
       "CRUK0002           0       1      0      0      0     0      0     0     0   \n",
       "CRUK0003           0       0      1      1      0     1      0     0     1   \n",
       "CRUK0004           0       0      0      0      0     0      0     0     1   \n",
       "CRUK0005           1       1      1      1      1     1      1     0     1   \n",
       "...              ...     ...    ...    ...    ...   ...    ...   ...   ...   \n",
       "CRUK0810           0       0      1      0      0     1      0     0     1   \n",
       "CRUK0817           0       0      0      0      0     0      0     0     1   \n",
       "CRUK0864           0       0      0      0      0     0      0     0     0   \n",
       "CRUK0881           1       1      1      0      1     1      1     1     1   \n",
       "CRUK0893           0       0      0      0      0     0      1     1     1   \n",
       "\n",
       "Hugo_Symbol  RYR3  SI  SNTG1  SORCS1  SORCS3  SPEF2  SPEG  SPTA1  SSPO  SYNE1  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001        1   1      0       1       0      0     0      1     1      1   \n",
       "CRUK0002        1   0      0       0       0      0     0      0     0      0   \n",
       "CRUK0003        0   0      0       0       0      0     0      0     0      0   \n",
       "CRUK0004        0   0      0       0       0      0     0      1     0      0   \n",
       "CRUK0005        0   1      1       1       1      1     0      0     1      0   \n",
       "...           ...  ..    ...     ...     ...    ...   ...    ...   ...    ...   \n",
       "CRUK0810        1   0      0       0       0      0     1      0     0      1   \n",
       "CRUK0817        0   0      0       0       0      0     1      1     0      0   \n",
       "CRUK0864        1   0      0       0       0      0     0      0     0      0   \n",
       "CRUK0881        1   1      1       1       1      1     1      1     1      1   \n",
       "CRUK0893        1   0      0       1       1      0     0      0     1      0   \n",
       "\n",
       "Hugo_Symbol  TP53  TRY2P  TTC40  TTN  UNC79  UNC80  USH2A  VCAN  WDFY4  XIRP2  \\\n",
       "patient_id                                                                      \n",
       "CRUK0001        1      1      0    1      0      1      1     0      1      0   \n",
       "CRUK0002        0      0      0    1      0      0      0     0      1      0   \n",
       "CRUK0003        0      0      0    1      1      0      1     1      0      0   \n",
       "CRUK0004        1      1      0    0      0      0      0     0      0      0   \n",
       "CRUK0005        1      1      0    1      0      1      1     0      1      1   \n",
       "...           ...    ...    ...  ...    ...    ...    ...   ...    ...    ...   \n",
       "CRUK0810        0      0      1    1      0      0      1     0      1      1   \n",
       "CRUK0817        1      0      0    0      0      0      0     0      0      0   \n",
       "CRUK0864        0      0      0    0      1      0      0     0      0      0   \n",
       "CRUK0881        1      0      1    1      0      1      1     1      1      1   \n",
       "CRUK0893        0      0      0    1      0      0      0     0      0      1   \n",
       "\n",
       "Hugo_Symbol  ZFHX4  ZNF536  \n",
       "patient_id                  \n",
       "CRUK0001         0       1  \n",
       "CRUK0002         0       0  \n",
       "CRUK0003         0       0  \n",
       "CRUK0004         0       0  \n",
       "CRUK0005         1       1  \n",
       "...            ...     ...  \n",
       "CRUK0810         1       0  \n",
       "CRUK0817         1       1  \n",
       "CRUK0864         0       0  \n",
       "CRUK0881         1       0  \n",
       "CRUK0893         0       0  \n",
       "\n",
       "[242 rows x 100 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LUAD_gene_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUAD_gene_matrix:\n",
      "Hugo_Symbol  ABCA13  ADAMTS12  ADCY2  AFF2  ANK2  APOB  ASTN1  BAI3  CACNA1B  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001          3         3      1     4     3     3      0     3        0   \n",
      "CRUK0002          0         0      0     0     0     0      3     0        0   \n",
      "CRUK0003          0         0      0     0     0     0      0     0        3   \n",
      "CRUK0004          3         0      1     0     0     0      3     0        0   \n",
      "CRUK0005          1         3      1     4     3     3      0     0        0   \n",
      "...             ...       ...    ...   ...   ...   ...    ...   ...      ...   \n",
      "CRUK0810          1         0      0     4     0     0      0     0        3   \n",
      "CRUK0817          0         1      0     0     0     0      0     0        0   \n",
      "CRUK0864          0         0      0     0     0     0      0     0        0   \n",
      "CRUK0881          0         1      0     0     1     1      3     0        0   \n",
      "CRUK0893          0         0      0     0     1     1      0     0        0   \n",
      "\n",
      "Hugo_Symbol  CACNA1C  CACNA1E  CCDC168  CDH23  CNTNAP2  CNTNAP5  COL11A1  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001           3        0        0      0        0        3        0   \n",
      "CRUK0002           0        0        0      0        0        0        0   \n",
      "CRUK0003           0        0        0      0        3        0        0   \n",
      "CRUK0004           0        0        3      0        1        0        0   \n",
      "CRUK0005           0        3        3      0        3        0        3   \n",
      "...              ...      ...      ...    ...      ...      ...      ...   \n",
      "CRUK0810           3        1        3      0        0        0        0   \n",
      "CRUK0817           0        3        0      1        0        0        0   \n",
      "CRUK0864           0        0        3      0        0        0        0   \n",
      "CRUK0881           3        1        3      0        1        3        1   \n",
      "CRUK0893           0        1        0      0        2        3        0   \n",
      "\n",
      "Hugo_Symbol  COL22A1  COL3A1  COL4A6  CPS1  CSMD1  CSMD2  CSMD3  CTNNA2  \\\n",
      "patient_id                                                                \n",
      "CRUK0001           3       3       4     0      3      3      3       3   \n",
      "CRUK0002           0       0       0     0      0      3      3       0   \n",
      "CRUK0003           0       0       0     0      0      0      3       0   \n",
      "CRUK0004           0       0       0     0      0      0      3       0   \n",
      "CRUK0005           3       3       0     0      0      1      3       0   \n",
      "...              ...     ...     ...   ...    ...    ...    ...     ...   \n",
      "CRUK0810           3       0       4     0      3      0      3       3   \n",
      "CRUK0817           3       0       0     0      0      0      0       0   \n",
      "CRUK0864           0       0       0     0      0      0      2       0   \n",
      "CRUK0881           0       1       0     1      3      1      3       3   \n",
      "CRUK0893           0       0       0     0      0      0      0       0   \n",
      "\n",
      "Hugo_Symbol  DCDC5  DMD  DNAH11  DNAH5  DNAH9  DOCK2  ELK2AP  EYS  FAM135B  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001         0    4       0      3      0      3       0    3        3   \n",
      "CRUK0002         0    0       0      3      3      2       0    4        0   \n",
      "CRUK0003         0    0       0      0      0      3       3    0        0   \n",
      "CRUK0004         3    0       0      0      0      0       0    0        1   \n",
      "CRUK0005         3    4       0      1      3      3       3    3        3   \n",
      "...            ...  ...     ...    ...    ...    ...     ...  ...      ...   \n",
      "CRUK0810         1    0       0      0      0      0       0    3        0   \n",
      "CRUK0817         3    0       0      0      0      0       0    0        0   \n",
      "CRUK0864         0    4       0      0      2      3       0    0        0   \n",
      "CRUK0881         1    4       1      1      3      1       0    1        3   \n",
      "CRUK0893         0    0       1      0      0      0       0    2        0   \n",
      "\n",
      "Hugo_Symbol  FAT3  FAT4  FBN2  FPGT-TNNI3K  FSIP2  GPR112  HDAC9  HMCN1  \\\n",
      "patient_id                                                                \n",
      "CRUK0001        0     3     3            3      3       0      3      3   \n",
      "CRUK0002        0     0     0            0      0       0      0      0   \n",
      "CRUK0003        0     0     0            0      0       0      0      0   \n",
      "CRUK0004        0     0     0            1      0       0      0      0   \n",
      "CRUK0005        3     3     3            3      3       4      0      3   \n",
      "...           ...   ...   ...          ...    ...     ...    ...    ...   \n",
      "CRUK0810        3     0     3            3      0       0      0      0   \n",
      "CRUK0817        1     0     0            0      3       0      0      0   \n",
      "CRUK0864        0     0     1            0      3       0      0      0   \n",
      "CRUK0881        3     0     3            2      1       4      0      3   \n",
      "CRUK0893        0     0     0            0      2       0      0      0   \n",
      "\n",
      "Hugo_Symbol  ITGA8  KIF1A  KRAS  LAMA1  LAMA2  LOC93432  LPHN3  LRP1B  LRP2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      3     0      3      3         0      0      0     3   \n",
      "CRUK0002         0      0     0      0      0         0      0      0     0   \n",
      "CRUK0003         0      0     0      0      0         3      0      1     3   \n",
      "CRUK0004         0      0     0      0      0         0      3      3     0   \n",
      "CRUK0005         0      3     3      1      0         3      3      1     3   \n",
      "...            ...    ...   ...    ...    ...       ...    ...    ...   ...   \n",
      "CRUK0810         0      0     1      0      0         3      0      3     0   \n",
      "CRUK0817         0      0     0      0      0         1      0      0     1   \n",
      "CRUK0864         0      0     1      0      2         0      0      1     0   \n",
      "CRUK0881         1      1     1      1      1         3      1      1     3   \n",
      "CRUK0893         1      0     1      0      1         0      0      0     0   \n",
      "\n",
      "Hugo_Symbol  LRRC7  MUC16  MUC19  MUC2  MYO18B  MYO7B  NALCN  NAV3  NCS1  NEB  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001         3      0      3     3       1      0      0     3     3    0   \n",
      "CRUK0002         0      3      0     0       0      0      0     0     0    0   \n",
      "CRUK0003         0      3      0     0       3      0      0     3     0    0   \n",
      "CRUK0004         0      3      0     0       0      0      0     0     0    0   \n",
      "CRUK0005         0      1      0     3       0      3      3     0     0    3   \n",
      "...            ...    ...    ...   ...     ...    ...    ...   ...   ...  ...   \n",
      "CRUK0810         0      1      3     0       0      0      1     1     0    0   \n",
      "CRUK0817         0      0      3     0       0      0      0     0     0    0   \n",
      "CRUK0864         0      2      0     0       0      0      0     0     0    0   \n",
      "CRUK0881         2      1      1     1       0      1      1     1     1    1   \n",
      "CRUK0893         0      0      0     3       0      0      0     0     0    0   \n",
      "\n",
      "Hugo_Symbol  NRXN1  OBSCN  OR4E2  OTOG  PAPPA2  PCDH15  PCLO  PIEZO2  PKHD1  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      3      3     0       0       0     3       0      3   \n",
      "CRUK0002         0      0      0     0       0       0     3       0      0   \n",
      "CRUK0003         0      3      0     0       0       0     3       0      0   \n",
      "CRUK0004         0      3      0     0       0       0     0       0      0   \n",
      "CRUK0005         3      3      3     1       3       3     0       3      3   \n",
      "...            ...    ...    ...   ...     ...     ...   ...     ...    ...   \n",
      "CRUK0810         3      0      0     0       0       0     0       3      0   \n",
      "CRUK0817         2      0      2     0       0       0     3       1      0   \n",
      "CRUK0864         0      0      0     3       0       0     0       0      0   \n",
      "CRUK0881         3      3      4     3       1       1     3       1      1   \n",
      "CRUK0893         1      0      1     4       0       0     0       0      0   \n",
      "\n",
      "Hugo_Symbol  PKHD1L1  PLXNA4  PTPRD  PTPRT  PXDNL  RELN  RIMS2  RYR1  RYR2  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001           0       3      0      0      0     3      1     3     3   \n",
      "CRUK0002           0       3      0      0      0     0      0     0     0   \n",
      "CRUK0003           0       0      3      1      0     3      0     0     3   \n",
      "CRUK0004           0       0      0      0      0     0      0     0     3   \n",
      "CRUK0005           3       3      3      3      3     3      3     0     3   \n",
      "...              ...     ...    ...    ...    ...   ...    ...   ...   ...   \n",
      "CRUK0810           0       0      1      0      0     1      0     0     3   \n",
      "CRUK0817           0       0      0      0      0     0      0     0     3   \n",
      "CRUK0864           0       0      0      0      0     0      0     0     0   \n",
      "CRUK0881           1       3      1      0      1     1      1     3     3   \n",
      "CRUK0893           0       0      0      0      0     0      1     3     3   \n",
      "\n",
      "Hugo_Symbol  RYR3  SI  SNTG1  SORCS1  SORCS3  SPEF2  SPEG  SPTA1  SSPO  SYNE1  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001        2   4      0       3       0      0     0      3     2      1   \n",
      "CRUK0002        3   0      0       0       0      0     0      0     0      0   \n",
      "CRUK0003        0   0      0       0       0      0     0      0     0      0   \n",
      "CRUK0004        0   0      0       0       0      0     0      3     0      0   \n",
      "CRUK0005        0   3      3       3       3      3     0      0     3      0   \n",
      "...           ...  ..    ...     ...     ...    ...   ...    ...   ...    ...   \n",
      "CRUK0810        2   0      0       0       0      0     3      0     0      3   \n",
      "CRUK0817        0   0      0       0       0      0     1      1     0      0   \n",
      "CRUK0864        1   0      0       0       0      0     0      0     0      0   \n",
      "CRUK0881        1   1      1       3       3      1     1      1     1      3   \n",
      "CRUK0893        4   0      0       3       4      0     0      0     2      0   \n",
      "\n",
      "Hugo_Symbol  TP53  TRY2P  TTC40  TTN  UNC79  UNC80  USH2A  VCAN  WDFY4  XIRP2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001        1      1      0    3      0      3      3     0      3      0   \n",
      "CRUK0002        0      0      0    3      0      0      0     0      3      0   \n",
      "CRUK0003        0      0      0    2      3      0      3     3      0      0   \n",
      "CRUK0004        1      3      0    0      0      0      0     0      0      0   \n",
      "CRUK0005        1      3      0    3      0      3      3     0      3      1   \n",
      "...           ...    ...    ...  ...    ...    ...    ...   ...    ...    ...   \n",
      "CRUK0810        0      0      2    3      0      0      3     0      3      1   \n",
      "CRUK0817        1      0      0    0      0      0      0     0      0      0   \n",
      "CRUK0864        0      0      0    0      3      0      0     0      0      0   \n",
      "CRUK0881        3      0      1    3      0      1      3     1      3      1   \n",
      "CRUK0893        0      0      0    1      0      0      0     0      0      3   \n",
      "\n",
      "Hugo_Symbol  ZFHX4  ZNF536  \n",
      "patient_id                  \n",
      "CRUK0001         0       3  \n",
      "CRUK0002         0       0  \n",
      "CRUK0003         0       0  \n",
      "CRUK0004         0       0  \n",
      "CRUK0005         3       1  \n",
      "...            ...     ...  \n",
      "CRUK0810         3       0  \n",
      "CRUK0817         1       1  \n",
      "CRUK0864         0       0  \n",
      "CRUK0881         3       0  \n",
      "CRUK0893         0       0  \n",
      "\n",
      "[242 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUAD_gene_matrix = LUAD_gene_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUAD_gene_matrix.columns:\n",
    "            if cumulative_LUAD_gene_matrix.loc[cumulative_LUAD_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUAD_gene_matrix.loc[cumulative_LUAD_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUAD_gene_matrix:\")\n",
    "print(cumulative_LUAD_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>ABCA13</th>\n",
       "      <th>ANK2</th>\n",
       "      <th>APOB</th>\n",
       "      <th>CACNA1C</th>\n",
       "      <th>CACNA1E</th>\n",
       "      <th>CCDC168</th>\n",
       "      <th>CNTNAP5</th>\n",
       "      <th>COL11A1</th>\n",
       "      <th>COL22A1</th>\n",
       "      <th>CSMD1</th>\n",
       "      <th>CSMD2</th>\n",
       "      <th>CSMD3</th>\n",
       "      <th>CTNNA2</th>\n",
       "      <th>DMD</th>\n",
       "      <th>DNAH9</th>\n",
       "      <th>DOCK2</th>\n",
       "      <th>FAT3</th>\n",
       "      <th>FBN2</th>\n",
       "      <th>FSIP2</th>\n",
       "      <th>LAMA2</th>\n",
       "      <th>LPHN3</th>\n",
       "      <th>LRP1B</th>\n",
       "      <th>MUC16</th>\n",
       "      <th>MUC19</th>\n",
       "      <th>NAV3</th>\n",
       "      <th>NCS1</th>\n",
       "      <th>NEB</th>\n",
       "      <th>NRXN1</th>\n",
       "      <th>OBSCN</th>\n",
       "      <th>OR4E2</th>\n",
       "      <th>OTOG</th>\n",
       "      <th>PAPPA2</th>\n",
       "      <th>PCDH15</th>\n",
       "      <th>PCLO</th>\n",
       "      <th>PTPRD</th>\n",
       "      <th>PXDNL</th>\n",
       "      <th>RELN</th>\n",
       "      <th>RIMS2</th>\n",
       "      <th>RYR1</th>\n",
       "      <th>RYR2</th>\n",
       "      <th>RYR3</th>\n",
       "      <th>SPTA1</th>\n",
       "      <th>SSPO</th>\n",
       "      <th>SYNE1</th>\n",
       "      <th>TP53</th>\n",
       "      <th>TTN</th>\n",
       "      <th>UNC80</th>\n",
       "      <th>USH2A</th>\n",
       "      <th>XIRP2</th>\n",
       "      <th>ZFHX4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>patient_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CRUK0001</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0002</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0003</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0004</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0005</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0864</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0881</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0893</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0059</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRUK0525</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>242 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Hugo_Symbol  ABCA13  ANK2  APOB  CACNA1C  CACNA1E  CCDC168  CNTNAP5  COL11A1  \\\n",
       "patient_id                                                                     \n",
       "CRUK0001          1     1     1        1        0        0        1        0   \n",
       "CRUK0002          0     0     0        0        0        0        0        0   \n",
       "CRUK0003          0     0     0        0        0        0        0        0   \n",
       "CRUK0004          1     0     0        0        0        1        0        0   \n",
       "CRUK0005          1     1     1        0        1        1        0        1   \n",
       "...             ...   ...   ...      ...      ...      ...      ...      ...   \n",
       "CRUK0864          0     0     0        0        0        1        0        0   \n",
       "CRUK0881          0     1     1        1        1        1        1        1   \n",
       "CRUK0893          0     1     1        0        1        0        1        0   \n",
       "CRUK0059          0     0     0        0        0        0        0        0   \n",
       "CRUK0525          0     0     0        0        0        0        0        0   \n",
       "\n",
       "Hugo_Symbol  COL22A1  CSMD1  CSMD2  CSMD3  CTNNA2  DMD  DNAH9  DOCK2  FAT3  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001           1      1      1      1       1    1      0      1     0   \n",
       "CRUK0002           0      0      1      1       0    0      1      1     0   \n",
       "CRUK0003           0      0      0      1       0    0      0      1     0   \n",
       "CRUK0004           0      0      0      1       0    0      0      0     0   \n",
       "CRUK0005           1      0      1      1       0    1      1      1     1   \n",
       "...              ...    ...    ...    ...     ...  ...    ...    ...   ...   \n",
       "CRUK0864           0      0      0      1       0    1      1      1     0   \n",
       "CRUK0881           0      1      1      1       1    1      1      1     1   \n",
       "CRUK0893           0      0      0      0       0    0      0      0     0   \n",
       "CRUK0059           0      0      0      0       0    0      0      0     0   \n",
       "CRUK0525           0      0      0      0       0    0      0      0     0   \n",
       "\n",
       "Hugo_Symbol  FBN2  FSIP2  LAMA2  LPHN3  LRP1B  MUC16  MUC19  NAV3  NCS1  NEB  \\\n",
       "patient_id                                                                     \n",
       "CRUK0001        1      1      1      0      0      0      1     1     1    0   \n",
       "CRUK0002        0      0      0      0      0      1      0     0     0    0   \n",
       "CRUK0003        0      0      0      0      1      1      0     1     0    0   \n",
       "CRUK0004        0      0      0      1      1      1      0     0     0    0   \n",
       "CRUK0005        1      1      0      1      1      1      0     0     0    1   \n",
       "...           ...    ...    ...    ...    ...    ...    ...   ...   ...  ...   \n",
       "CRUK0864        1      1      1      0      1      1      0     0     0    0   \n",
       "CRUK0881        1      1      1      1      1      1      1     1     1    1   \n",
       "CRUK0893        0      1      1      0      0      0      0     0     0    0   \n",
       "CRUK0059        0      0      0      0      0      0      0     0     0    0   \n",
       "CRUK0525        0      0      0      0      0      0      0     0     0    0   \n",
       "\n",
       "Hugo_Symbol  NRXN1  OBSCN  OR4E2  OTOG  PAPPA2  PCDH15  PCLO  PTPRD  PXDNL  \\\n",
       "patient_id                                                                   \n",
       "CRUK0001         0      1      1     0       0       0     1      0      0   \n",
       "CRUK0002         0      0      0     0       0       0     1      0      0   \n",
       "CRUK0003         0      1      0     0       0       0     1      1      0   \n",
       "CRUK0004         0      1      0     0       0       0     0      0      0   \n",
       "CRUK0005         1      1      1     1       1       1     0      1      1   \n",
       "...            ...    ...    ...   ...     ...     ...   ...    ...    ...   \n",
       "CRUK0864         0      0      0     1       0       0     0      0      0   \n",
       "CRUK0881         1      1      1     1       1       1     1      1      1   \n",
       "CRUK0893         1      0      1     1       0       0     0      0      0   \n",
       "CRUK0059         0      0      0     0       0       0     0      0      0   \n",
       "CRUK0525         0      0      0     0       0       0     0      0      0   \n",
       "\n",
       "Hugo_Symbol  RELN  RIMS2  RYR1  RYR2  RYR3  SPTA1  SSPO  SYNE1  TP53  TTN  \\\n",
       "patient_id                                                                  \n",
       "CRUK0001        1      1     1     1     1      1     1      1     1    1   \n",
       "CRUK0002        0      0     0     0     1      0     0      0     0    1   \n",
       "CRUK0003        1      0     0     1     0      0     0      0     0    1   \n",
       "CRUK0004        0      0     0     1     0      1     0      0     1    0   \n",
       "CRUK0005        1      1     0     1     0      0     1      0     1    1   \n",
       "...           ...    ...   ...   ...   ...    ...   ...    ...   ...  ...   \n",
       "CRUK0864        0      0     0     0     1      0     0      0     0    0   \n",
       "CRUK0881        1      1     1     1     1      1     1      1     1    1   \n",
       "CRUK0893        0      1     1     1     1      0     1      0     0    1   \n",
       "CRUK0059        0      0     0     0     0      0     0      0     0    0   \n",
       "CRUK0525        0      0     0     0     0      0     0      0     0    0   \n",
       "\n",
       "Hugo_Symbol  UNC80  USH2A  XIRP2  ZFHX4  \n",
       "patient_id                               \n",
       "CRUK0001         1      1      0      0  \n",
       "CRUK0002         0      0      0      0  \n",
       "CRUK0003         0      1      0      0  \n",
       "CRUK0004         0      0      0      0  \n",
       "CRUK0005         1      1      1      1  \n",
       "...            ...    ...    ...    ...  \n",
       "CRUK0864         0      0      0      0  \n",
       "CRUK0881         1      1      1      1  \n",
       "CRUK0893         0      0      1      0  \n",
       "CRUK0059         0      0      0      0  \n",
       "CRUK0525         0      0      0      0  \n",
       "\n",
       "[242 rows x 50 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LUAD_fifty_gene_matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUAD_fifty_gene_matrix:\n",
      "Hugo_Symbol  ABCA13  ANK2  APOB  CACNA1C  CACNA1E  CCDC168  CNTNAP5  COL11A1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001          3     3     3        3        0        0        3        0   \n",
      "CRUK0002          0     0     0        0        0        0        0        0   \n",
      "CRUK0003          0     0     0        0        0        0        0        0   \n",
      "CRUK0004          3     0     0        0        0        3        0        0   \n",
      "CRUK0005          1     3     3        0        3        3        0        3   \n",
      "...             ...   ...   ...      ...      ...      ...      ...      ...   \n",
      "CRUK0864          0     0     0        0        0        3        0        0   \n",
      "CRUK0881          0     1     1        3        1        3        3        1   \n",
      "CRUK0893          0     1     1        0        1        0        3        0   \n",
      "CRUK0059          0     0     0        0        0        0        0        0   \n",
      "CRUK0525          0     0     0        0        0        0        0        0   \n",
      "\n",
      "Hugo_Symbol  COL22A1  CSMD1  CSMD2  CSMD3  CTNNA2  DMD  DNAH9  DOCK2  FAT3  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001           3      3      3      3       3    4      0      3     0   \n",
      "CRUK0002           0      0      3      3       0    0      3      2     0   \n",
      "CRUK0003           0      0      0      3       0    0      0      3     0   \n",
      "CRUK0004           0      0      0      3       0    0      0      0     0   \n",
      "CRUK0005           3      0      1      3       0    4      3      3     3   \n",
      "...              ...    ...    ...    ...     ...  ...    ...    ...   ...   \n",
      "CRUK0864           0      0      0      2       0    4      2      3     0   \n",
      "CRUK0881           0      3      1      3       3    4      3      1     3   \n",
      "CRUK0893           0      0      0      0       0    0      0      0     0   \n",
      "CRUK0059           0      0      0      0       0    0      0      0     0   \n",
      "CRUK0525           0      0      0      0       0    0      0      0     0   \n",
      "\n",
      "Hugo_Symbol  FBN2  FSIP2  LAMA2  LPHN3  LRP1B  MUC16  MUC19  NAV3  NCS1  NEB  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001        3      3      3      0      0      0      3     3     3    0   \n",
      "CRUK0002        0      0      0      0      0      3      0     0     0    0   \n",
      "CRUK0003        0      0      0      0      1      3      0     3     0    0   \n",
      "CRUK0004        0      0      0      3      3      3      0     0     0    0   \n",
      "CRUK0005        3      3      0      3      1      1      0     0     0    3   \n",
      "...           ...    ...    ...    ...    ...    ...    ...   ...   ...  ...   \n",
      "CRUK0864        1      3      2      0      1      2      0     0     0    0   \n",
      "CRUK0881        3      1      1      1      1      1      1     1     1    1   \n",
      "CRUK0893        0      2      1      0      0      0      0     0     0    0   \n",
      "CRUK0059        0      0      0      0      0      0      0     0     0    0   \n",
      "CRUK0525        0      0      0      0      0      0      0     0     0    0   \n",
      "\n",
      "Hugo_Symbol  NRXN1  OBSCN  OR4E2  OTOG  PAPPA2  PCDH15  PCLO  PTPRD  PXDNL  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001         0      3      3     0       0       0     3      0      0   \n",
      "CRUK0002         0      0      0     0       0       0     3      0      0   \n",
      "CRUK0003         0      3      0     0       0       0     3      3      0   \n",
      "CRUK0004         0      3      0     0       0       0     0      0      0   \n",
      "CRUK0005         3      3      3     1       3       3     0      3      3   \n",
      "...            ...    ...    ...   ...     ...     ...   ...    ...    ...   \n",
      "CRUK0864         0      0      0     3       0       0     0      0      0   \n",
      "CRUK0881         3      3      4     3       1       1     3      1      1   \n",
      "CRUK0893         1      0      1     4       0       0     0      0      0   \n",
      "CRUK0059         0      0      0     0       0       0     0      0      0   \n",
      "CRUK0525         0      0      0     0       0       0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  RELN  RIMS2  RYR1  RYR2  RYR3  SPTA1  SSPO  SYNE1  TP53  TTN  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        3      1     3     3     2      3     2      1     1    3   \n",
      "CRUK0002        0      0     0     0     3      0     0      0     0    3   \n",
      "CRUK0003        3      0     0     3     0      0     0      0     0    2   \n",
      "CRUK0004        0      0     0     3     0      3     0      0     1    0   \n",
      "CRUK0005        3      3     0     3     0      0     3      0     1    3   \n",
      "...           ...    ...   ...   ...   ...    ...   ...    ...   ...  ...   \n",
      "CRUK0864        0      0     0     0     1      0     0      0     0    0   \n",
      "CRUK0881        1      1     3     3     1      1     1      3     3    3   \n",
      "CRUK0893        0      1     3     3     4      0     2      0     0    1   \n",
      "CRUK0059        0      0     0     0     0      0     0      0     0    0   \n",
      "CRUK0525        0      0     0     0     0      0     0      0     0    0   \n",
      "\n",
      "Hugo_Symbol  UNC80  USH2A  XIRP2  ZFHX4  \n",
      "patient_id                               \n",
      "CRUK0001         3      3      0      0  \n",
      "CRUK0002         0      0      0      0  \n",
      "CRUK0003         0      3      0      0  \n",
      "CRUK0004         0      0      0      0  \n",
      "CRUK0005         3      3      1      3  \n",
      "...            ...    ...    ...    ...  \n",
      "CRUK0864         0      0      0      0  \n",
      "CRUK0881         1      3      1      3  \n",
      "CRUK0893         0      0      3      0  \n",
      "CRUK0059         0      0      0      0  \n",
      "CRUK0525         0      0      0      0  \n",
      "\n",
      "[242 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUAD_fifty_gene_matrix = LUAD_fifty_gene_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUAD_fifty_gene_matrix.columns:\n",
    "            if cumulative_LUAD_fifty_gene_matrix.loc[cumulative_LUAD_fifty_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUAD_fifty_gene_matrix.loc[cumulative_LUAD_fifty_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUAD_fifty_gene_matrix:\")\n",
    "print(cumulative_LUAD_fifty_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUAD_twenty_gene_matrix:\n",
      "Hugo_Symbol  CACNA1E  COL22A1  CSMD1  CSMD3  DMD  LRP1B  MUC16  MUC19  NRXN1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001           0        3      3      3    4      0      0      3      0   \n",
      "CRUK0002           0        0      0      3    0      0      3      0      0   \n",
      "CRUK0003           0        0      0      3    0      1      3      0      0   \n",
      "CRUK0004           0        0      0      3    0      3      3      0      0   \n",
      "CRUK0005           3        3      0      3    4      1      1      0      3   \n",
      "...              ...      ...    ...    ...  ...    ...    ...    ...    ...   \n",
      "CRUK0010           0        0      0      0    0      0      0      0      0   \n",
      "CRUK0038           0        0      0      0    0      0      0      0      0   \n",
      "CRUK0059           0        0      0      0    0      0      0      0      0   \n",
      "CRUK0525           0        0      0      0    0      0      0      0      0   \n",
      "CRUK0386           0        0      0      0    0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  OBSCN  OR4E2  PCDH15  RYR1  RYR2  RYR3  SPTA1  SSPO  TTN  USH2A  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         3      3       0     3     3     2      3     2    3      3   \n",
      "CRUK0002         0      0       0     0     0     3      0     0    3      0   \n",
      "CRUK0003         3      0       0     0     3     0      0     0    2      3   \n",
      "CRUK0004         3      0       0     0     3     0      3     0    0      0   \n",
      "CRUK0005         3      3       3     0     3     0      0     3    3      3   \n",
      "...            ...    ...     ...   ...   ...   ...    ...   ...  ...    ...   \n",
      "CRUK0010         0      0       0     0     0     0      0     0    0      0   \n",
      "CRUK0038         0      0       0     0     0     0      0     0    0      0   \n",
      "CRUK0059         0      0       0     0     0     0      0     0    0      0   \n",
      "CRUK0525         0      0       0     0     0     0      0     0    0      0   \n",
      "CRUK0386         0      0       0     0     0     0      0     0    0      0   \n",
      "\n",
      "Hugo_Symbol  ZFHX4  \n",
      "patient_id          \n",
      "CRUK0001         0  \n",
      "CRUK0002         0  \n",
      "CRUK0003         0  \n",
      "CRUK0004         0  \n",
      "CRUK0005         3  \n",
      "...            ...  \n",
      "CRUK0010         0  \n",
      "CRUK0038         0  \n",
      "CRUK0059         0  \n",
      "CRUK0525         0  \n",
      "CRUK0386         0  \n",
      "\n",
      "[242 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUAD_twenty_gene_matrix = LUAD_twenty_gene_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUAD_twenty_gene_matrix.columns:\n",
    "            if cumulative_LUAD_twenty_gene_matrix.loc[cumulative_LUAD_twenty_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUAD_twenty_gene_matrix.loc[cumulative_LUAD_twenty_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUAD_twenty_gene_matrix:\")\n",
    "print(cumulative_LUAD_twenty_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUAD_driver_matrix:\n",
      "Hugo_Symbol  ACVR1B  AMER1  ANK2  APC  ARHGAP35  ARID1A  ARID1B  ARID2  ASXL2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0001          0      0     0    0         3       0       0      0      0   \n",
      "CRUK0002          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0003          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0004          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0005          0      0     0    0         0       0       0      0      0   \n",
      "...             ...    ...   ...  ...       ...     ...     ...    ...    ...   \n",
      "CRUK0810          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0817          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0864          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0881          0      0     0    0         0       0       0      0      0   \n",
      "CRUK0893          0      0     0    0         0       0       0      0      0   \n",
      "\n",
      "Hugo_Symbol  ATM  ATRX  BAP1  BCLAF1  BCOR  BRAF  CBLB  CDK12  CDKN2A  CHD4  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0002       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0003       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0004       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0005       0     4     0       0     0     1     0      0       0     0   \n",
      "...          ...   ...   ...     ...   ...   ...   ...    ...     ...   ...   \n",
      "CRUK0810       0     4     0       0     0     0     0      0       0     0   \n",
      "CRUK0817       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0864       0     0     0       0     0     0     0      0       0     0   \n",
      "CRUK0881       0     0     0       0     0     1     0      0       0     0   \n",
      "CRUK0893       1     0     0       0     0     0     0      0       0     0   \n",
      "\n",
      "Hugo_Symbol  CHD8  CIC  CMTR2  COL2A1  COL5A2  CREBBP  CTNNB1  CUL3  CUX1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0002        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0003        0    0      0       0       0       0       3     0     0   \n",
      "CRUK0004        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0005        0    0      1       0       0       0       0     0     0   \n",
      "...           ...  ...    ...     ...     ...     ...     ...   ...   ...   \n",
      "CRUK0810        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0817        0    0      0       0       0       0       0     0     0   \n",
      "CRUK0864        0    0      0       0       0       0       0     0     3   \n",
      "CRUK0881        0    0      0       0       0       1       0     3     0   \n",
      "CRUK0893        0    0      0       0       0       0       0     0     0   \n",
      "\n",
      "Hugo_Symbol  DDX3X  DICER1  DNAH12  DNM2  DNMT3A  DOT1L  EGFR  EP300  FANCM  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0       0       0     0       0      0     0      0      0   \n",
      "CRUK0002         0       0       0     0       0      0     0      3      0   \n",
      "CRUK0003         0       0       0     0       0      0     1      0      0   \n",
      "CRUK0004         0       0       0     0       0      0     1      0      0   \n",
      "CRUK0005         0       0       0     0       0      0     0      0      0   \n",
      "...            ...     ...     ...   ...     ...    ...   ...    ...    ...   \n",
      "CRUK0810         0       0       0     0       0      0     0      0      0   \n",
      "CRUK0817         0       0       0     0       0      0     0      3      0   \n",
      "CRUK0864         0       0       0     0       3      0     0      0      0   \n",
      "CRUK0881         4       0       3     0       0      0     0      0      0   \n",
      "CRUK0893         0       0       0     0       0      0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  FAT1  FLT4  GATA3  GRIN2A  IKZF1  KDM5C  KEAP1  KMT2B  KMT2C  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001        0     0      0       0      0      0      0      3      0   \n",
      "CRUK0002        0     0      0       0      3      0      0      0      0   \n",
      "CRUK0003        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0004        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0005        0     0      0       0      0      0      0      0      0   \n",
      "...           ...   ...    ...     ...    ...    ...    ...    ...    ...   \n",
      "CRUK0810        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0817        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0864        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0881        0     0      0       0      0      0      0      0      0   \n",
      "CRUK0893        0     0      0       0      0      0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  KRAS  MAP3K1  MEN1  MET  MGA  MTOR  NBN  NCOA6  NCOR1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0001         0     0       0     0    0    1     0    0      0      0   \n",
      "CRUK0002         0     0       0     0    1    0     0    0      0      0   \n",
      "CRUK0003         0     0       0     0    0    0     0    0      0      0   \n",
      "CRUK0004         0     0       0     0    0    0     0    0      0      0   \n",
      "CRUK0005         0     0       0     0    0    0     0    0      0      0   \n",
      "...            ...   ...     ...   ...  ...  ...   ...  ...    ...    ...   \n",
      "CRUK0810         0     1       0     0    0    0     0    0      0      0   \n",
      "CRUK0817         0     0       0     0    0    0     0    0      0      0   \n",
      "CRUK0864         0     1       0     0    0    0     0    0      0      0   \n",
      "CRUK0881         0     1       0     0    0    0     0    0      0      0   \n",
      "CRUK0893         0     1       0     0    0    0     0    0      0      0   \n",
      "\n",
      "Hugo_Symbol  NF1  NFE2L2  NOTCH2  NRAS  NRXN1  PBRM1  PIK3CA  PIK3R1  PLCG2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001       0       0       0     0      0      0       0       0      0   \n",
      "CRUK0002       3       0       0     0      0      0       0       0      0   \n",
      "CRUK0003       0       0       0     0      0      0       2       0      0   \n",
      "CRUK0004       0       0       0     0      0      0       0       0      0   \n",
      "CRUK0005       0       0       0     3      0      0       0       0      0   \n",
      "...          ...     ...     ...   ...    ...    ...     ...     ...    ...   \n",
      "CRUK0810       0       0       0     0      0      0       0       0      0   \n",
      "CRUK0817       1       0       0     0      0      0       0       0      0   \n",
      "CRUK0864       0       0       0     0      0      0       0       0      0   \n",
      "CRUK0881       1       0       0     0      0      3       0       0      0   \n",
      "CRUK0893       0       0       0     0      0      0       0       0      0   \n",
      "\n",
      "Hugo_Symbol  PLXNB2  PMS1  POLE  POT1  PPP3CA  PTEN  PTPRB  PTPRC  PTPRU  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0002          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0003          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0004          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0005          0     0     0     0       0     0      0      0      0   \n",
      "...             ...   ...   ...   ...     ...   ...    ...    ...    ...   \n",
      "CRUK0810          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0817          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0864          0     0     0     0       0     0      0      0      0   \n",
      "CRUK0881          0     1     0     0       3     0      0      0      0   \n",
      "CRUK0893          0     0     0     0       0     0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  RAD21  RASA1  RB1  RBM10  RNF43  RYR2  SERPINB13  SETD2  SETDB1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0001         0      0    0      0      0     0          0      0       0   \n",
      "CRUK0002         0      0    3      0      0     0          0      0       0   \n",
      "CRUK0003         0      0    0      0      0     0          0      0       0   \n",
      "CRUK0004         0      0    0      0      0     0          0      0       0   \n",
      "CRUK0005         0      0    0      0      0     0          0      0       0   \n",
      "...            ...    ...  ...    ...    ...   ...        ...    ...     ...   \n",
      "CRUK0810         0      0    0      0      0     0          0      0       0   \n",
      "CRUK0817         0      0    0      4      0     0          0      0       0   \n",
      "CRUK0864         0      0    0      4      0     0          0      0       0   \n",
      "CRUK0881         0      0    0      4      0     0          0      0       1   \n",
      "CRUK0893         0      0    0      0      0     0          0      0       0   \n",
      "\n",
      "Hugo_Symbol  SF3B1  SGK223  SMAD4  SMARCA4  SPEN  STAG2  STK11  TET2  TP53  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001         0       0      0        0     0      0      0     0     1   \n",
      "CRUK0002         0       0      0        0     0      0      0     0     0   \n",
      "CRUK0003         0       0      0        0     0      0      0     0     0   \n",
      "CRUK0004         0       0      3        0     0      0      0     0     1   \n",
      "CRUK0005         0       0      0        0     0      0      0     0     1   \n",
      "...            ...     ...    ...      ...   ...    ...    ...   ...   ...   \n",
      "CRUK0810         0       0      0        3     0      0      1     0     0   \n",
      "CRUK0817         0       0      0        0     0      0      3     0     1   \n",
      "CRUK0864         0       0      0        0     0      0      0     0     0   \n",
      "CRUK0881         0       0      0        0     0      0      0     3     3   \n",
      "CRUK0893         0       0      0        0     0      0      0     0     0   \n",
      "\n",
      "Hugo_Symbol  TP53BP1  TSC2  U2AF1  UBR5  WAS  WT1  ZFP36L1  ZMYM2  \n",
      "patient_id                                                         \n",
      "CRUK0001           0     0      0     0    0    0        0      0  \n",
      "CRUK0002           0     0      0     0    0    0        0      0  \n",
      "CRUK0003           0     0      0     0    0    0        0      0  \n",
      "CRUK0004           0     0      0     0    0    0        0      0  \n",
      "CRUK0005           0     0      0     0    0    0        0      0  \n",
      "...              ...   ...    ...   ...  ...  ...      ...    ...  \n",
      "CRUK0810           0     0      0     0    0    0        0      0  \n",
      "CRUK0817           0     0      0     0    0    0        0      0  \n",
      "CRUK0864           0     0      0     0    0    0        0      0  \n",
      "CRUK0881           0     0      0     0    0    0        0      0  \n",
      "CRUK0893           0     0      0     0    0    0        0      0  \n",
      "\n",
      "[242 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUAD_driver_matrix = LUAD_driver_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUAD_driver_matrix.columns:\n",
    "            if cumulative_LUAD_driver_matrix.loc[cumulative_LUAD_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUAD_driver_matrix.loc[cumulative_LUAD_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUAD_driver_matrix:\")\n",
    "print(cumulative_LUAD_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUAD_fifty_driver_matrix:\n",
      "Hugo_Symbol  APC  ARHGAP35  ARID1A  ARID2  ATM  ATRX  BAP1  BCLAF1  BCOR  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001       0         3       0      0    0     0     0       0     0   \n",
      "CRUK0002       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0003       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0004       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0005       0         0       0      0    0     4     0       0     0   \n",
      "...          ...       ...     ...    ...  ...   ...   ...     ...   ...   \n",
      "CRUK0817       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0864       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0881       0         0       0      0    0     0     0       0     0   \n",
      "CRUK0893       0         0       0      0    1     0     0       0     0   \n",
      "CRUK0764       0         0       0      0    0     0     0       0     0   \n",
      "\n",
      "Hugo_Symbol  BRAF  CDKN2A  CHD8  CMTR2  COL2A1  COL5A2  CREBBP  DICER1  \\\n",
      "patient_id                                                               \n",
      "CRUK0001        0       0     0      0       0       0       0       0   \n",
      "CRUK0002        0       0     0      0       0       0       0       0   \n",
      "CRUK0003        0       0     0      0       0       0       0       0   \n",
      "CRUK0004        0       0     0      0       0       0       0       0   \n",
      "CRUK0005        1       0     0      1       0       0       0       0   \n",
      "...           ...     ...   ...    ...     ...     ...     ...     ...   \n",
      "CRUK0817        0       0     0      0       0       0       0       0   \n",
      "CRUK0864        0       0     0      0       0       0       0       0   \n",
      "CRUK0881        1       0     0      0       0       0       1       0   \n",
      "CRUK0893        0       0     0      0       0       0       0       0   \n",
      "CRUK0764        0       0     0      0       0       0       0       0   \n",
      "\n",
      "Hugo_Symbol  DNAH12  DOT1L  EGFR  EP300  FANCM  FAT1  GRIN2A  IKZF1  KDM5C  \\\n",
      "patient_id                                                                   \n",
      "CRUK0001          0      0     0      0      0     0       0      0      0   \n",
      "CRUK0002          0      0     0      3      0     0       0      3      0   \n",
      "CRUK0003          0      0     1      0      0     0       0      0      0   \n",
      "CRUK0004          0      0     1      0      0     0       0      0      0   \n",
      "CRUK0005          0      0     0      0      0     0       0      0      0   \n",
      "...             ...    ...   ...    ...    ...   ...     ...    ...    ...   \n",
      "CRUK0817          0      0     0      3      0     0       0      0      0   \n",
      "CRUK0864          0      0     0      0      0     0       0      0      0   \n",
      "CRUK0881          3      0     0      0      0     0       0      0      0   \n",
      "CRUK0893          0      0     0      0      0     0       0      0      0   \n",
      "CRUK0764          0      0     0      0      0     0       0      0      0   \n",
      "\n",
      "Hugo_Symbol  KEAP1  KMT2C  KMT2D  KRAS  MGA  MTOR  NCOR1  NF1  PIK3CA  POLE  \\\n",
      "patient_id                                                                    \n",
      "CRUK0001         0      0      0     0    1     0      0    0       0     0   \n",
      "CRUK0002         0      0      0     0    0     0      0    3       0     0   \n",
      "CRUK0003         0      0      0     0    0     0      0    0       2     0   \n",
      "CRUK0004         0      0      0     0    0     0      0    0       0     0   \n",
      "CRUK0005         0      0      0     0    0     0      0    0       0     0   \n",
      "...            ...    ...    ...   ...  ...   ...    ...  ...     ...   ...   \n",
      "CRUK0817         0      0      0     0    0     0      0    1       0     0   \n",
      "CRUK0864         0      0      0     1    0     0      0    0       0     0   \n",
      "CRUK0881         0      0      0     1    0     0      0    1       0     0   \n",
      "CRUK0893         0      0      0     1    0     0      0    0       0     0   \n",
      "CRUK0764         0      0      0     0    0     0      0    0       0     0   \n",
      "\n",
      "Hugo_Symbol  POT1  PTPRB  PTPRC  RB1  RBM10  SETD2  SMAD4  SMARCA4  SPEN  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001        0      0      0    0      0      0      0        0     0   \n",
      "CRUK0002        0      0      0    3      0      0      0        0     0   \n",
      "CRUK0003        0      0      0    0      0      0      0        0     0   \n",
      "CRUK0004        0      0      0    0      0      0      3        0     0   \n",
      "CRUK0005        0      0      0    0      0      0      0        0     0   \n",
      "...           ...    ...    ...  ...    ...    ...    ...      ...   ...   \n",
      "CRUK0817        0      0      0    0      4      0      0        0     0   \n",
      "CRUK0864        0      0      0    0      4      0      0        0     0   \n",
      "CRUK0881        0      0      0    0      4      0      0        0     0   \n",
      "CRUK0893        0      0      0    0      0      0      0        0     0   \n",
      "CRUK0764        0      0      0    0      0      0      0        0     0   \n",
      "\n",
      "Hugo_Symbol  STK11  TET2  TP53  UBR5  ZFP36L1  \n",
      "patient_id                                     \n",
      "CRUK0001         0     0     1     0        0  \n",
      "CRUK0002         0     0     0     0        0  \n",
      "CRUK0003         0     0     0     0        0  \n",
      "CRUK0004         0     0     1     0        0  \n",
      "CRUK0005         0     0     1     0        0  \n",
      "...            ...   ...   ...   ...      ...  \n",
      "CRUK0817         3     0     1     0        0  \n",
      "CRUK0864         0     0     0     0        0  \n",
      "CRUK0881         0     3     3     0        0  \n",
      "CRUK0893         0     0     0     0        0  \n",
      "CRUK0764         0     0     0     0        0  \n",
      "\n",
      "[242 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUAD_fifty_driver_matrix = LUAD_fifty_driver_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUAD_fifty_driver_matrix.columns:\n",
    "            if cumulative_LUAD_fifty_driver_matrix.loc[cumulative_LUAD_fifty_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUAD_fifty_driver_matrix.loc[cumulative_LUAD_fifty_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUAD_fifty_driver_matrix:\")\n",
    "print(cumulative_LUAD_fifty_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUAD_twenty_driver_matrix:\n",
      "Hugo_Symbol  ARID2  ATM  ATRX  CDKN2A  EGFR  FAT1  GRIN2A  KEAP1  KMT2C  \\\n",
      "patient_id                                                                \n",
      "CRUK0001         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0002         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0003         0    0     0       0     1     0       0      0      0   \n",
      "CRUK0004         0    0     0       0     1     0       0      0      0   \n",
      "CRUK0005         0    0     4       0     0     0       0      0      0   \n",
      "...            ...  ...   ...     ...   ...   ...     ...    ...    ...   \n",
      "CRUK0045         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0041         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0742         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0764         0    0     0       0     0     0       0      0      0   \n",
      "CRUK0043         0    0     0       0     0     0       0      0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  KRAS  MGA  NCOR1  NF1  PIK3CA  PTPRB  RBM10  SMARCA4  \\\n",
      "patient_id                                                                 \n",
      "CRUK0001         0     0    1      0    0       0      0      0        0   \n",
      "CRUK0002         0     0    0      0    3       0      0      0        0   \n",
      "CRUK0003         0     0    0      0    0       2      0      0        0   \n",
      "CRUK0004         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0005         0     0    0      0    0       0      0      0        0   \n",
      "...            ...   ...  ...    ...  ...     ...    ...    ...      ...   \n",
      "CRUK0045         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0041         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0742         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0764         0     0    0      0    0       0      0      0        0   \n",
      "CRUK0043         0     0    0      0    0       0      0      0        0   \n",
      "\n",
      "Hugo_Symbol  STK11  TP53  \n",
      "patient_id                \n",
      "CRUK0001         0     1  \n",
      "CRUK0002         0     0  \n",
      "CRUK0003         0     0  \n",
      "CRUK0004         0     1  \n",
      "CRUK0005         0     1  \n",
      "...            ...   ...  \n",
      "CRUK0045         0     0  \n",
      "CRUK0041         0     0  \n",
      "CRUK0742         0     0  \n",
      "CRUK0764         0     0  \n",
      "CRUK0043         0     0  \n",
      "\n",
      "[242 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUAD_twenty_driver_matrix = LUAD_twenty_driver_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUAD_twenty_driver_matrix.columns:\n",
    "            if cumulative_LUAD_twenty_driver_matrix.loc[cumulative_LUAD_twenty_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUAD_twenty_driver_matrix.loc[cumulative_LUAD_twenty_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUAD_twenty_driver_matrix:\")\n",
    "print(cumulative_LUAD_twenty_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUAD_ten_driver_matrix:\n",
      "Hugo_Symbol  EGFR  FAT1  KEAP1  KRAS  MGA  NF1  RBM10  SMARCA4  STK11  TP53\n",
      "patient_id                                                                 \n",
      "CRUK0001        0     0      0     0    1    0      0        0      0     1\n",
      "CRUK0002        0     0      0     0    0    3      0        0      0     0\n",
      "CRUK0003        1     0      0     0    0    0      0        0      0     0\n",
      "CRUK0004        1     0      0     0    0    0      0        0      0     1\n",
      "CRUK0005        0     0      0     0    0    0      0        0      0     1\n",
      "...           ...   ...    ...   ...  ...  ...    ...      ...    ...   ...\n",
      "CRUK0764        0     0      0     0    0    0      0        0      0     0\n",
      "CRUK0193        0     0      0     0    0    0      0        0      0     0\n",
      "CRUK0043        0     0      0     0    0    0      0        0      0     0\n",
      "CRUK0730        0     0      0     0    0    0      0        0      0     0\n",
      "CRUK0253        0     0      0     0    0    0      0        0      0     0\n",
      "\n",
      "[242 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUAD_ten_driver_matrix = LUAD_ten_driver_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUAD_ten_driver_matrix.columns:\n",
    "            if cumulative_LUAD_ten_driver_matrix.loc[cumulative_LUAD_ten_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUAD_ten_driver_matrix.loc[cumulative_LUAD_ten_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUAD_ten_driver_matrix:\")\n",
    "print(cumulative_LUAD_ten_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add ploidy\n",
    "cumulative_LUAD_gene_matrix['Ploidy'] = cumulative_LUAD_gene_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUAD_gene_matrix = cumulative_LUAD_gene_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUAD_driver_matrix['Ploidy'] = cumulative_LUAD_driver_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUAD_driver_matrix = cumulative_LUAD_driver_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUAD_fifty_gene_matrix['Ploidy'] = cumulative_LUAD_fifty_gene_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUAD_fifty_gene_matrix = cumulative_LUAD_fifty_gene_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUAD_fifty_driver_matrix['Ploidy'] = cumulative_LUAD_fifty_driver_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUAD_fifty_driver_matrix = cumulative_LUAD_fifty_driver_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUAD_twenty_gene_matrix['Ploidy'] = cumulative_LUAD_twenty_gene_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUAD_twenty_gene_matrix = cumulative_LUAD_twenty_gene_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUAD_twenty_driver_matrix['Ploidy'] = cumulative_LUAD_twenty_driver_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUAD_twenty_driver_matrix = cumulative_LUAD_twenty_driver_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUAD_ten_driver_matrix['Ploidy'] = cumulative_LUAD_ten_driver_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUAD_ten_driver_matrix = cumulative_LUAD_ten_driver_matrix.dropna(subset=['Ploidy'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUSC_gene_matrix:\n",
      "Hugo_Symbol  ABCA13  ADAMTS12  ADAMTS16  ADGB  ANK1  ANK2  ANKS1B  APOB  BAI3  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062          0         0         1     0     3     3       3     0     0   \n",
      "CRUK0063          0         3         0     0     0     0       0     0     0   \n",
      "CRUK0064          0         0         1     0     0     0       1     3     0   \n",
      "CRUK0065          3         0         0     0     3     3       0     3     4   \n",
      "CRUK0066          1         3         1     1     3     0       0     0     0   \n",
      "...             ...       ...       ...   ...   ...   ...     ...   ...   ...   \n",
      "CRUK0772          3         0         0     2     0     0       0     2     0   \n",
      "CRUK0781          4         4         0     0     0     0       0     0     0   \n",
      "CRUK0784          0         1         0     1     0     0       0     0     0   \n",
      "CRUK0861          0         0         0     0     0     2       0     2     1   \n",
      "CRUK0872          0         0         0     3     0     0       1     1     0   \n",
      "\n",
      "Hugo_Symbol  BRINP3  CACNA1C  CACNA1E  CCDC168  CDH10  CNTNAP5  COL11A1  \\\n",
      "patient_id                                                                \n",
      "CRUK0062          0        3        3        0      1        0        0   \n",
      "CRUK0063          0        0        0        0      0        0        0   \n",
      "CRUK0064          0        0        0        0      0        0        0   \n",
      "CRUK0065          0        0        3        0      4        0        0   \n",
      "CRUK0066          4        0        1        0      1        3        3   \n",
      "...             ...      ...      ...      ...    ...      ...      ...   \n",
      "CRUK0772          0        0        0        0      0        0        1   \n",
      "CRUK0781          4        0        4        0      0        4        4   \n",
      "CRUK0784          0        1        0        3      0        1        1   \n",
      "CRUK0861          0        0        0        0      0        1        0   \n",
      "CRUK0872          0        0        0        0      0        0        3   \n",
      "\n",
      "Hugo_Symbol  COL22A1  COL5A2  COL6A5  CPS1  CSMD1  CSMD2  CSMD3  CTNNA2  CUBN  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062           1       0       0     0      0      3      1       1     0   \n",
      "CRUK0063           0       0       0     0      0      0      0       0     0   \n",
      "CRUK0064           1       0       0     1      0      3      0       0     0   \n",
      "CRUK0065           3       0       0     0      0      0      3       0     0   \n",
      "CRUK0066           1       1       1     0      0      0      1       0     0   \n",
      "...              ...     ...     ...   ...    ...    ...    ...     ...   ...   \n",
      "CRUK0772           1       0       0     0      0      0      0       0     1   \n",
      "CRUK0781           4       0       0     0      0      4      4       0     0   \n",
      "CRUK0784           0       0       1     1      0      3      2       0     0   \n",
      "CRUK0861           0       0       0     0      0      0      1       0     0   \n",
      "CRUK0872           1       1       1     0      0      0      1       0     1   \n",
      "\n",
      "Hugo_Symbol  DMD  DNAH11  DNAH5  DNAH6  DNAH8  DNAH9  DPP10  DPP6  EYS  \\\n",
      "patient_id                                                               \n",
      "CRUK0062       0       0      3      0      3      0      3     0    3   \n",
      "CRUK0063       0       0      3      0      3      3      0     0    0   \n",
      "CRUK0064       0       0      0      3      0      0      0     1    0   \n",
      "CRUK0065       4       0      4      0      0      4      0     4    4   \n",
      "CRUK0066       4       0      3      0      0      0      0     2    2   \n",
      "...          ...     ...    ...    ...    ...    ...    ...   ...  ...   \n",
      "CRUK0772       0       2      0      0      2      0      0     0    0   \n",
      "CRUK0781       0       0      4      4      0      0      0     0    4   \n",
      "CRUK0784       0       3      0      0      0      0      3     0    3   \n",
      "CRUK0861       0       0      0      2      0      1      0     0    3   \n",
      "CRUK0872       0       0      1      0      0      0      0     0    0   \n",
      "\n",
      "Hugo_Symbol  FAM135B  FAT3  FAT4  FBN2  FPGT-TNNI3K  FSIP2  HCN1  HMCN1  \\\n",
      "patient_id                                                                \n",
      "CRUK0062           3     0     0     3            3      3     3      0   \n",
      "CRUK0063           0     0     0     0            3      3     0      3   \n",
      "CRUK0064           0     1     0     0            0      0     1      3   \n",
      "CRUK0065           0     0     3     0            0      3     4      3   \n",
      "CRUK0066           0     0     0     0            0      0     3      0   \n",
      "...              ...   ...   ...   ...          ...    ...   ...    ...   \n",
      "CRUK0772           3     0     0     0            1      0     0      2   \n",
      "CRUK0781           4     0     0     4            0      0     0      0   \n",
      "CRUK0784           0     3     0     3            0      3     1      0   \n",
      "CRUK0861           0     0     0     4            0      0     0      0   \n",
      "CRUK0872           0     0     0     0            1      0     1      0   \n",
      "\n",
      "Hugo_Symbol  LAMA2  LOC93432  LPHN3  LRP1B  LRP2  LRRC7  MALRD1  MIR548N  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062         3         3      0      3     0      3       0        0   \n",
      "CRUK0063         0         0      3      3     0      0       0        0   \n",
      "CRUK0064         3         0      3      1     1      0       0        3   \n",
      "CRUK0065         0         0      0      0     4      0       0        4   \n",
      "CRUK0066         0         0      0      3     0      0       0        0   \n",
      "...            ...       ...    ...    ...   ...    ...     ...      ...   \n",
      "CRUK0772         2         0      0      1     2      0       1        0   \n",
      "CRUK0781         0         0      4      4     4      4       4        4   \n",
      "CRUK0784         3         0      1      1     0      0       2        1   \n",
      "CRUK0861         0         0      0      0     1      0       0        0   \n",
      "CRUK0872         0         0      1      1     1      0       0        0   \n",
      "\n",
      "Hugo_Symbol  MORC1  MROH2B  MUC16  MUC19  MYO16  NAV3  NCS1  NEB  NRXN1  \\\n",
      "patient_id                                                                \n",
      "CRUK0062         0       3      3      0      3     0     3    3      0   \n",
      "CRUK0063         0       0      0      0      0     0     0    0      0   \n",
      "CRUK0064         0       0      3      1      0     0     0    0      0   \n",
      "CRUK0065         0       0      3      4      3     4     4    4      0   \n",
      "CRUK0066         1       1      3      1      0     0     0    0      1   \n",
      "...            ...     ...    ...    ...    ...   ...   ...  ...    ...   \n",
      "CRUK0772         1       0      3      1      0     0     3    0      0   \n",
      "CRUK0781         0       0      4      0      0     0     0    0      0   \n",
      "CRUK0784         0       0      1      0      3     0     0    0      1   \n",
      "CRUK0861         0       0      1      0      0     0     0    1      0   \n",
      "CRUK0872         0       1      1      1      1     0     0    0      0   \n",
      "\n",
      "Hugo_Symbol  OBSCN  OR4E2  OTOG  OTOGL  PAPPA2  PCDH15  PCLO  PKHD1  PKHD1L1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062         3      3     0      0       3       3     0      1        3   \n",
      "CRUK0063         1      3     0      0       0       0     0      0        0   \n",
      "CRUK0064         0      1     0      0       0       0     0      3        0   \n",
      "CRUK0065         4      0     0      0       3       0     0      4        0   \n",
      "CRUK0066         0      0     0      0       0       4     0      1        3   \n",
      "...            ...    ...   ...    ...     ...     ...   ...    ...      ...   \n",
      "CRUK0772         0      1     0      0       0       0     0      4        0   \n",
      "CRUK0781         0      0     0      0       4       4     0      0        4   \n",
      "CRUK0784         1      0     0      0       0       0     0      1        1   \n",
      "CRUK0861         0      0     0      0       0       0     0      0        2   \n",
      "CRUK0872         0      0     0      3       0       1     0      0        0   \n",
      "\n",
      "Hugo_Symbol  PLEC  PLXNA4  PPFIA2  RELN  RIMS2  ROS1  RYR1  RYR2  RYR3  SI  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062        3       0       0     3      0     3     0     3     1   3   \n",
      "CRUK0063        0       0       3     0      0     0     1     0     0   3   \n",
      "CRUK0064        1       0       0     3      3     0     0     1     0   0   \n",
      "CRUK0065        3       4       4     4      0     3     3     4     4   1   \n",
      "CRUK0066        0       0       0     0      0     3     3     3     0   0   \n",
      "...           ...     ...     ...   ...    ...   ...   ...   ...   ...  ..   \n",
      "CRUK0772        3       0       0     0      0     0     0     1     0   0   \n",
      "CRUK0781        0       4       4     0      4     0     0     4     4   0   \n",
      "CRUK0784        0       3       0     1      0     3     1     3     0   1   \n",
      "CRUK0861        0       0       0     1      0     0     0     1     2   0   \n",
      "CRUK0872        0       0       0     1      0     3     0     1     0   1   \n",
      "\n",
      "Hugo_Symbol  SORCS1  SPEF2  SPHKAP  SPTA1  SSPO  STAB2  SYNE1  TENM3  THSD7B  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062          0      0       0      0     0      0      3      0       0   \n",
      "CRUK0063          0      0       0      0     0      0      0      3       0   \n",
      "CRUK0064          0      0       0      1     0      0      2      0       0   \n",
      "CRUK0065          0      3       0      3     0      0      3      0       0   \n",
      "CRUK0066          1      1       0      1     0      0      1      0       0   \n",
      "...             ...    ...     ...    ...   ...    ...    ...    ...     ...   \n",
      "CRUK0772          0      1       0      0     0      0      0      0       0   \n",
      "CRUK0781          0      0       0      4     4      0      0      0       4   \n",
      "CRUK0784          1      0       0      0     1      0      0      0       0   \n",
      "CRUK0861          0      1       0      2     1      0      1      0       0   \n",
      "CRUK0872          0      0       0      0     1      0      0      0       0   \n",
      "\n",
      "Hugo_Symbol  TNN  TNR  TP53  TRDN  TRIO  TTN  UNC13C  UNC80  USH2A  UTRN  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062       3    0     1     3     1    3       0      3      3     0   \n",
      "CRUK0063       0    3     1     0     0    3       0      0      0     0   \n",
      "CRUK0064       0    0     1     0     0    3       0      1      0     0   \n",
      "CRUK0065       0    3     4     0     4    3       0      3      3     4   \n",
      "CRUK0066       0    1     1     0     0    3       0      0      0     0   \n",
      "...          ...  ...   ...   ...   ...  ...     ...    ...    ...   ...   \n",
      "CRUK0772       0    0     1     0     0    1       0      3      3     0   \n",
      "CRUK0781       4    0     4     0     0    4       4      0      4     0   \n",
      "CRUK0784       0    1     2     1     0    1       0      3      1     0   \n",
      "CRUK0861       0    1     1     0     0    1       0      0      0     0   \n",
      "CRUK0872       0    1     1     0     0    3       0      1      1     0   \n",
      "\n",
      "Hugo_Symbol  WDFY4  XIRP2  ZFHX4  \n",
      "patient_id                        \n",
      "CRUK0062         0      0      1  \n",
      "CRUK0063         0      0      0  \n",
      "CRUK0064         3      0      3  \n",
      "CRUK0065         0      4      4  \n",
      "CRUK0066         0      2      3  \n",
      "...            ...    ...    ...  \n",
      "CRUK0772         0      2      0  \n",
      "CRUK0781         0      0      4  \n",
      "CRUK0784         0      0      3  \n",
      "CRUK0861         0      0      0  \n",
      "CRUK0872         0      0      1  \n",
      "\n",
      "[138 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "#repeat the same with LUSC\n",
    "\n",
    "# add timing\n",
    "cumulative_LUSC_gene_matrix = LUSC_gene_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUSC_gene_matrix.columns:\n",
    "            if cumulative_LUSC_gene_matrix.loc[cumulative_LUSC_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUSC_gene_matrix.loc[cumulative_LUSC_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUSC_gene_matrix:\")\n",
    "print(cumulative_LUSC_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUSC_fifty_gene_matrix:\n",
      "Hugo_Symbol  ANK2  APOB  CACNA1E  COL11A1  COL22A1  CSMD1  CSMD2  CSMD3  \\\n",
      "patient_id                                                                \n",
      "CRUK0062        3     0        3        0        1      0      3      1   \n",
      "CRUK0063        0     0        0        0        0      0      0      0   \n",
      "CRUK0064        0     3        0        0        1      0      3      0   \n",
      "CRUK0065        3     3        3        0        3      0      0      3   \n",
      "CRUK0066        0     0        1        3        1      0      0      1   \n",
      "...           ...   ...      ...      ...      ...    ...    ...    ...   \n",
      "CRUK0772        0     2        0        1        1      0      0      0   \n",
      "CRUK0781        0     0        4        4        4      0      4      4   \n",
      "CRUK0784        0     0        0        1        0      0      3      2   \n",
      "CRUK0861        2     2        0        0        0      0      0      1   \n",
      "CRUK0872        0     1        0        3        1      0      0      1   \n",
      "\n",
      "Hugo_Symbol  CTNNA2  DMD  DNAH5  DNAH6  DNAH9  DPP6  EYS  FAM135B  FSIP2  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062          1    0      3      0      0     0    3        3      3   \n",
      "CRUK0063          0    0      3      0      3     0    0        0      3   \n",
      "CRUK0064          0    0      0      3      0     1    0        0      0   \n",
      "CRUK0065          0    4      4      0      4     4    4        0      3   \n",
      "CRUK0066          0    4      3      0      0     2    2        0      0   \n",
      "...             ...  ...    ...    ...    ...   ...  ...      ...    ...   \n",
      "CRUK0772          0    0      0      0      0     0    0        3      0   \n",
      "CRUK0781          0    0      4      4      0     0    4        4      0   \n",
      "CRUK0784          0    0      0      0      0     0    3        0      3   \n",
      "CRUK0861          0    0      0      2      1     0    3        0      0   \n",
      "CRUK0872          0    0      1      0      0     0    0        0      0   \n",
      "\n",
      "Hugo_Symbol  HCN1  HMCN1  LPHN3  LRP1B  LRP2  MIR548N  MROH2B  MUC16  MUC19  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062        3      0      0      3     0        0       3      3      0   \n",
      "CRUK0063        0      3      3      3     0        0       0      0      0   \n",
      "CRUK0064        1      3      3      1     1        3       0      3      1   \n",
      "CRUK0065        4      3      0      0     4        4       0      3      4   \n",
      "CRUK0066        3      0      0      3     0        0       1      3      1   \n",
      "...           ...    ...    ...    ...   ...      ...     ...    ...    ...   \n",
      "CRUK0772        0      2      0      1     2        0       0      3      1   \n",
      "CRUK0781        0      0      4      4     4        4       0      4      0   \n",
      "CRUK0784        1      0      1      1     0        1       0      1      0   \n",
      "CRUK0861        0      0      0      0     1        0       0      1      0   \n",
      "CRUK0872        1      0      1      1     1        0       1      1      1   \n",
      "\n",
      "Hugo_Symbol  NAV3  NEB  NRXN1  OBSCN  OR4E2  PCDH15  PCLO  PKHD1  PKHD1L1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062        0    3      0      3      3       3     0      1        3   \n",
      "CRUK0063        0    0      0      1      3       0     0      0        0   \n",
      "CRUK0064        0    0      0      0      1       0     0      3        0   \n",
      "CRUK0065        4    4      0      4      0       0     0      4        0   \n",
      "CRUK0066        0    0      1      0      0       4     0      1        3   \n",
      "...           ...  ...    ...    ...    ...     ...   ...    ...      ...   \n",
      "CRUK0772        0    0      0      0      1       0     0      4        0   \n",
      "CRUK0781        0    0      0      0      0       4     0      0        4   \n",
      "CRUK0784        0    0      1      1      0       0     0      1        1   \n",
      "CRUK0861        0    1      0      0      0       0     0      0        2   \n",
      "CRUK0872        0    0      0      0      0       1     0      0        0   \n",
      "\n",
      "Hugo_Symbol  PLEC  RELN  RYR1  RYR2  RYR3  SI  SPTA1  SSPO  SYNE1  TP53  TTN  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062        3     3     0     3     1   3      0     0      3     1    3   \n",
      "CRUK0063        0     0     1     0     0   3      0     0      0     1    3   \n",
      "CRUK0064        1     3     0     1     0   0      1     0      2     1    3   \n",
      "CRUK0065        3     4     3     4     4   1      3     0      3     4    3   \n",
      "CRUK0066        0     0     3     3     0   0      1     0      1     1    3   \n",
      "...           ...   ...   ...   ...   ...  ..    ...   ...    ...   ...  ...   \n",
      "CRUK0772        3     0     0     1     0   0      0     0      0     1    1   \n",
      "CRUK0781        0     0     0     4     4   0      4     4      0     4    4   \n",
      "CRUK0784        0     1     1     3     0   1      0     1      0     2    1   \n",
      "CRUK0861        0     1     0     1     2   0      2     1      1     1    1   \n",
      "CRUK0872        0     1     0     1     0   1      0     1      0     1    3   \n",
      "\n",
      "Hugo_Symbol  UNC80  USH2A  XIRP2  ZFHX4  \n",
      "patient_id                               \n",
      "CRUK0062         3      3      0      1  \n",
      "CRUK0063         0      0      0      0  \n",
      "CRUK0064         1      0      0      3  \n",
      "CRUK0065         3      3      4      4  \n",
      "CRUK0066         0      0      2      3  \n",
      "...            ...    ...    ...    ...  \n",
      "CRUK0772         3      3      2      0  \n",
      "CRUK0781         0      4      0      4  \n",
      "CRUK0784         3      1      0      3  \n",
      "CRUK0861         0      0      0      0  \n",
      "CRUK0872         1      1      0      1  \n",
      "\n",
      "[138 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUSC_fifty_gene_matrix = LUSC_fifty_gene_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUSC_fifty_gene_matrix.columns:\n",
    "            if cumulative_LUSC_fifty_gene_matrix.loc[cumulative_LUSC_fifty_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUSC_fifty_gene_matrix.loc[cumulative_LUSC_fifty_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUSC_fifty_gene_matrix:\")\n",
    "print(cumulative_LUSC_fifty_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUSC_twenty_gene_matrix:\n",
      "Hugo_Symbol  COL11A1  COL22A1  CSMD3  DMD  LRP1B  MUC16  MUC19  OBSCN  OR4E2  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062           0        1      1    0      3      3      0      3      3   \n",
      "CRUK0063           0        0      0    0      3      0      0      1      3   \n",
      "CRUK0064           0        1      0    0      1      3      1      0      1   \n",
      "CRUK0065           0        3      3    4      0      3      4      4      0   \n",
      "CRUK0066           3        1      1    4      3      3      1      0      0   \n",
      "...              ...      ...    ...  ...    ...    ...    ...    ...    ...   \n",
      "CRUK0772           1        1      0    0      1      3      1      0      1   \n",
      "CRUK0781           4        4      4    0      4      4      0      0      0   \n",
      "CRUK0784           1        0      2    0      1      1      0      1      0   \n",
      "CRUK0861           0        0      1    0      0      1      0      0      0   \n",
      "CRUK0872           3        1      1    0      1      1      1      0      0   \n",
      "\n",
      "Hugo_Symbol  RELN  RYR2  RYR3  SI  SPTA1  SSPO  SYNE1  TP53  TTN  USH2A  ZFHX4  \n",
      "patient_id                                                                      \n",
      "CRUK0062        3     3     1   3      0     0      3     1    3      3      1  \n",
      "CRUK0063        0     0     0   3      0     0      0     1    3      0      0  \n",
      "CRUK0064        3     1     0   0      1     0      2     1    3      0      3  \n",
      "CRUK0065        4     4     4   1      3     0      3     4    3      3      4  \n",
      "CRUK0066        0     3     0   0      1     0      1     1    3      0      3  \n",
      "...           ...   ...   ...  ..    ...   ...    ...   ...  ...    ...    ...  \n",
      "CRUK0772        0     1     0   0      0     0      0     1    1      3      0  \n",
      "CRUK0781        0     4     4   0      4     4      0     4    4      4      4  \n",
      "CRUK0784        1     3     0   1      0     1      0     2    1      1      3  \n",
      "CRUK0861        1     1     2   0      2     1      1     1    1      0      0  \n",
      "CRUK0872        1     1     0   1      0     1      0     1    3      1      1  \n",
      "\n",
      "[138 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUSC_twenty_gene_matrix = LUSC_twenty_gene_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUSC_twenty_gene_matrix.columns:\n",
    "            if cumulative_LUSC_twenty_gene_matrix.loc[cumulative_LUSC_twenty_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUSC_twenty_gene_matrix.loc[cumulative_LUSC_twenty_gene_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUSC_twenty_gene_matrix:\")\n",
    "print(cumulative_LUSC_twenty_gene_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUSC_driver_matrix:\n",
      "Hugo_Symbol  ACVR1B  AMER1  APC  ARHGAP35  ARID1A  ARID2  ASXL1  ASXL2  ATM  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0063          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0064          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0065          0      0    0         0       3      0      0      0    0   \n",
      "CRUK0066          0      0    0         0       0      0      0      0    0   \n",
      "...             ...    ...  ...       ...     ...    ...    ...    ...  ...   \n",
      "CRUK0772          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0781          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0784          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0861          0      0    0         0       0      0      0      0    0   \n",
      "CRUK0872          0      0    0         0       0      0      0      0    0   \n",
      "\n",
      "Hugo_Symbol  ATR  ATRX  AXIN1  AXIN2  B2M  BCLAF1  BCOR  BRCA1  BRCA2  BRIP1  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062       0     4      0      0    0       0     0      0      0      0   \n",
      "CRUK0063       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0064       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0065       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0066       0     0      0      0    0       0     0      0      0      1   \n",
      "...          ...   ...    ...    ...  ...     ...   ...    ...    ...    ...   \n",
      "CRUK0772       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0781       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0784       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0861       0     0      0      0    0       0     0      0      0      0   \n",
      "CRUK0872       0     0      0      0    0       0     0      0      0      0   \n",
      "\n",
      "Hugo_Symbol  CASP8  CBLB  CDC73  CDK12  CDKN2A  CEP76  CHD4  CHD8  CHEK2  CIC  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062         0     0      0      0       0      0     3     0      0    0   \n",
      "CRUK0063         0     0      0      0       1      0     0     0      0    0   \n",
      "CRUK0064         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0065         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0066         0     0      0      0       1      0     0     0      0    0   \n",
      "...            ...   ...    ...    ...     ...    ...   ...   ...    ...  ...   \n",
      "CRUK0772         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0781         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0784         0     0      0      0       0      0     3     0      0    0   \n",
      "CRUK0861         0     0      0      0       0      0     0     0      0    0   \n",
      "CRUK0872         0     0      0      0       0      0     0     0      0    0   \n",
      "\n",
      "Hugo_Symbol  CMTR2  CNOT3  COL2A1  COL5A2  CREBBP  CUL3  CUX1  CYLD  DDX3X  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0063         0      0       0       0       0     0     0     3      0   \n",
      "CRUK0064         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0065         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0066         0      0       0       1       0     0     0     0      0   \n",
      "...            ...    ...     ...     ...     ...   ...   ...   ...    ...   \n",
      "CRUK0772         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0781         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0784         0      0       0       0       0     0     0     0      0   \n",
      "CRUK0861         0      0       0       0       0     2     0     0      0   \n",
      "CRUK0872         0      0       0       0       0     0     0     0      0   \n",
      "\n",
      "Hugo_Symbol  DICER1  DNAH12  DNM2  DNMT3A  EP300  EXT2  FANCM  FAT1  FBXW7  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0063          0       0     0       0      3     0      1     0      3   \n",
      "CRUK0064          0       0     0       0      0     0      0     3      0   \n",
      "CRUK0065          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0066          0       0     0       0      0     0      0     0      0   \n",
      "...             ...     ...   ...     ...    ...   ...    ...   ...    ...   \n",
      "CRUK0772          0       0     0       0      0     0      1     0      0   \n",
      "CRUK0781          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0784          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0861          0       0     0       0      0     0      0     0      0   \n",
      "CRUK0872          0       0     0       0      0     0      0     0      0   \n",
      "\n",
      "Hugo_Symbol  FGFR3  GATA3  HCN1  HIST1H4E  IKZF1  KDM5C  KDM6A  KEAP1  KMT2B  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0063         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0064         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0065         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0066         0      0     0         0      0      0      0      0      0   \n",
      "...            ...    ...   ...       ...    ...    ...    ...    ...    ...   \n",
      "CRUK0772         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0781         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0784         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0861         0      0     0         0      0      0      0      0      0   \n",
      "CRUK0872         0      0     0         0      0      0      0      0      1   \n",
      "\n",
      "Hugo_Symbol  KMT2C  KMT2D  KRAS  LATS1  MAP3K13  MBD1  MGA  MLH1  MTOR  NCOA6  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0063         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0064         0      0     0      0        0     0    0     3     0      0   \n",
      "CRUK0065         0      0     0      0        0     0    0     3     0      3   \n",
      "CRUK0066         0      0     0      0        0     0    0     0     0      0   \n",
      "...            ...    ...   ...    ...      ...   ...  ...   ...   ...    ...   \n",
      "CRUK0772         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0781         0      4     0      0        0     0    0     0     0      0   \n",
      "CRUK0784         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0861         0      0     0      0        0     0    0     0     0      0   \n",
      "CRUK0872         0      0     0      0        0     0    0     0     0      0   \n",
      "\n",
      "Hugo_Symbol  NCOR1  NF1  NF2  NFE2L2  NOTCH1  NOTCH2  PASK  PBRM1  PIK3CA  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062         0    0    0       0       0       0     0      0       0   \n",
      "CRUK0063         0    3    0       0       0       0     0      0       1   \n",
      "CRUK0064         0    0    0       0       0       0     0      0       0   \n",
      "CRUK0065         0    0    0       4       3       0     0      0       3   \n",
      "CRUK0066         0    0    0       0       1       0     0      0       0   \n",
      "...            ...  ...  ...     ...     ...     ...   ...    ...     ...   \n",
      "CRUK0772         0    0    0       1       0       0     0      0       0   \n",
      "CRUK0781         0    0    0       0       0       0     4      0       0   \n",
      "CRUK0784         0    0    0       0       0       0     0      0       0   \n",
      "CRUK0861         0    0    0       0       0       0     0      0       0   \n",
      "CRUK0872         0    0    0       0       0       0     0      0       0   \n",
      "\n",
      "Hugo_Symbol  PIK3R1  PLXNB2  PTEN  PTPRB  PTPRC  PTPRU  RASA1  RB1  RYR2  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062          0       3     0      0      0      0      0    0     0   \n",
      "CRUK0063          0       0     0      0      0      0      0    0     0   \n",
      "CRUK0064          0       0     0      0      0      0      0    0     0   \n",
      "CRUK0065          0       0     0      0      3      0      0    0     0   \n",
      "CRUK0066          0       0     0      0      0      0      0    0     0   \n",
      "...             ...     ...   ...    ...    ...    ...    ...  ...   ...   \n",
      "CRUK0772          0       0     0      0      0      0      0    1     0   \n",
      "CRUK0781          0       0     4      0      0      0      0    0     0   \n",
      "CRUK0784          0       0     0      0      0      0      0    0     0   \n",
      "CRUK0861          0       0     0      0      0      0      0    0     0   \n",
      "CRUK0872          0       0     0      1      0      0      0    0     0   \n",
      "\n",
      "Hugo_Symbol  SETD2  SETDB1  SGK223  SMAD4  SMARCA4  SPEN  STAG2  STK11  TET2  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0063         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0064         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0065         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0066         0       0       0      0        0     0      0      0     0   \n",
      "...            ...     ...     ...    ...      ...   ...    ...    ...   ...   \n",
      "CRUK0772         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0781         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0784         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0861         0       0       0      0        0     0      0      0     0   \n",
      "CRUK0872         0       0       0      0        0     0      0      0     0   \n",
      "\n",
      "Hugo_Symbol  TNFAIP3  TP53  TP53BP1  TSC2  UBR5  WRN  ZMYM2  \n",
      "patient_id                                                   \n",
      "CRUK0062           0     1        0     0     3    0      0  \n",
      "CRUK0063           0     1        0     0     0    0      0  \n",
      "CRUK0064           0     1        0     0     0    0      0  \n",
      "CRUK0065           0     4        0     0     3    0      0  \n",
      "CRUK0066           0     1        0     0     0    1      0  \n",
      "...              ...   ...      ...   ...   ...  ...    ...  \n",
      "CRUK0772           0     1        3     0     0    0      0  \n",
      "CRUK0781           0     4        0     0     0    0      0  \n",
      "CRUK0784           0     2        0     0     0    0      0  \n",
      "CRUK0861           0     1        0     0     0    0      0  \n",
      "CRUK0872           0     1        0     0     0    0      0  \n",
      "\n",
      "[138 rows x 100 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUSC_driver_matrix = LUSC_driver_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUSC_driver_matrix.columns:\n",
    "            if cumulative_LUSC_driver_matrix.loc[cumulative_LUSC_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUSC_driver_matrix.loc[cumulative_LUSC_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUSC_driver_matrix:\")\n",
    "print(cumulative_LUSC_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUSC_fifty_driver_matrix:\n",
      "Hugo_Symbol  ACVR1B  APC  ARHGAP35  ARID1A  ARID2  ASXL1  ATR  ATRX  B2M  \\\n",
      "patient_id                                                                 \n",
      "CRUK0062          0    0         0       0      0      0    0     4    0   \n",
      "CRUK0063          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0064          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0065          0    0         0       3      0      0    0     0    0   \n",
      "CRUK0066          0    0         0       0      0      0    0     0    0   \n",
      "...             ...  ...       ...     ...    ...    ...  ...   ...  ...   \n",
      "CRUK0772          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0781          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0784          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0861          0    0         0       0      0      0    0     0    0   \n",
      "CRUK0872          0    0         0       0      0      0    0     0    0   \n",
      "\n",
      "Hugo_Symbol  BCLAF1  BCOR  CBLB  CDKN2A  CHD4  CHD8  COL5A2  CREBBP  CUX1  \\\n",
      "patient_id                                                                  \n",
      "CRUK0062          0     0     0       0     3     0       0       0     0   \n",
      "CRUK0063          0     0     0       1     0     0       0       0     0   \n",
      "CRUK0064          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0065          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0066          0     0     0       1     0     0       1       0     0   \n",
      "...             ...   ...   ...     ...   ...   ...     ...     ...   ...   \n",
      "CRUK0772          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0781          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0784          0     0     0       0     3     0       0       0     0   \n",
      "CRUK0861          0     0     0       0     0     0       0       0     0   \n",
      "CRUK0872          0     0     0       0     0     0       0       0     0   \n",
      "\n",
      "Hugo_Symbol  CYLD  EP300  FANCM  FAT1  KEAP1  KMT2B  KMT2C  KMT2D  KRAS  \\\n",
      "patient_id                                                                \n",
      "CRUK0062        0      0      0     0      0      0      0      0     0   \n",
      "CRUK0063        3      3      1     0      0      0      0      0     0   \n",
      "CRUK0064        0      0      0     3      0      0      0      0     0   \n",
      "CRUK0065        0      0      0     0      0      0      0      0     0   \n",
      "CRUK0066        0      0      0     0      0      0      0      0     0   \n",
      "...           ...    ...    ...   ...    ...    ...    ...    ...   ...   \n",
      "CRUK0772        0      0      1     0      0      0      0      0     0   \n",
      "CRUK0781        0      0      0     0      0      0      0      4     0   \n",
      "CRUK0784        0      0      0     0      0      0      0      0     0   \n",
      "CRUK0861        0      0      0     0      0      0      0      0     0   \n",
      "CRUK0872        0      0      0     0      0      1      0      0     0   \n",
      "\n",
      "Hugo_Symbol  NCOA6  NCOR1  NF1  NFE2L2  NOTCH1  NOTCH2  PBRM1  PIK3CA  PLXNB2  \\\n",
      "patient_id                                                                      \n",
      "CRUK0062         0      0    0       0       0       0      0       0       3   \n",
      "CRUK0063         0      0    3       0       0       0      0       1       0   \n",
      "CRUK0064         0      0    0       0       0       0      0       0       0   \n",
      "CRUK0065         3      0    0       4       3       0      0       3       0   \n",
      "CRUK0066         0      0    0       0       1       0      0       0       0   \n",
      "...            ...    ...  ...     ...     ...     ...    ...     ...     ...   \n",
      "CRUK0772         0      0    0       1       0       0      0       0       0   \n",
      "CRUK0781         0      0    0       0       0       0      0       0       0   \n",
      "CRUK0784         0      0    0       0       0       0      0       0       0   \n",
      "CRUK0861         0      0    0       0       0       0      0       0       0   \n",
      "CRUK0872         0      0    0       0       0       0      0       0       0   \n",
      "\n",
      "Hugo_Symbol  PTEN  PTPRB  PTPRC  RASA1  RB1  SETD2  SETDB1  SMAD4  SMARCA4  \\\n",
      "patient_id                                                                   \n",
      "CRUK0062        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0063        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0064        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0065        0      0      3      0    0      0       0      0        0   \n",
      "CRUK0066        0      0      0      0    0      0       0      0        0   \n",
      "...           ...    ...    ...    ...  ...    ...     ...    ...      ...   \n",
      "CRUK0772        0      0      0      0    1      0       0      0        0   \n",
      "CRUK0781        4      0      0      0    0      0       0      0        0   \n",
      "CRUK0784        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0861        0      0      0      0    0      0       0      0        0   \n",
      "CRUK0872        0      1      0      0    0      0       0      0        0   \n",
      "\n",
      "Hugo_Symbol  SPEN  STK11  TET2  TP53  UBR5  \n",
      "patient_id                                  \n",
      "CRUK0062        0      0     0     1     3  \n",
      "CRUK0063        0      0     0     1     0  \n",
      "CRUK0064        0      0     0     1     0  \n",
      "CRUK0065        0      0     0     4     3  \n",
      "CRUK0066        0      0     0     1     0  \n",
      "...           ...    ...   ...   ...   ...  \n",
      "CRUK0772        0      0     0     1     0  \n",
      "CRUK0781        0      0     0     4     0  \n",
      "CRUK0784        0      0     0     2     0  \n",
      "CRUK0861        0      0     0     1     0  \n",
      "CRUK0872        0      0     0     1     0  \n",
      "\n",
      "[138 rows x 50 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUSC_fifty_driver_matrix = LUSC_fifty_driver_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUSC_fifty_driver_matrix.columns:\n",
    "            if cumulative_LUSC_fifty_driver_matrix.loc[cumulative_LUSC_fifty_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUSC_fifty_driver_matrix.loc[cumulative_LUSC_fifty_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUSC_fifty_driver_matrix:\")\n",
    "print(cumulative_LUSC_fifty_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUSC_twenty_driver_matrix:\n",
      "Hugo_Symbol  ARID1A  ARID2  CDKN2A  COL5A2  CREBBP  CUX1  FAT1  KEAP1  KMT2C  \\\n",
      "patient_id                                                                     \n",
      "CRUK0062          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0063          0      0       1       0       0     0     0      0      0   \n",
      "CRUK0064          0      0       0       0       0     0     3      0      0   \n",
      "CRUK0065          3      0       0       0       0     0     0      0      0   \n",
      "CRUK0066          0      0       1       1       0     0     0      0      0   \n",
      "...             ...    ...     ...     ...     ...   ...   ...    ...    ...   \n",
      "CRUK0784          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0861          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0872          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0331          0      0       0       0       0     0     0      0      0   \n",
      "CRUK0118          0      0       0       0       0     0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  KMT2D  NF1  NFE2L2  NOTCH1  PIK3CA  PLXNB2  PTEN  RASA1  SETD2  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062         0    0       0       0       0       3     0      0      0   \n",
      "CRUK0063         0    3       0       0       1       0     0      0      0   \n",
      "CRUK0064         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0065         0    0       4       3       3       0     0      0      0   \n",
      "CRUK0066         0    0       0       1       0       0     0      0      0   \n",
      "...            ...  ...     ...     ...     ...     ...   ...    ...    ...   \n",
      "CRUK0784         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0861         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0872         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0331         0    0       0       0       0       0     0      0      0   \n",
      "CRUK0118         0    0       0       0       0       0     0      0      0   \n",
      "\n",
      "Hugo_Symbol  TP53  UBR5  \n",
      "patient_id               \n",
      "CRUK0062        1     3  \n",
      "CRUK0063        1     0  \n",
      "CRUK0064        1     0  \n",
      "CRUK0065        4     3  \n",
      "CRUK0066        1     0  \n",
      "...           ...   ...  \n",
      "CRUK0784        2     0  \n",
      "CRUK0861        1     0  \n",
      "CRUK0872        1     0  \n",
      "CRUK0331        0     0  \n",
      "CRUK0118        0     0  \n",
      "\n",
      "[138 rows x 20 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUSC_twenty_driver_matrix = LUSC_twenty_driver_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUSC_twenty_driver_matrix.columns:\n",
    "            if cumulative_LUSC_twenty_driver_matrix.loc[cumulative_LUSC_twenty_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUSC_twenty_driver_matrix.loc[cumulative_LUSC_twenty_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUSC_twenty_driver_matrix:\")\n",
    "print(cumulative_LUSC_twenty_driver_matrix) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Transformed LUSC_ten_driver_matrix:\n",
      "Hugo_Symbol  CDKN2A  COL5A2  CUX1  FAT1  KEAP1  KMT2D  NFE2L2  PIK3CA  PTEN  \\\n",
      "patient_id                                                                    \n",
      "CRUK0062          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0063          1       0     0     0      0      0       0       1     0   \n",
      "CRUK0064          0       0     0     3      0      0       0       0     0   \n",
      "CRUK0065          0       0     0     0      0      0       4       3     0   \n",
      "CRUK0066          1       1     0     0      0      0       0       0     0   \n",
      "...             ...     ...   ...   ...    ...    ...     ...     ...   ...   \n",
      "CRUK0861          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0872          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0331          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0118          0       0     0     0      0      0       0       0     0   \n",
      "CRUK0084          0       0     0     0      0      0       0       0     0   \n",
      "\n",
      "Hugo_Symbol  TP53  \n",
      "patient_id         \n",
      "CRUK0062        1  \n",
      "CRUK0063        1  \n",
      "CRUK0064        1  \n",
      "CRUK0065        4  \n",
      "CRUK0066        1  \n",
      "...           ...  \n",
      "CRUK0861        1  \n",
      "CRUK0872        1  \n",
      "CRUK0331        0  \n",
      "CRUK0118        0  \n",
      "CRUK0084        0  \n",
      "\n",
      "[138 rows x 10 columns]\n"
     ]
    }
   ],
   "source": [
    "# add timing\n",
    "cumulative_LUSC_ten_driver_matrix = LUSC_ten_driver_matrix.copy()\n",
    "\n",
    "# Map values to the new gene_matrix_w_timing\n",
    "timing_mapping = {'early': 1, 'late': 2, 'subclonal': 3, 'Unknown': 4, 'NA': 4}\n",
    "\n",
    "# transform the gene_matrix_w_timing using the most_common_timing and timing_mapping\n",
    "for index, row in most_common_timing.iterrows():\n",
    "    try:\n",
    "        if row['Hugo_Symbol'] in cumulative_LUSC_ten_driver_matrix.columns:\n",
    "            if cumulative_LUSC_ten_driver_matrix.loc[cumulative_LUSC_ten_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']].values[0] != 0:\n",
    "                cumulative_LUSC_ten_driver_matrix.loc[cumulative_LUSC_ten_driver_matrix.index == row['patient_id'], row['Hugo_Symbol']] = timing_mapping[row['combTiming_SC']]\n",
    "    except IndexError:\n",
    "        pass\n",
    "\n",
    "# Display the resulting gene_matrix_w_timing after the transformation\n",
    "print(\"Transformed LUSC_ten_driver_matrix:\")\n",
    "print(cumulative_LUSC_ten_driver_matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add ploidy\n",
    "cumulative_LUSC_gene_matrix['Ploidy'] = cumulative_LUSC_gene_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUSC_gene_matrix = cumulative_LUSC_gene_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUSC_driver_matrix['Ploidy'] = cumulative_LUSC_driver_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUSC_driver_matrix = cumulative_LUSC_driver_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUSC_fifty_gene_matrix['Ploidy'] = cumulative_LUSC_fifty_gene_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUSC_fifty_gene_matrix = cumulative_LUSC_fifty_gene_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUSC_fifty_driver_matrix['Ploidy'] = cumulative_LUSC_fifty_driver_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUSC_fifty_driver_matrix = cumulative_LUSC_fifty_driver_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUSC_twenty_gene_matrix['Ploidy'] = cumulative_LUSC_twenty_gene_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUSC_twenty_gene_matrix = cumulative_LUSC_twenty_gene_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUSC_twenty_driver_matrix['Ploidy'] = cumulative_LUSC_twenty_driver_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUSC_twenty_driver_matrix = cumulative_LUSC_twenty_driver_matrix.dropna(subset=['Ploidy'])\n",
    "cumulative_LUSC_ten_driver_matrix['Ploidy'] = cumulative_LUSC_ten_driver_matrix.index.map(cna_table_no_duplicates.set_index('patient_id')['Ploidy'])\n",
    "cumulative_LUSC_ten_driver_matrix = cumulative_LUSC_ten_driver_matrix.dropna(subset=['Ploidy'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sort all cumulative matrices by patient_id\n",
    "cumulative_LUAD_gene_matrix = cumulative_LUAD_gene_matrix.sort_index()\n",
    "cumulative_LUAD_fifty_gene_matrix = cumulative_LUAD_fifty_gene_matrix.sort_index()\n",
    "cumulative_LUAD_twenty_gene_matrix = cumulative_LUAD_twenty_gene_matrix.sort_index()\n",
    "cumulative_LUAD_driver_matrix = cumulative_LUAD_driver_matrix.sort_index()\n",
    "cumulative_LUAD_fifty_driver_matrix = cumulative_LUAD_fifty_driver_matrix.sort_index()\n",
    "cumulative_LUAD_twenty_driver_matrix = cumulative_LUAD_twenty_driver_matrix.sort_index()\n",
    "cumulative_LUAD_ten_driver_matrix = cumulative_LUAD_ten_driver_matrix.sort_index()\n",
    "\n",
    "cumulative_LUSC_gene_matrix = cumulative_LUSC_gene_matrix.sort_index()\n",
    "cumulative_LUSC_fifty_gene_matrix = cumulative_LUSC_fifty_gene_matrix.sort_index()\n",
    "cumulative_LUSC_twenty_gene_matrix = cumulative_LUSC_twenty_gene_matrix.sort_index()\n",
    "cumulative_LUSC_driver_matrix = cumulative_LUSC_driver_matrix.sort_index()\n",
    "cumulative_LUSC_fifty_driver_matrix = cumulative_LUSC_fifty_driver_matrix.sort_index()\n",
    "cumulative_LUSC_twenty_driver_matrix = cumulative_LUSC_twenty_driver_matrix.sort_index()\n",
    "cumulative_LUSC_ten_driver_matrix = cumulative_LUSC_ten_driver_matrix.sort_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "# select clinical_df rows with patient_id in LUAD_patient_id\n",
    "clinical_df_LUAD_cumulative = clinical_df[clinical_df['cruk_id'].isin(cumulative_LUAD_driver_matrix.index)]\n",
    "# select clinical_df rows with patient_id in LUSC_patient_id\n",
    "clinical_df_LUSC_cumulative = clinical_df[clinical_df['cruk_id'].isin(cumulative_LUSC_driver_matrix.index)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sort clinical_df by patient_id\n",
    "clinical_df_LUAD_cumulative = clinical_df_LUAD_cumulative.sort_values(by=['cruk_id'])\n",
    "clinical_df_LUSC_cumulative = clinical_df_LUSC_cumulative.sort_values(by=['cruk_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[0 0 1 1 0 0 0 0 1 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 1 0 1 0 0 1 1 1 1 1 0\n",
      " 1 0 1 1 0 0 1 0 1 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 0 1 1 1 0 0\n",
      " 0 0 1 0 0 0 0 0 1 0 1 0 1 0 0 0 0 0 0 0 0 0 1 1 1 0 1 0 1 1 1 0 0 0 0 0 1\n",
      " 1 0 0 1 0 1 0 0 0 1 1 0 0 1 1 0 0 0 1 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 0 1 0\n",
      " 0 0 0 0 0 1 0 1 0 1 1 0 1 0 1 0 1 1 1 0 0 0 1 0 0 1 1 1 0 0 0 0 0 0 0 1 0\n",
      " 0 0 0 0 1 0 0 1 0 0 0 0 1 0 1 0 0 1 0 1 0 0 0 0 1 1 0 0 0 1 0 0 0 0 0 1 1\n",
      " 0 1 0]\n",
      "length of y for LUAD cumulative 225\n",
      "[1 1 0 1 0 0 0 0 0 0 0 1 1 0 0 1 0 0 1 0 0 0 0 0 1 1 1 0 1 0 0 1 1 0 0 1 0\n",
      " 0 0 1 0 0 0 1 1 1 1 0 1 1 1 0 1 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 0 1 0 0 1 1\n",
      " 1 1 0 0 1 1 1 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 0 0 1 1 0 0 1\n",
      " 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 1]\n",
      "length of y for LUSC cumulative 129\n"
     ]
    }
   ],
   "source": [
    "# make y for LUAD\n",
    "y_train_LUAD_cumulative = []\n",
    "for index, row in clinical_df_LUAD_cumulative.iterrows():\n",
    "    if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "        y_train_LUAD_cumulative.append(1)\n",
    "    else:\n",
    "        y_train_LUAD_cumulative.append(0)\n",
    "\n",
    "y_LUAD_cumulative = np.array(y_train_LUAD_cumulative)\n",
    "print(y_LUAD_cumulative)\n",
    "print(\"length of y for LUAD cumulative\", len(y_LUAD_cumulative))\n",
    "\n",
    "# make y for LUSC\n",
    "y_train_LUSC_cumulative = []\n",
    "for index, row in clinical_df_LUSC_cumulative.iterrows():\n",
    "    if row['first_dfs_any_event_rec.or.new.primary'] == 'recurrence':\n",
    "        y_train_LUSC_cumulative.append(1)\n",
    "    else:\n",
    "        y_train_LUSC_cumulative.append(0)\n",
    "\n",
    "y_LUSC_cumulative = np.array(y_train_LUSC_cumulative)\n",
    "print(y_LUSC_cumulative)\n",
    "print(\"length of y for LUSC cumulative\", len(y_LUSC_cumulative))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157, 101) (68, 101) (157,) (68,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (2,)}\n",
      "Best score:  0.5246320346320347\n",
      "Predictions:  [0 0 1 0 1 1 0 0 0 0 1 0 0 0 1 1 0 0 1 0 0 1 0 0 0 0 1 1 1 0 1 0 0 1 0 1 1\n",
      " 0 0 1 1 0 0 0 0 1 0 1 0 1 1 1 0 0 0 0 1 1 0 1 0 0 1 0 0 0 0 0]\n",
      "Actual:  [0 1 0 1 0 0 1 0 0 0 0 1 0 0 1 1 0 1 1 0 0 0 0 0 0 0 0 1 1 0 0 1 1 0 0 1 0\n",
      " 0 0 1 1 0 1 0 0 0 0 0 0 0 1 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 1 0]\n",
      "Balanced Accuracy:  0.5869565217391304\n",
      "ROC AUC:  0.5874505928853755\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.74      0.67      0.70        46\n",
      "   Recurrence       0.42      0.50      0.46        22\n",
      "\n",
      "     accuracy                           0.62        68\n",
      "    macro avg       0.58      0.59      0.58        68\n",
      " weighted avg       0.64      0.62      0.62        68\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from sklearn.neural_network import MLPClassifier\n",
    "from sklearn.model_selection import GridSearchCV\n",
    "\n",
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUAD_gene_matrix, y_LUAD_cumulative, test_size = 0.3, stratify = y_LUAD_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157, 51) (68, 51) (157,) (68,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1, 'hidden_layer_sizes': (2,)}\n",
      "Best score:  0.584112554112554\n",
      "Predictions:  [1 0 0 1 1 0 0 0 0 0 1 0 1 0 0 0 0 1 0 1 0 0 0 1 1 0 0 1 1 0 1 0 0 0 0 0 0\n",
      " 0 0 0 0 0 1 1 1 0 1 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 0 0 0 0]\n",
      "Actual:  [0 0 0 0 1 0 0 1 0 1 0 0 0 0 1 0 1 0 1 0 0 1 0 0 1 0 1 1 0 0 1 0 0 0 1 1 0\n",
      " 0 0 0 1 0 0 0 0 0 1 0 1 0 0 1 0 0 0 0 0 1 1 1 1 0 1 0 0 0 0 0]\n",
      "Balanced Accuracy:  0.4950592885375494\n",
      "ROC AUC:  0.4342885375494071\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.72      0.69        46\n",
      "   Recurrence       0.32      0.27      0.29        22\n",
      "\n",
      "     accuracy                           0.57        68\n",
      "    macro avg       0.49      0.50      0.49        68\n",
      " weighted avg       0.56      0.57      0.56        68\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUAD_fifty_gene_matrix, y_LUAD_cumulative, test_size = 0.3, stratify = y_LUAD_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157, 21) (68, 21) (157,) (68,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5936363636363637\n",
      "Predictions:  [1 0 0 0 0 0 1 0 0 0 0 0 0 0 1 0 0 1 0 0 1 0 1 0 1 1 0 0 0 0 0 1 1 0 1 0 0\n",
      " 0 1 0 1 1 1 1 0 1 0 1 1 0 1 0 1 0 0 0 0 1 1 0 0 0 0 0 1 0 1 0]\n",
      "Actual:  [0 0 0 1 0 0 0 1 1 1 0 0 1 0 1 1 0 0 1 1 0 0 0 1 0 1 0 0 0 0 1 0 0 0 1 0 0\n",
      " 0 0 1 1 0 0 1 0 1 0 1 0 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0]\n",
      "Balanced Accuracy:  0.5642292490118577\n",
      "ROC AUC:  0.5830039525691699\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.72      0.67      0.70        46\n",
      "   Recurrence       0.40      0.45      0.43        22\n",
      "\n",
      "     accuracy                           0.60        68\n",
      "    macro avg       0.56      0.56      0.56        68\n",
      " weighted avg       0.62      0.60      0.61        68\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUAD_twenty_gene_matrix, y_LUAD_cumulative, test_size = 0.3, stratify = y_LUAD_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157, 101) (68, 101) (157,) (68,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 10, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.5\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [1 0 0 1 0 0 1 1 0 0 1 0 1 0 0 1 0 0 1 0 0 0 0 0 1 0 1 0 0 0 0 0 0 1 0 0 1\n",
      " 0 0 0 0 0 1 0 0 1 0 1 0 0 0 0 0 0 0 1 0 1 1 1 0 1 1 0 0 0 1 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.5\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.68      1.00      0.81        46\n",
      "   Recurrence       0.00      0.00      0.00        22\n",
      "\n",
      "     accuracy                           0.68        68\n",
      "    macro avg       0.34      0.50      0.40        68\n",
      " weighted avg       0.46      0.68      0.55        68\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUAD_driver_matrix, y_LUAD_cumulative, test_size = 0.3, stratify = y_LUAD_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157, 51) (68, 51) (157,) (68,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.5\n",
      "Predictions:  [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0\n",
      " 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]\n",
      "Actual:  [0 0 1 0 1 0 0 0 0 0 0 0 0 1 1 0 0 0 1 1 0 1 0 1 0 0 1 1 0 0 1 1 0 0 0 1 0\n",
      " 0 0 0 1 0 1 1 1 0 0 0 1 0 0 0 1 0 0 1 0 0 1 0 0 0 1 0 0 0 0 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.4214426877470355\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.68      1.00      0.81        46\n",
      "   Recurrence       0.00      0.00      0.00        22\n",
      "\n",
      "     accuracy                           0.68        68\n",
      "    macro avg       0.34      0.50      0.40        68\n",
      " weighted avg       0.46      0.68      0.55        68\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/metrics/_classification.py:1497: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUAD_fifty_driver_matrix, y_LUAD_cumulative, test_size = 0.3, stratify = y_LUAD_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157, 21) (68, 21) (157,) (68,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5345887445887446\n",
      "Predictions:  [1 0 0 1 0 1 0 1 0 1 1 0 0 0 0 1 0 0 1 1 1 1 0 0 1 1 0 0 0 0 1 0 0 0 0 1 0\n",
      " 1 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 1 0 1 0 0 0 0 0 1 0 0 1 0 1 0]\n",
      "Actual:  [0 1 0 1 1 1 0 1 0 0 0 0 0 1 0 0 0 1 0 0 0 0 0 0 0 0 0 1 0 1 0 0 0 1 0 1 0\n",
      " 1 0 0 1 0 0 0 0 0 1 0 1 0 0 1 0 1 0 0 0 0 1 0 1 1 1 0 0 0 1 0]\n",
      "Balanced Accuracy:  0.5523715415019763\n",
      "ROC AUC:  0.4634387351778657\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.71      0.70      0.70        46\n",
      "   Recurrence       0.39      0.41      0.40        22\n",
      "\n",
      "     accuracy                           0.60        68\n",
      "    macro avg       0.55      0.55      0.55        68\n",
      " weighted avg       0.61      0.60      0.61        68\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUAD_twenty_driver_matrix, y_LUAD_cumulative, test_size = 0.3, stratify = y_LUAD_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157, 11) (68, 11) (157,) (68,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.585021645021645\n",
      "Predictions:  [0 0 1 1 1 0 0 1 1 0 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 1 0 1 0 0 0 0 1 0 0 0 0\n",
      " 0 0 1 1 0 1 1 0 0 0 1 1 0 0 0 0 1 0 1 1 0 0 0 0 1 0 0 0 0 1 1]\n",
      "Actual:  [1 1 0 1 1 0 1 0 0 1 0 0 0 0 0 0 0 1 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 0 1 0 0\n",
      " 0 0 0 1 0 0 1 1 0 1 1 1 0 0 1 0 0 1 1 0 1 1 0 0 0 0 0 0 1 0 0]\n",
      "Balanced Accuracy:  0.5296442687747036\n",
      "ROC AUC:  0.49752964426877466\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.70      0.70      0.70        46\n",
      "   Recurrence       0.36      0.36      0.36        22\n",
      "\n",
      "     accuracy                           0.59        68\n",
      "    macro avg       0.53      0.53      0.53        68\n",
      " weighted avg       0.59      0.59      0.59        68\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUAD_ten_driver_matrix, y_LUAD_cumulative, test_size = 0.3, stratify = y_LUAD_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(90, 101) (39, 101) (90,) (39,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5666666666666667\n",
      "Predictions:  [0 0 0 0 0 0 0 1 1 0 0 1 0 1 1 1 0 1 0 0 0 1 1 1 0 0 0 1 0 1 0 1 1 0 1 0 1\n",
      " 0 0]\n",
      "Actual:  [1 0 1 0 0 0 0 1 0 1 0 0 1 0 0 1 1 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 1 1 0 1 0\n",
      " 1 0]\n",
      "Balanced Accuracy:  0.4230769230769231\n",
      "ROC AUC:  0.3934911242603551\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.61      0.54      0.57        26\n",
      "   Recurrence       0.25      0.31      0.28        13\n",
      "\n",
      "     accuracy                           0.46        39\n",
      "    macro avg       0.43      0.42      0.42        39\n",
      " weighted avg       0.49      0.46      0.47        39\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUSC_gene_matrix, y_LUSC_cumulative, test_size = 0.3, stratify = y_LUSC_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(90, 51) (39, 51) (90,) (39,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.001, 'hidden_layer_sizes': (10,)}\n",
      "Best score:  0.5083333333333333\n",
      "Predictions:  [0 1 0 0 1 0 0 0 0 1 1 1 0 1 0 0 0 0 0 1 1 0 1 0 0 1 0 1 1 1 0 0 1 0 0 0 1\n",
      " 0 1]\n",
      "Actual:  [0 0 0 0 0 0 0 0 1 1 1 0 0 1 0 0 1 0 0 1 0 0 1 0 0 0 1 0 1 0 0 1 0 0 1 1 0\n",
      " 1 0]\n",
      "Balanced Accuracy:  0.5384615384615385\n",
      "ROC AUC:  0.47337278106508873\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.70      0.62      0.65        26\n",
      "   Recurrence       0.38      0.46      0.41        13\n",
      "\n",
      "     accuracy                           0.56        39\n",
      "    macro avg       0.54      0.54      0.53        39\n",
      " weighted avg       0.59      0.56      0.57        39\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUSC_fifty_gene_matrix, y_LUSC_cumulative, test_size = 0.3, stratify = y_LUSC_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(90, 21) (39, 21) (90,) (39,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5416666666666666\n",
      "Predictions:  [0 0 0 1 0 0 0 0 0 0 1 0 0 1 0 0 0 1 0 1 0 1 0 0 0 0 1 1 0 1 0 1 0 0 0 0 1\n",
      " 1 0]\n",
      "Actual:  [0 0 1 1 0 0 0 0 0 1 0 1 0 0 0 0 0 1 0 0 0 1 0 1 0 0 1 0 1 1 0 0 1 0 0 0 1\n",
      " 1 0]\n",
      "Balanced Accuracy:  0.6730769230769231\n",
      "ROC AUC:  0.6982248520710059\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.78      0.81      0.79        26\n",
      "   Recurrence       0.58      0.54      0.56        13\n",
      "\n",
      "     accuracy                           0.72        39\n",
      "    macro avg       0.68      0.67      0.68        39\n",
      " weighted avg       0.71      0.72      0.71        39\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUSC_twenty_gene_matrix, y_LUSC_cumulative, test_size = 0.3, stratify = y_LUSC_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(90, 101) (39, 101) (90,) (39,)\n",
      "Best parameters:  {'alpha': 1, 'hidden_layer_sizes': (15,)}\n",
      "Best score:  0.5416666666666666\n",
      "Predictions:  [0 1 0 0 0 0 0 0 1 0 1 0 0 0 1 0 1 1 0 0 0 1 0 0 0 1 0 0 1 0 0 0 0 0 1 0 1\n",
      " 0 0]\n",
      "Actual:  [0 0 1 0 1 0 0 0 1 0 1 0 0 0 0 0 1 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 0 1 1 0 0\n",
      " 1 1]\n",
      "Balanced Accuracy:  0.5192307692307692\n",
      "ROC AUC:  0.5473372781065088\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.68      0.73      0.70        26\n",
      "   Recurrence       0.36      0.31      0.33        13\n",
      "\n",
      "     accuracy                           0.59        39\n",
      "    macro avg       0.52      0.52      0.52        39\n",
      " weighted avg       0.57      0.59      0.58        39\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUSC_driver_matrix, y_LUSC_cumulative, test_size = 0.3, stratify = y_LUSC_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(90, 51) (39, 51) (90,) (39,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.01, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5666666666666667\n",
      "Predictions:  [1 0 1 1 0 0 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 0 0 0 0 1 1 1 0 0 0\n",
      " 1 0]\n",
      "Actual:  [1 1 0 0 0 1 1 0 1 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 0 1\n",
      " 0 1]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.5798816568047337\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.69      0.68        26\n",
      "   Recurrence       0.33      0.31      0.32        13\n",
      "\n",
      "     accuracy                           0.56        39\n",
      "    macro avg       0.50      0.50      0.50        39\n",
      " weighted avg       0.56      0.56      0.56        39\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUSC_fifty_driver_matrix, y_LUSC_cumulative, test_size = 0.3, stratify = y_LUSC_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(90, 21) (39, 21) (90,) (39,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n",
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 0.0001, 'hidden_layer_sizes': (1,)}\n",
      "Best score:  0.5333333333333334\n",
      "Predictions:  [0 1 0 0 1 0 0 0 1 0 1 0 1 0 0 1 0 0 0 0 0 0 1 1 0 0 1 0 0 0 1 0 0 0 1 0 0\n",
      " 1 0]\n",
      "Actual:  [1 0 0 1 1 0 0 0 0 1 0 1 0 0 1 0 1 0 1 0 0 0 1 0 1 0 1 0 0 0 0 0 1 0 0 0 0\n",
      " 1 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.40236686390532544\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.69      0.68        26\n",
      "   Recurrence       0.33      0.31      0.32        13\n",
      "\n",
      "     accuracy                           0.56        39\n",
      "    macro avg       0.50      0.50      0.50        39\n",
      " weighted avg       0.56      0.56      0.56        39\n",
      "\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUSC_twenty_driver_matrix, y_LUSC_cumulative, test_size = 0.3, stratify = y_LUSC_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(90, 11) (39, 11) (90,) (39,)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=2):\n",
      "ABNORMAL_TERMINATION_IN_LNSRCH.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Best parameters:  {'alpha': 1e-05, 'hidden_layer_sizes': (5,)}\n",
      "Best score:  0.5666666666666667\n",
      "Predictions:  [1 0 0 1 0 1 1 0 0 0 0 0 1 1 0 0 0 0 1 0 1 1 0 1 1 1 1 0 0 0 1 0 1 0 1 1 1\n",
      " 0 0]\n",
      "Actual:  [0 1 0 1 1 0 1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 0 1 0 0 0 0 0 0 1\n",
      " 1 0]\n",
      "Balanced Accuracy:  0.5\n",
      "ROC AUC:  0.49112426035502954\n",
      "               precision    recall  f1-score   support\n",
      "\n",
      "No Recurrence       0.67      0.54      0.60        26\n",
      "   Recurrence       0.33      0.46      0.39        13\n",
      "\n",
      "     accuracy                           0.51        39\n",
      "    macro avg       0.50      0.50      0.49        39\n",
      " weighted avg       0.56      0.51      0.53        39\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/sklearn/neural_network/_multilayer_perceptron.py:546: ConvergenceWarning: lbfgs failed to converge (status=1):\n",
      "STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n",
      "\n",
      "Increase the number of iterations (max_iter) or scale the data as shown in:\n",
      "    https://scikit-learn.org/stable/modules/preprocessing.html\n",
      "  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n"
     ]
    }
   ],
   "source": [
    "X_train, X_test, y_train, y_test = train_test_split(cumulative_LUSC_ten_driver_matrix, y_LUSC_cumulative, test_size = 0.3, stratify = y_LUSC_cumulative)\n",
    "print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)\n",
    "\n",
    "\n",
    "param_grid = {\n",
    "    'hidden_layer_sizes': [(1,), (2,), (5,), (10,), (15,)],\n",
    "    'alpha': [1e-5, 1e-4, 1e-3, 1e-2, 1e-1, 1, 10, 100]\n",
    "}\n",
    "\n",
    "clf = MLPClassifier(solver='lbfgs', random_state=1, max_iter=2000)\n",
    "\n",
    "grid_search = GridSearchCV(clf, param_grid, scoring='balanced_accuracy', n_jobs=-1)\n",
    "grid_search.fit(X_train, y_train)\n",
    "\n",
    "best_params = grid_search.best_params_\n",
    "print(\"Best parameters: \", grid_search.best_params_)\n",
    "print(\"Best score: \", grid_search.best_score_)\n",
    "\n",
    "predictions = grid_search.predict(X_test)\n",
    "\n",
    "target_names = ['No Recurrence', 'Recurrence']\n",
    "\n",
    "y_score = grid_search.predict_proba(X_test)[:, 1]\n",
    "roc_auc = roc_auc_score(y_test, y_score)\n",
    "\n",
    "print(\"Predictions: \",predictions)\n",
    "print(\"Actual: \",y_test)\n",
    "print(\"Balanced Accuracy: \",balanced_accuracy_score(y_test, predictions))\n",
    "print(\"ROC AUC: \",roc_auc)\n",
    "print(classification_report(y_test, predictions, target_names=target_names))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
